{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "from collections import Counter\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Выравниваем сырые данные:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[bwa_index] Pack FASTA... 0.00 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[bwa_index] 0.07 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.00 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.00 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 0.02 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index MG1655-K12.first400K.fasta\n",
      "[main] Real time: 0.106 sec; CPU: 0.109 sec\n"
     ]
    }
   ],
   "source": [
    "! bwa index MG1655-K12.first400K.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48301, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.27, 10.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.745 CPU sec, 2.708 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48357, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.22, 10.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.842 CPU sec, 2.703 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48345, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.35, 10.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.977 CPU sec, 2.851 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48337, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.37, 10.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.910 CPU sec, 2.754 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48338, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.35, 10.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.876 CPU sec, 2.727 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48367, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.33, 10.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 3.056 CPU sec, 2.881 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48287, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 221)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (179, 249)\n",
      "[M::mem_pestat] mean and std.dev: (214.23, 10.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (165, 263)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 3.135 CPU sec, 2.981 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48328, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.36, 10.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.994 CPU sec, 2.843 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48359, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.37, 10.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 3.081 CPU sec, 2.954 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48323, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.38, 10.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 3.084 CPU sec, 2.938 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48276, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.26, 10.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 3.114 CPU sec, 2.962 real sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48231, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.27, 10.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 3.103 CPU sec, 2.968 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48187, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.35, 10.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 3.192 CPU sec, 3.056 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48265, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.41, 10.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 3.182 CPU sec, 3.058 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48287, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.24, 10.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.909 CPU sec, 2.757 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48307, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.33, 10.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.834 CPU sec, 2.670 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48303, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.40, 10.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.772 CPU sec, 2.611 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48393, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.40, 10.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.815 CPU sec, 2.676 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48320, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.32, 10.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.875 CPU sec, 2.746 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48377, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.26, 10.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.887 CPU sec, 2.727 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48307, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.31, 10.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.787 CPU sec, 2.632 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48322, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 221)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (179, 249)\n",
      "[M::mem_pestat] mean and std.dev: (214.27, 10.06)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (165, 263)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.669 CPU sec, 2.507 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48361, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.23, 10.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_process_seqs] Processed 100000 reads in 2.669 CPU sec, 2.509 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48350, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.35, 10.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.710 CPU sec, 2.558 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48319, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.32, 10.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.657 CPU sec, 2.507 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48334, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.24, 10.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.622 CPU sec, 2.458 real sec\n",
      "[M::process] read 63204 sequences (6320400 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48289, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.28, 10.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.578 CPU sec, 2.445 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 30551, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.31, 10.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 63204 reads in 1.666 CPU sec, 1.550 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem MG1655-K12.first400K.fasta ecoli_400K_err_1.fastq ecoli_400K_err_2.fastq\n",
      "[main] Real time: 75.889 sec; CPU: 79.888 sec\n"
     ]
    }
   ],
   "source": [
    "! bwa mem MG1655-K12.first400K.fasta ecoli_400K_err_1.fastq ecoli_400K_err_2.fastq > alignment_raw.sam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Получаем и выравниваем данные из trimmomatic:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrimmomaticPE: Started with arguments:\n",
      " ecoli_400K_err_1.fastq ecoli_400K_err_2.fastq trim1_paired.fq trim1_unpaired.fq trim2_paired.fq trim2_unpaired.fq LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
      "Multiple cores found: Using 4 threads\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 1381602 Both Surviving: 1309635 (94,79%) Forward Only Surviving: 38225 (2,77%) Reverse Only Surviving: 28561 (2,07%) Dropped: 5181 (0,37%)\n",
      "TrimmomaticPE: Completed successfully\n"
     ]
    }
   ],
   "source": [
    "! java -jar /Users/olga/Downloads/Trimmomatic-0-2.39/trimmomatic-0.39.jar PE ecoli_400K_err_1.fastq ecoli_400K_err_2.fastq trim1_paired.fq trim1_unpaired.fq trim2_paired.fq trim2_unpaired.fq LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 108924 sequences (10000070 bp)...\n",
      "[M::process] read 109428 sequences (10000064 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 52778, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.25, 10.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 108924 reads in 2.363 CPU sec, 2.319 real sec\n",
      "[M::process] read 109606 sequences (10000071 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53060, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.26, 10.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 109428 reads in 2.549 CPU sec, 2.408 real sec\n",
      "[M::process] read 110054 sequences (10000174 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53130, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.35, 10.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 109606 reads in 2.387 CPU sec, 2.235 real sec\n",
      "[M::process] read 110554 sequences (10000134 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53392, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.36, 10.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 110054 reads in 2.299 CPU sec, 2.140 real sec\n",
      "[M::process] read 111350 sequences (10000032 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53623, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.31, 10.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 110554 reads in 2.462 CPU sec, 2.312 real sec\n",
      "[M::process] read 110210 sequences (10000082 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53989, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 221)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (179, 249)\n",
      "[M::mem_pestat] mean and std.dev: (214.24, 10.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (165, 263)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 111350 reads in 2.563 CPU sec, 2.397 real sec\n",
      "[M::process] read 111126 sequences (10000020 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53419, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.34, 10.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 110210 reads in 2.484 CPU sec, 2.325 real sec\n",
      "[M::process] read 111532 sequences (10000041 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53911, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.39, 10.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 111126 reads in 2.496 CPU sec, 2.473 real sec\n",
      "[M::process] read 111910 sequences (10000027 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54068, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.31, 10.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 111532 reads in 2.525 CPU sec, 2.380 real sec\n",
      "[M::process] read 112920 sequences (10000107 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54241, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.23, 10.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 111910 reads in 2.508 CPU sec, 2.344 real sec\n",
      "[M::process] read 111748 sequences (10000010 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54697, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.36, 10.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 112920 reads in 2.580 CPU sec, 2.422 real sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::process] read 108998 sequences (10000004 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54135, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.35, 10.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 111748 reads in 2.458 CPU sec, 2.285 real sec\n",
      "[M::process] read 108162 sequences (10000049 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 52813, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.29, 10.19)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 108998 reads in 2.522 CPU sec, 2.370 real sec\n",
      "[M::process] read 107370 sequences (10000170 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 52449, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.30, 10.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 108162 reads in 2.509 CPU sec, 2.349 real sec\n",
      "[M::process] read 106840 sequences (10000095 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 52049, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.42, 10.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 107370 reads in 2.430 CPU sec, 2.267 real sec\n",
      "[M::process] read 106290 sequences (10000041 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 51809, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.34, 10.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 106840 reads in 2.368 CPU sec, 2.201 real sec\n",
      "[M::process] read 106420 sequences (10000058 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 51532, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.27, 10.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 106290 reads in 2.293 CPU sec, 2.132 real sec\n",
      "[M::process] read 106396 sequences (10000123 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 51541, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.28, 10.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 106420 reads in 2.263 CPU sec, 2.108 real sec\n",
      "[M::process] read 106506 sequences (10000002 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 51530, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.28, 10.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 106396 reads in 2.425 CPU sec, 2.274 real sec\n",
      "[M::process] read 106098 sequences (10000172 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 51637, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.24, 10.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 106506 reads in 2.452 CPU sec, 2.289 real sec\n",
      "[M::process] read 105820 sequences (10000021 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 51411, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.32, 10.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 106098 reads in 2.393 CPU sec, 2.241 real sec\n",
      "[M::process] read 105544 sequences (10000051 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 51266, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.27, 10.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 105820 reads in 2.316 CPU sec, 2.155 real sec\n",
      "[M::process] read 105398 sequences (10000114 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 51086, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.26, 10.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_process_seqs] Processed 105544 reads in 2.253 CPU sec, 2.086 real sec\n",
      "[M::process] read 10066 sequences (959482 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 51080, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.27, 10.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 105398 reads in 2.211 CPU sec, 2.090 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 4867, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (206, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (174, 254)\n",
      "[M::mem_pestat] mean and std.dev: (214.23, 10.26)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (158, 270)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 10066 reads in 0.347 CPU sec, 0.226 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem MG1655-K12.first400K.fasta trim1_paired.fq trim2_paired.fq\n",
      "[main] Real time: 54.924 sec; CPU: 58.552 sec\n"
     ]
    }
   ],
   "source": [
    "! bwa mem MG1655-K12.first400K.fasta trim1_paired.fq trim2_paired.fq > alignment_trim.sam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Получаем и выравниваем данные из BayesHammer:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "== Warning ==  No assembly mode was specified! If you intend to assemble high-coverage multi-cell/isolate data, use '--isolate' option.\n",
      "\n",
      "\n",
      "Command line: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades.py\t-1\t/Users/olga/Documents/Bioinformatics Institute/NGS/ecoli_400K_err_1.fastq\t-2\t/Users/olga/Documents/Bioinformatics Institute/NGS/ecoli_400K_err_2.fastq\t-o\t/Users/olga/Documents/Bioinformatics Institute/NGS/spades_res\t--only-error-correction\t--threads\t8\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.15.2\n",
      "  Python version: 3.7.7\n",
      "  OS: Darwin-20.3.0-x86_64-i386-64bit\n",
      "\n",
      "Output dir: /Users/olga/Documents/Bioinformatics Institute/NGS/spades_res\n",
      "Mode: ONLY read error correction (without assembling)\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Standard mode\n",
      "  For multi-cell/isolate data we recommend to use '--isolate' option; for single-cell MDA data use '--sc'; for metagenomic data use '--meta'; for RNA-Seq use '--rna'.\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/Users/olga/Documents/Bioinformatics Institute/NGS/ecoli_400K_err_1.fastq']\n",
      "      right reads: ['/Users/olga/Documents/Bioinformatics Institute/NGS/ecoli_400K_err_2.fastq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Other parameters:\n",
      "  Dir for temp files: /Users/olga/Documents/Bioinformatics Institute/NGS/spades_res/tmp\n",
      "  Threads: 8\n",
      "  Memory limit (in Gb): 250\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /Users/olga/Documents/Bioinformatics Institute/NGS/spades_res/spades.log\n",
      "\n",
      "\n",
      "===== Before start started. \n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-hammer /Users/olga/Documents/Bioinformatics Institute/NGS/spades_res/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  76)   Loading config from /Users/olga/Documents/Bioinformatics Institute/NGS/spades_res/corrected/configs/config.info\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (hammer_tools.cpp          :  38)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 16 files using 1 threads. This might take a while.\n",
      "  0:00:00.002     1M / 2M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:00.002     1M / 2M    INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:00.002     1M / 2M    INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 4194304\n",
      "  0:00:00.002   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /Users/olga/Documents/Bioinformatics Institute/NGS/ecoli_400K_err_1.fastq\n",
      "  0:00:08.045   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 476992 reads\n",
      "  0:00:15.658   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 952384 reads\n",
      "  0:00:22.510   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 1381602 reads\n",
      "  0:00:22.510   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /Users/olga/Documents/Bioinformatics Institute/NGS/ecoli_400K_err_2.fastq\n",
      "  0:00:30.010   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 1868015 reads\n",
      "  0:00:37.380   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2350262 reads\n",
      "  0:00:43.937   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2763204 reads\n",
      "  0:00:43.937   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 2763204 reads processed\n",
      "  0:00:43.937     1M / 162M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:44.784     1M / 169M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 13488952 kmers in total.\n",
      "  0:00:44.784     1M / 169M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:45.892    11M / 173M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 13488952 kmers, 9754400 bytes occupied (5.78512 bits per kmer).\n",
      "  0:00:45.892    11M / 173M  INFO   K-mer Counting           (kmer_data.cpp             : 354)   Arranging kmers in hash map order\n",
      "  0:00:46.648   219M / 379M  INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:02:09.398   219M / 379M  INFO    General                 (main.cpp                  : 155)   Extracting clusters:\n",
      "  0:02:09.398   219M / 379M  INFO    General                 (concurrent_dsu.cpp        :  18)   Connecting to root\n",
      "  0:02:09.640   219M / 379M  INFO    General                 (concurrent_dsu.cpp        :  34)   Calculating counts\n",
      "  0:02:11.175   287M / 447M  INFO    General                 (concurrent_dsu.cpp        :  63)   Writing down entries\n",
      "  0:02:13.914   219M / 379M  INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 2187352\n",
      "  0:02:13.921   115M / 276M  INFO   K-mer Counting           (kmer_data.cpp             : 371)   Collecting K-mer information, this takes a while.\n",
      "  0:02:14.054   427M / 585M  INFO   K-mer Counting           (kmer_data.cpp             : 377)   Processing /Users/olga/Documents/Bioinformatics Institute/NGS/ecoli_400K_err_1.fastq\n",
      "  0:14:53.674   427M / 585M  INFO   K-mer Counting           (kmer_data.cpp             : 377)   Processing /Users/olga/Documents/Bioinformatics Institute/NGS/ecoli_400K_err_2.fastq\n",
      "  0:22:51.574   427M / 585M  INFO   K-mer Counting           (kmer_data.cpp             : 384)   Collection done, postprocessing.\n",
      "  0:22:51.659   427M / 585M  INFO   K-mer Counting           (kmer_data.cpp             : 398)   There are 13488952 kmers in total. Among them 9831116 (72.8827%) are singletons.\n",
      "  0:22:51.659   427M / 585M  INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:23:03.564   427M / 603M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering done. Total 1 non-read kmers were generated.\n",
      "  0:23:03.564   427M / 603M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)   Subclustering statistics:\n",
      "  0:23:03.564   427M / 603M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton hamming clusters: 1393328. Among them 2580 (0.185168%) are good\n",
      "  0:23:03.564   427M / 603M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total singleton subclusters: 4144. Among them 4141 (99.9276%) are good\n",
      "  0:23:03.564   427M / 603M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Total non-singleton subcluster centers: 855746. Among them 836856 (97.7926%) are good\n",
      "  0:23:03.564   427M / 603M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average size of non-trivial subcluster: 14.1346 kmers\n",
      "  0:23:03.564   427M / 603M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Average number of sub-clusters per non-singleton cluster: 1.08295\n",
      "  0:23:03.564   427M / 603M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Total solid k-mers: 843577\n",
      "  0:23:03.564   427M / 603M  INFO   Hamming Subclustering    (kmer_cluster.cpp          : 658)     Substitution probabilities: [4,4]((0.925364,0.0417853,0.0246913,0.008159),(0.0168245,0.960387,0.012269,0.0105199),(0.0104931,0.0122445,0.960404,0.0168586),(0.00814172,0.0246661,0.0417493,0.925443))\n",
      "  0:23:03.565   427M / 586M  INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:23:03.565   427M / 586M  INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 1 threads.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:24:11.264   427M / 586M  INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 45120 new k-mers.\n",
      "  0:25:18.904   427M / 586M  INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 1693 new k-mers.\n",
      "  0:26:25.950   427M / 586M  INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 63 new k-mers.\n",
      "  0:27:32.638   427M / 586M  INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 3 produced 32 new k-mers.\n",
      "  0:28:40.033   427M / 586M  INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 4 produced 0 new k-mers.\n",
      "  0:28:40.033   427M / 586M  INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:28:40.033   427M / 586M  INFO    General                 (hammer_tools.cpp          : 222)   Starting read correction in 1 threads.\n",
      "  0:28:40.033   427M / 586M  INFO    General                 (hammer_tools.cpp          : 235)   Correcting pair of reads: /Users/olga/Documents/Bioinformatics Institute/NGS/ecoli_400K_err_1.fastq and /Users/olga/Documents/Bioinformatics Institute/NGS/ecoli_400K_err_2.fastq\n",
      "  0:28:40.212   490M / 615M  INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 0 of 100000 reads.\n",
      "  0:28:45.501   497M / 617M  INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 0\n",
      "  0:28:45.726   497M / 617M  INFO    General                 (hammer_tools.cpp          : 187)   Written batch 0\n",
      "  0:28:45.872   497M / 617M  INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 1 of 100000 reads.\n",
      "  0:28:50.979   497M / 617M  INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 1\n",
      "  0:28:51.206   497M / 617M  INFO    General                 (hammer_tools.cpp          : 187)   Written batch 1\n",
      "  0:28:51.352   497M / 617M  INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 2 of 100000 reads.\n",
      "  0:28:56.539   497M / 617M  INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 2\n",
      "  0:28:56.773   497M / 617M  INFO    General                 (hammer_tools.cpp          : 187)   Written batch 2\n",
      "  0:28:56.924   497M / 617M  INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 3 of 100000 reads.\n",
      "  0:29:02.199   497M / 617M  INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 3\n",
      "  0:29:02.445   497M / 617M  INFO    General                 (hammer_tools.cpp          : 187)   Written batch 3\n",
      "  0:29:02.592   497M / 617M  INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 4 of 100000 reads.\n",
      "  0:29:07.818   497M / 617M  INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 4\n",
      "  0:29:08.069   497M / 617M  INFO    General                 (hammer_tools.cpp          : 187)   Written batch 4\n",
      "  0:29:08.230   497M / 617M  INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 5 of 100000 reads.\n",
      "  0:29:13.415   497M / 617M  INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 5\n",
      "  0:29:13.677   497M / 617M  INFO    General                 (hammer_tools.cpp          : 187)   Written batch 5\n",
      "  0:29:13.825   497M / 617M  INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 6 of 100000 reads.\n",
      "  0:29:19.108   497M / 617M  INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 6\n",
      "  0:29:19.374   497M / 617M  INFO    General                 (hammer_tools.cpp          : 187)   Written batch 6\n",
      "  0:29:19.519   497M / 617M  INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 7 of 100000 reads.\n",
      "  0:29:24.778   497M / 617M  INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 7\n",
      "  0:29:25.010   497M / 617M  INFO    General                 (hammer_tools.cpp          : 187)   Written batch 7\n",
      "  0:29:25.153   497M / 617M  INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 8 of 100000 reads.\n",
      "  0:29:30.440   497M / 617M  INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 8\n",
      "  0:29:30.665   497M / 617M  INFO    General                 (hammer_tools.cpp          : 187)   Written batch 8\n",
      "  0:29:30.808   497M / 617M  INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 9 of 100000 reads.\n",
      "  0:29:36.126   497M / 617M  INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 9\n",
      "  0:29:36.342   497M / 617M  INFO    General                 (hammer_tools.cpp          : 187)   Written batch 9\n",
      "  0:29:36.485   497M / 617M  INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 10 of 100000 reads.\n",
      "  0:29:41.815   497M / 617M  INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 10\n",
      "  0:29:42.041   497M / 617M  INFO    General                 (hammer_tools.cpp          : 187)   Written batch 10\n",
      "  0:29:42.184   497M / 617M  INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 11 of 100000 reads.\n",
      "  0:29:47.472   497M / 617M  INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 11\n",
      "  0:29:47.704   497M / 617M  INFO    General                 (hammer_tools.cpp          : 187)   Written batch 11\n",
      "  0:29:47.844   497M / 617M  INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 12 of 100000 reads.\n",
      "  0:29:53.109   497M / 617M  INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 12\n",
      "  0:29:53.325   497M / 617M  INFO    General                 (hammer_tools.cpp          : 187)   Written batch 12\n",
      "  0:29:53.441   497M / 617M  INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 13 of 81602 reads.\n",
      "  0:29:57.791   497M / 617M  INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 13\n",
      "  0:29:57.966   497M / 617M  INFO    General                 (hammer_tools.cpp          : 187)   Written batch 13\n",
      "  0:29:57.977   427M / 617M  INFO    General                 (hammer_tools.cpp          : 276)   Correction done. Changed 883777 bases in 651912 reads.\n",
      "  0:29:57.977   427M / 617M  INFO    General                 (hammer_tools.cpp          : 277)   Failed to correct 21932 bases out of 252262696.\n",
      "  0:29:58.004     1M / 205M  INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /Users/olga/Documents/Bioinformatics Institute/NGS/spades_res/corrected/corrected.yaml\n",
      "  0:29:58.004     1M / 205M  INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== corrected reads compression started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/miniconda3/bin/python /Users/olga/Downloads/SPAdes-3.15.2/share/spades/spades_pipeline/scripts/compress_all.py --input_file /Users/olga/Documents/Bioinformatics Institute/NGS/spades_res/corrected/corrected.yaml --ext_python_modules_home /Users/olga/Downloads/SPAdes-3.15.2/share/spades --max_threads 8 --output_dir /Users/olga/Documents/Bioinformatics Institute/NGS/spades_res/corrected --gzip_output\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "== Files to compress: ['/Users/olga/Documents/Bioinformatics Institute/NGS/spades_res/corrected/ecoli_400K_err_1.00.0_0.cor.fastq', '/Users/olga/Documents/Bioinformatics Institute/NGS/spades_res/corrected/ecoli_400K_err_2.00.0_0.cor.fastq', '/Users/olga/Documents/Bioinformatics Institute/NGS/spades_res/corrected/ecoli_400K_err__unpaired.00.0_0.cor.fastq']\n",
      "== Files compression is finished\n",
      "== Dataset yaml file is updated\n",
      "\n",
      "===== corrected reads compression finished. \n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Breaking scaffolds started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/miniconda3/bin/python /Users/olga/Downloads/SPAdes-3.15.2/share/spades/spades_pipeline/scripts/breaking_scaffolds_script.py --result_scaffolds_filename /Users/olga/Documents/Bioinformatics Institute/NGS/spades_res/scaffolds.fasta --misc_dir /Users/olga/Documents/Bioinformatics Institute/NGS/spades_res/misc --threshold_for_breaking_scaffolds 3\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "===== Breaking scaffolds finished. \r\n",
      "\r\n",
      "\r\n",
      "===== Terminate started. \r\n",
      "\r\n",
      "\r\n",
      "===== Terminate finished. \r\n",
      "\r\n",
      " * Corrected reads are in \"/Users/olga/Documents/Bioinformatics Institute/NGS/spades_res/corrected/\"\r\n",
      "\r\n",
      "======= SPAdes pipeline finished.\r\n",
      "\r\n",
      "SPAdes log can be found here: /Users/olga/Documents/Bioinformatics Institute/NGS/spades_res/spades.log\r\n",
      "\r\n",
      "Thank you for using SPAdes!\r\n"
     ]
    }
   ],
   "source": [
    "! /Users/olga/Downloads/SPAdes-3.15.2/bin/spades.py -1 ecoli_400K_err_1.fastq -2 ecoli_400K_err_2.fastq -o spades_res --only-error-correction --threads 8"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48352, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.27, 10.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.225 CPU sec, 2.054 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48401, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.22, 10.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.356 CPU sec, 2.097 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48399, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.35, 10.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.341 CPU sec, 2.087 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48377, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.38, 10.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.430 CPU sec, 2.136 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48371, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.32, 10.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.250 CPU sec, 1.971 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48414, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.36, 10.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.259 CPU sec, 1.982 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48317, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 221)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (179, 249)\n",
      "[M::mem_pestat] mean and std.dev: (214.24, 10.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (165, 263)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.290 CPU sec, 2.021 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48419, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.34, 10.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.477 CPU sec, 2.204 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48404, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.40, 10.07)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.372 CPU sec, 2.101 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48388, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.36, 10.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.423 CPU sec, 2.153 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48318, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.24, 10.18)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.257 CPU sec, 1.976 real sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48338, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.30, 10.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.421 CPU sec, 2.117 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48260, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.35, 10.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.370 CPU sec, 2.086 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48332, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.35, 10.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.461 CPU sec, 2.154 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48379, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.31, 10.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.435 CPU sec, 2.155 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48396, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.29, 10.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.286 CPU sec, 1.992 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48371, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.45, 10.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.227 CPU sec, 1.946 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48421, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.38, 10.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.254 CPU sec, 1.968 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48381, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 221)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (179, 249)\n",
      "[M::mem_pestat] mean and std.dev: (214.26, 10.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (165, 263)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.228 CPU sec, 1.947 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48426, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.32, 10.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.242 CPU sec, 1.958 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48336, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.24, 10.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.252 CPU sec, 1.967 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48423, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.24, 10.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.230 CPU sec, 1.949 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48404, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.32, 10.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_process_seqs] Processed 100000 reads in 2.243 CPU sec, 1.951 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48382, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.31, 10.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.235 CPU sec, 1.954 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48355, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.31, 10.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.227 CPU sec, 1.948 real sec\n",
      "[M::process] read 100000 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48327, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.25, 10.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.232 CPU sec, 1.949 real sec\n",
      "[M::process] read 16052 sequences (1605200 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48388, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.28, 10.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 100000 reads in 2.108 CPU sec, 1.956 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 7765, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (207, 214, 222)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 252)\n",
      "[M::mem_pestat] mean and std.dev: (214.28, 10.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (162, 267)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 16052 reads in 0.458 CPU sec, 0.338 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem MG1655-K12.first400K.fasta spades_res/corrected/ecoli_400K_err_1.00.0_0.cor.fastq.gz spades_res/corrected/ecoli_400K_err_2.00.0_0.cor.fastq.gz\n",
      "[main] Real time: 55.367 sec; CPU: 62.832 sec\n"
     ]
    }
   ],
   "source": [
    "! bwa mem MG1655-K12.first400K.fasta spades_res/corrected/ecoli_400K_err_1.00.0_0.cor.fastq.gz spades_res/corrected/ecoli_400K_err_2.00.0_0.cor.fastq.gz > alignment_bh.sam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Оценка качества исправления ошибок"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Описание метода: для каждой позиции референса записываем нуклеотиды из всех ридов, которые покрывают данную позицию; если данная позиция референса не покрыта ридами, то записываем 0. Далее считаем консенсус по каждой покрытой позиции референса - так считаем ошибки / совпадения в сырых и исправленных ридах. Удалённые ошибки / совпадения из сырых ридов считаем, смотря на не покрытые позиции референса в исправленных ридах. Статистика выводится в процентах."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# читаем референс\n",
    "ref_file = open('MG1655-K12.first400K.fasta', 'r')\n",
    "ref_length = 0\n",
    "ref_seq = ''\n",
    "ref_file.readline()\n",
    "for line in ref_file:\n",
    "    line = line.strip()\n",
    "    ref_length += len(line)\n",
    "    ref_seq += line"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_coverage_letters(sam_file_name, ref_seq, ref_length):    \n",
    "    sam_file = open(sam_file_name, 'r')\n",
    "    sam_file.readline()\n",
    "    sam_file.readline()\n",
    "    per_base_letters = [0]* ref_length\n",
    "    for line in sam_file:\n",
    "        line = line.strip().split()\n",
    "        start = int(line[3]) - 1\n",
    "        if start == -1: # unmapped read\n",
    "            continue\n",
    "        else:\n",
    "            read_length = len(line[9])\n",
    "            read_seq = line[9]\n",
    "            stop = min(start + read_length - 1, ref_length)\n",
    "            for pos in range(start, stop):\n",
    "                if per_base_letters[pos] == 0:\n",
    "                    per_base_letters[pos] = [list(read_seq)[pos - start]]\n",
    "                else:\n",
    "                    per_base_letters[pos].append(list(read_seq)[pos - start])\n",
    "    ref_file.close()\n",
    "    sam_file.close()\n",
    "    return per_base_letters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "coverage_letters_raw = get_coverage_letters('alignment_raw.sam', ref_seq, ref_length)\n",
    "coverage_letters_trim = get_coverage_letters('alignment_trim.sam', ref_seq, ref_length)\n",
    "coverage_letters_bh = get_coverage_letters('alignment_bh.sam', ref_seq, ref_length)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_correction_estimation(per_base_raw, per_base_corr, ref_seq):\n",
    "    data = np.zeros((2, 3))\n",
    "    for i in range(len(per_base_corr)):\n",
    "        if per_base_corr[i] == 0:\n",
    "            if per_base_raw[i] != 0:\n",
    "                counter = Counter(per_base_raw[i])\n",
    "                most_common = counter.most_common(1)[0][0]\n",
    "                if most_common != ref_seq[i] and most_common != 'N':\n",
    "                    data[0, 2] += 1\n",
    "                elif most_common == ref_seq[i] and most_common != 'N':\n",
    "                    data[1, 2] += 1\n",
    "        elif per_base_corr[i] != 0:\n",
    "            if per_base_raw[i] != 0:\n",
    "                counter_raw = Counter(per_base_raw[i])\n",
    "                counter_corr = Counter(per_base_corr[i])\n",
    "                most_common_raw = counter_raw.most_common(1)[0][0]\n",
    "                most_common_corr = counter_corr.most_common(1)[0][0]\n",
    "                if most_common_raw != 'N' and most_common_corr == 'N':\n",
    "                    if most_common_raw != ref_seq[i]:\n",
    "                        data[0, 2] += 1\n",
    "                    else:\n",
    "                        data[1, 2] += 1\n",
    "                elif most_common_raw != 'N' and most_common_corr != 'N':\n",
    "                    if most_common_raw != ref_seq[i] and most_common_corr != ref_seq[i]:\n",
    "                        data[0, 0] += 1\n",
    "                    elif most_common_raw != ref_seq[i] and most_common_corr == ref_seq[i]:\n",
    "                        data[0, 1] += 1\n",
    "                    elif most_common_raw == ref_seq[i] and most_common_corr != ref_seq[i]:\n",
    "                        data[1, 0] += 1\n",
    "                    elif most_common_raw == ref_seq[i] and most_common_corr == ref_seq[i]:\n",
    "                        data[1, 1] += 1\n",
    "    return (data / len(ref_seq)) * 100"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Данные trimmomatic:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  Error corr reads  Correct corr reads  Abscent corr reads\n",
      "Error raw data             0.00475               0.006                 0.0\n",
      "Correct raw data           0.00000              99.989                 0.0\n"
     ]
    }
   ],
   "source": [
    "data_trim = get_correction_estimation(coverage_letters_raw, coverage_letters_trim, ref_seq)\n",
    "data_trim_df = pd.DataFrame(data_trim, columns = ['Error corr reads', 'Correct corr reads', 'Abscent corr reads'], index = ['Error raw data', 'Correct raw data'])\n",
    "print(data_trim_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Видим, что процент незамеченных ошибок намного выше процента замеченных и исправленных ошибок. Исправленных правильных баз нет, все правильные нуклеотиды остались, что хорошо. Возможно, это связано с тем, что trimmomatic не исправляет ошибки, а удаляет риды и отдельные базы, из-за чего может \"смещаться\" консенсус. Возможно также, что нужно подобрать другие параметры и получить другие исправленные риды."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Данные BayesHammer:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                  Error corr reads  Correct corr reads  Abscent corr reads\n",
      "Error raw data             0.00475               0.000              0.0060\n",
      "Correct raw data           0.00050              99.503              0.4855\n"
     ]
    }
   ],
   "source": [
    "data_bh = get_correction_estimation(coverage_letters_raw, coverage_letters_bh, ref_seq)\n",
    "data_bh_df = pd.DataFrame(data_bh, columns = ['Error corr reads', 'Correct corr reads', 'Abscent corr reads'], index = ['Error raw data', 'Correct raw data'])\n",
    "print(data_bh_df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Видим, что исправленных ошибок в итоге не оказалось, наоборот, примерно полпроцента правильных баз были исправлены на неправильные. Все ошибки оказались незамеченными. Возможно, здесь тоже всё дело в параметрах программы."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
