{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 1. Сборка длинных ридов"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Для запуска Canu посмотрим на размер генома:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Genome size: 4639675\n"
     ]
    }
   ],
   "source": [
    "ref_len = 0\n",
    "f = open('./homework6/reference.fasta', 'r')\n",
    "for line in f:\n",
    "    if line.startswith('>'):\n",
    "        pass\n",
    "    else:\n",
    "        line = line.strip()\n",
    "        ref_len += len(line)\n",
    "print(f'Genome size: {ref_len}')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- Canu release v1.5\n",
      "-- Detected Java(TM) Runtime Environment '1.8.0_152-release' (from '/Users/olga/miniconda3/bin/java').\n",
      "-- Detected gnuplot version '5.4 patchlevel 1' (from 'gnuplot') and image format 'png'.\n",
      "-- Detected 8 CPUs and 16 gigabytes of memory.\n",
      "-- No grid engine detected, grid disabled.\n",
      "--\n",
      "-- Run   2 jobs concurrently using    8 GB and   4 CPUs for stage 'meryl'.\n",
      "-- Run   1 job  concurrently using    6 GB and   8 CPUs for stage 'mhap (cor)'.\n",
      "-- Run   1 job  concurrently using    8 GB and   8 CPUs for stage 'overlapper (obt)'.\n",
      "-- Run   1 job  concurrently using    8 GB and   8 CPUs for stage 'overlapper (utg)'.\n",
      "-- Run   4 jobs concurrently using    8 GB and   2 CPUs for stage 'falcon_sense'.\n",
      "-- Run   8 jobs concurrently using    2 GB and   1 CPU  for stage 'ovStore bucketizer'.\n",
      "-- Run   8 jobs concurrently using    8 GB and   1 CPU  for stage 'ovStore sorting'.\n",
      "-- Run   2 jobs concurrently using    2 GB and   4 CPUs for stage 'read error detection'.\n",
      "-- Run   8 jobs concurrently using    1 GB and   1 CPU  for stage 'overlap error adjustment'.\n",
      "-- Run   2 jobs concurrently using    8 GB and   4 CPUs for stage 'bogart'.\n",
      "-- Run   8 jobs concurrently using    2 GB and   1 CPU  for stage 'GFA alignment and processing'.\n",
      "-- Run   2 jobs concurrently using    8 GB and   4 CPUs for stage 'consensus'.\n",
      "--\n",
      "-- Generating assembly 'pacbio10' in '/Users/olga/Documents/BioinformaticsInstitute/NGS/pacbio10'\n",
      "--\n",
      "-- Parameters:\n",
      "--\n",
      "--  genomeSize        4700000\n",
      "--\n",
      "--  Overlap Generation Limits:\n",
      "--    corOvlErrorRate 0.2400 ( 24.00%)\n",
      "--    obtOvlErrorRate 0.0450 (  4.50%)\n",
      "--    utgOvlErrorRate 0.0450 (  4.50%)\n",
      "--\n",
      "--  Overlap Processing Limits:\n",
      "--    corErrorRate    0.3000 ( 30.00%)\n",
      "--    obtErrorRate    0.0450 (  4.50%)\n",
      "--    utgErrorRate    0.0450 (  4.50%)\n",
      "--    cnsErrorRate    0.0450 (  4.50%)\n",
      "--\n",
      "--\n",
      "-- BEGIN CORRECTION\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:33:31 2021 with 317.646 GB free disk space\n",
      "\n",
      "    cd correction\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperCreate \\\n",
      "      -minlength 1000 \\\n",
      "      -o ./pacbio10.gkpStore.BUILDING \\\n",
      "      ./pacbio10.gkpStore.gkp \\\n",
      "    > ./pacbio10.gkpStore.BUILDING.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:33:31 2021 (lickety-split) with 317.631 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- In gatekeeper store 'correction/pacbio10.gkpStore':\n",
      "--   Found 9478 reads.\n",
      "--   Found 53986092 bases (11.48 times coverage).\n",
      "--\n",
      "--   Read length histogram (one '*' equals 19.47 reads):\n",
      "--        0    999      0 \n",
      "--     1000   1999   1363 **********************************************************************\n",
      "--     2000   2999   1243 ***************************************************************\n",
      "--     3000   3999   1108 ********************************************************\n",
      "--     4000   4999    989 **************************************************\n",
      "--     5000   5999    967 *************************************************\n",
      "--     6000   6999    837 ******************************************\n",
      "--     7000   7999    697 ***********************************\n",
      "--     8000   8999    518 **************************\n",
      "--     9000   9999    503 *************************\n",
      "--    10000  10999    373 *******************\n",
      "--    11000  11999    274 **************\n",
      "--    12000  12999    200 **********\n",
      "--    13000  13999    172 ********\n",
      "--    14000  14999    109 *****\n",
      "--    15000  15999     71 ***\n",
      "--    16000  16999     34 *\n",
      "--    17000  17999     13 \n",
      "--    18000  18999      4 \n",
      "--    19000  19999      2 \n",
      "--    20000  20999      0 \n",
      "--    21000  21999      1 \n",
      "-- Finished stage 'cor-gatekeeper', reset canuIteration.\n",
      "-- Finished stage 'merylConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 11:33:32 2021 with 317.621 GB free disk space (1 processes; 2 concurrently)\n",
      "\n",
      "    cd correction/0-mercounts\n",
      "    ./meryl.sh 1 > ./meryl.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:33:46 2021 (14 seconds) with 317.414 GB free disk space\n",
      "----------------------------------------\n",
      "-- Meryl finished successfully.\n",
      "-- Finished stage 'merylCheck', reset canuIteration.\n",
      "--\n",
      "--  16-mers                                                                                           Fraction\n",
      "--    Occurrences   NumMers                                                                         Unique Total\n",
      "--       1-     1  43183063 *******************************************************************--> 0.9137 0.8020\n",
      "--       2-     2   2684728 ********************************************************************** 0.9705 0.9017\n",
      "--       3-     4   1162953 ******************************                                         0.9878 0.9471\n",
      "--       5-     7    205743 *****                                                                  0.9981 0.9858\n",
      "--       8-    11     15061                                                                        0.9997 0.9948\n",
      "--      12-    16      4656                                                                        0.9999 0.9965\n",
      "--      17-    22      1765                                                                        0.9999 0.9975\n",
      "--      23-    29       981                                                                        1.0000 0.9981\n",
      "--      30-    37       356                                                                        1.0000 0.9985\n",
      "--      38-    46       114                                                                        1.0000 0.9987\n",
      "--      47-    56        84                                                                        1.0000 0.9988\n",
      "--      57-    67        50                                                                        1.0000 0.9988\n",
      "--      68-    79        35                                                                        1.0000 0.9989\n",
      "--      80-    92        46                                                                        1.0000 0.9989\n",
      "--      93-   106        36                                                                        1.0000 0.9990\n",
      "--     107-   121        17                                                                        1.0000 0.9991\n",
      "--     122-   137        22                                                                        1.0000 0.9991\n",
      "--     138-   154        14                                                                        1.0000 0.9992\n",
      "--     155-   172         8                                                                        1.0000 0.9992\n",
      "--     173-   191         8                                                                        1.0000 0.9992\n",
      "--     192-   211         1                                                                        0.0000 0.9993\n",
      "--     212-   232         7                                                                        0.0000 0.9993\n",
      "--     233-   254        14                                                                        0.0000 0.9993\n",
      "--     255-   277        10                                                                        0.0000 0.9994\n",
      "--     278-   301         7                                                                        0.0000 0.9994\n",
      "--     302-   326         1                                                                        0.0000 0.9994\n",
      "--     327-   352         7                                                                        0.0000 0.9994\n",
      "--     353-   379         0                                                                        0.0000 0.0000\n",
      "--     380-   407         1                                                                        0.0000 0.9995\n",
      "--     408-   436         2                                                                        0.0000 0.9995\n",
      "--     437-   466         0                                                                        0.0000 0.0000\n",
      "--     467-   497         0                                                                        0.0000 0.0000\n",
      "--     498-   529         2                                                                        0.0000 0.9995\n",
      "--     530-   562         1                                                                        0.0000 0.9995\n",
      "--     563-   596         0                                                                        0.0000 0.0000\n",
      "--     597-   631         0                                                                        0.0000 0.0000\n",
      "--     632-   667         0                                                                        0.0000 0.0000\n",
      "--     668-   704         0                                                                        0.0000 0.0000\n",
      "--     705-   742         1                                                                        0.0000 0.9995\n",
      "--     743-   781         2                                                                        0.0000 0.9996\n",
      "--     782-   821         0                                                                        0.0000 0.0000\n",
      "--\n",
      "--       16494 (max occurrences)\n",
      "--    10660859 (total mers, non-unique)\n",
      "--     4076738 (distinct mers, non-unique)\n",
      "--    43183063 (unique mers)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- For mhap overlapping, set repeat k-mer threshold to 538.\n",
      "--\n",
      "-- Found 53843922 16-mers; 47259801 distinct and 43183063 unique.  Largest count 16494.\n",
      "-- Finished stage 'cor-meryl', reset canuIteration.\n",
      "--\n",
      "-- OVERLAPPER (mhap) (correction)\n",
      "--\n",
      "-- Set corMhapSensitivity=high based on read coverage of 11.\n",
      "--\n",
      "-- PARAMETERS: hashes=768, minMatches=2, threshold=0.73\n",
      "--\n",
      "-- Given 6 GB, can fit 9000 reads per block.\n",
      "-- For 3 blocks, set stride to 2 blocks.\n",
      "-- Logging partitioning to 'correction/1-overlapper/partitioning.log'.\n",
      "-- Configured 2 mhap precompute jobs.\n",
      "-- Configured 2 mhap overlap jobs.\n",
      "-- Finished stage 'cor-mhapConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 11:33:47 2021 with 317.603 GB free disk space (2 processes; 1 concurrently)\n",
      "\n",
      "    cd correction/1-overlapper\n",
      "    ./precompute.sh 1 > ./precompute.000001.out 2>&1\n",
      "    ./precompute.sh 2 > ./precompute.000002.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:38:22 2021 (275 seconds) with 317.334 GB free disk space\n",
      "----------------------------------------\n",
      "-- All 2 mhap precompute jobs finished successfully.\n",
      "-- Finished stage 'cor-mhapPrecomputeCheck', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 11:38:22 2021 with 317.334 GB free disk space (2 processes; 1 concurrently)\n",
      "\n",
      "    cd correction/1-overlapper\n",
      "    ./mhap.sh 1 > ./mhap.000001.out 2>&1\n",
      "    ./mhap.sh 2 > ./mhap.000002.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:38:38 2021 (16 seconds) with 317.332 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 2 mhap overlap output files.\n",
      "-- Finished stage 'cor-mhapCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:38:38 2021 with 317.332 GB free disk space\n",
      "\n",
      "    cd correction\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/ovStoreBuild \\\n",
      "     -O ./pacbio10.ovlStore.BUILDING \\\n",
      "     -G ./pacbio10.gkpStore \\\n",
      "     -M 2-8 \\\n",
      "     -L ./1-overlapper/ovljob.files \\\n",
      "     > ./pacbio10.ovlStore.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:38:38 2021 (lickety-split) with 317.327 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Overlap store 'correction/pacbio10.ovlStore' successfully constructed.\n",
      "--\n",
      "-- Purged 0.27 GB in 6 overlap output files and 2 directories.\n",
      "-- Overlap store 'correction/pacbio10.ovlStore' statistics not available (skipped in correction and trimming stages).\n",
      "-- Finished stage 'cor-createOverlapStore', reset canuIteration.\n",
      "-- Set corMinCoverage=0 based on read coverage of 11.\n",
      "-- Computing global filter scores 'correction/2-correction/pacbio10.globalScores'.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:38:38 2021 with 317.6 GB free disk space\n",
      "\n",
      "    cd correction/2-correction\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/filterCorrectionOverlaps \\\n",
      "      -G ../pacbio10.gkpStore \\\n",
      "      -O ../pacbio10.ovlStore \\\n",
      "      -S ./pacbio10.globalScores.WORKING \\\n",
      "      -c 40 \\\n",
      "      -l 0 \\\n",
      "    > ./pacbio10.globalScores.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:38:38 2021 (lickety-split) with 317.599 GB free disk space\n",
      "----------------------------------------\n",
      "--  PARAMETERS:\n",
      "--  ----------\n",
      "--  \n",
      "--       40 (expected coverage)\n",
      "--        0 (don't use overlaps shorter than this)\n",
      "--    0.000 (don't use overlaps with erate less than this)\n",
      "--    1.000 (don't use overlaps with erate more than this)\n",
      "--  \n",
      "--  OVERLAPS:\n",
      "--  --------\n",
      "--  \n",
      "--  IGNORED:\n",
      "--  \n",
      "--             0 (< 0.0000 fraction error)\n",
      "--             0 (> 0.4095 fraction error)\n",
      "--             0 (< 0 bases long)\n",
      "--             0 (> 2097151 bases long)\n",
      "--  \n",
      "--  FILTERED:\n",
      "--  \n",
      "--         19968 (too many overlaps, discard these shortest ones)\n",
      "--  \n",
      "--  EVIDENCE:\n",
      "--  \n",
      "--        149442 (longest overlaps)\n",
      "--  \n",
      "--  TOTAL:\n",
      "--  \n",
      "--        169410 (all overlaps)\n",
      "--  \n",
      "--  READS:\n",
      "--  -----\n",
      "--  \n",
      "--            87 (no overlaps)\n",
      "--          8824 (no overlaps filtered)\n",
      "--           386 (<  50% overlaps filtered)\n",
      "--           560 (<  80% overlaps filtered)\n",
      "--           567 (<  95% overlaps filtered)\n",
      "--           567 (< 100% overlaps filtered)\n",
      "--  \n",
      "-- Computing expected corrected read lengths 'correction/2-correction/pacbio10.estimate.log'.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:38:38 2021 with 317.599 GB free disk space\n",
      "\n",
      "    cd correction/2-correction\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/generateCorrectionLayouts \\\n",
      "      -G ../pacbio10.gkpStore \\\n",
      "      -O ../pacbio10.ovlStore \\\n",
      "      -S ./pacbio10.globalScores \\\n",
      "      -c 0 \\\n",
      "      -C 80 \\\n",
      "      -p ./pacbio10.estimate.WORKING\n",
      "\n",
      "-- Finished on Wed Apr 21 11:38:39 2021 (1 second) with 317.599 GB free disk space\n",
      "----------------------------------------\n",
      "-- Sorting reads by expected corrected length.\n",
      "-- Sorting reads by uncorrected length.\n",
      "-- Loading expected corrected read lengths.\n",
      "-- Picking longest corrected reads.\n",
      "-- Writing longest corrected reads to 'correction/2-correction/pacbio10.readsToCorrect'.\n",
      "-- Summarizing filter.\n",
      "--\n",
      "-- Reads to be corrected:\n",
      "--   9391 reads longer than 0 bp\n",
      "--   53519532 bp\n",
      "-- Expected corrected reads:\n",
      "--   9391 reads\n",
      "--   51084666 bp\n",
      "--   0 bp minimum length\n",
      "--   5440 bp mean length\n",
      "--   14421 bp n50 length\n",
      "-- Using overlaps no worse than 0.3 fraction error for correcting reads (from corErrorRate parameter).\n",
      "-- Finished stage 'cor-buildCorrectionLayouts', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 11:38:39 2021 with 317.599 GB free disk space (2 processes; 4 concurrently)\n",
      "\n",
      "    cd correction/2-correction\n",
      "    ./correctReads.sh 1 > ./correctReads.000001.out 2>&1\n",
      "    ./correctReads.sh 2 > ./correctReads.000002.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:41:53 2021 (194 seconds) with 317.511 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 2 read correction output files.\n",
      "-- Finished stage 'cor-generateCorrectedReads', reset canuIteration.\n",
      "-- Found 2 read correction output files.\n",
      "-- Finished stage 'cor-generateCorrectedReads', reset canuIteration.\n",
      "-- Concatenating correctReads output.\n",
      "-- Analyzing correctReads output.\n",
      "--\n",
      "-- Reads to be corrected:\n",
      "--   9391 reads longer than 0 bp\n",
      "--   53519532 bp\n",
      "-- Expected corrected reads:\n",
      "--   9391 reads\n",
      "--   51084666 bp\n",
      "--   0 bp minimum length\n",
      "--   5440 bp mean length\n",
      "--   14421 bp n50 length\n",
      "-- Actual corrected reads:\n",
      "--   9212 reads\n",
      "--   51240041 bp\n",
      "--   179 reads with 0 corrected blocks\n",
      "--   9212 reads with 1 corrected block\n",
      "--\n",
      "-- Purging correctReads output after merging to final output file.\n",
      "-- Purged 2 .dump.success sentinels.\n",
      "-- Purged 2 .fasta outputs.\n",
      "-- Purged 2 .err outputs.\n",
      "-- Purged 2 .out job log outputs.\n",
      "-- Finished stage 'cor-dumpCorrectedReads', reset canuIteration.\n",
      "--\n",
      "-- Corrected reads saved in 'pacbio10.correctedReads.fasta.gz'.\n",
      "--\n",
      "--\n",
      "-- BEGIN TRIMMING\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:41:54 2021 with 317.545 GB free disk space\n",
      "\n",
      "    cd trimming\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperCreate \\\n",
      "      -minlength 1000 \\\n",
      "      -o ./pacbio10.gkpStore.BUILDING \\\n",
      "      ./pacbio10.gkpStore.gkp \\\n",
      "    > ./pacbio10.gkpStore.BUILDING.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:41:55 2021 (1 second) with 317.53 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- In gatekeeper store 'trimming/pacbio10.gkpStore':\n",
      "--   Found 9212 reads.\n",
      "--   Found 51240041 bases (10.9 times coverage).\n",
      "--\n",
      "--   Read length histogram (one '*' equals 18.95 reads):\n",
      "--        0    999      0 \n",
      "--     1000   1999   1327 **********************************************************************\n",
      "--     2000   2999   1246 *****************************************************************\n",
      "--     3000   3999   1127 ***********************************************************\n",
      "--     4000   4999    994 ****************************************************\n",
      "--     5000   5999    950 **************************************************\n",
      "--     6000   6999    824 *******************************************\n",
      "--     7000   7999    667 ***********************************\n",
      "--     8000   8999    506 **************************\n",
      "--     9000   9999    461 ************************\n",
      "--    10000  10999    342 ******************\n",
      "--    11000  11999    263 *************\n",
      "--    12000  12999    186 *********\n",
      "--    13000  13999    138 *******\n",
      "--    14000  14999     93 ****\n",
      "--    15000  15999     51 **\n",
      "--    16000  16999     23 *\n",
      "--    17000  17999      8 \n",
      "--    18000  18999      3 \n",
      "--    19000  19999      2 \n",
      "--    20000  20999      1 \n",
      "-- Finished stage 'obt-gatekeeper', reset canuIteration.\n",
      "-- Finished stage 'merylConfigure', reset canuIteration.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 11:41:55 2021 with 317.53 GB free disk space (1 processes; 2 concurrently)\n",
      "\n",
      "    cd trimming/0-mercounts\n",
      "    ./meryl.sh 1 > ./meryl.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:42:09 2021 (14 seconds) with 317.403 GB free disk space\n",
      "----------------------------------------\n",
      "-- Meryl finished successfully.\n",
      "-- Finished stage 'merylCheck', reset canuIteration.\n",
      "--\n",
      "--  22-mers                                                                                           Fraction\n",
      "--    Occurrences   NumMers                                                                         Unique Total\n",
      "--       1-     1  18138400 *******************************************************************--> 0.7758 0.3553\n",
      "--       2-     2   1088894 *********************************************************              0.8224 0.3980\n",
      "--       3-     4   1077728 ********************************************************               0.8479 0.4331\n",
      "--       5-     7   1324315 ********************************************************************** 0.8878 0.5151\n",
      "--       8-    11   1238025 *****************************************************************      0.9417 0.6867\n",
      "--      12-    16    448321 ***********************                                                0.9853 0.8908\n",
      "--      17-    22     46862 **                                                                     0.9981 0.9745\n",
      "--      23-    29      5298                                                                        0.9993 0.9848\n",
      "--      30-    37      3184                                                                        0.9995 0.9872\n",
      "--      38-    46      2174                                                                        0.9996 0.9890\n",
      "--      47-    56      4323                                                                        0.9997 0.9913\n",
      "--      57-    67      1392                                                                        0.9999 0.9953\n",
      "--      68-    79       469                                                                        0.9999 0.9967\n",
      "--      80-    92       202                                                                        0.9999 0.9974\n",
      "--      93-   106       775                                                                        1.0000 0.9978\n",
      "--     107-   121       186                                                                        1.0000 0.9993\n",
      "--     122-   137        13                                                                        1.0000 0.9996\n",
      "--     138-   154         2                                                                        1.0000 0.9997\n",
      "--     155-   172        10                                                                        1.0000 0.9997\n",
      "--     173-   191         6                                                                        1.0000 0.9997\n",
      "--     192-   211         1                                                                        1.0000 0.9997\n",
      "--     212-   232         6                                                                        1.0000 0.9997\n",
      "--     233-   254         4                                                                        1.0000 0.9998\n",
      "--     255-   277         1                                                                        1.0000 0.9998\n",
      "--     278-   301         3                                                                        1.0000 0.9998\n",
      "--     302-   326         6                                                                        1.0000 0.9998\n",
      "--     327-   352         6                                                                        1.0000 0.9998\n",
      "--     353-   379         2                                                                        1.0000 0.9999\n",
      "--     380-   407         1                                                                        1.0000 0.9999\n",
      "--     408-   436         3                                                                        1.0000 0.9999\n",
      "--     437-   466         1                                                                        1.0000 0.9999\n",
      "--     467-   497         0                                                                        0.0000 0.0000\n",
      "--     498-   529         0                                                                        0.0000 0.0000\n",
      "--     530-   562         0                                                                        0.0000 0.0000\n",
      "--     563-   596         0                                                                        0.0000 0.0000\n",
      "--     597-   631         0                                                                        0.0000 0.0000\n",
      "--     632-   667         0                                                                        0.0000 0.0000\n",
      "--     668-   704         0                                                                        0.0000 0.0000\n",
      "--     705-   742         0                                                                        0.0000 0.0000\n",
      "--     743-   781         0                                                                        0.0000 0.0000\n",
      "--     782-   821         0                                                                        0.0000 0.0000\n",
      "--\n",
      "--        3877 (max occurrences)\n",
      "--    32908189 (total mers, non-unique)\n",
      "--     5242214 (distinct mers, non-unique)\n",
      "--    18138400 (unique mers)\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:42:09 2021 with 317.403 GB free disk space\n",
      "\n",
      "    cd trimming/0-mercounts\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/meryl \\\n",
      "      -Dt \\\n",
      "      -n 451 \\\n",
      "      -s ./pacbio10.ms22 \\\n",
      "    > ./pacbio10.ms22.frequentMers.fasta \\\n",
      "    2> ./pacbio10.ms22.frequentMers.fasta.err\n",
      "\n",
      "-- Finished on Wed Apr 21 11:42:10 2021 (1 second) with 317.403 GB free disk space\n",
      "----------------------------------------\n",
      "-- Reset obtOvlMerThreshold from auto to 451.\n",
      "--\n",
      "-- Found 51046589 22-mers; 23380614 distinct and 18138400 unique.  Largest count 3877.\n",
      "-- Finished stage 'obt-meryl', reset canuIteration.\n",
      "--\n",
      "-- OVERLAPPER (normal) (trimming) erate=0.045\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:42:10 2021 with 317.53 GB free disk space\n",
      "\n",
      "    cd trimming/1-overlapper\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/overlapInCorePartition \\\n",
      "     -g  ../pacbio10.gkpStore \\\n",
      "     -bl 100000000 \\\n",
      "     -bs 0 \\\n",
      "     -rs 2000000 \\\n",
      "     -rl 0 \\\n",
      "     -ol 500 \\\n",
      "     -o  ./pacbio10.partition \\\n",
      "    > ./pacbio10.partition.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:42:10 2021 (lickety-split) with 317.53 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Configured 1 overlapInCore jobs.\n",
      "-- Finished stage 'obt-overlapConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 11:42:10 2021 with 317.53 GB free disk space (1 processes; 1 concurrently)\n",
      "\n",
      "    cd trimming/1-overlapper\n",
      "    ./overlap.sh 1 > ./overlap.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:43:48 2021 (98 seconds) with 317.532 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 1 overlapInCore output files.\n",
      "--\n",
      "-- overlapInCore compute 'trimming/1-overlapper':\n",
      "--   kmer hits\n",
      "--     with no overlap           352996      352996 +- 0\n",
      "--     with an overlap            77620       77620 +- 0\n",
      "--\n",
      "--   overlaps                     77620       77620 +- 0\n",
      "--     contained                      0           0 +- 0\n",
      "--     dovetail                       0           0 +- 0\n",
      "--\n",
      "--   overlaps rejected\n",
      "--     multiple per pair              0           0 +- 0\n",
      "--     bad short window               0           0 +- 0\n",
      "--     bad long window                0           0 +- 0\n",
      "-- Finished stage 'obt-overlapCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:43:48 2021 with 317.532 GB free disk space\n",
      "\n",
      "    cd trimming\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/ovStoreBuild \\\n",
      "     -O ./pacbio10.ovlStore.BUILDING \\\n",
      "     -G ./pacbio10.gkpStore \\\n",
      "     -M 2-8 \\\n",
      "     -L ./1-overlapper/ovljob.files \\\n",
      "     > ./pacbio10.ovlStore.err 2>&1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "-- Finished on Wed Apr 21 11:43:48 2021 (lickety-split) with 317.529 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Overlap store 'trimming/pacbio10.ovlStore' successfully constructed.\n",
      "--\n",
      "-- Purged 0.001 GB in 3 overlap output files and 1 directories.\n",
      "-- Overlap store 'trimming/pacbio10.ovlStore' statistics not available (skipped in correction and trimming stages).\n",
      "-- Finished stage 'obt-createOverlapStore', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:43:48 2021 with 317.531 GB free disk space\n",
      "\n",
      "    cd trimming/3-overlapbasedtrimming\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/trimReads \\\n",
      "      -G  ../pacbio10.gkpStore \\\n",
      "      -O  ../pacbio10.ovlStore \\\n",
      "      -Co ./pacbio10.1.trimReads.clear \\\n",
      "      -e  0.045 \\\n",
      "      -minlength 1000 \\\n",
      "      -ol 1 \\\n",
      "      -oc 1 \\\n",
      "      -o  ./pacbio10.1.trimReads \\\n",
      "    >     ./pacbio10.1.trimReads.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:43:49 2021 (1 second) with 317.53 GB free disk space\n",
      "----------------------------------------\n",
      "--  PARAMETERS:\n",
      "--  ----------\n",
      "--     1000    (reads trimmed below this many bases are deleted)\n",
      "--   0.0450    (use overlaps at or below this fraction error)\n",
      "--        1    (break region if overlap is less than this long, for 'largest covered' algorithm)\n",
      "--        1    (break region if overlap coverage is less than this many read, for 'largest covered' algorithm)\n",
      "--  \n",
      "--  INPUT READS:\n",
      "--  -----------\n",
      "--    9212 reads     51240041 bases (reads processed)\n",
      "--       0 reads            0 bases (reads not processed, previously deleted)\n",
      "--       0 reads            0 bases (reads not processed, in a library where trimming isn't allowed)\n",
      "--  \n",
      "--  OUTPUT READS:\n",
      "--  ------------\n",
      "--    7692 reads     34589500 bases (trimmed reads output)\n",
      "--      55 reads       231015 bases (reads with no change, kept as is)\n",
      "--    1019 reads      4834874 bases (reads with no overlaps, deleted)\n",
      "--     446 reads      1601652 bases (reads with short trimmed length, deleted)\n",
      "--  \n",
      "--  TRIMMING DETAILS:\n",
      "--  ----------------\n",
      "--    6928 reads      6251970 bases (bases trimmed from the 5' end of a read)\n",
      "--    7345 reads      3731030 bases (bases trimmed from the 3' end of a read)\n",
      "-- Finished stage 'obt-trimReads', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:43:49 2021 with 317.53 GB free disk space\n",
      "\n",
      "    cd trimming/3-overlapbasedtrimming\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/splitReads \\\n",
      "      -G  ../pacbio10.gkpStore \\\n",
      "      -O  ../pacbio10.ovlStore \\\n",
      "      -Ci ./pacbio10.1.trimReads.clear \\\n",
      "      -Co ./pacbio10.2.splitReads.clear \\\n",
      "      -e  0.045 \\\n",
      "      -minlength 1000 \\\n",
      "      -o  ./pacbio10.2.splitReads \\\n",
      "    >     ./pacbio10.2.splitReads.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:43:49 2021 (lickety-split) with 317.53 GB free disk space\n",
      "----------------------------------------\n",
      "--  PARAMETERS:\n",
      "--  ----------\n",
      "--     1000    (reads trimmed below this many bases are deleted)\n",
      "--   0.0450    (use overlaps at or below this fraction error)\n",
      "--  INPUT READS:\n",
      "--  -----------\n",
      "--    7747 reads     44803515 bases (reads processed)\n",
      "--    1465 reads      6436526 bases (reads not processed, previously deleted)\n",
      "--       0 reads            0 bases (reads not processed, in a library where trimming isn't allowed)\n",
      "--  \n",
      "--  PROCESSED:\n",
      "--  --------\n",
      "--       0 reads            0 bases (no overlaps)\n",
      "--      19 reads        66774 bases (no coverage after adjusting for trimming done already)\n",
      "--       0 reads            0 bases (processed for chimera)\n",
      "--       0 reads            0 bases (processed for spur)\n",
      "--    7728 reads     44736741 bases (processed for subreads)\n",
      "--  \n",
      "--  READS WITH SIGNALS:\n",
      "--  ------------------\n",
      "--       0 reads            0 signals (number of 5' spur signal)\n",
      "--       0 reads            0 signals (number of 3' spur signal)\n",
      "--       0 reads            0 signals (number of chimera signal)\n",
      "--       0 reads            0 signals (number of subread signal)\n",
      "--  \n",
      "--  SIGNALS:\n",
      "--  -------\n",
      "--       0 reads            0 bases (size of 5' spur signal)\n",
      "--       0 reads            0 bases (size of 3' spur signal)\n",
      "--       0 reads            0 bases (size of chimera signal)\n",
      "--       0 reads            0 bases (size of subread signal)\n",
      "--  \n",
      "--  TRIMMING:\n",
      "--  --------\n",
      "--       0 reads            0 bases (trimmed from the 5' end of the read)\n",
      "--       0 reads            0 bases (trimmed from the 3' end of the read)\n",
      "-- Finished stage 'obt-splitReads', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:43:49 2021 with 317.53 GB free disk space\n",
      "\n",
      "    cd trimming\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperDumpFASTQ -fasta -nolibname \\\n",
      "      -G ./pacbio10.gkpStore \\\n",
      "      -c ./3-overlapbasedtrimming/pacbio10.2.splitReads.clear \\\n",
      "      -o ../pacbio10.trimmedReads.gz \\\n",
      "    >    ../pacbio10.trimmedReads.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:43:50 2021 (1 second) with 317.518 GB free disk space\n",
      "----------------------------------------\n",
      "-- Finished stage 'obt-dumpReads', reset canuIteration.\n",
      "--\n",
      "-- Trimmed reads saved in 'trimming/pacbio10.trimmedReads.fasta.gz'\n",
      "--\n",
      "--\n",
      "-- BEGIN ASSEMBLY\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:43:50 2021 with 317.518 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperCreate \\\n",
      "      -minlength 1000 \\\n",
      "      -o ./pacbio10.gkpStore.BUILDING \\\n",
      "      ./pacbio10.gkpStore.gkp \\\n",
      "    > ./pacbio10.gkpStore.BUILDING.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:43:50 2021 (lickety-split) with 317.508 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- In gatekeeper store 'unitigging/pacbio10.gkpStore':\n",
      "--   Found 7747 reads.\n",
      "--   Found 34820515 bases (7.4 times coverage).\n",
      "--\n",
      "--   Read length histogram (one '*' equals 22.74 reads):\n",
      "--        0    999      0 \n",
      "--     1000   1999   1592 **********************************************************************\n",
      "--     2000   2999   1380 ************************************************************\n",
      "--     3000   3999   1121 *************************************************\n",
      "--     4000   4999    943 *****************************************\n",
      "--     5000   5999    742 ********************************\n",
      "--     6000   6999    563 ************************\n",
      "--     7000   7999    430 ******************\n",
      "--     8000   8999    292 ************\n",
      "--     9000   9999    246 **********\n",
      "--    10000  10999    149 ******\n",
      "--    11000  11999    119 *****\n",
      "--    12000  12999     69 ***\n",
      "--    13000  13999     52 **\n",
      "--    14000  14999     28 *\n",
      "--    15000  15999     12 \n",
      "--    16000  16999      6 \n",
      "--    17000  17999      3 \n",
      "-- Finished stage 'utg-gatekeeper', reset canuIteration.\n",
      "-- Finished stage 'merylConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 11:43:51 2021 with 317.507 GB free disk space (1 processes; 2 concurrently)\n",
      "\n",
      "    cd unitigging/0-mercounts\n",
      "    ./meryl.sh 1 > ./meryl.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:44:00 2021 (9 seconds) with 317.454 GB free disk space\n",
      "----------------------------------------\n",
      "-- Meryl finished successfully.\n",
      "-- Finished stage 'merylCheck', reset canuIteration.\n",
      "--\n",
      "--  22-mers                                                                                           Fraction\n",
      "--    Occurrences   NumMers                                                                         Unique Total\n",
      "--       1-     1   4857166 *******************************************************************--> 0.5068 0.1401\n",
      "--       2-     2    907535 **************************************************                     0.6015 0.1925\n",
      "--       3-     4    981343 ******************************************************                 0.6575 0.2389\n",
      "--       5-     7   1259810 ********************************************************************** 0.7486 0.3520\n",
      "--       8-    11   1136914 ***************************************************************        0.8728 0.5906\n",
      "--      12-    16    388562 *********************                                                  0.9695 0.8633\n",
      "--      17-    22     35399 *                                                                      0.9962 0.9680\n",
      "--      23-    29      5160                                                                        0.9984 0.9797\n",
      "--      30-    37      2223                                                                        0.9989 0.9833\n",
      "--      38-    46      2789                                                                        0.9990 0.9849\n",
      "--      47-    56      4034                                                                        0.9994 0.9892\n",
      "--      57-    67       854                                                                        0.9998 0.9945\n",
      "--      68-    79       456                                                                        0.9998 0.9958\n",
      "--      80-    92       339                                                                        0.9999 0.9967\n",
      "--      93-   106       695                                                                        0.9999 0.9977\n",
      "--     107-   121        63                                                                        1.0000 0.9995\n",
      "--     122-   137         4                                                                        1.0000 0.9997\n",
      "--     138-   154         9                                                                        1.0000 0.9997\n",
      "--     155-   172         4                                                                        1.0000 0.9997\n",
      "--     173-   191         1                                                                        1.0000 0.9997\n",
      "--     192-   211         6                                                                        1.0000 0.9998\n",
      "--     212-   232         3                                                                        1.0000 0.9998\n",
      "--     233-   254         1                                                                        1.0000 0.9998\n",
      "--     255-   277         5                                                                        1.0000 0.9998\n",
      "--     278-   301         4                                                                        1.0000 0.9999\n",
      "--     302-   326         6                                                                        1.0000 0.9999\n",
      "--     327-   352         2                                                                        1.0000 0.9999\n",
      "--     353-   379         1                                                                        1.0000 1.0000\n",
      "--     380-   407         2                                                                        1.0000 1.0000\n",
      "--     408-   436         1                                                                        1.0000 1.0000\n",
      "--     437-   466         0                                                                        0.0000 0.0000\n",
      "--     467-   497         0                                                                        0.0000 0.0000\n",
      "--     498-   529         0                                                                        0.0000 0.0000\n",
      "--     530-   562         0                                                                        0.0000 0.0000\n",
      "--     563-   596         0                                                                        0.0000 0.0000\n",
      "--     597-   631         0                                                                        0.0000 0.0000\n",
      "--     632-   667         0                                                                        0.0000 0.0000\n",
      "--     668-   704         0                                                                        0.0000 0.0000\n",
      "--     705-   742         0                                                                        0.0000 0.0000\n",
      "--     743-   781         0                                                                        0.0000 0.0000\n",
      "--     782-   821         0                                                                        0.0000 0.0000\n",
      "--\n",
      "--         408 (max occurrences)\n",
      "--    29800662 (total mers, non-unique)\n",
      "--     4726225 (distinct mers, non-unique)\n",
      "--     4857166 (unique mers)\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:44:00 2021 with 317.454 GB free disk space\n",
      "\n",
      "    cd unitigging/0-mercounts\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/meryl \\\n",
      "      -Dt \\\n",
      "      -n 340 \\\n",
      "      -s ./pacbio10.ms22 \\\n",
      "    > ./pacbio10.ms22.frequentMers.fasta \\\n",
      "    2> ./pacbio10.ms22.frequentMers.fasta.err\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "-- Finished on Wed Apr 21 11:44:00 2021 (lickety-split) with 317.454 GB free disk space\n",
      "----------------------------------------\n",
      "-- Reset utgOvlMerThreshold from auto to 340.\n",
      "--\n",
      "-- Found 34657828 22-mers; 9583391 distinct and 4857166 unique.  Largest count 408.\n",
      "-- Finished stage 'utg-meryl', reset canuIteration.\n",
      "--\n",
      "-- OVERLAPPER (normal) (assembly) erate=0.045\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:44:00 2021 with 317.507 GB free disk space\n",
      "\n",
      "    cd unitigging/1-overlapper\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/overlapInCorePartition \\\n",
      "     -g  ../pacbio10.gkpStore \\\n",
      "     -bl 100000000 \\\n",
      "     -bs 0 \\\n",
      "     -rs 2000000 \\\n",
      "     -rl 0 \\\n",
      "     -ol 500 \\\n",
      "     -o  ./pacbio10.partition \\\n",
      "    > ./pacbio10.partition.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:44:00 2021 (lickety-split) with 317.507 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Configured 1 overlapInCore jobs.\n",
      "-- Finished stage 'utg-overlapConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 11:44:00 2021 with 317.507 GB free disk space (1 processes; 1 concurrently)\n",
      "\n",
      "    cd unitigging/1-overlapper\n",
      "    ./overlap.sh 1 > ./overlap.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:45:02 2021 (62 seconds) with 317.506 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 1 overlapInCore output files.\n",
      "--\n",
      "-- overlapInCore compute 'unitigging/1-overlapper':\n",
      "--   kmer hits\n",
      "--     with no overlap           224945      224945 +- 0\n",
      "--     with an overlap            55389       55389 +- 0\n",
      "--\n",
      "--   overlaps                     55389       55389 +- 0\n",
      "--     contained                  22217       22217 +- 0\n",
      "--     dovetail                   33172       33172 +- 0\n",
      "--\n",
      "--   overlaps rejected\n",
      "--     multiple per pair              0           0 +- 0\n",
      "--     bad short window               0           0 +- 0\n",
      "--     bad long window                0           0 +- 0\n",
      "-- Finished stage 'utg-overlapCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:45:03 2021 with 317.506 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/ovStoreBuild \\\n",
      "     -O ./pacbio10.ovlStore.BUILDING \\\n",
      "     -G ./pacbio10.gkpStore \\\n",
      "     -M 2-8 \\\n",
      "     -L ./1-overlapper/ovljob.files \\\n",
      "     > ./pacbio10.ovlStore.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:45:03 2021 (1 second) with 317.504 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Overlap store 'unitigging/pacbio10.ovlStore' successfully constructed.\n",
      "--\n",
      "-- Purged 0.001 GB in 3 overlap output files and 1 directories.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:45:03 2021 with 317.505 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/ovStoreStats \\\n",
      "     -G ./pacbio10.gkpStore \\\n",
      "     -O ./pacbio10.ovlStore \\\n",
      "     -o ./pacbio10.ovlStore \\\n",
      "     > ./pacbio10.ovlStore.summary.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:45:03 2021 (lickety-split) with 317.504 GB free disk space\n",
      "----------------------------------------\n",
      "-- category            reads     %          read length        feature size or coverage  analysis\n",
      "-- ----------------  -------  -------  ----------------------  ------------------------  --------------------\n",
      "-- middle-missing          6    0.08     5546.33 +- 2500.41        275.33 +- 206.93     (bad trimming)\n",
      "-- middle-hump            17    0.22     3132.29 +- 2152.12        176.41 +- 274.13     (bad trimming)\n",
      "-- no-5-prime            168    2.17     5568.47 +- 3148.98         90.57 +- 218.80     (bad trimming)\n",
      "-- no-3-prime            137    1.77     5247.82 +- 3028.19         82.38 +- 227.19     (bad trimming)\n",
      "-- \n",
      "-- low-coverage         2242   28.94     3083.23 +- 1967.76          4.96 +- 1.73       (easy to assemble, potential for lower quality consensus)\n",
      "-- unique               1109   14.32     3320.55 +- 1994.73         13.05 +- 2.93       (easy to assemble, perfect, yay)\n",
      "-- repeat-cont             1    0.01     1168.00 +- 0.00            94.87 +- 2.87       (potential for consensus errors, no impact on assembly)\n",
      "-- repeat-dove             0    0.00        0.00 +- 0.00             0.00 +- 0.00       (hard to assemble, likely won't assemble correctly or even at all)\n",
      "-- \n",
      "-- span-repeat          1764   22.77     6031.15 +- 3088.52       2608.24 +- 2522.99    (read spans a large repeat, usually easy to assemble)\n",
      "-- uniq-repeat-cont     1853   23.92     4279.03 +- 2150.88                             (should be uniquely placed, low potential for consensus errors, no impact on assembly)\n",
      "-- uniq-repeat-dove      426    5.50     9116.88 +- 2699.11                             (will end contigs, potential to misassemble)\n",
      "-- uniq-anchor             0    0.00        0.00 +- 0.00             0.00 +- 0.00       (repeat read, with unique section, probable bad read)\n",
      "--\n",
      "-- Overlap store 'unitigging/pacbio10.ovlStore' contains:\n",
      "--\n",
      "--   category            reads     %          read length        feature size or coverage  analysis\n",
      "--   ----------------  -------  -------  ----------------------  ------------------------  --------------------\n",
      "--   middle-missing          6    0.08     5546.33 +- 2500.41        275.33 +- 206.93     (bad trimming)\n",
      "--   middle-hump            17    0.22     3132.29 +- 2152.12        176.41 +- 274.13     (bad trimming)\n",
      "--   no-5-prime            168    2.17     5568.47 +- 3148.98         90.57 +- 218.80     (bad trimming)\n",
      "--   no-3-prime            137    1.77     5247.82 +- 3028.19         82.38 +- 227.19     (bad trimming)\n",
      "--   \n",
      "--   low-coverage         2242   28.94     3083.23 +- 1967.76          4.96 +- 1.73       (easy to assemble, potential for lower quality consensus)\n",
      "--   unique               1109   14.32     3320.55 +- 1994.73         13.05 +- 2.93       (easy to assemble, perfect, yay)\n",
      "--   repeat-cont             1    0.01     1168.00 +- 0.00            94.87 +- 2.87       (potential for consensus errors, no impact on assembly)\n",
      "--   repeat-dove             0    0.00        0.00 +- 0.00             0.00 +- 0.00       (hard to assemble, likely won't assemble correctly or even at all)\n",
      "--   \n",
      "--   span-repeat          1764   22.77     6031.15 +- 3088.52       2608.24 +- 2522.99    (read spans a large repeat, usually easy to assemble)\n",
      "--   uniq-repeat-cont     1853   23.92     4279.03 +- 2150.88                             (should be uniquely placed, low potential for consensus errors, no impact on assembly)\n",
      "--   uniq-repeat-dove      426    5.50     9116.88 +- 2699.11                             (will end contigs, potential to misassemble)\n",
      "--   uniq-anchor             0    0.00        0.00 +- 0.00             0.00 +- 0.00       (repeat read, with unique section, probable bad read)\n",
      "-- Finished stage 'utg-createOverlapStore', reset canuIteration.\n",
      "\n",
      "Configure RED for 2gb memory with batches of at most 5000 reads and (unlimited) bases.\n",
      "\n",
      "RED job   1 from read         1 to read      5000 -   0.566 GB for    5000 reads -   0.272 GB for  22486299 olaps -   0.001 GB for evidence\n",
      "RED job   2 from read      5001 to read      7747 -   0.311 GB for    2747 reads -   0.149 GB for  12334216 olaps -   0.000 GB for evidence\n",
      "-- Finished stage 'readErrorDetectionConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 11:45:03 2021 with 317.504 GB free disk space (2 processes; 2 concurrently)\n",
      "\n",
      "    cd unitigging/3-overlapErrorAdjustment\n",
      "    ./red.sh 1 > ./red.000001.out 2>&1\n",
      "    ./red.sh 2 > ./red.000002.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:45:06 2021 (3 seconds) with 317.504 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 2 read error detection output files.\n",
      "-- Finished stage 'readErrorDetectionCheck', reset canuIteration.\n",
      "\n",
      "Configure OEA for 1gb memory with batches of at most 25000 reads and (unlimited) bases.\n",
      "Configured 1 jobs.\n",
      "\n",
      "OEA job   1 from read         1 to read      7747 - 33.2 bases +  2.4 adjusts +  0.2 reads +  3.4 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 367.3 MB\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- Finished stage 'overlapErrorAdjustmentConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 11:45:06 2021 with 317.504 GB free disk space (1 processes; 8 concurrently)\n",
      "\n",
      "    cd unitigging/3-overlapErrorAdjustment\n",
      "    ./oea.sh 1 > ./oea.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:45:09 2021 (3 seconds) with 317.503 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 1 overlap error adjustment output files.\n",
      "-- Finished stage 'overlapErrorAdjustmentCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:45:09 2021 with 317.503 GB free disk space\n",
      "\n",
      "    cd unitigging/3-overlapErrorAdjustment\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/ovStoreBuild \\\n",
      "      -G ../pacbio10.gkpStore \\\n",
      "      -O ../pacbio10.ovlStore \\\n",
      "      -evalues \\\n",
      "      -L ./oea.files \\\n",
      "    > ./oea.apply.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:45:09 2021 (lickety-split) with 317.503 GB free disk space\n",
      "----------------------------------------\n",
      "-- No report available.\n",
      "-- Finished stage 'updateOverlapStore', reset canuIteration.\n",
      "-- Finished stage 'unitig', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 11:45:09 2021 with 317.503 GB free disk space (1 processes; 2 concurrently)\n",
      "\n",
      "    cd unitigging/4-unitigger\n",
      "    ./unitigger.sh 1 > ./unitigger.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:45:10 2021 (1 second) with 317.499 GB free disk space\n",
      "----------------------------------------\n",
      "-- Unitigger finished successfully.\n",
      "-- Found, in version 1, after unitig construction:\n",
      "--   contigs:      94 sequences, total length 3545599 bp (including 0 repeats of total length 0 bp).\n",
      "--   bubbles:      0 sequences, total length 0 bp.\n",
      "--   unassembled:  1988 sequences, total length 7368940 bp.\n",
      "--\n",
      "-- Contig sizes based on genome size --\n",
      "--            NG (bp)  LG (contigs)    sum (bp)\n",
      "--         ----------  ------------  ----------\n",
      "--     10       88405             4      475878\n",
      "--     20       73434            10      949171\n",
      "--     30       56292            18     1443666\n",
      "--     40       48634            27     1905901\n",
      "--     50       41887            37     2351578\n",
      "--     60       26665            52     2843751\n",
      "--     70       16885            74     3304503\n",
      "--\n",
      "-- Finished stage 'unitigCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:45:10 2021 with 317.499 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperPartition \\\n",
      "      -G ./pacbio10.gkpStore \\\n",
      "      -T ./pacbio10.ctgStore 1 \\\n",
      "      -b 15000 \\\n",
      "      -p 8 \\\n",
      "    > ./pacbio10.ctgStore/partitionedReads.log 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:45:10 2021 (lickety-split) with 317.49 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:45:10 2021 with 317.49 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperPartition \\\n",
      "      -G ./pacbio10.gkpStore \\\n",
      "      -T ./pacbio10.utgStore 1 \\\n",
      "      -b 15000 \\\n",
      "      -p 8 \\\n",
      "    > ./pacbio10.utgStore/partitionedReads.log 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:45:10 2021 (lickety-split) with 317.483 GB free disk space\n",
      "----------------------------------------\n",
      "-- Using fast alignment for consensus (iteration '0').\n",
      "-- Configured 1 contig and 1 unitig consensus jobs.\n",
      "-- Finished stage 'consensusConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 11:45:10 2021 with 317.483 GB free disk space (2 processes; 2 concurrently)\n",
      "\n",
      "    cd unitigging/5-consensus\n",
      "    ./consensus.sh 1 > ./consensus.000001.out 2>&1\n",
      "    ./consensus.sh 2 > ./consensus.000002.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:46:38 2021 (88 seconds) with 317.449 GB free disk space\n",
      "----------------------------------------\n",
      "-- Using slow alignment for consensus (iteration '1').\n",
      "-- Configured 1 contig and 1 unitig consensus jobs.\n",
      "-- All 2 consensus jobs finished successfully.\n",
      "-- Finished stage 'consensusCheck', reset canuIteration.\n",
      "-- Using fast alignment for consensus (iteration '0').\n",
      "-- Configured 1 contig and 1 unitig consensus jobs.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:46:38 2021 with 317.449 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreLoad \\\n",
      "      -G ./pacbio10.gkpStore \\\n",
      "      -T ./pacbio10.ctgStore 2 \\\n",
      "      -L ./5-consensus/ctgcns.files \\\n",
      "    > ./5-consensus/ctgcns.files.ctgStoreLoad.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:46:39 2021 (1 second) with 317.428 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:46:39 2021 with 317.428 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreLoad \\\n",
      "      -G ./pacbio10.gkpStore \\\n",
      "      -T ./pacbio10.utgStore 2 \\\n",
      "      -L ./5-consensus/utgcns.files \\\n",
      "    > ./5-consensus/utgcns.files.utgStoreLoad.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:46:39 2021 (lickety-split) with 317.421 GB free disk space\n",
      "----------------------------------------\n",
      "-- Purging consensus output after loading to ctgStore and/or utgStore.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:46:39 2021 with 317.421 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./pacbio10.gkpStore \\\n",
      "      -T ./pacbio10.ctgStore 2 \\\n",
      "      -sizes -s 4700000 \\\n",
      "    > ./pacbio10.ctgStore/seqDB.v002.sizes.txt\n",
      "\n",
      "-- Finished on Wed Apr 21 11:46:39 2021 (lickety-split) with 317.421 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found, in version 2, after consensus generation:\n",
      "--   contigs:      94 sequences, total length 3540004 bp (including 0 repeats of total length 0 bp).\n",
      "--   bubbles:      0 sequences, total length 0 bp.\n",
      "--   unassembled:  1988 sequences, total length 7369166 bp.\n",
      "--\n",
      "-- Contig sizes based on genome size --\n",
      "--            NG (bp)  LG (contigs)    sum (bp)\n",
      "--         ----------  ------------  ----------\n",
      "--     10       88305             4      476380\n",
      "--     20       73437            10      950210\n",
      "--     30       55904            18     1443586\n",
      "--     40       48609            27     1904536\n",
      "--     50       41894            37     2350626\n",
      "--     60       25880            52     2841570\n",
      "--     70       16885            74     3300809\n",
      "--\n",
      "-- Finished stage 'consensusLoad', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:46:39 2021 with 317.421 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreCoverageStat \\\n",
      "      -G ./pacbio10.gkpStore \\\n",
      "      -T ./pacbio10.ctgStore 2 \\\n",
      "      -s 4700000 \\\n",
      "      -o ./pacbio10.ctgStore.coverageStat \\\n",
      "    > ./pacbio10.ctgStore.coverageStat.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:46:39 2021 (lickety-split) with 317.421 GB free disk space\n",
      "----------------------------------------\n",
      "-- Finished stage 'consensusAnalyze', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:46:39 2021 with 317.421 GB free disk space\n",
      "\n",
      "    cd unitigging/4-unitigger\n",
      "    ./alignGFA.sh\n",
      "GFA alignments updated.\n",
      "\n",
      "-- Finished on Wed Apr 21 11:46:40 2021 (1 second) with 317.421 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:46:40 2021 with 317.421 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./unitigging/pacbio10.gkpStore \\\n",
      "      -T ./unitigging/pacbio10.ctgStore 2 \\\n",
      "      -o ./pacbio10.contigs \\\n",
      "      -layout \\\n",
      "    > ./pacbio10.contigs.layout.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:46:40 2021 (lickety-split) with 317.4 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:46:40 2021 with 317.4 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./unitigging/pacbio10.gkpStore \\\n",
      "      -T ./unitigging/pacbio10.utgStore 2 \\\n",
      "      -o ./pacbio10.unitigs \\\n",
      "      -layout \\\n",
      "    > ./pacbio10.unitigs.layout.err 2>&1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "-- Finished on Wed Apr 21 11:46:40 2021 (lickety-split) with 317.392 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:46:40 2021 with 317.392 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./unitigging/pacbio10.gkpStore \\\n",
      "      -T ./unitigging/pacbio10.ctgStore 2 \\\n",
      "      -consensus -fasta \\\n",
      "      -unassembled \\\n",
      "    > ./pacbio10.unassembled.fasta\n",
      "    2> ./pacbio10.unassembled.err\n",
      "\n",
      "-- Finished on Wed Apr 21 11:46:40 2021 (lickety-split) with 317.385 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:46:40 2021 with 317.385 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./unitigging/pacbio10.gkpStore \\\n",
      "      -T ./unitigging/pacbio10.ctgStore 2 \\\n",
      "      -consensus -fasta \\\n",
      "      -contigs \\\n",
      "    > ./pacbio10.contigs.fasta\n",
      "    2> ./pacbio10.contigs.err\n",
      "\n",
      "-- Finished on Wed Apr 21 11:46:40 2021 (lickety-split) with 317.382 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:46:40 2021 with 317.382 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./unitigging/pacbio10.gkpStore \\\n",
      "      -T ./unitigging/pacbio10.utgStore 2 \\\n",
      "      -consensus -fasta \\\n",
      "      -contigs \\\n",
      "    > ./pacbio10.unitigs.fasta\n",
      "    2> ./pacbio10.unitigs.err\n",
      "\n",
      "-- Finished on Wed Apr 21 11:46:40 2021 (lickety-split) with 317.379 GB free disk space\n",
      "----------------------------------------\n",
      "-- Finished stage 'generateOutputs', reset canuIteration.\n",
      "--\n",
      "-- Assembly 'pacbio10' finished in '/Users/olga/Documents/BioinformaticsInstitute/NGS/pacbio10'.\n",
      "--\n",
      "-- Summary saved in 'unitigging.html'.\n",
      "--\n",
      "-- Sequences saved:\n",
      "--   Contigs       -> 'pacbio10.contigs.fasta'\n",
      "--   Unassembled   -> 'pacbio10.unassembled.fasta'\n",
      "--   Unitigs       -> 'pacbio10.unitigs.fasta'\n",
      "--\n",
      "-- Read layouts saved:\n",
      "--   Contigs       -> 'pacbio10.contigs.layout'.\n",
      "--   Unitigs       -> 'pacbio10.unitigs.layout'.\n",
      "--\n",
      "-- Graphs saved:\n",
      "--   Contigs       -> 'pacbio10.contigs.gfa'.\n",
      "--   Unitigs       -> 'pacbio10.unitigs.gfa'.\n",
      "--\n",
      "-- Bye.\n",
      "-- Finished stage 'outputSequence', reset canuIteration.\n"
     ]
    }
   ],
   "source": [
    "! /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/bin/canu -p pacbio10 -d pacbio10 genomesize=4.7m -pacbio-raw ./homework6/pacbio_10x.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- Canu release v1.5\n",
      "-- Detected Java(TM) Runtime Environment '1.8.0_152-release' (from '/Users/olga/miniconda3/bin/java').\n",
      "-- Detected gnuplot version '5.4 patchlevel 1' (from 'gnuplot') and image format 'png'.\n",
      "-- Detected 8 CPUs and 16 gigabytes of memory.\n",
      "-- No grid engine detected, grid disabled.\n",
      "--\n",
      "-- Run   2 jobs concurrently using    8 GB and   4 CPUs for stage 'meryl'.\n",
      "-- Run   1 job  concurrently using    6 GB and   8 CPUs for stage 'mhap (cor)'.\n",
      "-- Run   1 job  concurrently using    8 GB and   8 CPUs for stage 'overlapper (obt)'.\n",
      "-- Run   1 job  concurrently using    8 GB and   8 CPUs for stage 'overlapper (utg)'.\n",
      "-- Run   4 jobs concurrently using    8 GB and   2 CPUs for stage 'falcon_sense'.\n",
      "-- Run   8 jobs concurrently using    2 GB and   1 CPU  for stage 'ovStore bucketizer'.\n",
      "-- Run   8 jobs concurrently using    8 GB and   1 CPU  for stage 'ovStore sorting'.\n",
      "-- Run   2 jobs concurrently using    2 GB and   4 CPUs for stage 'read error detection'.\n",
      "-- Run   8 jobs concurrently using    1 GB and   1 CPU  for stage 'overlap error adjustment'.\n",
      "-- Run   2 jobs concurrently using    8 GB and   4 CPUs for stage 'bogart'.\n",
      "-- Run   8 jobs concurrently using    2 GB and   1 CPU  for stage 'GFA alignment and processing'.\n",
      "-- Run   2 jobs concurrently using    8 GB and   4 CPUs for stage 'consensus'.\n",
      "--\n",
      "-- Generating assembly 'pacbio20' in '/Users/olga/Documents/BioinformaticsInstitute/NGS/pacbio20'\n",
      "--\n",
      "-- Parameters:\n",
      "--\n",
      "--  genomeSize        4700000\n",
      "--\n",
      "--  Overlap Generation Limits:\n",
      "--    corOvlErrorRate 0.2400 ( 24.00%)\n",
      "--    obtOvlErrorRate 0.0450 (  4.50%)\n",
      "--    utgOvlErrorRate 0.0450 (  4.50%)\n",
      "--\n",
      "--  Overlap Processing Limits:\n",
      "--    corErrorRate    0.3000 ( 30.00%)\n",
      "--    obtErrorRate    0.0450 (  4.50%)\n",
      "--    utgErrorRate    0.0450 (  4.50%)\n",
      "--    cnsErrorRate    0.0450 (  4.50%)\n",
      "--\n",
      "--\n",
      "-- BEGIN CORRECTION\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:47:21 2021 with 317.268 GB free disk space\n",
      "\n",
      "    cd correction\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperCreate \\\n",
      "      -minlength 1000 \\\n",
      "      -o ./pacbio20.gkpStore.BUILDING \\\n",
      "      ./pacbio20.gkpStore.gkp \\\n",
      "    > ./pacbio20.gkpStore.BUILDING.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:47:23 2021 (2 seconds) with 317.237 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- In gatekeeper store 'correction/pacbio20.gkpStore':\n",
      "--   Found 18875 reads.\n",
      "--   Found 107977839 bases (22.97 times coverage).\n",
      "--\n",
      "--   Read length histogram (one '*' equals 38.01 reads):\n",
      "--        0    999      0 \n",
      "--     1000   1999   2661 **********************************************************************\n",
      "--     2000   2999   2488 *****************************************************************\n",
      "--     3000   3999   2243 ***********************************************************\n",
      "--     4000   4999   2011 ****************************************************\n",
      "--     5000   5999   1859 ************************************************\n",
      "--     6000   6999   1606 ******************************************\n",
      "--     7000   7999   1365 ***********************************\n",
      "--     8000   8999   1183 *******************************\n",
      "--     9000   9999    928 ************************\n",
      "--    10000  10999    756 *******************\n",
      "--    11000  11999    547 **************\n",
      "--    12000  12999    425 ***********\n",
      "--    13000  13999    319 ********\n",
      "--    14000  14999    207 *****\n",
      "--    15000  15999    156 ****\n",
      "--    16000  16999     66 *\n",
      "--    17000  17999     31 \n",
      "--    18000  18999     14 \n",
      "--    19000  19999      5 \n",
      "--    20000  20999      2 \n",
      "--    21000  21999      1 \n",
      "--    22000  22999      2 \n",
      "-- Finished stage 'cor-gatekeeper', reset canuIteration.\n",
      "-- Finished stage 'merylConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 11:47:23 2021 with 317.237 GB free disk space (1 processes; 2 concurrently)\n",
      "\n",
      "    cd correction/0-mercounts\n",
      "    ./meryl.sh 1 > ./meryl.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:47:53 2021 (30 seconds) with 316.89 GB free disk space\n",
      "----------------------------------------\n",
      "-- Meryl finished successfully.\n",
      "-- Finished stage 'merylCheck', reset canuIteration.\n",
      "--\n",
      "--  16-mers                                                                                           Fraction\n",
      "--    Occurrences   NumMers                                                                         Unique Total\n",
      "--       1-     1  78998584 *******************************************************************--> 0.8920 0.7335\n",
      "--       2-     2   5787125 ********************************************************************** 0.9573 0.8410\n",
      "--       3-     4   2383846 ****************************                                           0.9742 0.8828\n",
      "--       5-     7   1163583 **************                                                         0.9909 0.9430\n",
      "--       8-    11    203140 **                                                                     0.9986 0.9850\n",
      "--      12-    16     17764                                                                        0.9997 0.9942\n",
      "--      17-    22      6262                                                                        0.9999 0.9958\n",
      "--      23-    29      3054                                                                        0.9999 0.9968\n",
      "--      30-    37      1315                                                                        1.0000 0.9975\n",
      "--      38-    46       937                                                                        1.0000 0.9978\n",
      "--      47-    56       546                                                                        1.0000 0.9982\n",
      "--      57-    67       234                                                                        1.0000 0.9984\n",
      "--      68-    79       108                                                                        1.0000 0.9986\n",
      "--      80-    92        83                                                                        1.0000 0.9986\n",
      "--      93-   106        67                                                                        1.0000 0.9987\n",
      "--     107-   121        40                                                                        1.0000 0.9988\n",
      "--     122-   137        29                                                                        1.0000 0.9988\n",
      "--     138-   154        33                                                                        1.0000 0.9988\n",
      "--     155-   172        26                                                                        1.0000 0.9989\n",
      "--     173-   191        30                                                                        1.0000 0.9989\n",
      "--     192-   211        19                                                                        1.0000 0.9990\n",
      "--     212-   232        23                                                                        1.0000 0.9990\n",
      "--     233-   254        13                                                                        1.0000 0.9991\n",
      "--     255-   277        18                                                                        1.0000 0.9991\n",
      "--     278-   301        11                                                                        1.0000 0.9991\n",
      "--     302-   326         8                                                                        1.0000 0.9992\n",
      "--     327-   352         8                                                                        1.0000 0.9992\n",
      "--     353-   379         5                                                                        0.0000 0.9992\n",
      "--     380-   407         4                                                                        0.0000 0.9992\n",
      "--     408-   436         2                                                                        0.0000 0.9992\n",
      "--     437-   466         3                                                                        0.0000 0.9992\n",
      "--     467-   497         4                                                                        0.0000 0.9993\n",
      "--     498-   529         7                                                                        1.0000 0.9993\n",
      "--     530-   562        10                                                                        1.0000 0.9993\n",
      "--     563-   596         5                                                                        0.0000 0.9994\n",
      "--     597-   631         5                                                                        0.0000 0.9994\n",
      "--     632-   667         3                                                                        0.0000 0.9994\n",
      "--     668-   704         2                                                                        0.0000 0.9994\n",
      "--     705-   742         3                                                                        0.0000 0.9994\n",
      "--     743-   781         2                                                                        0.0000 0.9995\n",
      "--     782-   821         1                                                                        0.0000 0.9995\n",
      "--\n",
      "--       33917 (max occurrences)\n",
      "--    28696130 (total mers, non-unique)\n",
      "--     9568390 (distinct mers, non-unique)\n",
      "--    78998584 (unique mers)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- For mhap overlapping, set repeat k-mer threshold to 1076.\n",
      "--\n",
      "-- Found 107694714 16-mers; 88566974 distinct and 78998584 unique.  Largest count 33917.\n",
      "-- Finished stage 'cor-meryl', reset canuIteration.\n",
      "--\n",
      "-- OVERLAPPER (mhap) (correction)\n",
      "--\n",
      "-- Set corMhapSensitivity=high based on read coverage of 22.\n",
      "--\n",
      "-- PARAMETERS: hashes=768, minMatches=2, threshold=0.73\n",
      "--\n",
      "-- Given 6 GB, can fit 9000 reads per block.\n",
      "-- For 4 blocks, set stride to 2 blocks.\n",
      "-- Logging partitioning to 'correction/1-overlapper/partitioning.log'.\n",
      "-- Configured 3 mhap precompute jobs.\n",
      "-- Configured 3 mhap overlap jobs.\n",
      "-- Finished stage 'cor-mhapConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 11:47:54 2021 with 317.236 GB free disk space (3 processes; 1 concurrently)\n",
      "\n",
      "    cd correction/1-overlapper\n",
      "    ./precompute.sh 1 > ./precompute.000001.out 2>&1\n",
      "    ./precompute.sh 2 > ./precompute.000002.out 2>&1\n",
      "    ./precompute.sh 3 > ./precompute.000003.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:58:16 2021 (622 seconds) with 316.862 GB free disk space\n",
      "----------------------------------------\n",
      "-- All 3 mhap precompute jobs finished successfully.\n",
      "-- Finished stage 'cor-mhapPrecomputeCheck', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 11:58:16 2021 with 316.862 GB free disk space (3 processes; 1 concurrently)\n",
      "\n",
      "    cd correction/1-overlapper\n",
      "    ./mhap.sh 1 > ./mhap.000001.out 2>&1\n",
      "    ./mhap.sh 2 > ./mhap.000002.out 2>&1\n",
      "    ./mhap.sh 3 > ./mhap.000003.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:58:58 2021 (42 seconds) with 316.856 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 3 mhap overlap output files.\n",
      "-- Finished stage 'cor-mhapCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:58:58 2021 with 316.855 GB free disk space\n",
      "\n",
      "    cd correction\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/ovStoreBuild \\\n",
      "     -O ./pacbio20.ovlStore.BUILDING \\\n",
      "     -G ./pacbio20.gkpStore \\\n",
      "     -M 2-8 \\\n",
      "     -L ./1-overlapper/ovljob.files \\\n",
      "     > ./pacbio20.ovlStore.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:58:59 2021 (1 second) with 316.842 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Overlap store 'correction/pacbio20.ovlStore' successfully constructed.\n",
      "--\n",
      "-- Purged 0.541 GB in 9 overlap output files and 2 directories.\n",
      "-- Overlap store 'correction/pacbio20.ovlStore' statistics not available (skipped in correction and trimming stages).\n",
      "-- Finished stage 'cor-createOverlapStore', reset canuIteration.\n",
      "-- Set corMinCoverage=4 based on read coverage of 22.\n",
      "-- Computing global filter scores 'correction/2-correction/pacbio20.globalScores'.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:58:59 2021 with 317.388 GB free disk space\n",
      "\n",
      "    cd correction/2-correction\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/filterCorrectionOverlaps \\\n",
      "      -G ../pacbio20.gkpStore \\\n",
      "      -O ../pacbio20.ovlStore \\\n",
      "      -S ./pacbio20.globalScores.WORKING \\\n",
      "      -c 40 \\\n",
      "      -l 0 \\\n",
      "    > ./pacbio20.globalScores.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 11:58:59 2021 (lickety-split) with 317.387 GB free disk space\n",
      "----------------------------------------\n",
      "--  PARAMETERS:\n",
      "--  ----------\n",
      "--  \n",
      "--       40 (expected coverage)\n",
      "--        0 (don't use overlaps shorter than this)\n",
      "--    0.000 (don't use overlaps with erate less than this)\n",
      "--    1.000 (don't use overlaps with erate more than this)\n",
      "--  \n",
      "--  OVERLAPS:\n",
      "--  --------\n",
      "--  \n",
      "--  IGNORED:\n",
      "--  \n",
      "--             0 (< 0.0000 fraction error)\n",
      "--             0 (> 0.4095 fraction error)\n",
      "--             0 (< 0 bases long)\n",
      "--             0 (> 2097151 bases long)\n",
      "--  \n",
      "--  FILTERED:\n",
      "--  \n",
      "--        151685 (too many overlaps, discard these shortest ones)\n",
      "--  \n",
      "--  EVIDENCE:\n",
      "--  \n",
      "--        506837 (longest overlaps)\n",
      "--  \n",
      "--  TOTAL:\n",
      "--  \n",
      "--        658522 (all overlaps)\n",
      "--  \n",
      "--  READS:\n",
      "--  -----\n",
      "--  \n",
      "--            74 (no overlaps)\n",
      "--         14257 (no overlaps filtered)\n",
      "--          3549 (<  50% overlaps filtered)\n",
      "--          4345 (<  80% overlaps filtered)\n",
      "--          4544 (<  95% overlaps filtered)\n",
      "--          4544 (< 100% overlaps filtered)\n",
      "--  \n",
      "-- Computing expected corrected read lengths 'correction/2-correction/pacbio20.estimate.log'.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 11:58:59 2021 with 317.387 GB free disk space\n",
      "\n",
      "    cd correction/2-correction\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/generateCorrectionLayouts \\\n",
      "      -G ../pacbio20.gkpStore \\\n",
      "      -O ../pacbio20.ovlStore \\\n",
      "      -S ./pacbio20.globalScores \\\n",
      "      -c 4 \\\n",
      "      -C 80 \\\n",
      "      -p ./pacbio20.estimate.WORKING\n",
      "\n",
      "-- Finished on Wed Apr 21 11:58:59 2021 (lickety-split) with 317.379 GB free disk space\n",
      "----------------------------------------\n",
      "-- Sorting reads by expected corrected length.\n",
      "-- Sorting reads by uncorrected length.\n",
      "-- Loading expected corrected read lengths.\n",
      "-- Picking longest corrected reads.\n",
      "-- Writing longest corrected reads to 'correction/2-correction/pacbio20.readsToCorrect'.\n",
      "-- Summarizing filter.\n",
      "--\n",
      "-- Reads to be corrected:\n",
      "--   18801 reads longer than 0 bp\n",
      "--   107485976 bp\n",
      "-- Expected corrected reads:\n",
      "--   18801 reads\n",
      "--   93566864 bp\n",
      "--   0 bp minimum length\n",
      "--   4977 bp mean length\n",
      "--   14849 bp n50 length\n",
      "-- Using overlaps no worse than 0.3 fraction error for correcting reads (from corErrorRate parameter).\n",
      "-- Finished stage 'cor-buildCorrectionLayouts', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 11:59:00 2021 with 317.375 GB free disk space (4 processes; 4 concurrently)\n",
      "\n",
      "    cd correction/2-correction\n",
      "    ./correctReads.sh 1 > ./correctReads.000001.out 2>&1\n",
      "    ./correctReads.sh 2 > ./correctReads.000002.out 2>&1\n",
      "    ./correctReads.sh 3 > ./correctReads.000003.out 2>&1\n",
      "    ./correctReads.sh 4 > ./correctReads.000004.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:05:05 2021 (365 seconds) with 317.226 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 4 read correction output files.\n",
      "-- Finished stage 'cor-generateCorrectedReads', reset canuIteration.\n",
      "-- Found 4 read correction output files.\n",
      "-- Finished stage 'cor-generateCorrectedReads', reset canuIteration.\n",
      "-- Concatenating correctReads output.\n",
      "-- Analyzing correctReads output.\n",
      "--\n",
      "-- Reads to be corrected:\n",
      "--   18801 reads longer than 0 bp\n",
      "--   107485976 bp\n",
      "-- Expected corrected reads:\n",
      "--   18801 reads\n",
      "--   93566864 bp\n",
      "--   0 bp minimum length\n",
      "--   4977 bp mean length\n",
      "--   14849 bp n50 length\n",
      "-- Actual corrected reads:\n",
      "--   16551 reads\n",
      "--   79737405 bp\n",
      "--   2721 reads with 0 corrected blocks\n",
      "--   15624 reads with 1 corrected block\n",
      "--   442 reads with 2 corrected blocks\n",
      "--   13 reads with 3 corrected blocks\n",
      "--   1 reads with 4 corrected blocks\n",
      "--\n",
      "-- Purging correctReads output after merging to final output file.\n",
      "-- Purged 4 .dump.success sentinels.\n",
      "-- Purged 4 .fasta outputs.\n",
      "-- Purged 4 .err outputs.\n",
      "-- Purged 4 .out job log outputs.\n",
      "-- Finished stage 'cor-dumpCorrectedReads', reset canuIteration.\n",
      "--\n",
      "-- Corrected reads saved in 'pacbio20.correctedReads.fasta.gz'.\n",
      "--\n",
      "--\n",
      "-- BEGIN TRIMMING\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:05:07 2021 with 317.277 GB free disk space\n",
      "\n",
      "    cd trimming\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperCreate \\\n",
      "      -minlength 1000 \\\n",
      "      -o ./pacbio20.gkpStore.BUILDING \\\n",
      "      ./pacbio20.gkpStore.gkp \\\n",
      "    > ./pacbio20.gkpStore.BUILDING.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:05:08 2021 (1 second) with 317.255 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- In gatekeeper store 'trimming/pacbio20.gkpStore':\n",
      "--   Found 16551 reads.\n",
      "--   Found 79737405 bases (16.96 times coverage).\n",
      "--\n",
      "--   Read length histogram (one '*' equals 44.88 reads):\n",
      "--        0    999      0 \n",
      "--     1000   1999   3142 **********************************************************************\n",
      "--     2000   2999   2653 ***********************************************************\n",
      "--     3000   3999   2320 ***************************************************\n",
      "--     4000   4999   1898 ******************************************\n",
      "--     5000   5999   1634 ************************************\n",
      "--     6000   6999   1274 ****************************\n",
      "--     7000   7999   1028 **********************\n",
      "--     8000   8999    773 *****************\n",
      "--     9000   9999    617 *************\n",
      "--    10000  10999    413 *********\n",
      "--    11000  11999    293 ******\n",
      "--    12000  12999    184 ****\n",
      "--    13000  13999    165 ***\n",
      "--    14000  14999     89 *\n",
      "--    15000  15999     50 *\n",
      "--    16000  16999     14 \n",
      "--    17000  17999      4 \n",
      "-- Finished stage 'obt-gatekeeper', reset canuIteration.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- Finished stage 'merylConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 12:05:08 2021 with 317.254 GB free disk space (1 processes; 2 concurrently)\n",
      "\n",
      "    cd trimming/0-mercounts\n",
      "    ./meryl.sh 1 > ./meryl.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:05:30 2021 (22 seconds) with 317.173 GB free disk space\n",
      "----------------------------------------\n",
      "-- Meryl finished successfully.\n",
      "-- Finished stage 'merylCheck', reset canuIteration.\n",
      "--\n",
      "--  22-mers                                                                                           Fraction\n",
      "--    Occurrences   NumMers                                                                         Unique Total\n",
      "--       1-     1   6268685 *******************************************************************--> 0.5251 0.0790\n",
      "--       2-     2    681057 *********************************                                      0.5822 0.0961\n",
      "--       3-     4    461977 **********************                                                 0.6060 0.1069\n",
      "--       5-     7    447197 **********************                                                 0.6332 0.1251\n",
      "--       8-    11    842421 *****************************************                              0.6731 0.1675\n",
      "--      12-    16   1419677 ********************************************************************** 0.7512 0.2923\n",
      "--      17-    22   1271202 **************************************************************         0.8706 0.5612\n",
      "--      23-    29    462679 **********************                                                 0.9646 0.8453\n",
      "--      30-    37     55082 **                                                                     0.9945 0.9622\n",
      "--      38-    46     12908                                                                        0.9978 0.9783\n",
      "--      47-    56      4433                                                                        0.9988 0.9847\n",
      "--      57-    67      2238                                                                        0.9991 0.9872\n",
      "--      68-    79      2363                                                                        0.9993 0.9890\n",
      "--      80-    92      2628                                                                        0.9995 0.9912\n",
      "--      93-   106       997                                                                        0.9997 0.9940\n",
      "--     107-   121       717                                                                        0.9998 0.9951\n",
      "--     122-   137       399                                                                        0.9999 0.9961\n",
      "--     138-   154       169                                                                        0.9999 0.9967\n",
      "--     155-   172       251                                                                        0.9999 0.9971\n",
      "--     173-   191       582                                                                        0.9999 0.9976\n",
      "--     192-   211       219                                                                        1.0000 0.9990\n",
      "--     212-   232        26                                                                        1.0000 0.9995\n",
      "--     233-   254        10                                                                        1.0000 0.9995\n",
      "--     255-   277         8                                                                        1.0000 0.9995\n",
      "--     278-   301         2                                                                        1.0000 0.9996\n",
      "--     302-   326         1                                                                        1.0000 0.9996\n",
      "--     327-   352         4                                                                        1.0000 0.9996\n",
      "--     353-   379         1                                                                        1.0000 0.9996\n",
      "--     380-   407         3                                                                        1.0000 0.9996\n",
      "--     408-   436         2                                                                        1.0000 0.9996\n",
      "--     437-   466         3                                                                        1.0000 0.9996\n",
      "--     467-   497         1                                                                        1.0000 0.9996\n",
      "--     498-   529         3                                                                        1.0000 0.9997\n",
      "--     530-   562         1                                                                        1.0000 0.9997\n",
      "--     563-   596         0                                                                        0.0000 0.0000\n",
      "--     597-   631         6                                                                        1.0000 0.9997\n",
      "--     632-   667         1                                                                        1.0000 0.9997\n",
      "--     668-   704         1                                                                        1.0000 0.9997\n",
      "--     705-   742         3                                                                        1.0000 0.9997\n",
      "--     743-   781         0                                                                        0.0000 0.0000\n",
      "--     782-   821         6                                                                        1.0000 0.9998\n",
      "--\n",
      "--        1170 (max occurrences)\n",
      "--    73121149 (total mers, non-unique)\n",
      "--     5669292 (distinct mers, non-unique)\n",
      "--     6268685 (unique mers)\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:05:30 2021 with 317.173 GB free disk space\n",
      "\n",
      "    cd trimming/0-mercounts\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/meryl \\\n",
      "      -Dt \\\n",
      "      -n 558 \\\n",
      "      -s ./pacbio20.ms22 \\\n",
      "    > ./pacbio20.ms22.frequentMers.fasta \\\n",
      "    2> ./pacbio20.ms22.frequentMers.fasta.err\n",
      "\n",
      "-- Finished on Wed Apr 21 12:05:31 2021 (1 second) with 317.173 GB free disk space\n",
      "----------------------------------------\n",
      "-- Reset obtOvlMerThreshold from auto to 558.\n",
      "--\n",
      "-- Found 79389834 22-mers; 11937977 distinct and 6268685 unique.  Largest count 1170.\n",
      "-- Finished stage 'obt-meryl', reset canuIteration.\n",
      "--\n",
      "-- OVERLAPPER (normal) (trimming) erate=0.045\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:05:31 2021 with 317.253 GB free disk space\n",
      "\n",
      "    cd trimming/1-overlapper\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/overlapInCorePartition \\\n",
      "     -g  ../pacbio20.gkpStore \\\n",
      "     -bl 100000000 \\\n",
      "     -bs 0 \\\n",
      "     -rs 2000000 \\\n",
      "     -rl 0 \\\n",
      "     -ol 500 \\\n",
      "     -o  ./pacbio20.partition \\\n",
      "    > ./pacbio20.partition.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:05:31 2021 (lickety-split) with 317.253 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Configured 1 overlapInCore jobs.\n",
      "-- Finished stage 'obt-overlapConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 12:05:31 2021 with 317.253 GB free disk space (1 processes; 1 concurrently)\n",
      "\n",
      "    cd trimming/1-overlapper\n",
      "    ./overlap.sh 1 > ./overlap.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:08:15 2021 (164 seconds) with 317.231 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 1 overlapInCore output files.\n",
      "--\n",
      "-- overlapInCore compute 'trimming/1-overlapper':\n",
      "--   kmer hits\n",
      "--     with no overlap           893646      893646 +- 0\n",
      "--     with an overlap           311961      311961 +- 0\n",
      "--\n",
      "--   overlaps                    311961      311961 +- 0\n",
      "--     contained                      0           0 +- 0\n",
      "--     dovetail                       0           0 +- 0\n",
      "--\n",
      "--   overlaps rejected\n",
      "--     multiple per pair              0           0 +- 0\n",
      "--     bad short window               0           0 +- 0\n",
      "--     bad long window                0           0 +- 0\n",
      "-- Finished stage 'obt-overlapCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:08:15 2021 with 317.231 GB free disk space\n",
      "\n",
      "    cd trimming\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/ovStoreBuild \\\n",
      "     -O ./pacbio20.ovlStore.BUILDING \\\n",
      "     -G ./pacbio20.gkpStore \\\n",
      "     -M 2-8 \\\n",
      "     -L ./1-overlapper/ovljob.files \\\n",
      "     > ./pacbio20.ovlStore.err 2>&1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "-- Finished on Wed Apr 21 12:08:15 2021 (lickety-split) with 317.219 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Overlap store 'trimming/pacbio20.ovlStore' successfully constructed.\n",
      "--\n",
      "-- Purged 0.006 GB in 3 overlap output files and 1 directories.\n",
      "-- Overlap store 'trimming/pacbio20.ovlStore' statistics not available (skipped in correction and trimming stages).\n",
      "-- Finished stage 'obt-createOverlapStore', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:08:15 2021 with 317.226 GB free disk space\n",
      "\n",
      "    cd trimming/3-overlapbasedtrimming\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/trimReads \\\n",
      "      -G  ../pacbio20.gkpStore \\\n",
      "      -O  ../pacbio20.ovlStore \\\n",
      "      -Co ./pacbio20.1.trimReads.clear \\\n",
      "      -e  0.045 \\\n",
      "      -minlength 1000 \\\n",
      "      -ol 1 \\\n",
      "      -oc 1 \\\n",
      "      -o  ./pacbio20.1.trimReads \\\n",
      "    >     ./pacbio20.1.trimReads.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:08:16 2021 (1 second) with 317.221 GB free disk space\n",
      "----------------------------------------\n",
      "--  PARAMETERS:\n",
      "--  ----------\n",
      "--     1000    (reads trimmed below this many bases are deleted)\n",
      "--   0.0450    (use overlaps at or below this fraction error)\n",
      "--        1    (break region if overlap is less than this long, for 'largest covered' algorithm)\n",
      "--        1    (break region if overlap coverage is less than this many read, for 'largest covered' algorithm)\n",
      "--  \n",
      "--  INPUT READS:\n",
      "--  -----------\n",
      "--   16551 reads     79737405 bases (reads processed)\n",
      "--       0 reads            0 bases (reads not processed, previously deleted)\n",
      "--       0 reads            0 bases (reads not processed, in a library where trimming isn't allowed)\n",
      "--  \n",
      "--  OUTPUT READS:\n",
      "--  ------------\n",
      "--   14062 reads     68323812 bases (trimmed reads output)\n",
      "--    2407 reads      9807090 bases (reads with no change, kept as is)\n",
      "--       5 reads         6036 bases (reads with no overlaps, deleted)\n",
      "--      77 reads        94148 bases (reads with short trimmed length, deleted)\n",
      "--  \n",
      "--  TRIMMING DETAILS:\n",
      "--  ----------------\n",
      "--    9737 reads       963067 bases (bases trimmed from the 5' end of a read)\n",
      "--   10775 reads       543252 bases (bases trimmed from the 3' end of a read)\n",
      "-- Finished stage 'obt-trimReads', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:08:16 2021 with 317.221 GB free disk space\n",
      "\n",
      "    cd trimming/3-overlapbasedtrimming\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/splitReads \\\n",
      "      -G  ../pacbio20.gkpStore \\\n",
      "      -O  ../pacbio20.ovlStore \\\n",
      "      -Ci ./pacbio20.1.trimReads.clear \\\n",
      "      -Co ./pacbio20.2.splitReads.clear \\\n",
      "      -e  0.045 \\\n",
      "      -minlength 1000 \\\n",
      "      -o  ./pacbio20.2.splitReads \\\n",
      "    >     ./pacbio20.2.splitReads.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:08:16 2021 (lickety-split) with 317.221 GB free disk space\n",
      "----------------------------------------\n",
      "--  PARAMETERS:\n",
      "--  ----------\n",
      "--     1000    (reads trimmed below this many bases are deleted)\n",
      "--   0.0450    (use overlaps at or below this fraction error)\n",
      "--  INPUT READS:\n",
      "--  -----------\n",
      "--   16469 reads     79637221 bases (reads processed)\n",
      "--      82 reads       100184 bases (reads not processed, previously deleted)\n",
      "--       0 reads            0 bases (reads not processed, in a library where trimming isn't allowed)\n",
      "--  \n",
      "--  PROCESSED:\n",
      "--  --------\n",
      "--       0 reads            0 bases (no overlaps)\n",
      "--       1 reads         1323 bases (no coverage after adjusting for trimming done already)\n",
      "--       0 reads            0 bases (processed for chimera)\n",
      "--       0 reads            0 bases (processed for spur)\n",
      "--   16468 reads     79635898 bases (processed for subreads)\n",
      "--  \n",
      "--  READS WITH SIGNALS:\n",
      "--  ------------------\n",
      "--       0 reads            0 signals (number of 5' spur signal)\n",
      "--       0 reads            0 signals (number of 3' spur signal)\n",
      "--       0 reads            0 signals (number of chimera signal)\n",
      "--      10 reads           10 signals (number of subread signal)\n",
      "--  \n",
      "--  SIGNALS:\n",
      "--  -------\n",
      "--       0 reads            0 bases (size of 5' spur signal)\n",
      "--       0 reads            0 bases (size of 3' spur signal)\n",
      "--       0 reads            0 bases (size of chimera signal)\n",
      "--      10 reads         1752 bases (size of subread signal)\n",
      "--  \n",
      "--  TRIMMING:\n",
      "--  --------\n",
      "--       4 reads         5968 bases (trimmed from the 5' end of the read)\n",
      "--       6 reads        10062 bases (trimmed from the 3' end of the read)\n",
      "-- Finished stage 'obt-splitReads', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:08:16 2021 with 317.221 GB free disk space\n",
      "\n",
      "    cd trimming\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperDumpFASTQ -fasta -nolibname \\\n",
      "      -G ./pacbio20.gkpStore \\\n",
      "      -c ./3-overlapbasedtrimming/pacbio20.2.splitReads.clear \\\n",
      "      -o ../pacbio20.trimmedReads.gz \\\n",
      "    >    ../pacbio20.trimmedReads.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:08:17 2021 (1 second) with 317.195 GB free disk space\n",
      "----------------------------------------\n",
      "-- Finished stage 'obt-dumpReads', reset canuIteration.\n",
      "--\n",
      "-- Trimmed reads saved in 'trimming/pacbio20.trimmedReads.fasta.gz'\n",
      "--\n",
      "--\n",
      "-- BEGIN ASSEMBLY\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:08:18 2021 with 317.195 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperCreate \\\n",
      "      -minlength 1000 \\\n",
      "      -o ./pacbio20.gkpStore.BUILDING \\\n",
      "      ./pacbio20.gkpStore.gkp \\\n",
      "    > ./pacbio20.gkpStore.BUILDING.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:08:18 2021 (lickety-split) with 317.172 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- In gatekeeper store 'unitigging/pacbio20.gkpStore':\n",
      "--   Found 16469 reads.\n",
      "--   Found 78114872 bases (16.62 times coverage).\n",
      "--\n",
      "--   Read length histogram (one '*' equals 45.92 reads):\n",
      "--        0    999      0 \n",
      "--     1000   1999   3215 **********************************************************************\n",
      "--     2000   2999   2709 **********************************************************\n",
      "--     3000   3999   2326 **************************************************\n",
      "--     4000   4999   1856 ****************************************\n",
      "--     5000   5999   1619 ***********************************\n",
      "--     6000   6999   1270 ***************************\n",
      "--     7000   7999    994 *********************\n",
      "--     8000   8999    742 ****************\n",
      "--     9000   9999    591 ************\n",
      "--    10000  10999    389 ********\n",
      "--    11000  11999    285 ******\n",
      "--    12000  12999    176 ***\n",
      "--    13000  13999    143 ***\n",
      "--    14000  14999     89 *\n",
      "--    15000  15999     49 *\n",
      "--    16000  16999     13 \n",
      "--    17000  17999      3 \n",
      "-- Finished stage 'utg-gatekeeper', reset canuIteration.\n",
      "-- Finished stage 'merylConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 12:08:19 2021 with 317.172 GB free disk space (1 processes; 2 concurrently)\n",
      "\n",
      "    cd unitigging/0-mercounts\n",
      "    ./meryl.sh 1 > ./meryl.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:08:38 2021 (19 seconds) with 317.139 GB free disk space\n",
      "----------------------------------------\n",
      "-- Meryl finished successfully.\n",
      "-- Finished stage 'merylCheck', reset canuIteration.\n",
      "--\n",
      "--  22-mers                                                                                           Fraction\n",
      "--    Occurrences   NumMers                                                                         Unique Total\n",
      "--       1-     1   5327252 *******************************************************************--> 0.4862 0.0685\n",
      "--       2-     2    659350 ********************************                                       0.5463 0.0855\n",
      "--       3-     4    457982 **********************                                                 0.5720 0.0963\n",
      "--       5-     7    450569 **********************                                                 0.6016 0.1149\n",
      "--       8-    11    851188 *****************************************                              0.6456 0.1588\n",
      "--      12-    16   1429647 ********************************************************************** 0.7316 0.2874\n",
      "--      17-    22   1257055 *************************************************************          0.8620 0.5625\n",
      "--      23-    29    446672 *********************                                                  0.9630 0.8484\n",
      "--      30-    37     50274 **                                                                     0.9944 0.9631\n",
      "--      38-    46     12981                                                                        0.9976 0.9782\n",
      "--      47-    56      4221                                                                        0.9987 0.9847\n",
      "--      57-    67      2432                                                                        0.9991 0.9871\n",
      "--      68-    79      2213                                                                        0.9993 0.9891\n",
      "--      80-    92      2576                                                                        0.9995 0.9912\n",
      "--      93-   106       978                                                                        0.9997 0.9940\n",
      "--     107-   121       752                                                                        0.9998 0.9952\n",
      "--     122-   137       342                                                                        0.9999 0.9962\n",
      "--     138-   154       197                                                                        0.9999 0.9967\n",
      "--     155-   172       312                                                                        0.9999 0.9971\n",
      "--     173-   191       578                                                                        0.9999 0.9978\n",
      "--     192-   211       145                                                                        1.0000 0.9992\n",
      "--     212-   232        10                                                                        1.0000 0.9995\n",
      "--     233-   254         9                                                                        1.0000 0.9995\n",
      "--     255-   277         7                                                                        1.0000 0.9996\n",
      "--     278-   301         2                                                                        1.0000 0.9996\n",
      "--     302-   326         4                                                                        1.0000 0.9996\n",
      "--     327-   352         1                                                                        1.0000 0.9996\n",
      "--     353-   379         1                                                                        1.0000 0.9996\n",
      "--     380-   407         3                                                                        1.0000 0.9996\n",
      "--     408-   436         5                                                                        1.0000 0.9996\n",
      "--     437-   466         1                                                                        1.0000 0.9996\n",
      "--     467-   497         0                                                                        0.0000 0.0000\n",
      "--     498-   529         3                                                                        1.0000 0.9997\n",
      "--     530-   562         1                                                                        1.0000 0.9997\n",
      "--     563-   596         0                                                                        0.0000 0.0000\n",
      "--     597-   631         6                                                                        1.0000 0.9997\n",
      "--     632-   667         1                                                                        1.0000 0.9997\n",
      "--     668-   704         1                                                                        1.0000 0.9997\n",
      "--     705-   742         3                                                                        1.0000 0.9997\n",
      "--     743-   781         3                                                                        1.0000 0.9998\n",
      "--     782-   821         5                                                                        1.0000 0.9998\n",
      "--\n",
      "--        1162 (max occurrences)\n",
      "--    72441771 (total mers, non-unique)\n",
      "--     5630542 (distinct mers, non-unique)\n",
      "--     5327252 (unique mers)\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:08:38 2021 with 317.139 GB free disk space\n",
      "\n",
      "    cd unitigging/0-mercounts\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/meryl \\\n",
      "      -Dt \\\n",
      "      -n 548 \\\n",
      "      -s ./pacbio20.ms22 \\\n",
      "    > ./pacbio20.ms22.frequentMers.fasta \\\n",
      "    2> ./pacbio20.ms22.frequentMers.fasta.err\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "-- Finished on Wed Apr 21 12:08:39 2021 (1 second) with 317.139 GB free disk space\n",
      "----------------------------------------\n",
      "-- Reset utgOvlMerThreshold from auto to 548.\n",
      "--\n",
      "-- Found 77769023 22-mers; 10957794 distinct and 5327252 unique.  Largest count 1162.\n",
      "-- Finished stage 'utg-meryl', reset canuIteration.\n",
      "--\n",
      "-- OVERLAPPER (normal) (assembly) erate=0.045\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:08:39 2021 with 317.202 GB free disk space\n",
      "\n",
      "    cd unitigging/1-overlapper\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/overlapInCorePartition \\\n",
      "     -g  ../pacbio20.gkpStore \\\n",
      "     -bl 100000000 \\\n",
      "     -bs 0 \\\n",
      "     -rs 2000000 \\\n",
      "     -rl 0 \\\n",
      "     -ol 500 \\\n",
      "     -o  ./pacbio20.partition \\\n",
      "    > ./pacbio20.partition.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:08:39 2021 (lickety-split) with 317.202 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Configured 1 overlapInCore jobs.\n",
      "-- Finished stage 'utg-overlapConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 12:08:39 2021 with 317.202 GB free disk space (1 processes; 1 concurrently)\n",
      "\n",
      "    cd unitigging/1-overlapper\n",
      "    ./overlap.sh 1 > ./overlap.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:11:14 2021 (155 seconds) with 317.169 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 1 overlapInCore output files.\n",
      "--\n",
      "-- overlapInCore compute 'unitigging/1-overlapper':\n",
      "--   kmer hits\n",
      "--     with no overlap           916699      916699 +- 0\n",
      "--     with an overlap           259360      259360 +- 0\n",
      "--\n",
      "--   overlaps                    259360      259360 +- 0\n",
      "--     contained                 101891      101891 +- 0\n",
      "--     dovetail                  157469      157469 +- 0\n",
      "--\n",
      "--   overlaps rejected\n",
      "--     multiple per pair              0           0 +- 0\n",
      "--     bad short window               0           0 +- 0\n",
      "--     bad long window                0           0 +- 0\n",
      "-- Finished stage 'utg-overlapCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:11:14 2021 with 317.169 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/ovStoreBuild \\\n",
      "     -O ./pacbio20.ovlStore.BUILDING \\\n",
      "     -G ./pacbio20.gkpStore \\\n",
      "     -M 2-8 \\\n",
      "     -L ./1-overlapper/ovljob.files \\\n",
      "     > ./pacbio20.ovlStore.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:11:14 2021 (lickety-split) with 317.159 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Overlap store 'unitigging/pacbio20.ovlStore' successfully constructed.\n",
      "--\n",
      "-- Purged 0.004 GB in 3 overlap output files and 1 directories.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:11:14 2021 with 317.164 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/ovStoreStats \\\n",
      "     -G ./pacbio20.gkpStore \\\n",
      "     -O ./pacbio20.ovlStore \\\n",
      "     -o ./pacbio20.ovlStore \\\n",
      "     > ./pacbio20.ovlStore.summary.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:11:14 2021 (lickety-split) with 317.164 GB free disk space\n",
      "----------------------------------------\n",
      "-- category            reads     %          read length        feature size or coverage  analysis\n",
      "-- ----------------  -------  -------  ----------------------  ------------------------  --------------------\n",
      "-- middle-missing          2    0.01     3331.50 +- 690.84         576.00 +- 16.97      (bad trimming)\n",
      "-- middle-hump             1    0.01     2932.00 +- 0.00           293.00 +- 0.00       (bad trimming)\n",
      "-- no-5-prime             20    0.12     6486.00 +- 3613.93         56.45 +- 154.68     (bad trimming)\n",
      "-- no-3-prime             19    0.12     5086.11 +- 2845.85         30.16 +- 90.67      (bad trimming)\n",
      "-- \n",
      "-- low-coverage          233    1.41     1992.61 +- 1064.47          5.74 +- 1.71       (easy to assemble, potential for lower quality consensus)\n",
      "-- unique              13381   81.25     4572.41 +- 2946.75         19.12 +- 5.12       (easy to assemble, perfect, yay)\n",
      "-- repeat-cont             8    0.05     1358.25 +- 273.83          96.80 +- 41.02      (potential for consensus errors, no impact on assembly)\n",
      "-- repeat-dove             0    0.00        0.00 +- 0.00             0.00 +- 0.00       (hard to assemble, likely won't assemble correctly or even at all)\n",
      "-- \n",
      "-- span-repeat           965    5.86     6603.80 +- 3454.67       1727.27 +- 1974.30    (read spans a large repeat, usually easy to assemble)\n",
      "-- uniq-repeat-cont     1397    8.48     4206.66 +- 2166.89                             (should be uniquely placed, low potential for consensus errors, no impact on assembly)\n",
      "-- uniq-repeat-dove      441    2.68     8991.40 +- 2884.17                             (will end contigs, potential to misassemble)\n",
      "-- uniq-anchor             1    0.01     4569.00 +- 0.00          1098.00 +- 0.00       (repeat read, with unique section, probable bad read)\n",
      "--\n",
      "-- Overlap store 'unitigging/pacbio20.ovlStore' contains:\n",
      "--\n",
      "--   category            reads     %          read length        feature size or coverage  analysis\n",
      "--   ----------------  -------  -------  ----------------------  ------------------------  --------------------\n",
      "--   middle-missing          2    0.01     3331.50 +- 690.84         576.00 +- 16.97      (bad trimming)\n",
      "--   middle-hump             1    0.01     2932.00 +- 0.00           293.00 +- 0.00       (bad trimming)\n",
      "--   no-5-prime             20    0.12     6486.00 +- 3613.93         56.45 +- 154.68     (bad trimming)\n",
      "--   no-3-prime             19    0.12     5086.11 +- 2845.85         30.16 +- 90.67      (bad trimming)\n",
      "--   \n",
      "--   low-coverage          233    1.41     1992.61 +- 1064.47          5.74 +- 1.71       (easy to assemble, potential for lower quality consensus)\n",
      "--   unique              13381   81.25     4572.41 +- 2946.75         19.12 +- 5.12       (easy to assemble, perfect, yay)\n",
      "--   repeat-cont             8    0.05     1358.25 +- 273.83          96.80 +- 41.02      (potential for consensus errors, no impact on assembly)\n",
      "--   repeat-dove             0    0.00        0.00 +- 0.00             0.00 +- 0.00       (hard to assemble, likely won't assemble correctly or even at all)\n",
      "--   \n",
      "--   span-repeat           965    5.86     6603.80 +- 3454.67       1727.27 +- 1974.30    (read spans a large repeat, usually easy to assemble)\n",
      "--   uniq-repeat-cont     1397    8.48     4206.66 +- 2166.89                             (should be uniquely placed, low potential for consensus errors, no impact on assembly)\n",
      "--   uniq-repeat-dove      441    2.68     8991.40 +- 2884.17                             (will end contigs, potential to misassemble)\n",
      "--   uniq-anchor             1    0.01     4569.00 +- 0.00          1098.00 +- 0.00       (repeat read, with unique section, probable bad read)\n",
      "-- Finished stage 'utg-createOverlapStore', reset canuIteration.\n",
      "\n",
      "Configure RED for 2gb memory with batches of at most 5000 reads and (unlimited) bases.\n",
      "\n",
      "RED job   1 from read         1 to read      5000 -   1.005 GB for    5000 reads -   0.290 GB for  23948007 olaps -   0.002 GB for evidence\n",
      "RED job   2 from read      5001 to read     10000 -   0.990 GB for    5000 reads -   0.285 GB for  23572013 olaps -   0.002 GB for evidence\n",
      "RED job   3 from read     10001 to read     15000 -   0.987 GB for    5000 reads -   0.285 GB for  23515782 olaps -   0.002 GB for evidence\n",
      "RED job   4 from read     15001 to read     16469 -   0.297 GB for    1469 reads -   0.086 GB for   7079070 olaps -   0.001 GB for evidence\n",
      "-- Finished stage 'readErrorDetectionConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 12:11:15 2021 with 317.164 GB free disk space (4 processes; 2 concurrently)\n",
      "\n",
      "    cd unitigging/3-overlapErrorAdjustment\n",
      "    ./red.sh 1 > ./red.000001.out 2>&1\n",
      "    ./red.sh 2 > ./red.000002.out 2>&1\n",
      "    ./red.sh 3 > ./red.000003.out 2>&1\n",
      "    ./red.sh 4 > ./red.000004.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:11:21 2021 (6 seconds) with 317.162 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 4 read error detection output files.\n",
      "-- Finished stage 'readErrorDetectionCheck', reset canuIteration.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Configure OEA for 1gb memory with batches of at most 25000 reads and (unlimited) bases.\n",
      "Configured 1 jobs.\n",
      "\n",
      "OEA job   1 from read         1 to read     16469 - 74.5 bases +  3.4 adjusts +  0.5 reads + 15.8 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 422.2 MB\n",
      "\n",
      "-- Finished stage 'overlapErrorAdjustmentConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 12:11:21 2021 with 317.162 GB free disk space (1 processes; 8 concurrently)\n",
      "\n",
      "    cd unitigging/3-overlapErrorAdjustment\n",
      "    ./oea.sh 1 > ./oea.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:11:28 2021 (7 seconds) with 317.162 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 1 overlap error adjustment output files.\n",
      "-- Finished stage 'overlapErrorAdjustmentCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:11:28 2021 with 317.162 GB free disk space\n",
      "\n",
      "    cd unitigging/3-overlapErrorAdjustment\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/ovStoreBuild \\\n",
      "      -G ../pacbio20.gkpStore \\\n",
      "      -O ../pacbio20.ovlStore \\\n",
      "      -evalues \\\n",
      "      -L ./oea.files \\\n",
      "    > ./oea.apply.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:11:28 2021 (lickety-split) with 317.161 GB free disk space\n",
      "----------------------------------------\n",
      "-- No report available.\n",
      "-- Finished stage 'updateOverlapStore', reset canuIteration.\n",
      "-- Finished stage 'unitig', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 12:11:28 2021 with 317.161 GB free disk space (1 processes; 2 concurrently)\n",
      "\n",
      "    cd unitigging/4-unitigger\n",
      "    ./unitigger.sh 1 > ./unitigger.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:11:30 2021 (2 seconds) with 317.152 GB free disk space\n",
      "----------------------------------------\n",
      "-- Unitigger finished successfully.\n",
      "-- Found, in version 1, after unitig construction:\n",
      "--   contigs:      22 sequences, total length 4589251 bp (including 0 repeats of total length 0 bp).\n",
      "--   bubbles:      0 sequences, total length 0 bp.\n",
      "--   unassembled:  1352 sequences, total length 4806664 bp.\n",
      "--\n",
      "-- Contig sizes based on genome size --\n",
      "--            NG (bp)  LG (contigs)    sum (bp)\n",
      "--         ----------  ------------  ----------\n",
      "--     10      750063             1      750063\n",
      "--     20      516327             2     1266390\n",
      "--     30      465535             3     1731925\n",
      "--     40      453970             4     2185895\n",
      "--     50      272595             5     2458490\n",
      "--     60      240929             7     2970820\n",
      "--     70      211882             9     3404229\n",
      "--     80      164370            11     3767456\n",
      "--     90      106774            15     4242900\n",
      "--\n",
      "-- Finished stage 'unitigCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:11:30 2021 with 317.152 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperPartition \\\n",
      "      -G ./pacbio20.gkpStore \\\n",
      "      -T ./pacbio20.ctgStore 1 \\\n",
      "      -b 15000 \\\n",
      "      -p 8 \\\n",
      "    > ./pacbio20.ctgStore/partitionedReads.log 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:11:30 2021 (lickety-split) with 317.13 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:11:30 2021 with 317.13 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperPartition \\\n",
      "      -G ./pacbio20.gkpStore \\\n",
      "      -T ./pacbio20.utgStore 1 \\\n",
      "      -b 15000 \\\n",
      "      -p 8 \\\n",
      "    > ./pacbio20.utgStore/partitionedReads.log 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:11:30 2021 (lickety-split) with 317.111 GB free disk space\n",
      "----------------------------------------\n",
      "-- Using fast alignment for consensus (iteration '0').\n",
      "-- Configured 3 contig and 2 unitig consensus jobs.\n",
      "-- Finished stage 'consensusConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 12:11:30 2021 with 317.111 GB free disk space (5 processes; 2 concurrently)\n",
      "\n",
      "    cd unitigging/5-consensus\n",
      "    ./consensus.sh 1 > ./consensus.000001.out 2>&1\n",
      "    ./consensus.sh 2 > ./consensus.000002.out 2>&1\n",
      "    ./consensus.sh 3 > ./consensus.000003.out 2>&1\n",
      "    ./consensus.sh 4 > ./consensus.000004.out 2>&1\n",
      "    ./consensus.sh 5 > ./consensus.000005.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:13:53 2021 (143 seconds) with 317.069 GB free disk space\n",
      "----------------------------------------\n",
      "-- Using slow alignment for consensus (iteration '1').\n",
      "-- Configured 3 contig and 2 unitig consensus jobs.\n",
      "-- All 5 consensus jobs finished successfully.\n",
      "-- Finished stage 'consensusCheck', reset canuIteration.\n",
      "-- Using fast alignment for consensus (iteration '0').\n",
      "-- Configured 3 contig and 2 unitig consensus jobs.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:13:53 2021 with 317.069 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreLoad \\\n",
      "      -G ./pacbio20.gkpStore \\\n",
      "      -T ./pacbio20.ctgStore 2 \\\n",
      "      -L ./5-consensus/ctgcns.files \\\n",
      "    > ./5-consensus/ctgcns.files.ctgStoreLoad.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:13:53 2021 (lickety-split) with 317.051 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:13:53 2021 with 317.051 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreLoad \\\n",
      "      -G ./pacbio20.gkpStore \\\n",
      "      -T ./pacbio20.utgStore 2 \\\n",
      "      -L ./5-consensus/utgcns.files \\\n",
      "    > ./5-consensus/utgcns.files.utgStoreLoad.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:13:53 2021 (lickety-split) with 317.041 GB free disk space\n",
      "----------------------------------------\n",
      "-- Purging consensus output after loading to ctgStore and/or utgStore.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:13:53 2021 with 317.041 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./pacbio20.gkpStore \\\n",
      "      -T ./pacbio20.ctgStore 2 \\\n",
      "      -sizes -s 4700000 \\\n",
      "    > ./pacbio20.ctgStore/seqDB.v002.sizes.txt\n",
      "\n",
      "-- Finished on Wed Apr 21 12:13:53 2021 (lickety-split) with 317.041 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found, in version 2, after consensus generation:\n",
      "--   contigs:      22 sequences, total length 4598820 bp (including 0 repeats of total length 0 bp).\n",
      "--   bubbles:      0 sequences, total length 0 bp.\n",
      "--   unassembled:  1352 sequences, total length 4806720 bp.\n",
      "--\n",
      "-- Contig sizes based on genome size --\n",
      "--            NG (bp)  LG (contigs)    sum (bp)\n",
      "--         ----------  ------------  ----------\n",
      "--     10      751732             1      751732\n",
      "--     20      517261             2     1268993\n",
      "--     30      466710             3     1735703\n",
      "--     40      454808             4     2190511\n",
      "--     50      273171             5     2463682\n",
      "--     60      241359             7     2977182\n",
      "--     70      212266             9     3411365\n",
      "--     80      164632            11     3775320\n",
      "--     90      107061            15     4251819\n",
      "--\n",
      "-- Finished stage 'consensusLoad', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:13:53 2021 with 317.041 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreCoverageStat \\\n",
      "      -G ./pacbio20.gkpStore \\\n",
      "      -T ./pacbio20.ctgStore 2 \\\n",
      "      -s 4700000 \\\n",
      "      -o ./pacbio20.ctgStore.coverageStat \\\n",
      "    > ./pacbio20.ctgStore.coverageStat.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:13:53 2021 (lickety-split) with 317.041 GB free disk space\n",
      "----------------------------------------\n",
      "-- Finished stage 'consensusAnalyze', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:13:53 2021 with 317.041 GB free disk space\n",
      "\n",
      "    cd unitigging/4-unitigger\n",
      "    ./alignGFA.sh\n",
      "GFA alignments updated.\n",
      "\n",
      "-- Finished on Wed Apr 21 12:13:54 2021 (1 second) with 317.041 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:13:54 2021 with 317.041 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./unitigging/pacbio20.gkpStore \\\n",
      "      -T ./unitigging/pacbio20.ctgStore 2 \\\n",
      "      -o ./pacbio20.contigs \\\n",
      "      -layout \\\n",
      "    > ./pacbio20.contigs.layout.err 2>&1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "-- Finished on Wed Apr 21 12:13:54 2021 (lickety-split) with 317.022 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:13:54 2021 with 317.022 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./unitigging/pacbio20.gkpStore \\\n",
      "      -T ./unitigging/pacbio20.utgStore 2 \\\n",
      "      -o ./pacbio20.unitigs \\\n",
      "      -layout \\\n",
      "    > ./pacbio20.unitigs.layout.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:13:54 2021 (lickety-split) with 317.012 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:13:54 2021 with 317.012 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./unitigging/pacbio20.gkpStore \\\n",
      "      -T ./unitigging/pacbio20.ctgStore 2 \\\n",
      "      -consensus -fasta \\\n",
      "      -unassembled \\\n",
      "    > ./pacbio20.unassembled.fasta\n",
      "    2> ./pacbio20.unassembled.err\n",
      "\n",
      "-- Finished on Wed Apr 21 12:13:54 2021 (lickety-split) with 317.007 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:13:54 2021 with 317.007 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./unitigging/pacbio20.gkpStore \\\n",
      "      -T ./unitigging/pacbio20.ctgStore 2 \\\n",
      "      -consensus -fasta \\\n",
      "      -contigs \\\n",
      "    > ./pacbio20.contigs.fasta\n",
      "    2> ./pacbio20.contigs.err\n",
      "\n",
      "-- Finished on Wed Apr 21 12:13:54 2021 (lickety-split) with 317.003 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:13:54 2021 with 317.003 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./unitigging/pacbio20.gkpStore \\\n",
      "      -T ./unitigging/pacbio20.utgStore 2 \\\n",
      "      -consensus -fasta \\\n",
      "      -contigs \\\n",
      "    > ./pacbio20.unitigs.fasta\n",
      "    2> ./pacbio20.unitigs.err\n",
      "\n",
      "-- Finished on Wed Apr 21 12:13:54 2021 (lickety-split) with 316.998 GB free disk space\n",
      "----------------------------------------\n",
      "-- Finished stage 'generateOutputs', reset canuIteration.\n",
      "--\n",
      "-- Assembly 'pacbio20' finished in '/Users/olga/Documents/BioinformaticsInstitute/NGS/pacbio20'.\n",
      "--\n",
      "-- Summary saved in 'unitigging.html'.\n",
      "--\n",
      "-- Sequences saved:\n",
      "--   Contigs       -> 'pacbio20.contigs.fasta'\n",
      "--   Unassembled   -> 'pacbio20.unassembled.fasta'\n",
      "--   Unitigs       -> 'pacbio20.unitigs.fasta'\n",
      "--\n",
      "-- Read layouts saved:\n",
      "--   Contigs       -> 'pacbio20.contigs.layout'.\n",
      "--   Unitigs       -> 'pacbio20.unitigs.layout'.\n",
      "--\n",
      "-- Graphs saved:\n",
      "--   Contigs       -> 'pacbio20.contigs.gfa'.\n",
      "--   Unitigs       -> 'pacbio20.unitigs.gfa'.\n",
      "--\n",
      "-- Bye.\n",
      "-- Finished stage 'outputSequence', reset canuIteration.\n"
     ]
    }
   ],
   "source": [
    "! /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/bin/canu -p pacbio20 -d pacbio20 genomesize=4.7m -pacbio-raw ./homework6/pacbio_20x.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- Canu release v1.5\n",
      "-- Detected Java(TM) Runtime Environment '1.8.0_152-release' (from '/Users/olga/miniconda3/bin/java').\n",
      "-- Detected gnuplot version '5.4 patchlevel 1' (from 'gnuplot') and image format 'png'.\n",
      "-- Detected 8 CPUs and 16 gigabytes of memory.\n",
      "-- No grid engine detected, grid disabled.\n",
      "--\n",
      "-- Run   2 jobs concurrently using    8 GB and   4 CPUs for stage 'meryl'.\n",
      "-- Run   1 job  concurrently using    6 GB and   8 CPUs for stage 'mhap (cor)'.\n",
      "-- Run   1 job  concurrently using    8 GB and   8 CPUs for stage 'overlapper (obt)'.\n",
      "-- Run   1 job  concurrently using    8 GB and   8 CPUs for stage 'overlapper (utg)'.\n",
      "-- Run   4 jobs concurrently using    8 GB and   2 CPUs for stage 'falcon_sense'.\n",
      "-- Run   8 jobs concurrently using    2 GB and   1 CPU  for stage 'ovStore bucketizer'.\n",
      "-- Run   8 jobs concurrently using    8 GB and   1 CPU  for stage 'ovStore sorting'.\n",
      "-- Run   2 jobs concurrently using    2 GB and   4 CPUs for stage 'read error detection'.\n",
      "-- Run   8 jobs concurrently using    1 GB and   1 CPU  for stage 'overlap error adjustment'.\n",
      "-- Run   2 jobs concurrently using    8 GB and   4 CPUs for stage 'bogart'.\n",
      "-- Run   8 jobs concurrently using    2 GB and   1 CPU  for stage 'GFA alignment and processing'.\n",
      "-- Run   2 jobs concurrently using    8 GB and   4 CPUs for stage 'consensus'.\n",
      "--\n",
      "-- Generating assembly 'pacbio40' in '/Users/olga/Documents/BioinformaticsInstitute/NGS/pacbio40'\n",
      "--\n",
      "-- Parameters:\n",
      "--\n",
      "--  genomeSize        4700000\n",
      "--\n",
      "--  Overlap Generation Limits:\n",
      "--    corOvlErrorRate 0.2400 ( 24.00%)\n",
      "--    obtOvlErrorRate 0.0450 (  4.50%)\n",
      "--    utgOvlErrorRate 0.0450 (  4.50%)\n",
      "--\n",
      "--  Overlap Processing Limits:\n",
      "--    corErrorRate    0.3000 ( 30.00%)\n",
      "--    obtErrorRate    0.0450 (  4.50%)\n",
      "--    utgErrorRate    0.0450 (  4.50%)\n",
      "--    cnsErrorRate    0.0450 (  4.50%)\n",
      "--\n",
      "--\n",
      "-- BEGIN CORRECTION\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:14:13 2021 with 316.998 GB free disk space\n",
      "\n",
      "    cd correction\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperCreate \\\n",
      "      -minlength 1000 \\\n",
      "      -o ./pacbio40.gkpStore.BUILDING \\\n",
      "      ./pacbio40.gkpStore.gkp \\\n",
      "    > ./pacbio40.gkpStore.BUILDING.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:14:16 2021 (3 seconds) with 316.937 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- In gatekeeper store 'correction/pacbio40.gkpStore':\n",
      "--   Found 37828 reads.\n",
      "--   Found 216693367 bases (46.1 times coverage).\n",
      "--\n",
      "--   Read length histogram (one '*' equals 76.04 reads):\n",
      "--        0    999      0 \n",
      "--     1000   1999   5323 **********************************************************************\n",
      "--     2000   2999   4969 *****************************************************************\n",
      "--     3000   3999   4455 **********************************************************\n",
      "--     4000   4999   4022 ****************************************************\n",
      "--     5000   5999   3742 *************************************************\n",
      "--     6000   6999   3296 *******************************************\n",
      "--     7000   7999   2760 ************************************\n",
      "--     8000   8999   2315 ******************************\n",
      "--     9000   9999   1865 ************************\n",
      "--    10000  10999   1482 *******************\n",
      "--    11000  11999   1086 **************\n",
      "--    12000  12999    864 ***********\n",
      "--    13000  13999    676 ********\n",
      "--    14000  14999    458 ******\n",
      "--    15000  15999    307 ****\n",
      "--    16000  16999    117 *\n",
      "--    17000  17999     47 \n",
      "--    18000  18999     26 \n",
      "--    19000  19999     12 \n",
      "--    20000  20999      3 \n",
      "--    21000  21999      2 \n",
      "--    22000  22999      1 \n",
      "-- Finished stage 'cor-gatekeeper', reset canuIteration.\n",
      "-- Finished stage 'merylConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 12:14:16 2021 with 316.936 GB free disk space (1 processes; 2 concurrently)\n",
      "\n",
      "    cd correction/0-mercounts\n",
      "    ./meryl.sh 1 > ./meryl.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:15:18 2021 (62 seconds) with 316.289 GB free disk space\n",
      "----------------------------------------\n",
      "-- Meryl finished successfully.\n",
      "-- Finished stage 'merylCheck', reset canuIteration.\n",
      "--\n",
      "--  16-mers                                                                                           Fraction\n",
      "--    Occurrences   NumMers                                                                         Unique Total\n",
      "--       1-     1 140339517 *******************************************************************--> 0.8602 0.6493\n",
      "--       2-     2  14761620 ********************************************************************** 0.9507 0.7859\n",
      "--       3-     4   4036568 *******************                                                    0.9687 0.8268\n",
      "--       5-     7   2036860 *********                                                              0.9800 0.8643\n",
      "--       8-    11   1518393 *******                                                                0.9912 0.9228\n",
      "--      12-    16    386692 *                                                                      0.9982 0.9760\n",
      "--      17-    22     41859                                                                        0.9997 0.9921\n",
      "--      23-    29      9123                                                                        0.9999 0.9947\n",
      "--      30-    37      4910                                                                        0.9999 0.9957\n",
      "--      38-    46      3609                                                                        0.9999 0.9964\n",
      "--      47-    56      2002                                                                        1.0000 0.9971\n",
      "--      57-    67      1047                                                                        1.0000 0.9975\n",
      "--      68-    79       785                                                                        1.0000 0.9978\n",
      "--      80-    92       718                                                                        1.0000 0.9981\n",
      "--      93-   106       438                                                                        1.0000 0.9984\n",
      "--     107-   121       187                                                                        1.0000 0.9986\n",
      "--     122-   137       105                                                                        1.0000 0.9986\n",
      "--     138-   154        69                                                                        1.0000 0.9987\n",
      "--     155-   172        52                                                                        1.0000 0.9987\n",
      "--     173-   191        46                                                                        1.0000 0.9988\n",
      "--     192-   211        43                                                                        1.0000 0.9988\n",
      "--     212-   232        27                                                                        1.0000 0.9989\n",
      "--     233-   254        22                                                                        1.0000 0.9989\n",
      "--     255-   277        19                                                                        1.0000 0.9989\n",
      "--     278-   301        18                                                                        1.0000 0.9989\n",
      "--     302-   326        19                                                                        1.0000 0.9990\n",
      "--     327-   352        22                                                                        1.0000 0.9990\n",
      "--     353-   379        17                                                                        1.0000 0.9990\n",
      "--     380-   407        16                                                                        1.0000 0.9991\n",
      "--     408-   436        18                                                                        1.0000 0.9991\n",
      "--     437-   466        11                                                                        1.0000 0.9991\n",
      "--     467-   497         8                                                                        1.0000 0.9991\n",
      "--     498-   529         8                                                                        1.0000 0.9992\n",
      "--     530-   562        12                                                                        1.0000 0.9992\n",
      "--     563-   596         9                                                                        1.0000 0.9992\n",
      "--     597-   631         7                                                                        1.0000 0.9992\n",
      "--     632-   667         2                                                                        1.0000 0.9993\n",
      "--     668-   704         5                                                                        1.0000 0.9993\n",
      "--     705-   742         3                                                                        1.0000 0.9993\n",
      "--     743-   781         4                                                                        1.0000 0.9993\n",
      "--     782-   821         1                                                                        0.0000 0.9993\n",
      "--\n",
      "--       62861 (max occurrences)\n",
      "--    75786430 (total mers, non-unique)\n",
      "--    22805430 (distinct mers, non-unique)\n",
      "--   140339517 (unique mers)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- For mhap overlapping, set repeat k-mer threshold to 2161.\n",
      "--\n",
      "-- Found 216125947 16-mers; 163144947 distinct and 140339517 unique.  Largest count 62861.\n",
      "-- Finished stage 'cor-meryl', reset canuIteration.\n",
      "--\n",
      "-- OVERLAPPER (mhap) (correction)\n",
      "--\n",
      "-- Set corMhapSensitivity=normal based on read coverage of 46.\n",
      "--\n",
      "-- PARAMETERS: hashes=512, minMatches=3, threshold=0.78\n",
      "--\n",
      "-- Given 6 GB, can fit 18000 reads per block.\n",
      "-- For 4 blocks, set stride to 2 blocks.\n",
      "-- Logging partitioning to 'correction/1-overlapper/partitioning.log'.\n",
      "-- Configured 3 mhap precompute jobs.\n",
      "-- Configured 3 mhap overlap jobs.\n",
      "-- Finished stage 'cor-mhapConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 12:15:20 2021 with 316.934 GB free disk space (3 processes; 1 concurrently)\n",
      "\n",
      "    cd correction/1-overlapper\n",
      "    ./precompute.sh 1 > ./precompute.000001.out 2>&1\n",
      "    ./precompute.sh 2 > ./precompute.000002.out 2>&1\n",
      "    ./precompute.sh 3 > ./precompute.000003.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:28:29 2021 (789 seconds) with 315.923 GB free disk space\n",
      "----------------------------------------\n",
      "-- All 3 mhap precompute jobs finished successfully.\n",
      "-- Finished stage 'cor-mhapPrecomputeCheck', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 12:28:29 2021 with 315.923 GB free disk space (3 processes; 1 concurrently)\n",
      "\n",
      "    cd correction/1-overlapper\n",
      "    ./mhap.sh 1 > ./mhap.000001.out 2>&1\n",
      "    ./mhap.sh 2 > ./mhap.000002.out 2>&1\n",
      "    ./mhap.sh 3 > ./mhap.000003.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:29:28 2021 (59 seconds) with 315.907 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 3 mhap overlap output files.\n",
      "-- Finished stage 'cor-mhapCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:29:28 2021 with 315.907 GB free disk space\n",
      "\n",
      "    cd correction\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/ovStoreBuild \\\n",
      "     -O ./pacbio40.ovlStore.BUILDING \\\n",
      "     -G ./pacbio40.gkpStore \\\n",
      "     -M 2-8 \\\n",
      "     -L ./1-overlapper/ovljob.files \\\n",
      "     > ./pacbio40.ovlStore.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:29:29 2021 (1 second) with 315.88 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Overlap store 'correction/pacbio40.ovlStore' successfully constructed.\n",
      "--\n",
      "-- Purged 1.013 GB in 9 overlap output files and 2 directories.\n",
      "-- Overlap store 'correction/pacbio40.ovlStore' statistics not available (skipped in correction and trimming stages).\n",
      "-- Finished stage 'cor-createOverlapStore', reset canuIteration.\n",
      "-- Set corMinCoverage=4 based on read coverage of 46.\n",
      "-- Computing global filter scores 'correction/2-correction/pacbio40.globalScores'.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:29:29 2021 with 316.894 GB free disk space\n",
      "\n",
      "    cd correction/2-correction\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/filterCorrectionOverlaps \\\n",
      "      -G ../pacbio40.gkpStore \\\n",
      "      -O ../pacbio40.ovlStore \\\n",
      "      -S ./pacbio40.globalScores.WORKING \\\n",
      "      -c 40 \\\n",
      "      -l 0 \\\n",
      "    > ./pacbio40.globalScores.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:29:29 2021 (lickety-split) with 316.891 GB free disk space\n",
      "----------------------------------------\n",
      "--  PARAMETERS:\n",
      "--  ----------\n",
      "--  \n",
      "--       40 (expected coverage)\n",
      "--        0 (don't use overlaps shorter than this)\n",
      "--    0.000 (don't use overlaps with erate less than this)\n",
      "--    1.000 (don't use overlaps with erate more than this)\n",
      "--  \n",
      "--  OVERLAPS:\n",
      "--  --------\n",
      "--  \n",
      "--  IGNORED:\n",
      "--  \n",
      "--             0 (< 0.0000 fraction error)\n",
      "--             0 (> 0.4095 fraction error)\n",
      "--             0 (< 0 bases long)\n",
      "--             0 (> 2097151 bases long)\n",
      "--  \n",
      "--  FILTERED:\n",
      "--  \n",
      "--        328175 (too many overlaps, discard these shortest ones)\n",
      "--  \n",
      "--  EVIDENCE:\n",
      "--  \n",
      "--       1019617 (longest overlaps)\n",
      "--  \n",
      "--  TOTAL:\n",
      "--  \n",
      "--       1347792 (all overlaps)\n",
      "--  \n",
      "--  READS:\n",
      "--  -----\n",
      "--  \n",
      "--          1088 (no overlaps)\n",
      "--         23671 (no overlaps filtered)\n",
      "--         10882 (<  50% overlaps filtered)\n",
      "--         12921 (<  80% overlaps filtered)\n",
      "--         13069 (<  95% overlaps filtered)\n",
      "--         13069 (< 100% overlaps filtered)\n",
      "--  \n",
      "-- Computing expected corrected read lengths 'correction/2-correction/pacbio40.estimate.log'.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:29:29 2021 with 316.891 GB free disk space\n",
      "\n",
      "    cd correction/2-correction\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/generateCorrectionLayouts \\\n",
      "      -G ../pacbio40.gkpStore \\\n",
      "      -O ../pacbio40.ovlStore \\\n",
      "      -S ./pacbio40.globalScores \\\n",
      "      -c 4 \\\n",
      "      -C 80 \\\n",
      "      -p ./pacbio40.estimate.WORKING\n",
      "\n",
      "-- Finished on Wed Apr 21 12:29:30 2021 (1 second) with 316.89 GB free disk space\n",
      "----------------------------------------\n",
      "-- Sorting reads by expected corrected length.\n",
      "-- Sorting reads by uncorrected length.\n",
      "-- Loading expected corrected read lengths.\n",
      "-- Picking longest corrected reads.\n",
      "-- Writing longest corrected reads to 'correction/2-correction/pacbio40.readsToCorrect'.\n",
      "-- Summarizing filter.\n",
      "--\n",
      "-- Reads to be corrected:\n",
      "--   36740 reads longer than 3170 bp\n",
      "--   210090556 bp\n",
      "-- Expected corrected reads:\n",
      "--   36740 reads\n",
      "--   170668607 bp\n",
      "--   0 bp minimum length\n",
      "--   4645 bp mean length\n",
      "--   15436 bp n50 length\n",
      "-- Using overlaps no worse than 0.3 fraction error for correcting reads (from corErrorRate parameter).\n",
      "-- Finished stage 'cor-buildCorrectionLayouts', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 12:29:31 2021 with 316.888 GB free disk space (8 processes; 4 concurrently)\n",
      "\n",
      "    cd correction/2-correction\n",
      "    ./correctReads.sh 1 > ./correctReads.000001.out 2>&1\n",
      "    ./correctReads.sh 2 > ./correctReads.000002.out 2>&1\n",
      "    ./correctReads.sh 3 > ./correctReads.000003.out 2>&1\n",
      "    ./correctReads.sh 4 > ./correctReads.000004.out 2>&1\n",
      "    ./correctReads.sh 5 > ./correctReads.000005.out 2>&1\n",
      "    ./correctReads.sh 6 > ./correctReads.000006.out 2>&1\n",
      "    ./correctReads.sh 7 > ./correctReads.000007.out 2>&1\n",
      "    ./correctReads.sh 8 > ./correctReads.000008.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:44:27 2021 (896 seconds) with 316.654 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 8 read correction output files.\n",
      "-- Finished stage 'cor-generateCorrectedReads', reset canuIteration.\n",
      "-- Found 8 read correction output files.\n",
      "-- Finished stage 'cor-generateCorrectedReads', reset canuIteration.\n",
      "-- Concatenating correctReads output.\n",
      "-- Analyzing correctReads output.\n",
      "--\n",
      "-- Reads to be corrected:\n",
      "--   36740 reads longer than 3170 bp\n",
      "--   210090556 bp\n",
      "-- Expected corrected reads:\n",
      "--   36740 reads\n",
      "--   170668607 bp\n",
      "--   0 bp minimum length\n",
      "--   4645 bp mean length\n",
      "--   15436 bp n50 length\n",
      "-- Actual corrected reads:\n",
      "--   31021 reads\n",
      "--   154816686 bp\n",
      "--   6287 reads with 0 corrected blocks\n",
      "--   29894 reads with 1 corrected block\n",
      "--   550 reads with 2 corrected blocks\n",
      "--   9 reads with 3 corrected blocks\n",
      "--\n",
      "-- Purging correctReads output after merging to final output file.\n",
      "-- Purged 8 .dump.success sentinels.\n",
      "-- Purged 8 .fasta outputs.\n",
      "-- Purged 8 .err outputs.\n",
      "-- Purged 8 .out job log outputs.\n",
      "-- Finished stage 'cor-dumpCorrectedReads', reset canuIteration.\n",
      "--\n",
      "-- Corrected reads saved in 'pacbio40.correctedReads.fasta.gz'.\n",
      "--\n",
      "--\n",
      "-- BEGIN TRIMMING\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:44:31 2021 with 316.67 GB free disk space\n",
      "\n",
      "    cd trimming\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperCreate \\\n",
      "      -minlength 1000 \\\n",
      "      -o ./pacbio40.gkpStore.BUILDING \\\n",
      "      ./pacbio40.gkpStore.gkp \\\n",
      "    > ./pacbio40.gkpStore.BUILDING.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:44:33 2021 (2 seconds) with 316.625 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- In gatekeeper store 'trimming/pacbio40.gkpStore':\n",
      "--   Found 31021 reads.\n",
      "--   Found 154816686 bases (32.93 times coverage).\n",
      "--\n",
      "--   Read length histogram (one '*' equals 75.62 reads):\n",
      "--        0    999      0 \n",
      "--     1000   1999   5294 **********************************************************************\n",
      "--     2000   2999   4811 ***************************************************************\n",
      "--     3000   3999   4197 *******************************************************\n",
      "--     4000   4999   3738 *************************************************\n",
      "--     5000   5999   3184 ******************************************\n",
      "--     6000   6999   2567 *********************************\n",
      "--     7000   7999   2018 **************************\n",
      "--     8000   8999   1510 *******************\n",
      "--     9000   9999   1158 ***************\n",
      "--    10000  10999    831 **********\n",
      "--    11000  11999    612 ********\n",
      "--    12000  12999    454 ******\n",
      "--    13000  13999    314 ****\n",
      "--    14000  14999    188 **\n",
      "--    15000  15999    105 *\n",
      "--    16000  16999     33 \n",
      "--    17000  17999      6 \n",
      "--    18000  18999      1 \n",
      "-- Finished stage 'obt-gatekeeper', reset canuIteration.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- Finished stage 'merylConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 12:44:33 2021 with 316.624 GB free disk space (1 processes; 2 concurrently)\n",
      "\n",
      "    cd trimming/0-mercounts\n",
      "    ./meryl.sh 1 > ./meryl.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:45:09 2021 (36 seconds) with 316.528 GB free disk space\n",
      "----------------------------------------\n",
      "-- Meryl finished successfully.\n",
      "-- Finished stage 'merylCheck', reset canuIteration.\n",
      "--\n",
      "--  22-mers                                                                                           Fraction\n",
      "--    Occurrences   NumMers                                                                         Unique Total\n",
      "--       1-     1   8971785 *******************************************************************--> 0.6036 0.0582\n",
      "--       2-     2    754243 **********************************                                     0.6543 0.0680\n",
      "--       3-     4    366412 ****************                                                       0.6710 0.0728\n",
      "--       5-     7    148845 ******                                                                 0.6836 0.0781\n",
      "--       8-    11     86779 ****                                                                   0.6907 0.0828\n",
      "--      12-    16    160647 *******                                                                0.6963 0.0884\n",
      "--      17-    22    591188 ***************************                                            0.7095 0.1081\n",
      "--      23-    29   1371994 ***************************************************************        0.7561 0.2022\n",
      "--      30-    37   1509499 ********************************************************************** 0.8526 0.4548\n",
      "--      38-    46    734990 **********************************                                     0.9480 0.7686\n",
      "--      47-    56    130828 ******                                                                 0.9908 0.9416\n",
      "--      57-    67     12281                                                                        0.9977 0.9755\n",
      "--      68-    79      6423                                                                        0.9984 0.9795\n",
      "--      80-    92      4648                                                                        0.9988 0.9825\n",
      "--      93-   106      2665                                                                        0.9991 0.9851\n",
      "--     107-   121      2060                                                                        0.9993 0.9867\n",
      "--     122-   137      1750                                                                        0.9994 0.9882\n",
      "--     138-   154      1663                                                                        0.9995 0.9897\n",
      "--     155-   172      1016                                                                        0.9996 0.9913\n",
      "--     173-   191      1182                                                                        0.9997 0.9923\n",
      "--     192-   211       651                                                                        0.9998 0.9937\n",
      "--     212-   232       429                                                                        0.9998 0.9945\n",
      "--     233-   254       429                                                                        0.9999 0.9951\n",
      "--     255-   277       396                                                                        0.9999 0.9958\n",
      "--     278-   301       141                                                                        0.9999 0.9965\n",
      "--     302-   326       148                                                                        0.9999 0.9967\n",
      "--     327-   352       242                                                                        0.9999 0.9970\n",
      "--     353-   379       502                                                                        0.9999 0.9976\n",
      "--     380-   407       257                                                                        1.0000 0.9987\n",
      "--     408-   436        46                                                                        1.0000 0.9994\n",
      "--     437-   466         4                                                                        1.0000 0.9995\n",
      "--     467-   497         5                                                                        1.0000 0.9995\n",
      "--     498-   529         2                                                                        1.0000 0.9995\n",
      "--     530-   562         0                                                                        0.0000 0.0000\n",
      "--     563-   596         1                                                                        1.0000 0.9995\n",
      "--     597-   631         2                                                                        1.0000 0.9995\n",
      "--     632-   667         5                                                                        1.0000 0.9996\n",
      "--     668-   704         1                                                                        1.0000 0.9996\n",
      "--     705-   742         1                                                                        1.0000 0.9996\n",
      "--     743-   781         0                                                                        0.0000 0.0000\n",
      "--     782-   821         5                                                                        1.0000 0.9996\n",
      "--\n",
      "--        2311 (max occurrences)\n",
      "--   145193460 (total mers, non-unique)\n",
      "--     5892421 (distinct mers, non-unique)\n",
      "--     8971785 (unique mers)\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:45:10 2021 with 316.528 GB free disk space\n",
      "\n",
      "    cd trimming/0-mercounts\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/meryl \\\n",
      "      -Dt \\\n",
      "      -n 528 \\\n",
      "      -s ./pacbio40.ms22 \\\n",
      "    > ./pacbio40.ms22.frequentMers.fasta \\\n",
      "    2> ./pacbio40.ms22.frequentMers.fasta.err\n",
      "\n",
      "-- Finished on Wed Apr 21 12:45:10 2021 (lickety-split) with 316.528 GB free disk space\n",
      "----------------------------------------\n",
      "-- Reset obtOvlMerThreshold from auto to 528.\n",
      "--\n",
      "-- Found 154165245 22-mers; 14864206 distinct and 8971785 unique.  Largest count 2311.\n",
      "-- Finished stage 'obt-meryl', reset canuIteration.\n",
      "--\n",
      "-- OVERLAPPER (normal) (trimming) erate=0.045\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:45:10 2021 with 316.624 GB free disk space\n",
      "\n",
      "    cd trimming/1-overlapper\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/overlapInCorePartition \\\n",
      "     -g  ../pacbio40.gkpStore \\\n",
      "     -bl 100000000 \\\n",
      "     -bs 0 \\\n",
      "     -rs 2000000 \\\n",
      "     -rl 0 \\\n",
      "     -ol 500 \\\n",
      "     -o  ./pacbio40.partition \\\n",
      "    > ./pacbio40.partition.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:45:10 2021 (lickety-split) with 316.624 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Configured 2 overlapInCore jobs.\n",
      "-- Finished stage 'obt-overlapConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 12:45:10 2021 with 316.624 GB free disk space (2 processes; 1 concurrently)\n",
      "\n",
      "    cd trimming/1-overlapper\n",
      "    ./overlap.sh 1 > ./overlap.000001.out 2>&1\n",
      "    ./overlap.sh 2 > ./overlap.000002.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:52:26 2021 (436 seconds) with 316.594 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 2 overlapInCore output files.\n",
      "--\n",
      "-- overlapInCore compute 'trimming/1-overlapper':\n",
      "--   kmer hits\n",
      "--     with no overlap          2076865   1038432.5 +- 255209.686\n",
      "--     with an overlap          1130520      565260 +- 134832.535\n",
      "--\n",
      "--   overlaps                   1130520      565260 +- 134832.535\n",
      "--     contained                      0           0 +- 0\n",
      "--     dovetail                       0           0 +- 0\n",
      "--\n",
      "--   overlaps rejected\n",
      "--     multiple per pair              0           0 +- 0\n",
      "--     bad short window               0           0 +- 0\n",
      "--     bad long window                0           0 +- 0\n",
      "-- Finished stage 'obt-overlapCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:52:26 2021 with 316.594 GB free disk space\n",
      "\n",
      "    cd trimming\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/ovStoreBuild \\\n",
      "     -O ./pacbio40.ovlStore.BUILDING \\\n",
      "     -G ./pacbio40.gkpStore \\\n",
      "     -M 2-8 \\\n",
      "     -L ./1-overlapper/ovljob.files \\\n",
      "     > ./pacbio40.ovlStore.err 2>&1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "-- Finished on Wed Apr 21 12:52:27 2021 (1 second) with 316.551 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Overlap store 'trimming/pacbio40.ovlStore' successfully constructed.\n",
      "--\n",
      "-- Purged 0.021 GB in 6 overlap output files and 1 directories.\n",
      "-- Overlap store 'trimming/pacbio40.ovlStore' statistics not available (skipped in correction and trimming stages).\n",
      "-- Finished stage 'obt-createOverlapStore', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:52:27 2021 with 316.574 GB free disk space\n",
      "\n",
      "    cd trimming/3-overlapbasedtrimming\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/trimReads \\\n",
      "      -G  ../pacbio40.gkpStore \\\n",
      "      -O  ../pacbio40.ovlStore \\\n",
      "      -Co ./pacbio40.1.trimReads.clear \\\n",
      "      -e  0.045 \\\n",
      "      -minlength 1000 \\\n",
      "      -ol 1 \\\n",
      "      -oc 1 \\\n",
      "      -o  ./pacbio40.1.trimReads \\\n",
      "    >     ./pacbio40.1.trimReads.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:52:29 2021 (2 seconds) with 316.571 GB free disk space\n",
      "----------------------------------------\n",
      "--  PARAMETERS:\n",
      "--  ----------\n",
      "--     1000    (reads trimmed below this many bases are deleted)\n",
      "--   0.0450    (use overlaps at or below this fraction error)\n",
      "--        1    (break region if overlap is less than this long, for 'largest covered' algorithm)\n",
      "--        1    (break region if overlap coverage is less than this many read, for 'largest covered' algorithm)\n",
      "--  \n",
      "--  INPUT READS:\n",
      "--  -----------\n",
      "--   31021 reads    154816686 bases (reads processed)\n",
      "--       0 reads            0 bases (reads not processed, previously deleted)\n",
      "--       0 reads            0 bases (reads not processed, in a library where trimming isn't allowed)\n",
      "--  \n",
      "--  OUTPUT READS:\n",
      "--  ------------\n",
      "--   25682 reads    130788934 bases (trimmed reads output)\n",
      "--    5216 reads     21587998 bases (reads with no change, kept as is)\n",
      "--       3 reads         3946 bases (reads with no overlaps, deleted)\n",
      "--     120 reads       138273 bases (reads with short trimmed length, deleted)\n",
      "--  \n",
      "--  TRIMMING DETAILS:\n",
      "--  ----------------\n",
      "--   16715 reads      1542733 bases (bases trimmed from the 5' end of a read)\n",
      "--   19829 reads       754802 bases (bases trimmed from the 3' end of a read)\n",
      "-- Finished stage 'obt-trimReads', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:52:29 2021 with 316.571 GB free disk space\n",
      "\n",
      "    cd trimming/3-overlapbasedtrimming\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/splitReads \\\n",
      "      -G  ../pacbio40.gkpStore \\\n",
      "      -O  ../pacbio40.ovlStore \\\n",
      "      -Ci ./pacbio40.1.trimReads.clear \\\n",
      "      -Co ./pacbio40.2.splitReads.clear \\\n",
      "      -e  0.045 \\\n",
      "      -minlength 1000 \\\n",
      "      -o  ./pacbio40.2.splitReads \\\n",
      "    >     ./pacbio40.2.splitReads.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:52:29 2021 (lickety-split) with 316.571 GB free disk space\n",
      "----------------------------------------\n",
      "--  PARAMETERS:\n",
      "--  ----------\n",
      "--     1000    (reads trimmed below this many bases are deleted)\n",
      "--   0.0450    (use overlaps at or below this fraction error)\n",
      "--  INPUT READS:\n",
      "--  -----------\n",
      "--   30898 reads    154674467 bases (reads processed)\n",
      "--     123 reads       142219 bases (reads not processed, previously deleted)\n",
      "--       0 reads            0 bases (reads not processed, in a library where trimming isn't allowed)\n",
      "--  \n",
      "--  PROCESSED:\n",
      "--  --------\n",
      "--       0 reads            0 bases (no overlaps)\n",
      "--       0 reads            0 bases (no coverage after adjusting for trimming done already)\n",
      "--       0 reads            0 bases (processed for chimera)\n",
      "--       0 reads            0 bases (processed for spur)\n",
      "--   30898 reads    154674467 bases (processed for subreads)\n",
      "--  \n",
      "--  READS WITH SIGNALS:\n",
      "--  ------------------\n",
      "--       0 reads            0 signals (number of 5' spur signal)\n",
      "--       0 reads            0 signals (number of 3' spur signal)\n",
      "--       0 reads            0 signals (number of chimera signal)\n",
      "--      32 reads           34 signals (number of subread signal)\n",
      "--  \n",
      "--  SIGNALS:\n",
      "--  -------\n",
      "--       0 reads            0 bases (size of 5' spur signal)\n",
      "--       0 reads            0 bases (size of 3' spur signal)\n",
      "--       0 reads            0 bases (size of chimera signal)\n",
      "--      34 reads        10474 bases (size of subread signal)\n",
      "--  \n",
      "--  TRIMMING:\n",
      "--  --------\n",
      "--      19 reads        27223 bases (trimmed from the 5' end of the read)\n",
      "--      13 reads        29678 bases (trimmed from the 3' end of the read)\n",
      "-- Finished stage 'obt-splitReads', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:52:29 2021 with 316.571 GB free disk space\n",
      "\n",
      "    cd trimming\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperDumpFASTQ -fasta -nolibname \\\n",
      "      -G ./pacbio40.gkpStore \\\n",
      "      -c ./3-overlapbasedtrimming/pacbio40.2.splitReads.clear \\\n",
      "      -o ../pacbio40.trimmedReads.gz \\\n",
      "    >    ../pacbio40.trimmedReads.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:52:32 2021 (3 seconds) with 316.519 GB free disk space\n",
      "----------------------------------------\n",
      "-- Finished stage 'obt-dumpReads', reset canuIteration.\n",
      "--\n",
      "-- Trimmed reads saved in 'trimming/pacbio40.trimmedReads.fasta.gz'\n",
      "--\n",
      "--\n",
      "-- BEGIN ASSEMBLY\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:52:32 2021 with 316.519 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperCreate \\\n",
      "      -minlength 1000 \\\n",
      "      -o ./pacbio40.gkpStore.BUILDING \\\n",
      "      ./pacbio40.gkpStore.gkp \\\n",
      "    > ./pacbio40.gkpStore.BUILDING.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:52:33 2021 (1 second) with 316.474 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- In gatekeeper store 'unitigging/pacbio40.gkpStore':\n",
      "--   Found 30896 reads.\n",
      "--   Found 152318325 bases (32.4 times coverage).\n",
      "--\n",
      "--   Read length histogram (one '*' equals 77.17 reads):\n",
      "--        0    999      0 \n",
      "--     1000   1999   5402 **********************************************************************\n",
      "--     2000   2999   4883 ***************************************************************\n",
      "--     3000   3999   4218 ******************************************************\n",
      "--     4000   4999   3724 ************************************************\n",
      "--     5000   5999   3140 ****************************************\n",
      "--     6000   6999   2542 ********************************\n",
      "--     7000   7999   1948 *************************\n",
      "--     8000   8999   1470 *******************\n",
      "--     9000   9999   1113 **************\n",
      "--    10000  10999    809 **********\n",
      "--    11000  11999    608 *******\n",
      "--    12000  12999    426 *****\n",
      "--    13000  13999    299 ***\n",
      "--    14000  14999    181 **\n",
      "--    15000  15999     96 *\n",
      "--    16000  16999     31 \n",
      "--    17000  17999      5 \n",
      "--    18000  18999      1 \n",
      "-- Finished stage 'utg-gatekeeper', reset canuIteration.\n",
      "-- Finished stage 'merylConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 12:52:34 2021 with 316.473 GB free disk space (1 processes; 2 concurrently)\n",
      "\n",
      "    cd unitigging/0-mercounts\n",
      "    ./meryl.sh 1 > ./meryl.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:53:10 2021 (36 seconds) with 316.397 GB free disk space\n",
      "----------------------------------------\n",
      "-- Meryl finished successfully.\n",
      "-- Finished stage 'merylCheck', reset canuIteration.\n",
      "--\n",
      "--  22-mers                                                                                           Fraction\n",
      "--    Occurrences   NumMers                                                                         Unique Total\n",
      "--       1-     1   7420453 *******************************************************************--> 0.5590 0.0489\n",
      "--       2-     2    728524 *********************************                                      0.6139 0.0585\n",
      "--       3-     4    357728 ****************                                                       0.6321 0.0633\n",
      "--       5-     7    147171 ******                                                                 0.6460 0.0686\n",
      "--       8-    11     86817 ****                                                                   0.6538 0.0733\n",
      "--      12-    16    165437 *******                                                                0.6601 0.0791\n",
      "--      17-    22    605939 ****************************                                           0.6753 0.0996\n",
      "--      23-    29   1390213 ****************************************************************       0.7287 0.1974\n",
      "--      30-    37   1500488 ********************************************************************** 0.8379 0.4570\n",
      "--      38-    46    713397 *********************************                                      0.9438 0.7731\n",
      "--      47-    56    122303 *****                                                                  0.9902 0.9433\n",
      "--      57-    67     12222                                                                        0.9975 0.9754\n",
      "--      68-    79      6167                                                                        0.9982 0.9795\n",
      "--      80-    92      4773                                                                        0.9987 0.9824\n",
      "--      93-   106      2524                                                                        0.9990 0.9850\n",
      "--     107-   121      2141                                                                        0.9992 0.9866\n",
      "--     122-   137      1683                                                                        0.9994 0.9882\n",
      "--     138-   154      1631                                                                        0.9995 0.9897\n",
      "--     155-   172       952                                                                        0.9996 0.9912\n",
      "--     173-   191      1170                                                                        0.9997 0.9922\n",
      "--     192-   211       650                                                                        0.9998 0.9936\n",
      "--     212-   232       439                                                                        0.9998 0.9944\n",
      "--     233-   254       442                                                                        0.9998 0.9951\n",
      "--     255-   277       387                                                                        0.9999 0.9958\n",
      "--     278-   301       135                                                                        0.9999 0.9965\n",
      "--     302-   326       139                                                                        0.9999 0.9967\n",
      "--     327-   352       255                                                                        0.9999 0.9970\n",
      "--     353-   379       488                                                                        0.9999 0.9976\n",
      "--     380-   407       267                                                                        1.0000 0.9987\n",
      "--     408-   436        33                                                                        1.0000 0.9994\n",
      "--     437-   466         6                                                                        1.0000 0.9995\n",
      "--     467-   497         2                                                                        1.0000 0.9995\n",
      "--     498-   529         2                                                                        1.0000 0.9995\n",
      "--     530-   562         0                                                                        0.0000 0.0000\n",
      "--     563-   596         1                                                                        1.0000 0.9995\n",
      "--     597-   631         3                                                                        1.0000 0.9995\n",
      "--     632-   667         4                                                                        1.0000 0.9996\n",
      "--     668-   704         2                                                                        1.0000 0.9996\n",
      "--     705-   742         0                                                                        0.0000 0.0000\n",
      "--     743-   781         0                                                                        0.0000 0.0000\n",
      "--     782-   821         5                                                                        1.0000 0.9996\n",
      "--\n",
      "--        2292 (max occurrences)\n",
      "--   144249056 (total mers, non-unique)\n",
      "--     5854581 (distinct mers, non-unique)\n",
      "--     7420453 (unique mers)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:53:10 2021 with 316.397 GB free disk space\n",
      "\n",
      "    cd unitigging/0-mercounts\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/meryl \\\n",
      "      -Dt \\\n",
      "      -n 527 \\\n",
      "      -s ./pacbio40.ms22 \\\n",
      "    > ./pacbio40.ms22.frequentMers.fasta \\\n",
      "    2> ./pacbio40.ms22.frequentMers.fasta.err\n",
      "\n",
      "-- Finished on Wed Apr 21 12:53:10 2021 (lickety-split) with 316.397 GB free disk space\n",
      "----------------------------------------\n",
      "-- Reset utgOvlMerThreshold from auto to 527.\n",
      "--\n",
      "-- Found 151669509 22-mers; 13275034 distinct and 7420453 unique.  Largest count 2292.\n",
      "-- Finished stage 'utg-meryl', reset canuIteration.\n",
      "--\n",
      "-- OVERLAPPER (normal) (assembly) erate=0.045\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 12:53:10 2021 with 316.477 GB free disk space\n",
      "\n",
      "    cd unitigging/1-overlapper\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/overlapInCorePartition \\\n",
      "     -g  ../pacbio40.gkpStore \\\n",
      "     -bl 100000000 \\\n",
      "     -bs 0 \\\n",
      "     -rs 2000000 \\\n",
      "     -rl 0 \\\n",
      "     -ol 500 \\\n",
      "     -o  ./pacbio40.partition \\\n",
      "    > ./pacbio40.partition.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 12:53:10 2021 (lickety-split) with 316.477 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Configured 2 overlapInCore jobs.\n",
      "-- Finished stage 'utg-overlapConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 12:53:11 2021 with 316.477 GB free disk space (2 processes; 1 concurrently)\n",
      "\n",
      "    cd unitigging/1-overlapper\n",
      "    ./overlap.sh 1 > ./overlap.000001.out 2>&1\n",
      "    ./overlap.sh 2 > ./overlap.000002.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:00:50 2021 (459 seconds) with 316.464 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 2 overlapInCore output files.\n",
      "--\n",
      "-- overlapInCore compute 'unitigging/1-overlapper':\n",
      "--   kmer hits\n",
      "--     with no overlap          2220480     1110240 +- 240931.079\n",
      "--     with an overlap           932589    466294.5 +- 91556.893\n",
      "--\n",
      "--   overlaps                    932589    466294.5 +- 91556.893\n",
      "--     contained                 356183    178091.5 +- 35887.79\n",
      "--     dovetail                  576406      288203 +- 55669.102\n",
      "--\n",
      "--   overlaps rejected\n",
      "--     multiple per pair              0           0 +- 0\n",
      "--     bad short window               0           0 +- 0\n",
      "--     bad long window                0           0 +- 0\n",
      "-- Finished stage 'utg-overlapCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:00:50 2021 with 316.464 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/ovStoreBuild \\\n",
      "     -O ./pacbio40.ovlStore.BUILDING \\\n",
      "     -G ./pacbio40.gkpStore \\\n",
      "     -M 2-8 \\\n",
      "     -L ./1-overlapper/ovljob.files \\\n",
      "     > ./pacbio40.ovlStore.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:00:51 2021 (1 second) with 316.428 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Overlap store 'unitigging/pacbio40.ovlStore' successfully constructed.\n",
      "--\n",
      "-- Purged 0.016 GB in 6 overlap output files and 1 directories.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:00:51 2021 with 316.446 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/ovStoreStats \\\n",
      "     -G ./pacbio40.gkpStore \\\n",
      "     -O ./pacbio40.ovlStore \\\n",
      "     -o ./pacbio40.ovlStore \\\n",
      "     > ./pacbio40.ovlStore.summary.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:00:51 2021 (lickety-split) with 316.445 GB free disk space\n",
      "----------------------------------------\n",
      "-- category            reads     %          read length        feature size or coverage  analysis\n",
      "-- ----------------  -------  -------  ----------------------  ------------------------  --------------------\n",
      "-- middle-missing         14    0.05     6019.07 +- 2080.36        281.50 +- 368.44     (bad trimming)\n",
      "-- middle-hump             0    0.00        0.00 +- 0.00             0.00 +- 0.00       (bad trimming)\n",
      "-- no-5-prime              1    0.00     1702.00 +- 0.00           560.00 +- 0.00       (bad trimming)\n",
      "-- no-3-prime              0    0.00        0.00 +- 0.00             0.00 +- 0.00       (bad trimming)\n",
      "-- \n",
      "-- low-coverage            6    0.02     1806.33 +- 291.65           4.88 +- 1.66       (easy to assemble, potential for lower quality consensus)\n",
      "-- unique              28013   90.67     4790.84 +- 3027.54         31.97 +- 6.96       (easy to assemble, perfect, yay)\n",
      "-- repeat-cont            70    0.23     1975.03 +- 857.70          86.31 +- 51.31      (potential for consensus errors, no impact on assembly)\n",
      "-- repeat-dove             0    0.00        0.00 +- 0.00             0.00 +- 0.00       (hard to assemble, likely won't assemble correctly or even at all)\n",
      "-- \n",
      "-- span-repeat          2252    7.29     6581.43 +- 3319.39       1698.59 +- 1935.78    (read spans a large repeat, usually easy to assemble)\n",
      "-- uniq-repeat-cont      492    1.59     5165.35 +- 2646.76                             (should be uniquely placed, low potential for consensus errors, no impact on assembly)\n",
      "-- uniq-repeat-dove       44    0.14    11289.25 +- 2882.42                             (will end contigs, potential to misassemble)\n",
      "-- uniq-anchor             4    0.01     4533.00 +- 3269.81       1376.00 +- 2470.11    (repeat read, with unique section, probable bad read)\n",
      "--\n",
      "-- Overlap store 'unitigging/pacbio40.ovlStore' contains:\n",
      "--\n",
      "--   category            reads     %          read length        feature size or coverage  analysis\n",
      "--   ----------------  -------  -------  ----------------------  ------------------------  --------------------\n",
      "--   middle-missing         14    0.05     6019.07 +- 2080.36        281.50 +- 368.44     (bad trimming)\n",
      "--   middle-hump             0    0.00        0.00 +- 0.00             0.00 +- 0.00       (bad trimming)\n",
      "--   no-5-prime              1    0.00     1702.00 +- 0.00           560.00 +- 0.00       (bad trimming)\n",
      "--   no-3-prime              0    0.00        0.00 +- 0.00             0.00 +- 0.00       (bad trimming)\n",
      "--   \n",
      "--   low-coverage            6    0.02     1806.33 +- 291.65           4.88 +- 1.66       (easy to assemble, potential for lower quality consensus)\n",
      "--   unique              28013   90.67     4790.84 +- 3027.54         31.97 +- 6.96       (easy to assemble, perfect, yay)\n",
      "--   repeat-cont            70    0.23     1975.03 +- 857.70          86.31 +- 51.31      (potential for consensus errors, no impact on assembly)\n",
      "--   repeat-dove             0    0.00        0.00 +- 0.00             0.00 +- 0.00       (hard to assemble, likely won't assemble correctly or even at all)\n",
      "--   \n",
      "--   span-repeat          2252    7.29     6581.43 +- 3319.39       1698.59 +- 1935.78    (read spans a large repeat, usually easy to assemble)\n",
      "--   uniq-repeat-cont      492    1.59     5165.35 +- 2646.76                             (should be uniquely placed, low potential for consensus errors, no impact on assembly)\n",
      "--   uniq-repeat-dove       44    0.14    11289.25 +- 2882.42                             (will end contigs, potential to misassemble)\n",
      "--   uniq-anchor             4    0.01     4533.00 +- 3269.81       1376.00 +- 2470.11    (repeat read, with unique section, probable bad read)\n",
      "-- Finished stage 'utg-createOverlapStore', reset canuIteration.\n",
      "\n",
      "Configure RED for 2gb memory with batches of at most 5000 reads and (unlimited) bases.\n",
      "\n",
      "RED job   1 from read         1 to read      4118 -   1.501 GB for    4118 reads -   0.253 GB for  20885935 olaps -   0.003 GB for evidence\n",
      "RED job   2 from read      4119 to read      8399 -   1.500 GB for    4281 reads -   0.253 GB for  20880313 olaps -   0.003 GB for evidence\n",
      "RED job   3 from read      8400 to read     12679 -   1.500 GB for    4280 reads -   0.253 GB for  20878241 olaps -   0.003 GB for evidence\n",
      "RED job   4 from read     12680 to read     16974 -   1.500 GB for    4295 reads -   0.253 GB for  20880565 olaps -   0.003 GB for evidence\n",
      "RED job   5 from read     16975 to read     21168 -   1.500 GB for    4194 reads -   0.253 GB for  20879605 olaps -   0.003 GB for evidence\n",
      "RED job   6 from read     21169 to read     25418 -   1.500 GB for    4250 reads -   0.253 GB for  20879568 olaps -   0.003 GB for evidence\n",
      "RED job   7 from read     25419 to read     29661 -   1.501 GB for    4243 reads -   0.253 GB for  20886508 olaps -   0.003 GB for evidence\n",
      "RED job   8 from read     29662 to read     30896 -   0.442 GB for    1235 reads -   0.074 GB for   6147590 olaps -   0.001 GB for evidence\n",
      "-- Finished stage 'readErrorDetectionConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 13:00:51 2021 with 316.445 GB free disk space (8 processes; 2 concurrently)\n",
      "\n",
      "    cd unitigging/3-overlapErrorAdjustment\n",
      "    ./red.sh 1 > ./red.000001.out 2>&1\n",
      "    ./red.sh 2 > ./red.000002.out 2>&1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    ./red.sh 3 > ./red.000003.out 2>&1\n",
      "    ./red.sh 4 > ./red.000004.out 2>&1\n",
      "    ./red.sh 5 > ./red.000005.out 2>&1\n",
      "    ./red.sh 6 > ./red.000006.out 2>&1\n",
      "    ./red.sh 7 > ./red.000007.out 2>&1\n",
      "    ./red.sh 8 > ./red.000008.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:01:11 2021 (20 seconds) with 316.443 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 8 read error detection output files.\n",
      "-- Finished stage 'readErrorDetectionCheck', reset canuIteration.\n",
      "\n",
      "Configure OEA for 1gb memory with batches of at most 25000 reads and (unlimited) bases.\n",
      "Configured 2 jobs.\n",
      "\n",
      "OEA job   1 from read         1 to read     25000 - 117.4 bases +  5.2 adjusts +  0.8 reads + 46.0 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 497.4 MB\n",
      "OEA job   2 from read     25001 to read     30896 - 27.8 bases +  5.2 adjusts +  0.2 reads + 10.9 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 372.1 MB\n",
      "\n",
      "-- Finished stage 'overlapErrorAdjustmentConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 13:01:11 2021 with 316.443 GB free disk space (2 processes; 8 concurrently)\n",
      "\n",
      "    cd unitigging/3-overlapErrorAdjustment\n",
      "    ./oea.sh 1 > ./oea.000001.out 2>&1\n",
      "    ./oea.sh 2 > ./oea.000002.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:01:27 2021 (16 seconds) with 316.44 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 2 overlap error adjustment output files.\n",
      "-- Finished stage 'overlapErrorAdjustmentCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:01:27 2021 with 316.44 GB free disk space\n",
      "\n",
      "    cd unitigging/3-overlapErrorAdjustment\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/ovStoreBuild \\\n",
      "      -G ../pacbio40.gkpStore \\\n",
      "      -O ../pacbio40.ovlStore \\\n",
      "      -evalues \\\n",
      "      -L ./oea.files \\\n",
      "    > ./oea.apply.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:01:27 2021 (lickety-split) with 316.436 GB free disk space\n",
      "----------------------------------------\n",
      "-- No report available.\n",
      "-- Finished stage 'updateOverlapStore', reset canuIteration.\n",
      "-- Finished stage 'unitig', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 13:01:27 2021 with 316.436 GB free disk space (1 processes; 2 concurrently)\n",
      "\n",
      "    cd unitigging/4-unitigger\n",
      "    ./unitigger.sh 1 > ./unitigger.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:01:30 2021 (3 seconds) with 316.42 GB free disk space\n",
      "----------------------------------------\n",
      "-- Unitigger finished successfully.\n",
      "-- Found, in version 1, after unitig construction:\n",
      "--   contigs:      1 sequences, total length 4632247 bp (including 0 repeats of total length 0 bp).\n",
      "--   bubbles:      0 sequences, total length 0 bp.\n",
      "--   unassembled:  1929 sequences, total length 6093088 bp.\n",
      "--\n",
      "-- Contig sizes based on genome size --\n",
      "--            NG (bp)  LG (contigs)    sum (bp)\n",
      "--         ----------  ------------  ----------\n",
      "--     10     4632247             1     4632247\n",
      "--     20     4632247             1     4632247\n",
      "--     30     4632247             1     4632247\n",
      "--     40     4632247             1     4632247\n",
      "--     50     4632247             1     4632247\n",
      "--     60     4632247             1     4632247\n",
      "--     70     4632247             1     4632247\n",
      "--     80     4632247             1     4632247\n",
      "--     90     4632247             1     4632247\n",
      "--\n",
      "-- Finished stage 'unitigCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:01:30 2021 with 316.42 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperPartition \\\n",
      "      -G ./pacbio40.gkpStore \\\n",
      "      -T ./pacbio40.ctgStore 1 \\\n",
      "      -b 15000 \\\n",
      "      -p 8 \\\n",
      "    > ./pacbio40.ctgStore/partitionedReads.log 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:01:30 2021 (lickety-split) with 316.379 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:01:30 2021 with 316.379 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperPartition \\\n",
      "      -G ./pacbio40.gkpStore \\\n",
      "      -T ./pacbio40.utgStore 1 \\\n",
      "      -b 15000 \\\n",
      "      -p 8 \\\n",
      "    > ./pacbio40.utgStore/partitionedReads.log 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:01:30 2021 (lickety-split) with 316.339 GB free disk space\n",
      "----------------------------------------\n",
      "-- Using fast alignment for consensus (iteration '0').\n",
      "-- Configured 2 contig and 1 unitig consensus jobs.\n",
      "-- Finished stage 'consensusConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 13:01:30 2021 with 316.339 GB free disk space (3 processes; 2 concurrently)\n",
      "\n",
      "    cd unitigging/5-consensus\n",
      "    ./consensus.sh 1 > ./consensus.000001.out 2>&1\n",
      "    ./consensus.sh 2 > ./consensus.000002.out 2>&1\n",
      "    ./consensus.sh 3 > ./consensus.000003.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:06:33 2021 (303 seconds) with 316.295 GB free disk space\n",
      "----------------------------------------\n",
      "-- Using slow alignment for consensus (iteration '1').\n",
      "-- Configured 2 contig and 1 unitig consensus jobs.\n",
      "-- All 3 consensus jobs finished successfully.\n",
      "-- Finished stage 'consensusCheck', reset canuIteration.\n",
      "-- Using fast alignment for consensus (iteration '0').\n",
      "-- Configured 2 contig and 1 unitig consensus jobs.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:06:33 2021 with 316.295 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreLoad \\\n",
      "      -G ./pacbio40.gkpStore \\\n",
      "      -T ./pacbio40.ctgStore 2 \\\n",
      "      -L ./5-consensus/ctgcns.files \\\n",
      "    > ./5-consensus/ctgcns.files.ctgStoreLoad.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:06:34 2021 (1 second) with 316.273 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:06:34 2021 with 316.273 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreLoad \\\n",
      "      -G ./pacbio40.gkpStore \\\n",
      "      -T ./pacbio40.utgStore 2 \\\n",
      "      -L ./5-consensus/utgcns.files \\\n",
      "    > ./5-consensus/utgcns.files.utgStoreLoad.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:06:34 2021 (lickety-split) with 316.263 GB free disk space\n",
      "----------------------------------------\n",
      "-- Purging consensus output after loading to ctgStore and/or utgStore.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:06:34 2021 with 316.263 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./pacbio40.gkpStore \\\n",
      "      -T ./pacbio40.ctgStore 2 \\\n",
      "      -sizes -s 4700000 \\\n",
      "    > ./pacbio40.ctgStore/seqDB.v002.sizes.txt\n",
      "\n",
      "-- Finished on Wed Apr 21 13:06:34 2021 (lickety-split) with 316.263 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found, in version 2, after consensus generation:\n",
      "--   contigs:      1 sequences, total length 4641141 bp (including 0 repeats of total length 0 bp).\n",
      "--   bubbles:      0 sequences, total length 0 bp.\n",
      "--   unassembled:  1929 sequences, total length 6093090 bp.\n",
      "--\n",
      "-- Contig sizes based on genome size --\n",
      "--            NG (bp)  LG (contigs)    sum (bp)\n",
      "--         ----------  ------------  ----------\n",
      "--     10     4641141             1     4641141\n",
      "--     20     4641141             1     4641141\n",
      "--     30     4641141             1     4641141\n",
      "--     40     4641141             1     4641141\n",
      "--     50     4641141             1     4641141\n",
      "--     60     4641141             1     4641141\n",
      "--     70     4641141             1     4641141\n",
      "--     80     4641141             1     4641141\n",
      "--     90     4641141             1     4641141\n",
      "--\n",
      "-- Finished stage 'consensusLoad', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:06:34 2021 with 316.263 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreCoverageStat \\\n",
      "      -G ./pacbio40.gkpStore \\\n",
      "      -T ./pacbio40.ctgStore 2 \\\n",
      "      -s 4700000 \\\n",
      "      -o ./pacbio40.ctgStore.coverageStat \\\n",
      "    > ./pacbio40.ctgStore.coverageStat.err 2>&1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "-- Finished on Wed Apr 21 13:06:34 2021 (lickety-split) with 316.263 GB free disk space\n",
      "----------------------------------------\n",
      "-- Finished stage 'consensusAnalyze', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:06:34 2021 with 316.263 GB free disk space\n",
      "\n",
      "    cd unitigging/4-unitigger\n",
      "    ./alignGFA.sh\n",
      "GFA alignments updated.\n",
      "\n",
      "-- Finished on Wed Apr 21 13:06:34 2021 (lickety-split) with 316.263 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:06:34 2021 with 316.263 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./unitigging/pacbio40.gkpStore \\\n",
      "      -T ./unitigging/pacbio40.ctgStore 2 \\\n",
      "      -o ./pacbio40.contigs \\\n",
      "      -layout \\\n",
      "    > ./pacbio40.contigs.layout.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:06:35 2021 (1 second) with 316.24 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:06:35 2021 with 316.24 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./unitigging/pacbio40.gkpStore \\\n",
      "      -T ./unitigging/pacbio40.utgStore 2 \\\n",
      "      -o ./pacbio40.unitigs \\\n",
      "      -layout \\\n",
      "    > ./pacbio40.unitigs.layout.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:06:35 2021 (lickety-split) with 316.228 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:06:35 2021 with 316.228 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./unitigging/pacbio40.gkpStore \\\n",
      "      -T ./unitigging/pacbio40.ctgStore 2 \\\n",
      "      -consensus -fasta \\\n",
      "      -unassembled \\\n",
      "    > ./pacbio40.unassembled.fasta\n",
      "    2> ./pacbio40.unassembled.err\n",
      "\n",
      "-- Finished on Wed Apr 21 13:06:35 2021 (lickety-split) with 316.222 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:06:35 2021 with 316.222 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./unitigging/pacbio40.gkpStore \\\n",
      "      -T ./unitigging/pacbio40.ctgStore 2 \\\n",
      "      -consensus -fasta \\\n",
      "      -contigs \\\n",
      "    > ./pacbio40.contigs.fasta\n",
      "    2> ./pacbio40.contigs.err\n",
      "\n",
      "-- Finished on Wed Apr 21 13:06:35 2021 (lickety-split) with 316.218 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:06:35 2021 with 316.218 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./unitigging/pacbio40.gkpStore \\\n",
      "      -T ./unitigging/pacbio40.utgStore 2 \\\n",
      "      -consensus -fasta \\\n",
      "      -contigs \\\n",
      "    > ./pacbio40.unitigs.fasta\n",
      "    2> ./pacbio40.unitigs.err\n",
      "\n",
      "-- Finished on Wed Apr 21 13:06:35 2021 (lickety-split) with 316.213 GB free disk space\n",
      "----------------------------------------\n",
      "-- Finished stage 'generateOutputs', reset canuIteration.\n",
      "--\n",
      "-- Assembly 'pacbio40' finished in '/Users/olga/Documents/BioinformaticsInstitute/NGS/pacbio40'.\n",
      "--\n",
      "-- Summary saved in 'unitigging.html'.\n",
      "--\n",
      "-- Sequences saved:\n",
      "--   Contigs       -> 'pacbio40.contigs.fasta'\n",
      "--   Unassembled   -> 'pacbio40.unassembled.fasta'\n",
      "--   Unitigs       -> 'pacbio40.unitigs.fasta'\n",
      "--\n",
      "-- Read layouts saved:\n",
      "--   Contigs       -> 'pacbio40.contigs.layout'.\n",
      "--   Unitigs       -> 'pacbio40.unitigs.layout'.\n",
      "--\n",
      "-- Graphs saved:\n",
      "--   Contigs       -> 'pacbio40.contigs.gfa'.\n",
      "--   Unitigs       -> 'pacbio40.unitigs.gfa'.\n",
      "--\n",
      "-- Bye.\n",
      "-- Finished stage 'outputSequence', reset canuIteration.\n"
     ]
    }
   ],
   "source": [
    "! /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/bin/canu -p pacbio40 -d pacbio40 genomesize=4.7m -pacbio-raw ./homework6/pacbio_40x.fq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- Canu release v1.5\n",
      "-- Detected Java(TM) Runtime Environment '1.8.0_152-release' (from '/Users/olga/miniconda3/bin/java').\n",
      "-- Detected gnuplot version '5.4 patchlevel 1' (from 'gnuplot') and image format 'png'.\n",
      "-- Detected 8 CPUs and 16 gigabytes of memory.\n",
      "-- No grid engine detected, grid disabled.\n",
      "--\n",
      "-- Run   2 jobs concurrently using    8 GB and   4 CPUs for stage 'meryl'.\n",
      "-- Run   1 job  concurrently using    6 GB and   8 CPUs for stage 'mhap (cor)'.\n",
      "-- Run   1 job  concurrently using    8 GB and   8 CPUs for stage 'overlapper (obt)'.\n",
      "-- Run   1 job  concurrently using    8 GB and   8 CPUs for stage 'overlapper (utg)'.\n",
      "-- Run   4 jobs concurrently using    8 GB and   2 CPUs for stage 'falcon_sense'.\n",
      "-- Run   8 jobs concurrently using    2 GB and   1 CPU  for stage 'ovStore bucketizer'.\n",
      "-- Run   8 jobs concurrently using    8 GB and   1 CPU  for stage 'ovStore sorting'.\n",
      "-- Run   2 jobs concurrently using    2 GB and   4 CPUs for stage 'read error detection'.\n",
      "-- Run   8 jobs concurrently using    1 GB and   1 CPU  for stage 'overlap error adjustment'.\n",
      "-- Run   2 jobs concurrently using    8 GB and   4 CPUs for stage 'bogart'.\n",
      "-- Run   8 jobs concurrently using    2 GB and   1 CPU  for stage 'GFA alignment and processing'.\n",
      "-- Run   2 jobs concurrently using    8 GB and   4 CPUs for stage 'consensus'.\n",
      "--\n",
      "-- Generating assembly 'pacbio80' in '/Users/olga/Documents/BioinformaticsInstitute/NGS/pacbio80'\n",
      "--\n",
      "-- Parameters:\n",
      "--\n",
      "--  genomeSize        4700000\n",
      "--\n",
      "--  Overlap Generation Limits:\n",
      "--    corOvlErrorRate 0.2400 ( 24.00%)\n",
      "--    obtOvlErrorRate 0.0450 (  4.50%)\n",
      "--    utgOvlErrorRate 0.0450 (  4.50%)\n",
      "--\n",
      "--  Overlap Processing Limits:\n",
      "--    corErrorRate    0.3000 ( 30.00%)\n",
      "--    obtErrorRate    0.0450 (  4.50%)\n",
      "--    utgErrorRate    0.0450 (  4.50%)\n",
      "--    cnsErrorRate    0.0450 (  4.50%)\n",
      "--\n",
      "--\n",
      "-- BEGIN CORRECTION\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:08:17 2021 with 316.157 GB free disk space\n",
      "\n",
      "    cd correction\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperCreate \\\n",
      "      -minlength 1000 \\\n",
      "      -o ./pacbio80.gkpStore.BUILDING \\\n",
      "      ./pacbio80.gkpStore.gkp \\\n",
      "    > ./pacbio80.gkpStore.BUILDING.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:08:22 2021 (5 seconds) with 316.022 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- In gatekeeper store 'correction/pacbio80.gkpStore':\n",
      "--   Found 75415 reads.\n",
      "--   Found 432651023 bases (92.05 times coverage).\n",
      "--\n",
      "--   Read length histogram (one '*' equals 153.01 reads):\n",
      "--        0    999      0 \n",
      "--     1000   1999  10711 **********************************************************************\n",
      "--     2000   2999   9757 ***************************************************************\n",
      "--     3000   3999   8796 *********************************************************\n",
      "--     4000   4999   8116 *****************************************************\n",
      "--     5000   5999   7447 ************************************************\n",
      "--     6000   6999   6482 ******************************************\n",
      "--     7000   7999   5554 ************************************\n",
      "--     8000   8999   4601 ******************************\n",
      "--     9000   9999   3774 ************************\n",
      "--    10000  10999   2931 *******************\n",
      "--    11000  11999   2239 **************\n",
      "--    12000  12999   1726 ***********\n",
      "--    13000  13999   1341 ********\n",
      "--    14000  14999    904 *****\n",
      "--    15000  15999    584 ***\n",
      "--    16000  16999    263 *\n",
      "--    17000  17999    104 \n",
      "--    18000  18999     46 \n",
      "--    19000  19999     24 \n",
      "--    20000  20999      8 \n",
      "--    21000  21999      4 \n",
      "--    22000  22999      3 \n",
      "-- Finished stage 'cor-gatekeeper', reset canuIteration.\n",
      "-- Finished stage 'merylConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 13:08:23 2021 with 316.02 GB free disk space (1 processes; 2 concurrently)\n",
      "\n",
      "    cd correction/0-mercounts\n",
      "    ./meryl.sh 1 > ./meryl.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:10:30 2021 (127 seconds) with 314.874 GB free disk space\n",
      "----------------------------------------\n",
      "-- Meryl finished successfully.\n",
      "-- Finished stage 'merylCheck', reset canuIteration.\n",
      "--\n",
      "--  16-mers                                                                                           Fraction\n",
      "--    Occurrences   NumMers                                                                         Unique Total\n",
      "--       1-     1 233771624 *******************************************************************--> 0.8050 0.5417\n",
      "--       2-     2  37075255 ********************************************************************** 0.9327 0.7136\n",
      "--       3-     4  12620809 ***********************                                                0.9650 0.7789\n",
      "--       5-     7   2336457 ****                                                                   0.9806 0.8240\n",
      "--       8-    11   1336854 **                                                                     0.9852 0.8450\n",
      "--      12-    16   1845407 ***                                                                    0.9901 0.8796\n",
      "--      17-    22   1096296 **                                                                     0.9961 0.9394\n",
      "--      23-    29    249910                                                                        0.9991 0.9802\n",
      "--      30-    37     36557                                                                        0.9998 0.9913\n",
      "--      38-    46     11700                                                                        0.9999 0.9936\n",
      "--      47-    56      5767                                                                        0.9999 0.9947\n",
      "--      57-    67      3885                                                                        0.9999 0.9953\n",
      "--      68-    79      3741                                                                        1.0000 0.9959\n",
      "--      80-    92      2810                                                                        1.0000 0.9965\n",
      "--      93-   106      1578                                                                        1.0000 0.9970\n",
      "--     107-   121       842                                                                        1.0000 0.9974\n",
      "--     122-   137       550                                                                        1.0000 0.9976\n",
      "--     138-   154       480                                                                        1.0000 0.9978\n",
      "--     155-   172       504                                                                        1.0000 0.9979\n",
      "--     173-   191       492                                                                        1.0000 0.9981\n",
      "--     192-   211       341                                                                        1.0000 0.9983\n",
      "--     212-   232       159                                                                        1.0000 0.9985\n",
      "--     233-   254        91                                                                        1.0000 0.9985\n",
      "--     255-   277        78                                                                        1.0000 0.9986\n",
      "--     278-   301        47                                                                        1.0000 0.9986\n",
      "--     302-   326        54                                                                        1.0000 0.9987\n",
      "--     327-   352        30                                                                        1.0000 0.9987\n",
      "--     353-   379        30                                                                        1.0000 0.9987\n",
      "--     380-   407        35                                                                        1.0000 0.9988\n",
      "--     408-   436        33                                                                        1.0000 0.9988\n",
      "--     437-   466        18                                                                        1.0000 0.9988\n",
      "--     467-   497        15                                                                        1.0000 0.9988\n",
      "--     498-   529        13                                                                        1.0000 0.9989\n",
      "--     530-   562        14                                                                        1.0000 0.9989\n",
      "--     563-   596        14                                                                        1.0000 0.9989\n",
      "--     597-   631        14                                                                        1.0000 0.9989\n",
      "--     632-   667        18                                                                        1.0000 0.9989\n",
      "--     668-   704        20                                                                        1.0000 0.9990\n",
      "--     705-   742        10                                                                        1.0000 0.9990\n",
      "--     743-   781        12                                                                        1.0000 0.9990\n",
      "--     782-   821         8                                                                        1.0000 0.9990\n",
      "--\n",
      "--      132776 (max occurrences)\n",
      "--   197748174 (total mers, non-unique)\n",
      "--    56631094 (distinct mers, non-unique)\n",
      "--   233771624 (unique mers)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-- For mhap overlapping, set repeat k-mer threshold to 4315.\n",
      "--\n",
      "-- Found 431519798 16-mers; 290402718 distinct and 233771624 unique.  Largest count 132776.\n",
      "-- Finished stage 'cor-meryl', reset canuIteration.\n",
      "--\n",
      "-- OVERLAPPER (mhap) (correction)\n",
      "--\n",
      "-- Set corMhapSensitivity=low based on read coverage of 92.\n",
      "--\n",
      "-- PARAMETERS: hashes=256, minMatches=3, threshold=0.8\n",
      "--\n",
      "-- Given 6 GB, can fit 18000 reads per block.\n",
      "-- For 6 blocks, set stride to 2 blocks.\n",
      "-- Logging partitioning to 'correction/1-overlapper/partitioning.log'.\n",
      "-- Configured 5 mhap precompute jobs.\n",
      "-- Configured 7 mhap overlap jobs.\n",
      "-- Finished stage 'cor-mhapConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 13:10:35 2021 with 316.019 GB free disk space (5 processes; 1 concurrently)\n",
      "\n",
      "    cd correction/1-overlapper\n",
      "    ./precompute.sh 1 > ./precompute.000001.out 2>&1\n",
      "    ./precompute.sh 2 > ./precompute.000002.out 2>&1\n",
      "    ./precompute.sh 3 > ./precompute.000003.out 2>&1\n",
      "    ./precompute.sh 4 > ./precompute.000004.out 2>&1\n",
      "    ./precompute.sh 5 > ./precompute.000005.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:25:08 2021 (873 seconds) with 314.817 GB free disk space\n",
      "----------------------------------------\n",
      "-- All 5 mhap precompute jobs finished successfully.\n",
      "-- Finished stage 'cor-mhapPrecomputeCheck', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 13:25:08 2021 with 314.817 GB free disk space (7 processes; 1 concurrently)\n",
      "\n",
      "    cd correction/1-overlapper\n",
      "    ./mhap.sh 1 > ./mhap.000001.out 2>&1\n",
      "    ./mhap.sh 2 > ./mhap.000002.out 2>&1\n",
      "    ./mhap.sh 3 > ./mhap.000003.out 2>&1\n",
      "    ./mhap.sh 4 > ./mhap.000004.out 2>&1\n",
      "    ./mhap.sh 5 > ./mhap.000005.out 2>&1\n",
      "    ./mhap.sh 6 > ./mhap.000006.out 2>&1\n",
      "    ./mhap.sh 7 > ./mhap.000007.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:26:47 2021 (99 seconds) with 314.762 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 7 mhap overlap output files.\n",
      "-- Finished stage 'cor-mhapCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:26:47 2021 with 314.762 GB free disk space\n",
      "\n",
      "    cd correction\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/ovStoreBuild \\\n",
      "     -O ./pacbio80.ovlStore.BUILDING \\\n",
      "     -G ./pacbio80.gkpStore \\\n",
      "     -M 2-8 \\\n",
      "     -L ./1-overlapper/ovljob.files \\\n",
      "     > ./pacbio80.ovlStore.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:26:48 2021 (1 second) with 314.71 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Overlap store 'correction/pacbio80.ovlStore' successfully constructed.\n",
      "--\n",
      "-- Purged 1.299 GB in 19 overlap output files and 2 directories.\n",
      "-- Overlap store 'correction/pacbio80.ovlStore' statistics not available (skipped in correction and trimming stages).\n",
      "-- Finished stage 'cor-createOverlapStore', reset canuIteration.\n",
      "-- Set corMinCoverage=4 based on read coverage of 92.\n",
      "-- Computing global filter scores 'correction/2-correction/pacbio80.globalScores'.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:26:48 2021 with 316.014 GB free disk space\n",
      "\n",
      "    cd correction/2-correction\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/filterCorrectionOverlaps \\\n",
      "      -G ../pacbio80.gkpStore \\\n",
      "      -O ../pacbio80.ovlStore \\\n",
      "      -S ./pacbio80.globalScores.WORKING \\\n",
      "      -c 40 \\\n",
      "      -l 0 \\\n",
      "    > ./pacbio80.globalScores.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:26:49 2021 (1 second) with 316.007 GB free disk space\n",
      "----------------------------------------\n",
      "--  PARAMETERS:\n",
      "--  ----------\n",
      "--  \n",
      "--       40 (expected coverage)\n",
      "--        0 (don't use overlaps shorter than this)\n",
      "--    0.000 (don't use overlaps with erate less than this)\n",
      "--    1.000 (don't use overlaps with erate more than this)\n",
      "--  \n",
      "--  OVERLAPS:\n",
      "--  --------\n",
      "--  \n",
      "--  IGNORED:\n",
      "--  \n",
      "--             0 (< 0.0000 fraction error)\n",
      "--             0 (> 0.4095 fraction error)\n",
      "--             0 (< 0 bases long)\n",
      "--             0 (> 2097151 bases long)\n",
      "--  \n",
      "--  FILTERED:\n",
      "--  \n",
      "--        816363 (too many overlaps, discard these shortest ones)\n",
      "--  \n",
      "--  EVIDENCE:\n",
      "--  \n",
      "--       1841077 (longest overlaps)\n",
      "--  \n",
      "--  TOTAL:\n",
      "--  \n",
      "--       2657440 (all overlaps)\n",
      "--  \n",
      "--  READS:\n",
      "--  -----\n",
      "--  \n",
      "--          4856 (no overlaps)\n",
      "--         43806 (no overlaps filtered)\n",
      "--         19758 (<  50% overlaps filtered)\n",
      "--         26473 (<  80% overlaps filtered)\n",
      "--         26753 (<  95% overlaps filtered)\n",
      "--         26753 (< 100% overlaps filtered)\n",
      "--  \n",
      "-- Computing expected corrected read lengths 'correction/2-correction/pacbio80.estimate.log'.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:26:49 2021 with 316.007 GB free disk space\n",
      "\n",
      "    cd correction/2-correction\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/generateCorrectionLayouts \\\n",
      "      -G ../pacbio80.gkpStore \\\n",
      "      -O ../pacbio80.ovlStore \\\n",
      "      -S ./pacbio80.globalScores \\\n",
      "      -c 4 \\\n",
      "      -C 80 \\\n",
      "      -p ./pacbio80.estimate.WORKING\n",
      "\n",
      "-- Finished on Wed Apr 21 13:26:49 2021 (lickety-split) with 316.005 GB free disk space\n",
      "----------------------------------------\n",
      "-- Sorting reads by expected corrected length.\n",
      "-- Sorting reads by uncorrected length.\n",
      "-- Loading expected corrected read lengths.\n",
      "-- Picking longest corrected reads.\n",
      "-- Writing longest corrected reads to 'correction/2-correction/pacbio80.readsToCorrect'.\n",
      "-- Summarizing filter.\n",
      "--\n",
      "-- Reads to be corrected:\n",
      "--   21881 reads longer than 7948 bp\n",
      "--   201281849 bp\n",
      "-- Expected corrected reads:\n",
      "--   21881 reads\n",
      "--   188002998 bp\n",
      "--   5621 bp minimum length\n",
      "--   8592 bp mean length\n",
      "--   15790 bp n50 length\n",
      "-- Using overlaps no worse than 0.3 fraction error for correcting reads (from corErrorRate parameter).\n",
      "-- Finished stage 'cor-buildCorrectionLayouts', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 13:26:51 2021 with 316.001 GB free disk space (16 processes; 4 concurrently)\n",
      "\n",
      "    cd correction/2-correction\n",
      "    ./correctReads.sh 1 > ./correctReads.000001.out 2>&1\n",
      "    ./correctReads.sh 2 > ./correctReads.000002.out 2>&1\n",
      "    ./correctReads.sh 3 > ./correctReads.000003.out 2>&1\n",
      "    ./correctReads.sh 4 > ./correctReads.000004.out 2>&1\n",
      "    ./correctReads.sh 5 > ./correctReads.000005.out 2>&1\n",
      "    ./correctReads.sh 6 > ./correctReads.000006.out 2>&1\n",
      "    ./correctReads.sh 7 > ./correctReads.000007.out 2>&1\n",
      "    ./correctReads.sh 8 > ./correctReads.000008.out 2>&1\n",
      "    ./correctReads.sh 9 > ./correctReads.000009.out 2>&1\n",
      "    ./correctReads.sh 10 > ./correctReads.000010.out 2>&1\n",
      "    ./correctReads.sh 11 > ./correctReads.000011.out 2>&1\n",
      "    ./correctReads.sh 12 > ./correctReads.000012.out 2>&1\n",
      "    ./correctReads.sh 13 > ./correctReads.000013.out 2>&1\n",
      "    ./correctReads.sh 14 > ./correctReads.000014.out 2>&1\n",
      "    ./correctReads.sh 15 > ./correctReads.000015.out 2>&1\n",
      "    ./correctReads.sh 16 > ./correctReads.000016.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:50:55 2021 (1444 seconds) with 315.527 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 16 read correction output files.\n",
      "-- Finished stage 'cor-generateCorrectedReads', reset canuIteration.\n",
      "-- Found 16 read correction output files.\n",
      "-- Finished stage 'cor-generateCorrectedReads', reset canuIteration.\n",
      "-- Concatenating correctReads output.\n",
      "-- Analyzing correctReads output.\n",
      "--\n",
      "-- Reads to be corrected:\n",
      "--   21881 reads longer than 7948 bp\n",
      "--   201281849 bp\n",
      "-- Expected corrected reads:\n",
      "--   21881 reads\n",
      "--   188002998 bp\n",
      "--   5621 bp minimum length\n",
      "--   8592 bp mean length\n",
      "--   15790 bp n50 length\n",
      "-- Actual corrected reads:\n",
      "--   22021 reads\n",
      "--   177594207 bp\n",
      "--   116 reads with 0 corrected blocks\n",
      "--   21511 reads with 1 corrected block\n",
      "--   252 reads with 2 corrected blocks\n",
      "--   2 reads with 3 corrected blocks\n",
      "--\n",
      "-- Purging correctReads output after merging to final output file.\n",
      "-- Purged 16 .dump.success sentinels.\n",
      "-- Purged 16 .fasta outputs.\n",
      "-- Purged 16 .err outputs.\n",
      "-- Purged 16 .out job log outputs.\n",
      "-- Finished stage 'cor-dumpCorrectedReads', reset canuIteration.\n",
      "--\n",
      "-- Corrected reads saved in 'pacbio80.correctedReads.fasta.gz'.\n",
      "--\n",
      "--\n",
      "-- BEGIN TRIMMING\n",
      "--\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:50:59 2021 with 315.648 GB free disk space\n",
      "\n",
      "    cd trimming\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperCreate \\\n",
      "      -minlength 1000 \\\n",
      "      -o ./pacbio80.gkpStore.BUILDING \\\n",
      "      ./pacbio80.gkpStore.gkp \\\n",
      "    > ./pacbio80.gkpStore.BUILDING.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:51:00 2021 (1 second) with 315.601 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- In gatekeeper store 'trimming/pacbio80.gkpStore':\n",
      "--   Found 22021 reads.\n",
      "--   Found 177594207 bases (37.78 times coverage).\n",
      "--\n",
      "--   Read length histogram (one '*' equals 66.77 reads):\n",
      "--        0    999      0 \n",
      "--     1000   1999    162 **\n",
      "--     2000   2999    140 **\n",
      "--     3000   3999    192 **\n",
      "--     4000   4999    317 ****\n",
      "--     5000   5999   3430 ***************************************************\n",
      "--     6000   6999   4674 **********************************************************************\n",
      "--     7000   7999   3677 *******************************************************\n",
      "--     8000   8999   2765 *****************************************\n",
      "--     9000   9999   2177 ********************************\n",
      "--    10000  10999   1551 ***********************\n",
      "--    11000  11999   1088 ****************\n",
      "--    12000  12999    772 ***********\n",
      "--    13000  13999    537 ********\n",
      "--    14000  14999    306 ****\n",
      "--    15000  15999    166 **\n",
      "--    16000  16999     56 \n",
      "--    17000  17999     10 \n",
      "--    18000  18999      1 \n",
      "-- Finished stage 'obt-gatekeeper', reset canuIteration.\n",
      "-- Finished stage 'merylConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 13:51:01 2021 with 315.6 GB free disk space (1 processes; 2 concurrently)\n",
      "\n",
      "    cd trimming/0-mercounts\n",
      "    ./meryl.sh 1 > ./meryl.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:51:44 2021 (43 seconds) with 315.52 GB free disk space\n",
      "----------------------------------------\n",
      "-- Meryl finished successfully.\n",
      "-- Finished stage 'merylCheck', reset canuIteration.\n",
      "--\n",
      "--  22-mers                                                                                           Fraction\n",
      "--    Occurrences   NumMers                                                                         Unique Total\n",
      "--       1-     1   7933603 *******************************************************************--> 0.5888 0.0448\n",
      "--       2-     2    586144 ***************************                                            0.6323 0.0514\n",
      "--       3-     4    258755 ************                                                           0.6455 0.0544\n",
      "--       5-     7     92357 ****                                                                   0.6548 0.0575\n",
      "--       8-    11     44482 **                                                                     0.6595 0.0599\n",
      "--      12-    16     57599 **                                                                     0.6623 0.0621\n",
      "--      17-    22    274255 ************                                                           0.6676 0.0683\n",
      "--      23-    29    878268 *****************************************                              0.6926 0.1085\n",
      "--      30-    37   1497209 ********************************************************************** 0.7639 0.2568\n",
      "--      38-    46   1206353 ********************************************************               0.8755 0.5493\n",
      "--      47-    56    518118 ************************                                               0.9588 0.8183\n",
      "--      57-    67     84096 ***                                                                    0.9917 0.9461\n",
      "--      68-    79     16956                                                                        0.9970 0.9709\n",
      "--      80-    92      5328                                                                        0.9981 0.9768\n",
      "--      93-   106      4825                                                                        0.9985 0.9795\n",
      "--     107-   121      4417                                                                        0.9988 0.9821\n",
      "--     122-   137       959                                                                        0.9991 0.9849\n",
      "--     138-   154       844                                                                        0.9992 0.9855\n",
      "--     155-   172      1149                                                                        0.9993 0.9863\n",
      "--     173-   191      2982                                                                        0.9993 0.9873\n",
      "--     192-   211      2612                                                                        0.9996 0.9905\n",
      "--     212-   232       590                                                                        0.9997 0.9933\n",
      "--     233-   254       409                                                                        0.9998 0.9940\n",
      "--     255-   277       531                                                                        0.9998 0.9946\n",
      "--     278-   301       123                                                                        0.9999 0.9953\n",
      "--     302-   326       400                                                                        0.9999 0.9956\n",
      "--     327-   352       115                                                                        0.9999 0.9962\n",
      "--     353-   379        44                                                                        0.9999 0.9965\n",
      "--     380-   407       113                                                                        0.9999 0.9966\n",
      "--     408-   436       302                                                                        0.9999 0.9968\n",
      "--     437-   466       466                                                                        0.9999 0.9976\n",
      "--     467-   497       235                                                                        1.0000 0.9987\n",
      "--     498-   529        55                                                                        1.0000 0.9994\n",
      "--     530-   562         8                                                                        1.0000 0.9995\n",
      "--     563-   596         1                                                                        1.0000 0.9995\n",
      "--     597-   631         1                                                                        1.0000 0.9995\n",
      "--     632-   667         2                                                                        1.0000 0.9995\n",
      "--     668-   704         2                                                                        1.0000 0.9995\n",
      "--     705-   742         2                                                                        1.0000 0.9996\n",
      "--     743-   781         5                                                                        1.0000 0.9996\n",
      "--     782-   821         1                                                                        1.0000 0.9996\n",
      "--\n",
      "--        2633 (max occurrences)\n",
      "--   169198163 (total mers, non-unique)\n",
      "--     5541159 (distinct mers, non-unique)\n",
      "--     7933603 (unique mers)\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:51:44 2021 with 315.52 GB free disk space\n",
      "\n",
      "    cd trimming/0-mercounts\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/meryl \\\n",
      "      -Dt \\\n",
      "      -n 789 \\\n",
      "      -s ./pacbio80.ms22 \\\n",
      "    > ./pacbio80.ms22.frequentMers.fasta \\\n",
      "    2> ./pacbio80.ms22.frequentMers.fasta.err\n",
      "\n",
      "-- Finished on Wed Apr 21 13:51:44 2021 (lickety-split) with 315.52 GB free disk space\n",
      "----------------------------------------\n",
      "-- Reset obtOvlMerThreshold from auto to 789.\n",
      "--\n",
      "-- Found 177131766 22-mers; 13474762 distinct and 7933603 unique.  Largest count 2633.\n",
      "-- Finished stage 'obt-meryl', reset canuIteration.\n",
      "--\n",
      "-- OVERLAPPER (normal) (trimming) erate=0.045\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 13:51:45 2021 with 315.6 GB free disk space\n",
      "\n",
      "    cd trimming/1-overlapper\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/overlapInCorePartition \\\n",
      "     -g  ../pacbio80.gkpStore \\\n",
      "     -bl 100000000 \\\n",
      "     -bs 0 \\\n",
      "     -rs 2000000 \\\n",
      "     -rl 0 \\\n",
      "     -ol 500 \\\n",
      "     -o  ./pacbio80.partition \\\n",
      "    > ./pacbio80.partition.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 13:51:45 2021 (lickety-split) with 315.6 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Configured 2 overlapInCore jobs.\n",
      "-- Finished stage 'obt-overlapConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 13:51:45 2021 with 315.6 GB free disk space (2 processes; 1 concurrently)\n",
      "\n",
      "    cd trimming/1-overlapper\n",
      "    ./overlap.sh 1 > ./overlap.000001.out 2>&1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    ./overlap.sh 2 > ./overlap.000002.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 14:00:33 2021 (528 seconds) with 315.577 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 2 overlapInCore output files.\n",
      "--\n",
      "-- overlapInCore compute 'trimming/1-overlapper':\n",
      "--   kmer hits\n",
      "--     with no overlap          3060771   1530385.5 +- 786850.748\n",
      "--     with an overlap          1070421    535210.5 +- 269780.328\n",
      "--\n",
      "--   overlaps                   1070421    535210.5 +- 269780.328\n",
      "--     contained                      0           0 +- 0\n",
      "--     dovetail                       0           0 +- 0\n",
      "--\n",
      "--   overlaps rejected\n",
      "--     multiple per pair              0           0 +- 0\n",
      "--     bad short window               0           0 +- 0\n",
      "--     bad long window                0           0 +- 0\n",
      "-- Finished stage 'obt-overlapCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:00:33 2021 with 315.577 GB free disk space\n",
      "\n",
      "    cd trimming\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/ovStoreBuild \\\n",
      "     -O ./pacbio80.ovlStore.BUILDING \\\n",
      "     -G ./pacbio80.gkpStore \\\n",
      "     -M 2-8 \\\n",
      "     -L ./1-overlapper/ovljob.files \\\n",
      "     > ./pacbio80.ovlStore.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 14:00:33 2021 (lickety-split) with 315.537 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Overlap store 'trimming/pacbio80.ovlStore' successfully constructed.\n",
      "--\n",
      "-- Purged 0.02 GB in 6 overlap output files and 1 directories.\n",
      "-- Overlap store 'trimming/pacbio80.ovlStore' statistics not available (skipped in correction and trimming stages).\n",
      "-- Finished stage 'obt-createOverlapStore', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:00:33 2021 with 315.558 GB free disk space\n",
      "\n",
      "    cd trimming/3-overlapbasedtrimming\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/trimReads \\\n",
      "      -G  ../pacbio80.gkpStore \\\n",
      "      -O  ../pacbio80.ovlStore \\\n",
      "      -Co ./pacbio80.1.trimReads.clear \\\n",
      "      -e  0.045 \\\n",
      "      -minlength 1000 \\\n",
      "      -ol 1 \\\n",
      "      -oc 1 \\\n",
      "      -o  ./pacbio80.1.trimReads \\\n",
      "    >     ./pacbio80.1.trimReads.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 14:00:35 2021 (2 seconds) with 315.557 GB free disk space\n",
      "----------------------------------------\n",
      "--  PARAMETERS:\n",
      "--  ----------\n",
      "--     1000    (reads trimmed below this many bases are deleted)\n",
      "--   0.0450    (use overlaps at or below this fraction error)\n",
      "--        1    (break region if overlap is less than this long, for 'largest covered' algorithm)\n",
      "--        1    (break region if overlap coverage is less than this many read, for 'largest covered' algorithm)\n",
      "--  \n",
      "--  INPUT READS:\n",
      "--  -----------\n",
      "--   22021 reads    177594207 bases (reads processed)\n",
      "--       0 reads            0 bases (reads not processed, previously deleted)\n",
      "--       0 reads            0 bases (reads not processed, in a library where trimming isn't allowed)\n",
      "--  \n",
      "--  OUTPUT READS:\n",
      "--  ------------\n",
      "--   19385 reads    154545726 bases (trimmed reads output)\n",
      "--    2632 reads     20907729 bases (reads with no change, kept as is)\n",
      "--       0 reads            0 bases (reads with no overlaps, deleted)\n",
      "--       4 reads         4097 bases (reads with short trimmed length, deleted)\n",
      "--  \n",
      "--  TRIMMING DETAILS:\n",
      "--  ----------------\n",
      "--   12068 reads      1361737 bases (bases trimmed from the 5' end of a read)\n",
      "--   16211 reads       774918 bases (bases trimmed from the 3' end of a read)\n",
      "-- Finished stage 'obt-trimReads', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:00:35 2021 with 315.556 GB free disk space\n",
      "\n",
      "    cd trimming/3-overlapbasedtrimming\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/splitReads \\\n",
      "      -G  ../pacbio80.gkpStore \\\n",
      "      -O  ../pacbio80.ovlStore \\\n",
      "      -Ci ./pacbio80.1.trimReads.clear \\\n",
      "      -Co ./pacbio80.2.splitReads.clear \\\n",
      "      -e  0.045 \\\n",
      "      -minlength 1000 \\\n",
      "      -o  ./pacbio80.2.splitReads \\\n",
      "    >     ./pacbio80.2.splitReads.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 14:00:36 2021 (1 second) with 315.556 GB free disk space\n",
      "----------------------------------------\n",
      "--  PARAMETERS:\n",
      "--  ----------\n",
      "--     1000    (reads trimmed below this many bases are deleted)\n",
      "--   0.0450    (use overlaps at or below this fraction error)\n",
      "--  INPUT READS:\n",
      "--  -----------\n",
      "--   22017 reads    177590110 bases (reads processed)\n",
      "--       4 reads         4097 bases (reads not processed, previously deleted)\n",
      "--       0 reads            0 bases (reads not processed, in a library where trimming isn't allowed)\n",
      "--  \n",
      "--  PROCESSED:\n",
      "--  --------\n",
      "--       0 reads            0 bases (no overlaps)\n",
      "--       0 reads            0 bases (no coverage after adjusting for trimming done already)\n",
      "--       0 reads            0 bases (processed for chimera)\n",
      "--       0 reads            0 bases (processed for spur)\n",
      "--   22017 reads    177590110 bases (processed for subreads)\n",
      "--  \n",
      "--  READS WITH SIGNALS:\n",
      "--  ------------------\n",
      "--       0 reads            0 signals (number of 5' spur signal)\n",
      "--       0 reads            0 signals (number of 3' spur signal)\n",
      "--       0 reads            0 signals (number of chimera signal)\n",
      "--      20 reads           22 signals (number of subread signal)\n",
      "--  \n",
      "--  SIGNALS:\n",
      "--  -------\n",
      "--       0 reads            0 bases (size of 5' spur signal)\n",
      "--       0 reads            0 bases (size of 3' spur signal)\n",
      "--       0 reads            0 bases (size of chimera signal)\n",
      "--      22 reads         2994 bases (size of subread signal)\n",
      "--  \n",
      "--  TRIMMING:\n",
      "--  --------\n",
      "--       5 reads        18220 bases (trimmed from the 5' end of the read)\n",
      "--      15 reads        38930 bases (trimmed from the 3' end of the read)\n",
      "-- Finished stage 'obt-splitReads', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:00:36 2021 with 315.556 GB free disk space\n",
      "\n",
      "    cd trimming\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperDumpFASTQ -fasta -nolibname \\\n",
      "      -G ./pacbio80.gkpStore \\\n",
      "      -c ./3-overlapbasedtrimming/pacbio80.2.splitReads.clear \\\n",
      "      -o ../pacbio80.trimmedReads.gz \\\n",
      "    >    ../pacbio80.trimmedReads.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 14:00:38 2021 (2 seconds) with 315.498 GB free disk space\n",
      "----------------------------------------\n",
      "-- Finished stage 'obt-dumpReads', reset canuIteration.\n",
      "--\n",
      "-- Trimmed reads saved in 'trimming/pacbio80.trimmedReads.fasta.gz'\n",
      "--\n",
      "--\n",
      "-- BEGIN ASSEMBLY\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:00:38 2021 with 315.498 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperCreate \\\n",
      "      -minlength 1000 \\\n",
      "      -o ./pacbio80.gkpStore.BUILDING \\\n",
      "      ./pacbio80.gkpStore.gkp \\\n",
      "    > ./pacbio80.gkpStore.BUILDING.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 14:00:40 2021 (2 seconds) with 315.45 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- In gatekeeper store 'unitigging/pacbio80.gkpStore':\n",
      "--   Found 22017 reads.\n",
      "--   Found 175396305 bases (37.31 times coverage).\n",
      "--\n",
      "--   Read length histogram (one '*' equals 65.58 reads):\n",
      "--        0    999      0 \n",
      "--     1000   1999    169 **\n",
      "--     2000   2999    162 **\n",
      "--     3000   3999    278 ****\n",
      "--     4000   4999    504 *******\n",
      "--     5000   5999   3552 ******************************************************\n",
      "--     6000   6999   4591 **********************************************************************\n",
      "--     7000   7999   3637 *******************************************************\n",
      "--     8000   8999   2674 ****************************************\n",
      "--     9000   9999   2100 ********************************\n",
      "--    10000  10999   1526 ***********************\n",
      "--    11000  11999   1059 ****************\n",
      "--    12000  12999    738 ***********\n",
      "--    13000  13999    510 *******\n",
      "--    14000  14999    296 ****\n",
      "--    15000  15999    161 **\n",
      "--    16000  16999     50 \n",
      "--    17000  17999      9 \n",
      "--    18000  18999      1 \n",
      "-- Finished stage 'utg-gatekeeper', reset canuIteration.\n",
      "-- Finished stage 'merylConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 14:00:40 2021 with 315.449 GB free disk space (1 processes; 2 concurrently)\n",
      "\n",
      "    cd unitigging/0-mercounts\n",
      "    ./meryl.sh 1 > ./meryl.000001.out 2>&1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "-- Finished on Wed Apr 21 14:01:18 2021 (38 seconds) with 315.372 GB free disk space\n",
      "----------------------------------------\n",
      "-- Meryl finished successfully.\n",
      "-- Finished stage 'merylCheck', reset canuIteration.\n",
      "--\n",
      "--  22-mers                                                                                           Fraction\n",
      "--    Occurrences   NumMers                                                                         Unique Total\n",
      "--       1-     1   6548773 *******************************************************************--> 0.5429 0.0374\n",
      "--       2-     2    566458 **************************                                             0.5898 0.0439\n",
      "--       3-     4    253439 ***********                                                            0.6043 0.0469\n",
      "--       5-     7     91440 ****                                                                   0.6146 0.0500\n",
      "--       8-    11     44238 **                                                                     0.6197 0.0524\n",
      "--      12-    16     59520 **                                                                     0.6228 0.0546\n",
      "--      17-    22    282672 *************                                                          0.6290 0.0612\n",
      "--      23-    29    889619 *****************************************                              0.6576 0.1028\n",
      "--      30-    37   1505729 ********************************************************************** 0.7383 0.2549\n",
      "--      38-    46   1194650 *******************************************************                0.8635 0.5524\n",
      "--      47-    56    501543 ***********************                                                0.9555 0.8218\n",
      "--      57-    67     81894 ***                                                                    0.9909 0.9464\n",
      "--      68-    79     16797                                                                        0.9967 0.9708\n",
      "--      80-    92      5214                                                                        0.9978 0.9767\n",
      "--      93-   106      4893                                                                        0.9983 0.9794\n",
      "--     107-   121      4296                                                                        0.9987 0.9821\n",
      "--     122-   137       920                                                                        0.9990 0.9847\n",
      "--     138-   154       889                                                                        0.9991 0.9854\n",
      "--     155-   172      1119                                                                        0.9992 0.9861\n",
      "--     173-   191      3007                                                                        0.9993 0.9872\n",
      "--     192-   211      2581                                                                        0.9995 0.9904\n",
      "--     212-   232       576                                                                        0.9997 0.9933\n",
      "--     233-   254       419                                                                        0.9998 0.9939\n",
      "--     255-   277       520                                                                        0.9998 0.9946\n",
      "--     278-   301       131                                                                        0.9998 0.9953\n",
      "--     302-   326       393                                                                        0.9999 0.9955\n",
      "--     327-   352       113                                                                        0.9999 0.9962\n",
      "--     353-   379        47                                                                        0.9999 0.9964\n",
      "--     380-   407       116                                                                        0.9999 0.9965\n",
      "--     408-   436       318                                                                        0.9999 0.9968\n",
      "--     437-   466       446                                                                        0.9999 0.9976\n",
      "--     467-   497       252                                                                        1.0000 0.9987\n",
      "--     498-   529        36                                                                        1.0000 0.9994\n",
      "--     530-   562         8                                                                        1.0000 0.9995\n",
      "--     563-   596         1                                                                        1.0000 0.9995\n",
      "--     597-   631         1                                                                        1.0000 0.9995\n",
      "--     632-   667         2                                                                        1.0000 0.9995\n",
      "--     668-   704         2                                                                        1.0000 0.9995\n",
      "--     705-   742         3                                                                        1.0000 0.9996\n",
      "--     743-   781         4                                                                        1.0000 0.9996\n",
      "--     782-   821         1                                                                        1.0000 0.9996\n",
      "--\n",
      "--        2623 (max occurrences)\n",
      "--   168385175 (total mers, non-unique)\n",
      "--     5514353 (distinct mers, non-unique)\n",
      "--     6548773 (unique mers)\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:01:18 2021 with 315.372 GB free disk space\n",
      "\n",
      "    cd unitigging/0-mercounts\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/meryl \\\n",
      "      -Dt \\\n",
      "      -n 786 \\\n",
      "      -s ./pacbio80.ms22 \\\n",
      "    > ./pacbio80.ms22.frequentMers.fasta \\\n",
      "    2> ./pacbio80.ms22.frequentMers.fasta.err\n",
      "\n",
      "-- Finished on Wed Apr 21 14:01:19 2021 (1 second) with 315.372 GB free disk space\n",
      "----------------------------------------\n",
      "-- Reset utgOvlMerThreshold from auto to 786.\n",
      "--\n",
      "-- Found 174933948 22-mers; 12063126 distinct and 6548773 unique.  Largest count 2623.\n",
      "-- Finished stage 'utg-meryl', reset canuIteration.\n",
      "--\n",
      "-- OVERLAPPER (normal) (assembly) erate=0.045\n",
      "--\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:01:19 2021 with 315.452 GB free disk space\n",
      "\n",
      "    cd unitigging/1-overlapper\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/overlapInCorePartition \\\n",
      "     -g  ../pacbio80.gkpStore \\\n",
      "     -bl 100000000 \\\n",
      "     -bs 0 \\\n",
      "     -rs 2000000 \\\n",
      "     -rl 0 \\\n",
      "     -ol 500 \\\n",
      "     -o  ./pacbio80.partition \\\n",
      "    > ./pacbio80.partition.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 14:01:19 2021 (lickety-split) with 315.452 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Configured 2 overlapInCore jobs.\n",
      "-- Finished stage 'utg-overlapConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 14:01:19 2021 with 315.452 GB free disk space (2 processes; 1 concurrently)\n",
      "\n",
      "    cd unitigging/1-overlapper\n",
      "    ./overlap.sh 1 > ./overlap.000001.out 2>&1\n",
      "    ./overlap.sh 2 > ./overlap.000002.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 14:12:02 2021 (643 seconds) with 315.428 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 2 overlapInCore output files.\n",
      "--\n",
      "-- overlapInCore compute 'unitigging/1-overlapper':\n",
      "--   kmer hits\n",
      "--     with no overlap          3286466     1643233 +- 803089.455\n",
      "--     with an overlap           805343    402671.5 +- 198438.911\n",
      "--\n",
      "--   overlaps                    805343    402671.5 +- 198438.911\n",
      "--     contained                 150109     75054.5 +- 36221.544\n",
      "--     dovetail                  655234      327617 +- 162217.366\n",
      "--\n",
      "--   overlaps rejected\n",
      "--     multiple per pair              0           0 +- 0\n",
      "--     bad short window               0           0 +- 0\n",
      "--     bad long window                0           0 +- 0\n",
      "-- Finished stage 'utg-overlapCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:12:02 2021 with 315.428 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/ovStoreBuild \\\n",
      "     -O ./pacbio80.ovlStore.BUILDING \\\n",
      "     -G ./pacbio80.gkpStore \\\n",
      "     -M 2-8 \\\n",
      "     -L ./1-overlapper/ovljob.files \\\n",
      "     > ./pacbio80.ovlStore.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 14:12:03 2021 (1 second) with 315.397 GB free disk space\n",
      "----------------------------------------\n",
      "--\n",
      "-- Overlap store 'unitigging/pacbio80.ovlStore' successfully constructed.\n",
      "--\n",
      "-- Purged 0.014 GB in 6 overlap output files and 1 directories.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:12:03 2021 with 315.412 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/ovStoreStats \\\n",
      "     -G ./pacbio80.gkpStore \\\n",
      "     -O ./pacbio80.ovlStore \\\n",
      "     -o ./pacbio80.ovlStore \\\n",
      "     > ./pacbio80.ovlStore.summary.err 2>&1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "-- Finished on Wed Apr 21 14:12:03 2021 (lickety-split) with 315.412 GB free disk space\n",
      "----------------------------------------\n",
      "-- category            reads     %          read length        feature size or coverage  analysis\n",
      "-- ----------------  -------  -------  ----------------------  ------------------------  --------------------\n",
      "-- middle-missing          4    0.02     6007.25 +- 1931.94        249.50 +- 145.75     (bad trimming)\n",
      "-- middle-hump             0    0.00        0.00 +- 0.00             0.00 +- 0.00       (bad trimming)\n",
      "-- no-5-prime              1    0.00     6134.00 +- 0.00           353.00 +- 0.00       (bad trimming)\n",
      "-- no-3-prime              0    0.00        0.00 +- 0.00             0.00 +- 0.00       (bad trimming)\n",
      "-- \n",
      "-- low-coverage            0    0.00        0.00 +- 0.00             0.00 +- 0.00       (easy to assemble, potential for lower quality consensus)\n",
      "-- unique              15089   68.53     7795.84 +- 2470.97         34.50 +- 6.75       (easy to assemble, perfect, yay)\n",
      "-- repeat-cont           328    1.49     7114.44 +- 2067.88         63.99 +- 7.87       (potential for consensus errors, no impact on assembly)\n",
      "-- repeat-dove             8    0.04    13125.25 +- 1223.02         65.11 +- 6.72       (hard to assemble, likely won't assemble correctly or even at all)\n",
      "-- \n",
      "-- span-repeat          4153   18.86     8489.82 +- 2578.21       2868.29 +- 2597.58    (read spans a large repeat, usually easy to assemble)\n",
      "-- uniq-repeat-cont     2074    9.42     7662.89 +- 1993.04                             (should be uniquely placed, low potential for consensus errors, no impact on assembly)\n",
      "-- uniq-repeat-dove      230    1.04    12911.49 +- 1662.56                             (will end contigs, potential to misassemble)\n",
      "-- uniq-anchor           130    0.59     9042.34 +- 2869.95       2269.86 +- 2388.91    (repeat read, with unique section, probable bad read)\n",
      "--\n",
      "-- Overlap store 'unitigging/pacbio80.ovlStore' contains:\n",
      "--\n",
      "--   category            reads     %          read length        feature size or coverage  analysis\n",
      "--   ----------------  -------  -------  ----------------------  ------------------------  --------------------\n",
      "--   middle-missing          4    0.02     6007.25 +- 1931.94        249.50 +- 145.75     (bad trimming)\n",
      "--   middle-hump             0    0.00        0.00 +- 0.00             0.00 +- 0.00       (bad trimming)\n",
      "--   no-5-prime              1    0.00     6134.00 +- 0.00           353.00 +- 0.00       (bad trimming)\n",
      "--   no-3-prime              0    0.00        0.00 +- 0.00             0.00 +- 0.00       (bad trimming)\n",
      "--   \n",
      "--   low-coverage            0    0.00        0.00 +- 0.00             0.00 +- 0.00       (easy to assemble, potential for lower quality consensus)\n",
      "--   unique              15089   68.53     7795.84 +- 2470.97         34.50 +- 6.75       (easy to assemble, perfect, yay)\n",
      "--   repeat-cont           328    1.49     7114.44 +- 2067.88         63.99 +- 7.87       (potential for consensus errors, no impact on assembly)\n",
      "--   repeat-dove             8    0.04    13125.25 +- 1223.02         65.11 +- 6.72       (hard to assemble, likely won't assemble correctly or even at all)\n",
      "--   \n",
      "--   span-repeat          4153   18.86     8489.82 +- 2578.21       2868.29 +- 2597.58    (read spans a large repeat, usually easy to assemble)\n",
      "--   uniq-repeat-cont     2074    9.42     7662.89 +- 1993.04                             (should be uniquely placed, low potential for consensus errors, no impact on assembly)\n",
      "--   uniq-repeat-dove      230    1.04    12911.49 +- 1662.56                             (will end contigs, potential to misassemble)\n",
      "--   uniq-anchor           130    0.59     9042.34 +- 2869.95       2269.86 +- 2388.91    (repeat read, with unique section, probable bad read)\n",
      "-- Finished stage 'utg-createOverlapStore', reset canuIteration.\n",
      "\n",
      "Configure RED for 2gb memory with batches of at most 5000 reads and (unlimited) bases.\n",
      "\n",
      "RED job   1 from read         1 to read      2285 -   1.500 GB for    2285 reads -   0.224 GB for  18494759 olaps -   0.002 GB for evidence\n",
      "RED job   2 from read      2286 to read      4610 -   1.500 GB for    2325 reads -   0.224 GB for  18494362 olaps -   0.002 GB for evidence\n",
      "RED job   3 from read      4611 to read      6941 -   1.501 GB for    2331 reads -   0.224 GB for  18496594 olaps -   0.002 GB for evidence\n",
      "RED job   4 from read      6942 to read      9292 -   1.500 GB for    2351 reads -   0.224 GB for  18492633 olaps -   0.002 GB for evidence\n",
      "RED job   5 from read      9293 to read     11639 -   1.500 GB for    2347 reads -   0.224 GB for  18492211 olaps -   0.002 GB for evidence\n",
      "RED job   6 from read     11640 to read     13976 -   1.500 GB for    2337 reads -   0.224 GB for  18494615 olaps -   0.002 GB for evidence\n",
      "RED job   7 from read     13977 to read     16296 -   1.500 GB for    2320 reads -   0.224 GB for  18489538 olaps -   0.002 GB for evidence\n",
      "RED job   8 from read     16297 to read     18592 -   1.500 GB for    2296 reads -   0.224 GB for  18490144 olaps -   0.002 GB for evidence\n",
      "RED job   9 from read     18593 to read     20896 -   1.501 GB for    2304 reads -   0.224 GB for  18496231 olaps -   0.002 GB for evidence\n",
      "RED job  10 from read     20897 to read     22017 -   0.727 GB for    1121 reads -   0.108 GB for   8955218 olaps -   0.001 GB for evidence\n",
      "-- Finished stage 'readErrorDetectionConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 14:12:03 2021 with 315.412 GB free disk space (10 processes; 2 concurrently)\n",
      "\n",
      "    cd unitigging/3-overlapErrorAdjustment\n",
      "    ./red.sh 1 > ./red.000001.out 2>&1\n",
      "    ./red.sh 2 > ./red.000002.out 2>&1\n",
      "    ./red.sh 3 > ./red.000003.out 2>&1\n",
      "    ./red.sh 4 > ./red.000004.out 2>&1\n",
      "    ./red.sh 5 > ./red.000005.out 2>&1\n",
      "    ./red.sh 6 > ./red.000006.out 2>&1\n",
      "    ./red.sh 7 > ./red.000007.out 2>&1\n",
      "    ./red.sh 8 > ./red.000008.out 2>&1\n",
      "    ./red.sh 9 > ./red.000009.out 2>&1\n",
      "    ./red.sh 10 > ./red.000010.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 14:12:32 2021 (29 seconds) with 315.408 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 10 read error detection output files.\n",
      "-- Finished stage 'readErrorDetectionCheck', reset canuIteration.\n",
      "\n",
      "Configure OEA for 1gb memory with batches of at most 25000 reads and (unlimited) bases.\n",
      "Configured 1 jobs.\n",
      "\n",
      "OEA job   1 from read         1 to read     22017 - 167.3 bases +  4.2 adjusts +  0.7 reads + 49.2 olaps +  8.0 fseq/rseq + 32.0 fadj/radj + 32.0 work + 256.0 misc = 549.3 MB\n",
      "\n",
      "-- Finished stage 'overlapErrorAdjustmentConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 14:12:33 2021 with 315.408 GB free disk space (1 processes; 8 concurrently)\n",
      "\n",
      "    cd unitigging/3-overlapErrorAdjustment\n",
      "    ./oea.sh 1 > ./oea.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 14:12:56 2021 (23 seconds) with 315.405 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found 1 overlap error adjustment output files.\n",
      "-- Finished stage 'overlapErrorAdjustmentCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:12:56 2021 with 315.405 GB free disk space\n",
      "\n",
      "    cd unitigging/3-overlapErrorAdjustment\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/ovStoreBuild \\\n",
      "      -G ../pacbio80.gkpStore \\\n",
      "      -O ../pacbio80.ovlStore \\\n",
      "      -evalues \\\n",
      "      -L ./oea.files \\\n",
      "    > ./oea.apply.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 14:12:56 2021 (lickety-split) with 315.402 GB free disk space\n",
      "----------------------------------------\n",
      "-- No report available.\n",
      "-- Finished stage 'updateOverlapStore', reset canuIteration.\n",
      "-- Finished stage 'unitig', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 14:12:56 2021 with 315.402 GB free disk space (1 processes; 2 concurrently)\n",
      "\n",
      "    cd unitigging/4-unitigger\n",
      "    ./unitigger.sh 1 > ./unitigger.000001.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 14:12:59 2021 (3 seconds) with 315.389 GB free disk space\n",
      "----------------------------------------\n",
      "-- Unitigger finished successfully.\n",
      "-- Found, in version 1, after unitig construction:\n",
      "--   contigs:      1 sequences, total length 4642080 bp (including 0 repeats of total length 0 bp).\n",
      "--   bubbles:      0 sequences, total length 0 bp.\n",
      "--   unassembled:  529 sequences, total length 3334979 bp.\n",
      "--\n",
      "-- Contig sizes based on genome size --\n",
      "--            NG (bp)  LG (contigs)    sum (bp)\n",
      "--         ----------  ------------  ----------\n",
      "--     10     4642080             1     4642080\n",
      "--     20     4642080             1     4642080\n",
      "--     30     4642080             1     4642080\n",
      "--     40     4642080             1     4642080\n",
      "--     50     4642080             1     4642080\n",
      "--     60     4642080             1     4642080\n",
      "--     70     4642080             1     4642080\n",
      "--     80     4642080             1     4642080\n",
      "--     90     4642080             1     4642080\n",
      "--\n",
      "-- Finished stage 'unitigCheck', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:12:59 2021 with 315.389 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperPartition \\\n",
      "      -G ./pacbio80.gkpStore \\\n",
      "      -T ./pacbio80.ctgStore 1 \\\n",
      "      -b 15000 \\\n",
      "      -p 8 \\\n",
      "    > ./pacbio80.ctgStore/partitionedReads.log 2>&1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "-- Finished on Wed Apr 21 14:12:59 2021 (lickety-split) with 315.344 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:12:59 2021 with 315.344 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/gatekeeperPartition \\\n",
      "      -G ./pacbio80.gkpStore \\\n",
      "      -T ./pacbio80.utgStore 1 \\\n",
      "      -b 15000 \\\n",
      "      -p 8 \\\n",
      "    > ./pacbio80.utgStore/partitionedReads.log 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 14:12:59 2021 (lickety-split) with 315.3 GB free disk space\n",
      "----------------------------------------\n",
      "-- Using fast alignment for consensus (iteration '0').\n",
      "-- Configured 2 contig and 1 unitig consensus jobs.\n",
      "-- Finished stage 'consensusConfigure', reset canuIteration.\n",
      "--\n",
      "-- Running jobs.  First attempt out of 2.\n",
      "----------------------------------------\n",
      "-- Starting concurrent execution on Wed Apr 21 14:13:00 2021 with 315.3 GB free disk space (3 processes; 2 concurrently)\n",
      "\n",
      "    cd unitigging/5-consensus\n",
      "    ./consensus.sh 1 > ./consensus.000001.out 2>&1\n",
      "    ./consensus.sh 2 > ./consensus.000002.out 2>&1\n",
      "    ./consensus.sh 3 > ./consensus.000003.out 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 14:18:40 2021 (340 seconds) with 315.238 GB free disk space\n",
      "----------------------------------------\n",
      "-- Using slow alignment for consensus (iteration '1').\n",
      "-- Configured 2 contig and 1 unitig consensus jobs.\n",
      "-- All 3 consensus jobs finished successfully.\n",
      "-- Finished stage 'consensusCheck', reset canuIteration.\n",
      "-- Using fast alignment for consensus (iteration '0').\n",
      "-- Configured 2 contig and 1 unitig consensus jobs.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:18:40 2021 with 315.238 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreLoad \\\n",
      "      -G ./pacbio80.gkpStore \\\n",
      "      -T ./pacbio80.ctgStore 2 \\\n",
      "      -L ./5-consensus/ctgcns.files \\\n",
      "    > ./5-consensus/ctgcns.files.ctgStoreLoad.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 14:18:40 2021 (lickety-split) with 315.222 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:18:40 2021 with 315.222 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreLoad \\\n",
      "      -G ./pacbio80.gkpStore \\\n",
      "      -T ./pacbio80.utgStore 2 \\\n",
      "      -L ./5-consensus/utgcns.files \\\n",
      "    > ./5-consensus/utgcns.files.utgStoreLoad.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 14:18:41 2021 (1 second) with 315.213 GB free disk space\n",
      "----------------------------------------\n",
      "-- Purging consensus output after loading to ctgStore and/or utgStore.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:18:41 2021 with 315.213 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./pacbio80.gkpStore \\\n",
      "      -T ./pacbio80.ctgStore 2 \\\n",
      "      -sizes -s 4700000 \\\n",
      "    > ./pacbio80.ctgStore/seqDB.v002.sizes.txt\n",
      "\n",
      "-- Finished on Wed Apr 21 14:18:41 2021 (lickety-split) with 315.213 GB free disk space\n",
      "----------------------------------------\n",
      "-- Found, in version 2, after consensus generation:\n",
      "--   contigs:      1 sequences, total length 4649321 bp (including 0 repeats of total length 0 bp).\n",
      "--   bubbles:      0 sequences, total length 0 bp.\n",
      "--   unassembled:  529 sequences, total length 3334979 bp.\n",
      "--\n",
      "-- Contig sizes based on genome size --\n",
      "--            NG (bp)  LG (contigs)    sum (bp)\n",
      "--         ----------  ------------  ----------\n",
      "--     10     4649321             1     4649321\n",
      "--     20     4649321             1     4649321\n",
      "--     30     4649321             1     4649321\n",
      "--     40     4649321             1     4649321\n",
      "--     50     4649321             1     4649321\n",
      "--     60     4649321             1     4649321\n",
      "--     70     4649321             1     4649321\n",
      "--     80     4649321             1     4649321\n",
      "--     90     4649321             1     4649321\n",
      "--\n",
      "-- Finished stage 'consensusLoad', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:18:41 2021 with 315.213 GB free disk space\n",
      "\n",
      "    cd unitigging\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreCoverageStat \\\n",
      "      -G ./pacbio80.gkpStore \\\n",
      "      -T ./pacbio80.ctgStore 2 \\\n",
      "      -s 4700000 \\\n",
      "      -o ./pacbio80.ctgStore.coverageStat \\\n",
      "    > ./pacbio80.ctgStore.coverageStat.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 14:18:41 2021 (lickety-split) with 315.213 GB free disk space\n",
      "----------------------------------------\n",
      "-- Finished stage 'consensusAnalyze', reset canuIteration.\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:18:41 2021 with 315.213 GB free disk space\n",
      "\n",
      "    cd unitigging/4-unitigger\n",
      "    ./alignGFA.sh\n",
      "GFA alignments updated.\n",
      "\n",
      "-- Finished on Wed Apr 21 14:18:41 2021 (lickety-split) with 315.213 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:18:41 2021 with 315.213 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./unitigging/pacbio80.gkpStore \\\n",
      "      -T ./unitigging/pacbio80.ctgStore 2 \\\n",
      "      -o ./pacbio80.contigs \\\n",
      "      -layout \\\n",
      "    > ./pacbio80.contigs.layout.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 14:18:41 2021 (lickety-split) with 315.195 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:18:41 2021 with 315.195 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./unitigging/pacbio80.gkpStore \\\n",
      "      -T ./unitigging/pacbio80.utgStore 2 \\\n",
      "      -o ./pacbio80.unitigs \\\n",
      "      -layout \\\n",
      "    > ./pacbio80.unitigs.layout.err 2>&1\n",
      "\n",
      "-- Finished on Wed Apr 21 14:18:41 2021 (lickety-split) with 315.184 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:18:41 2021 with 315.184 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./unitigging/pacbio80.gkpStore \\\n",
      "      -T ./unitigging/pacbio80.ctgStore 2 \\\n",
      "      -consensus -fasta \\\n",
      "      -unassembled \\\n",
      "    > ./pacbio80.unassembled.fasta\n",
      "    2> ./pacbio80.unassembled.err\n",
      "\n",
      "-- Finished on Wed Apr 21 14:18:42 2021 (1 second) with 315.181 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:18:42 2021 with 315.181 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./unitigging/pacbio80.gkpStore \\\n",
      "      -T ./unitigging/pacbio80.ctgStore 2 \\\n",
      "      -consensus -fasta \\\n",
      "      -contigs \\\n",
      "    > ./pacbio80.contigs.fasta\n",
      "    2> ./pacbio80.contigs.err\n",
      "\n",
      "-- Finished on Wed Apr 21 14:18:42 2021 (lickety-split) with 315.177 GB free disk space\n",
      "----------------------------------------\n",
      "----------------------------------------\n",
      "-- Starting command on Wed Apr 21 14:18:42 2021 with 315.177 GB free disk space\n",
      "\n",
      "    cd .\n",
      "    /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/libexec/tgStoreDump \\\n",
      "      -G ./unitigging/pacbio80.gkpStore \\\n",
      "      -T ./unitigging/pacbio80.utgStore 2 \\\n",
      "      -consensus -fasta \\\n",
      "      -contigs \\\n",
      "    > ./pacbio80.unitigs.fasta\n",
      "    2> ./pacbio80.unitigs.err\n",
      "\n",
      "-- Finished on Wed Apr 21 14:18:42 2021 (lickety-split) with 315.172 GB free disk space\n",
      "----------------------------------------\n",
      "-- Finished stage 'generateOutputs', reset canuIteration.\n",
      "--\n",
      "-- Assembly 'pacbio80' finished in '/Users/olga/Documents/BioinformaticsInstitute/NGS/pacbio80'.\n",
      "--\n",
      "-- Summary saved in 'unitigging.html'.\n",
      "--\n",
      "-- Sequences saved:\n",
      "--   Contigs       -> 'pacbio80.contigs.fasta'\n",
      "--   Unassembled   -> 'pacbio80.unassembled.fasta'\n",
      "--   Unitigs       -> 'pacbio80.unitigs.fasta'\n",
      "--\n",
      "-- Read layouts saved:\n",
      "--   Contigs       -> 'pacbio80.contigs.layout'.\n",
      "--   Unitigs       -> 'pacbio80.unitigs.layout'.\n",
      "--\n",
      "-- Graphs saved:\n",
      "--   Contigs       -> 'pacbio80.contigs.gfa'.\n",
      "--   Unitigs       -> 'pacbio80.unitigs.gfa'.\n",
      "--\n",
      "-- Bye.\n",
      "-- Finished stage 'outputSequence', reset canuIteration.\n"
     ]
    }
   ],
   "source": [
    "! /Users/olga/miniconda3/pkgs/canu-1.5-pl5.22.0_1/bin/canu -p pacbio80 -d pacbio80 genomesize=4.7m -pacbio-raw ./homework6/pacbio_80x.fq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2. Сборка коротких ридов"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Собираем с помощью Spades"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "== Warning ==  output dir is not empty! Please, clean output directory before run.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "== Warning ==  No assembly mode was specified! If you intend to assemble high-coverage multi-cell/isolate data, use '--isolate' option.\n",
      "\n",
      "\n",
      "Command line: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades.py\t-1\t/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq\t-2\t/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq\t-o\t/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.15.2\n",
      "  Python version: 3.7.7\n",
      "  OS: Darwin-20.3.0-x86_64-i386-64bit\n",
      "\n",
      "Output dir: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Standard mode\n",
      "  For multi-cell/isolate data we recommend to use '--isolate' option; for single-cell MDA data use '--sc'; for metagenomic data use '--meta'; for RNA-Seq use '--rna'.\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq']\n",
      "      right reads: ['/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: automatic selection based on read length\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 250\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/spades.log\n",
      "\n",
      "/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq: max reads length: 100\n",
      "/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq: max reads length: 100\n",
      "\n",
      "Reads length: 100\n",
      "\n",
      "\n",
      "===== Before start started. \n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-hammer /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  76)   Loading config from /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/corrected/configs/config.info\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (hammer_tools.cpp          :  38)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 16 files using 1 threads. This might take a while.\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 4194304\n",
      "  0:00:00.001   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq\n",
      "  0:00:10.366   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 467314 reads\n",
      "  0:00:20.064   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 935009 reads\n",
      "  0:00:29.916   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 1402861 reads\n",
      "  0:00:39.701   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 1870646 reads\n",
      "  0:00:49.555   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2339377 reads\n",
      "  0:00:53.132   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2500000 reads\n",
      "  0:00:53.132   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq\n",
      "  0:01:03.113   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2974795 reads\n",
      "  0:01:32.809   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 4400968 reads\n",
      "  0:01:45.296   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 5000000 reads processed\n",
      "  0:01:45.296     1M / 162M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:01:51.391     1M / 170M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 38919786 kmers in total.\n",
      "  0:01:51.391     1M / 170M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:55.551    30M / 195M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 38919786 kmers, 28121944 bytes occupied (5.78049 bits per kmer).\n",
      "  0:01:55.552    30M / 195M  INFO   K-mer Counting           (kmer_data.cpp             : 354)   Arranging kmers in hash map order\n",
      "  0:01:58.133   631M / 789M  INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:06:08.578   631M / 789M  INFO    General                 (main.cpp                  : 155)   Extracting clusters:\n",
      "  0:06:08.578   631M / 789M  INFO    General                 (concurrent_dsu.cpp        :  18)   Connecting to root\n",
      "  0:06:09.303   631M / 789M  INFO    General                 (concurrent_dsu.cpp        :  34)   Calculating counts\n",
      "  0:06:21.915   903M / 1061M INFO    General                 (concurrent_dsu.cpp        :  63)   Writing down entries\n",
      "  0:06:35.554   631M / 789M  INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 13651605\n",
      "  0:06:35.584   330M / 492M  INFO   K-mer Counting           (kmer_data.cpp             : 371)   Collecting K-mer information, this takes a while.\n",
      "  0:06:36.001  1223M / 1383M INFO   K-mer Counting           (kmer_data.cpp             : 377)   Processing /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq\n",
      "  0:11:20.671  1223M / 1383M INFO   K-mer Counting           (kmer_data.cpp             : 377)   Processing /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq\n",
      "  0:15:38.327  1223M / 1383M INFO   K-mer Counting           (kmer_data.cpp             : 384)   Collection done, postprocessing.\n",
      "  0:15:38.566  1223M / 1383M INFO   K-mer Counting           (kmer_data.cpp             : 398)   There are 38919786 kmers in total. Among them 27377044 (70.3422%) are singletons.\n",
      "  0:15:38.566  1223M / 1383M INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:16:21.104  1223M / 1487M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering done. Total 26 non-read kmers were generated.\n",
      "  0:16:21.104  1223M / 1487M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)   Subclustering statistics:\n",
      "  0:16:21.104  1223M / 1487M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton hamming clusters: 5541662. Among them 963110 (17.3794%) are good\n",
      "  0:16:21.104  1223M / 1487M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total singleton subclusters: 13747. Among them 13599 (98.9234%) are good\n",
      "  0:16:21.104  1223M / 1487M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Total non-singleton subcluster centers: 8197112. Among them 6855795 (83.6367%) are good\n",
      "  0:16:21.104  1223M / 1487M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average size of non-trivial subcluster: 4.07194 kmers\n",
      "  0:16:21.104  1223M / 1487M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Average number of sub-clusters per non-singleton cluster: 1.01244\n",
      "  0:16:21.104  1223M / 1487M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Total solid k-mers: 7832504\n",
      "  0:16:21.104  1223M / 1487M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 658)     Substitution probabilities: [4,4]((0.928457,0.0421442,0.022541,0.00685828),(0.0141904,0.967828,0.0090854,0.00889584),(0.00886597,0.00903879,0.967867,0.0142285),(0.00683121,0.022511,0.0420995,0.928558))\n",
      "  0:16:21.113  1223M / 1383M INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:16:21.113  1223M / 1383M INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 1 threads.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:18:48.066  1223M / 1383M INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 39977 new k-mers.\n",
      "  0:21:14.786  1223M / 1383M INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 790 new k-mers.\n",
      "  0:23:41.283  1223M / 1383M INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 38 new k-mers.\n",
      "  0:39:46.864  1223M / 1383M INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 3 produced 0 new k-mers.\n",
      "  0:39:46.864  1223M / 1383M INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:39:46.864  1223M / 1383M INFO    General                 (hammer_tools.cpp          : 222)   Starting read correction in 1 threads.\n",
      "  0:39:46.864  1223M / 1383M INFO    General                 (hammer_tools.cpp          : 235)   Correcting pair of reads: /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq and /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq\n",
      "  0:39:47.036  1289M / 1394M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 0 of 100000 reads.\n",
      "  0:39:53.546  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 0\n",
      "  0:39:53.764  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 0\n",
      "  0:39:53.914  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 1 of 100000 reads.\n",
      "  0:40:00.644  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 1\n",
      "  0:40:00.884  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 1\n",
      "  0:40:01.033  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 2 of 100000 reads.\n",
      "  0:40:07.689  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 2\n",
      "  0:40:07.923  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 2\n",
      "  0:40:08.075  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 3 of 100000 reads.\n",
      "  0:40:14.884  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 3\n",
      "  0:40:15.117  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 3\n",
      "  0:40:15.266  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 4 of 100000 reads.\n",
      "  0:40:21.776  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 4\n",
      "  0:40:22.008  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 4\n",
      "  0:40:22.148  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 5 of 100000 reads.\n",
      "  0:40:28.380  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 5\n",
      "  0:40:28.609  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 5\n",
      "  0:40:28.749  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 6 of 100000 reads.\n",
      "  0:40:35.014  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 6\n",
      "  0:40:35.244  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 6\n",
      "  0:40:35.384  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 7 of 100000 reads.\n",
      "  0:40:41.607  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 7\n",
      "  0:40:41.827  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 7\n",
      "  0:40:41.969  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 8 of 100000 reads.\n",
      "  0:40:48.175  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 8\n",
      "  0:40:48.397  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 8\n",
      "  0:40:48.538  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 9 of 100000 reads.\n",
      "  0:40:54.800  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 9\n",
      "  0:40:55.031  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 9\n",
      "  0:40:55.173  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 10 of 100000 reads.\n",
      "  0:41:01.396  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 10\n",
      "  0:41:01.630  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 10\n",
      "  0:41:01.771  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 11 of 100000 reads.\n",
      "  0:41:08.557  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 11\n",
      "  0:41:08.794  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 11\n",
      "  0:41:08.948  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 12 of 100000 reads.\n",
      "  0:41:15.399  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 12\n",
      "  0:41:15.629  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 12\n",
      "  0:41:15.776  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 13 of 100000 reads.\n",
      "  0:41:22.052  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 13\n",
      "  0:41:22.284  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 13\n",
      "  0:41:22.428  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 14 of 100000 reads.\n",
      "  0:41:28.711  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 14\n",
      "  0:41:28.946  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 14\n",
      "  0:41:29.092  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 15 of 100000 reads.\n",
      "  0:41:35.502  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 15\n",
      "  0:41:35.752  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 15\n",
      "  0:41:35.898  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 16 of 100000 reads.\n",
      "  0:41:42.724  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 16\n",
      "  0:41:42.968  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 16\n",
      "  0:41:43.123  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 17 of 100000 reads.\n",
      "  0:41:49.737  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 17\n",
      "  0:41:50.003  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 17\n",
      "  0:41:50.147  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 18 of 100000 reads.\n",
      "  0:41:56.796  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 18\n",
      "  0:41:57.033  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 18\n",
      "  0:41:57.178  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 19 of 100000 reads.\n",
      "  0:42:03.471  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 19\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:42:03.711  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 19\n",
      "  0:42:03.858  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 20 of 100000 reads.\n",
      "  0:42:10.454  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 20\n",
      "  0:42:10.690  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 20\n",
      "  0:42:10.835  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 21 of 100000 reads.\n",
      "  0:42:17.148  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 21\n",
      "  0:42:17.382  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 21\n",
      "  0:42:17.527  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 22 of 100000 reads.\n",
      "  0:42:23.840  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 22\n",
      "  0:42:24.075  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 22\n",
      "  0:42:24.222  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 23 of 100000 reads.\n",
      "  0:42:30.482  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 23\n",
      "  0:42:30.722  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 23\n",
      "  0:42:30.867  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 24 of 100000 reads.\n",
      "  0:42:37.501  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 24\n",
      "  0:42:37.741  1293M / 1395M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 24\n",
      "  0:42:37.749  1223M / 1395M INFO    General                 (hammer_tools.cpp          : 276)   Correction done. Changed 1639742 bases in 1194871 reads.\n",
      "  0:42:37.749  1223M / 1395M INFO    General                 (hammer_tools.cpp          : 277)   Failed to correct 44220 bases out of 458864877.\n",
      "  0:42:37.832     1M / 208M  INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/corrected/corrected.yaml\n",
      "  0:42:37.832     1M / 208M  INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== corrected reads compression started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/miniconda3/bin/python /Users/olga/Downloads/SPAdes-3.15.2/share/spades/spades_pipeline/scripts/compress_all.py --input_file /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/corrected/corrected.yaml --ext_python_modules_home /Users/olga/Downloads/SPAdes-3.15.2/share/spades --max_threads 16 --output_dir /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/corrected --gzip_output\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "== Files to compress: ['/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/corrected/illumina.100x.1.00.0_0.cor.fastq', '/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/corrected/illumina.100x.2.00.0_0.cor.fastq', '/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/corrected/illumina.100x._unpaired.00.0_0.cor.fastq']\n",
      "== Files compression is finished\n",
      "== Dataset yaml file is updated\n",
      "\n",
      "===== corrected reads compression finished. \n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started. \n",
      "\n",
      "\n",
      "===== K21 started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-core /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K21/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  :  99)   Loaded config from /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K21/configs/config.info\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 109)   Assembling dataset (/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/dataset.info) with K=21\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  72)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  73)   Converting paired reads\n",
      "  0:00:00.061    22M / 22M   INFO    General                 (binary_converter.cpp      :  96)   16384 reads processed\n",
      "  0:00:00.128    25M / 25M   INFO    General                 (binary_converter.cpp      :  96)   32768 reads processed\n",
      "  0:00:00.333    22M / 25M   INFO    General                 (binary_converter.cpp      :  96)   65536 reads processed\n",
      "  0:00:00.652    25M / 31M   INFO    General                 (binary_converter.cpp      :  96)   131072 reads processed\n",
      "  0:00:01.366    21M / 31M   INFO    General                 (binary_converter.cpp      :  96)   262144 reads processed\n",
      "  0:00:02.655    24M / 31M   INFO    General                 (binary_converter.cpp      :  96)   524288 reads processed\n",
      "  0:00:05.252    28M / 31M   INFO    General                 (binary_converter.cpp      :  96)   1048576 reads processed\n",
      "  0:00:10.512    28M / 31M   INFO    General                 (binary_converter.cpp      :  96)   2097152 reads processed\n",
      "  0:00:12.352    19M / 31M   INFO    General                 (binary_converter.cpp      : 111)   2448672 reads written\n",
      "  0:00:12.352     1M / 31M   INFO    General                 (read_converter.cpp        :  81)   Converting single reads\n",
      "  0:00:12.385    10M / 31M   INFO    General                 (binary_converter.cpp      :  96)   16384 reads processed\n",
      "  0:00:12.414    12M / 31M   INFO    General                 (binary_converter.cpp      :  96)   32768 reads processed\n",
      "  0:00:12.486     9M / 31M   INFO    General                 (binary_converter.cpp      : 111)   48607 reads written\n",
      "  0:00:12.486     1M / 31M   INFO    General                 (read_converter.cpp        :  87)   Converting merged reads\n",
      "  0:00:12.486     9M / 31M   INFO    General                 (binary_converter.cpp      : 111)   0 reads written\n",
      "  0:00:12.487     1M / 31M   INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:12.488     1M / 32M   INFO    General                 (construction.cpp          : 153)   Max read length 100\n",
      "  0:00:12.488     1M / 32M   INFO    General                 (construction.cpp          : 159)   Average read length 92.7147\n",
      "  0:00:12.488     1M / 32M   INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:12.488     1M / 32M   INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:12.488     1M / 32M   INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:12.488     1M / 32M   INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:12.488     1M / 32M   INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 6710886\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:22.243   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 1873811 reads\n",
      "  0:00:31.704   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 3734243 reads\n",
      "  0:00:41.172   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 5595232 reads\n",
      "  0:00:50.596   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 7456112 reads\n",
      "  0:01:00.022   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 9318931 reads\n",
      "  0:01:03.133   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 9891902 reads\n",
      "  0:01:03.133     1M / 161M  INFO    General                 (kmer_splitters.hpp        : 131)   Used 9891902 reads\n",
      "  0:01:03.165     1M / 161M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:01:03.639     1M / 161M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5715581 kmers in total.\n",
      "  0:01:03.639     1M / 161M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:01:03.640     1M / 161M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:01:03.640     1M / 161M  INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:01:03.641     1M / 161M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:01:03.641     1M / 161M  INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:01:03.641     1M / 161M  INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 6710886\n",
      "  0:01:05.438   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 194)   Processed 5715581 kmers\n",
      "  0:01:05.438   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 199)   Used 5715581 kmers.\n",
      "  0:01:05.438     1M / 204M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:01:05.618     1M / 204M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5708731 kmers in total.\n",
      "  0:01:05.618     1M / 204M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:06.056     5M / 204M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5708731 kmers, 4130592 bytes occupied (5.78846 bits per kmer).\n",
      "  0:01:06.056     5M / 204M  INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n",
      "  0:01:06.200    11M / 204M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:01:07.992    11M / 204M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:01:07.993    11M / 161M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Early tip clipping (id: construction:early_tip_clipper)\n",
      "  0:01:07.993    11M / 161M  INFO    General                 (construction.cpp          : 301)   Early tip clipper length bound set as (RL - K)\n",
      "  0:01:07.993    11M / 161M  INFO   Early tip clipping       (early_simplification.hpp  :  47)   Early tip clipping\n",
      "  0:01:09.701    21M / 204M  INFO   Early tip clipping       (early_simplification.hpp  :  82)   #tipped junctions: 123493\n",
      "  0:01:09.745    21M / 204M  INFO   Early tip clipping       (early_simplification.hpp  :  93)   Clipped tips: 124241\n",
      "  0:01:09.746    11M / 161M  INFO   Early tip clipping       (early_simplification.hpp  :  49)   978561 22-mers were removed by early tip clipper\n",
      "  0:01:09.746    11M / 161M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:01:09.750    11M / 161M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:01:12.001    13M / 205M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 25730 sequences extracted\n",
      "  0:01:12.770    13M / 161M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:01:13.169    13M / 205M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 4 loops collected\n",
      "  0:01:13.171    13M / 161M  INFO    General                 (debruijn_graph_constructor: 487)   Total 51468 edges to create\n",
      "  0:01:13.171    15M / 161M  INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:01:13.185    16M / 161M  INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:01:13.187    16M / 161M  INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:01:13.188    16M / 161M  INFO    General                 (debruijn_graph_constructor: 503)   Total 18884 vertices to create\n",
      "  0:01:13.188    18M / 161M  INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:01:13.190    16M / 161M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:01:13.190    16M / 161M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:13.624    20M / 161M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5715581 kmers, 4135544 bytes occupied (5.78845 bits per kmer).\n",
      "  0:01:13.628    42M / 168M  INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:02:10.219    42M / 168M  INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:02:11.660    42M / 168M  INFO    General                 (coverage_filling.hpp      :  82)   Processed 51447 edges\n",
      "  0:02:11.673     6M / 168M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:02:11.673     6M / 168M  INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 16\n",
      "  0:02:11.673     6M / 168M  INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 76\n",
      "  0:02:11.673     6M / 168M  INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 77. Coverage mad: 14.826\n",
      "  0:02:11.673     6M / 168M  INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:02:11.738     6M / 168M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:02:11.900     6M / 168M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:02:12.435     6M / 168M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:02:13.668     6M / 168M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:02:13.802     6M / 168M  INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 76.252. Fitted coverage std. dev: 14.5028\n",
      "  0:02:13.804     6M / 168M  INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.863705\n",
      "  0:02:13.804     6M / 168M  INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 37\n",
      "  0:02:13.804     6M / 168M  INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 37\n",
      "  0:02:13.804     6M / 168M  INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 4507902\n",
      "  0:02:13.804     6M / 168M  INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 76.252\n",
      "  0:02:13.804     6M / 168M  INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 37\n",
      "  0:02:13.804     6M / 168M  INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:02:13.804     6M / 168M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:02:13.804     6M / 168M  INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:02:13.804     6M / 168M  INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:02:13.804     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:02:13.806     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:02:13.806     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:02:13.825     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 2371 times\n",
      "  0:02:13.825     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n",
      "  0:02:13.897     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 4029 times\n",
      "  0:02:13.897     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:02:13.899     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 308 times\n",
      "  0:02:13.899     6M / 168M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:02:13.899     6M / 168M  INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:02:13.899     6M / 168M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:02:13.899     6M / 168M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:02:13.899     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:13.901     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 128 times\n",
      "  0:02:13.901     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:13.970     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 1810 times\n",
      "  0:02:13.970     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:13.975     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 143 times\n",
      "  0:02:13.975     6M / 168M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:02:13.975     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:13.975     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 5 times\n",
      "  0:02:13.975     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:13.977     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 36 times\n",
      "  0:02:13.977     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:13.978     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 68 times\n",
      "  0:02:13.978     6M / 168M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:02:13.978     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:13.978     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 3 times\n",
      "  0:02:13.978     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:13.979     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 23 times\n",
      "  0:02:13.979     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:13.979     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 16 times\n",
      "  0:02:13.979     6M / 168M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:02:13.979     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:13.979     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:13.979     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 7 times\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 10 times\n",
      "  0:02:13.980     6M / 168M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 2 times\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 4 times\n",
      "  0:02:13.980     6M / 168M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 4 times\n",
      "  0:02:13.980     6M / 168M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 1 times\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 2 times\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 3 times\n",
      "  0:02:13.980     6M / 168M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 7 times\n",
      "  0:02:13.980     6M / 168M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:02:13.980     6M / 168M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:13.980     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:13.981     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 6 times\n",
      "  0:02:13.981     6M / 168M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:02:13.981     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:13.981     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:13.981     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:13.983     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 3 times\n",
      "  0:02:13.983     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:13.984     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:13.984     6M / 168M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:02:13.984     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:13.984     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:13.984     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:13.984     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:13.984     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:13.984     6M / 168M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:13.984     6M / 168M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
      "  0:02:13.984     6M / 168M  INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\n",
      "  0:02:13.984     6M / 169M  INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\n",
      "  0:02:13.984     6M / 169M  INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\n",
      "  0:02:13.984     6M / 169M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:02:13.984     6M / 169M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:13.984     6M / 169M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:13.984     6M / 169M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:13.986     6M / 169M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:13.986     6M / 169M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:13.987     6M / 169M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:13.987     6M / 169M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:13.989     6M / 169M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:13.989     6M / 169M  INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:02:13.989     6M / 169M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:02:13.990     6M / 169M  INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 7 times\n",
      "  0:02:13.990     6M / 169M  INFO    General                 (simplification.cpp        : 491)   Counting average coverage\n",
      "  0:02:13.990     6M / 169M  INFO    General                 (simplification.cpp        : 497)   Average coverage = 78.0189\n",
      "  0:02:13.990     6M / 169M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:02:13.990     6M / 169M  INFO    General                 (read_converter.cpp        : 102)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K21/simplified_contigs\n",
      "  0:02:13.993     8M / 169M  INFO    General                 (binary_converter.cpp      : 111)   2571 reads written\n",
      "  0:02:13.994     6M / 169M  INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  0:02:13.994     1M / 169M  INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 2 minutes 13 seconds\n",
      "\n",
      "===== K21 finished. \n",
      "\n",
      "\n",
      "===== K33 started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-core /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K33/configs/config.info\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  :  99)   Loaded config from /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K33/configs/config.info\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 109)   Assembling dataset (/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/dataset.info) with K=33\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 118)   Contigs from previous K will be used: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K21/simplified_contigs\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 153)   Max read length 100\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 159)   Average read length 92.7147\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 3355443\n",
      "  0:00:05.697   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 1132634 reads\n",
      "  0:00:11.043   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 2249495 reads\n",
      "  0:00:16.376   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 3368100 reads\n",
      "  0:00:21.715   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 4487032 reads\n",
      "  0:00:27.026   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 5607299 reads\n",
      "  0:00:32.331   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 6726813 reads\n",
      "  0:00:37.645   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 7846654 reads\n",
      "  0:00:42.971   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 8966769 reads\n",
      "  0:00:48.079   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 9897044 reads\n",
      "  0:00:48.080     1M / 158M  INFO    General                 (kmer_splitters.hpp        : 131)   Used 9897044 reads\n",
      "  0:00:48.109     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:49.119     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5756125 kmers in total.\n",
      "  0:00:49.119     1M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:00:49.121     1M / 158M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:00:49.121     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:49.121     1M / 158M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:49.121     1M / 158M  INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:49.121     1M / 158M  INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 3355443\n",
      "  0:00:51.287   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 194)   Processed 5756125 kmers\n",
      "  0:00:51.287   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 199)   Used 5756125 kmers.\n",
      "  0:00:51.287     1M / 246M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:51.601     1M / 246M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5754745 kmers in total.\n",
      "  0:00:51.601     1M / 246M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:52.120     5M / 246M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5754745 kmers, 4163816 bytes occupied (5.78836 bits per kmer).\n",
      "  0:00:52.120     5M / 246M  INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n",
      "  0:00:52.383    11M / 246M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:00:54.286    11M / 246M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:54.287    11M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Early tip clipping (id: construction:early_tip_clipper)\n",
      "  0:00:54.287    11M / 158M  INFO    General                 (construction.cpp          : 301)   Early tip clipper length bound set as (RL - K)\n",
      "  0:00:54.287    11M / 158M  INFO   Early tip clipping       (early_simplification.hpp  :  47)   Early tip clipping\n",
      "  0:00:56.033    21M / 246M  INFO   Early tip clipping       (early_simplification.hpp  :  82)   #tipped junctions: 124481\n",
      "  0:00:56.073    21M / 246M  INFO   Early tip clipping       (early_simplification.hpp  :  93)   Clipped tips: 125222\n",
      "  0:00:56.075    11M / 158M  INFO   Early tip clipping       (early_simplification.hpp  :  49)   1029171 34-mers were removed by early tip clipper\n",
      "  0:00:56.075    11M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:00:56.083    11M / 158M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:00:58.343    12M / 248M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 15754 sequences extracted\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:59.173    12M / 160M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:00:59.598    12M / 248M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 2 loops collected\n",
      "  0:00:59.600    12M / 160M  INFO    General                 (debruijn_graph_constructor: 487)   Total 31512 edges to create\n",
      "  0:00:59.600    14M / 160M  INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:00:59.607    14M / 160M  INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:00:59.609    14M / 160M  INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:00:59.609    14M / 160M  INFO    General                 (debruijn_graph_constructor: 503)   Total 14376 vertices to create\n",
      "  0:00:59.609    16M / 160M  INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:00:59.611    15M / 160M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:00:59.611    15M / 160M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:00.138    19M / 160M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5756125 kmers, 4164864 bytes occupied (5.78843 bits per kmer).\n",
      "  0:01:00.140    41M / 163M  INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:01:51.049    41M / 163M  INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:01:52.543    41M / 163M  INFO    General                 (coverage_filling.hpp      :  82)   Processed 31510 edges\n",
      "  0:01:52.562     5M / 163M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:01:52.562     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 13\n",
      "  0:01:52.562     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 64\n",
      "  0:01:52.562     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 64. Coverage mad: 14.826\n",
      "  0:01:52.562     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:01:52.617     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:01:52.756     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:01:53.213     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:01:54.267     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:01:54.931     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 63.4837. Fitted coverage std. dev: 13.4198\n",
      "  0:01:54.933     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.693899\n",
      "  0:01:54.933     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 29\n",
      "  0:01:54.933     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 29\n",
      "  0:01:54.933     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 4507034\n",
      "  0:01:54.933     5M / 163M  INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 63.4837\n",
      "  0:01:54.933     5M / 163M  INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 29\n",
      "  0:01:54.933     5M / 163M  INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:01:54.933     5M / 163M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:01:54.933     5M / 163M  INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:01:54.933     5M / 163M  INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:01:54.933     5M / 163M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:01:54.934     5M / 163M  INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:01:54.934     5M / 163M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n",
      "  0:01:54.966     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 2699 times\n",
      "  0:01:54.967     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n",
      "  0:01:55.004     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 1350 times\n",
      "  0:01:55.005     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:01:55.007     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 1603 times\n",
      "  0:01:55.007     5M / 164M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:01:55.007     5M / 164M  INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:01:55.007     5M / 164M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:01:55.007     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:01:55.007     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:55.009     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 59 times\n",
      "  0:01:55.009     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:55.076     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 1018 times\n",
      "  0:01:55.076     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:55.078     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 42 times\n",
      "  0:01:55.078     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:01:55.078     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:55.078     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 5 times\n",
      "  0:01:55.078     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:55.079     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 17 times\n",
      "  0:01:55.079     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:55.082     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 52 times\n",
      "  0:01:55.083     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:01:55.083     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:55.083     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 1 times\n",
      "  0:01:55.083     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:55.086     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 27 times\n",
      "  0:01:55.086     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:55.086     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 10 times\n",
      "  0:01:55.086     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:01:55.086     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:55.086     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:55.086     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:55.086     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 9 times\n",
      "  0:01:55.086     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:55.087     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 8 times\n",
      "  0:01:55.087     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:01:55.087     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:55.087     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:55.087     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:55.087     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 5 times\n",
      "  0:01:55.087     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 6 times\n",
      "  0:01:55.088     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:01:55.088     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 3 times\n",
      "  0:01:55.088     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:01:55.088     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:01:55.088     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:01:55.088     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:55.088     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:55.089     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 2 times\n",
      "  0:01:55.089     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:55.089     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:01:55.089     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:01:55.089     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:55.089     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:55.089     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:55.089     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:55.089     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:55.089     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:01:55.090     5M / 164M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
      "  0:01:55.090     5M / 164M  INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\n",
      "  0:01:55.090     5M / 165M  INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\n",
      "  0:01:55.090     5M / 165M  INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\n",
      "  0:01:55.090     5M / 165M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:01:55.090     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:55.090     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:55.090     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:55.090     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:55.091     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:55.091     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:55.091     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:55.091     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:55.091     5M / 165M  INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:01:55.092     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:01:55.092     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 5 times\n",
      "  0:01:55.092     5M / 165M  INFO    General                 (simplification.cpp        : 491)   Counting average coverage\n",
      "  0:01:55.092     5M / 165M  INFO    General                 (simplification.cpp        : 497)   Average coverage = 64.7522\n",
      "  0:01:55.092     5M / 165M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:01:55.092     5M / 165M  INFO    General                 (read_converter.cpp        : 102)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K33/simplified_contigs\n",
      "  0:01:55.095     7M / 165M  INFO    General                 (binary_converter.cpp      : 111)   1188 reads written\n",
      "  0:01:55.096     5M / 165M  INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  0:01:55.096     1M / 165M  INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 1 minutes 55 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "===== K33 finished. \n",
      "\n",
      "\n",
      "===== K55 started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-core /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  :  99)   Loaded config from /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/configs/config.info\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 109)   Assembling dataset (/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/dataset.info) with K=55\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 242)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 118)   Contigs from previous K will be used: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K33/simplified_contigs\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 153)   Max read length 100\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 159)   Average read length 92.7147\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 3355443\n",
      "  0:00:05.912   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 1768213 reads\n",
      "  0:00:11.463   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 3518079 reads\n",
      "  0:00:16.998   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 5268519 reads\n",
      "  0:00:22.560   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 7021543 reads\n",
      "  0:00:28.085   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 8774696 reads\n",
      "  0:00:32.360   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 9894278 reads\n",
      "  0:00:32.360     1M / 158M  INFO    General                 (kmer_splitters.hpp        : 131)   Used 9894278 reads\n",
      "  0:00:32.394     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:33.112     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5692221 kmers in total.\n",
      "  0:00:33.112     1M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:00:33.114     1M / 158M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:00:33.114     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:33.114     1M / 158M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:33.114     1M / 158M  INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:33.114     1M / 158M  INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 3355443\n",
      "  0:00:35.243   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 194)   Processed 5692221 kmers\n",
      "  0:00:35.243   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 199)   Used 5692221 kmers.\n",
      "  0:00:35.243     1M / 245M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:35.556     1M / 245M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5698724 kmers in total.\n",
      "  0:00:35.556     1M / 245M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:36.080     5M / 245M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5698724 kmers, 4123376 bytes occupied (5.78849 bits per kmer).\n",
      "  0:00:36.080     5M / 245M  INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n",
      "  0:00:36.346    11M / 245M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:00:38.247    11M / 245M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:38.248    11M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:00:38.256    11M / 158M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:00:41.225    23M / 248M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 247400 sequences extracted\n",
      "  0:00:42.306    23M / 161M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:00:42.722    23M / 248M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:42.727    23M / 161M  INFO    General                 (debruijn_graph_constructor: 487)   Total 494800 edges to create\n",
      "  0:00:42.727    42M / 161M  INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:00:42.866    50M / 172M  INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:00:42.901    50M / 172M  INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:00:42.905    52M / 172M  INFO    General                 (debruijn_graph_constructor: 503)   Total 253903 vertices to create\n",
      "  0:00:42.905    76M / 172M  INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:42.944    62M / 172M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:00:42.944    62M / 172M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:43.462    66M / 172M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5692221 kmers, 4118704 bytes occupied (5.78854 bits per kmer).\n",
      "  0:00:43.463    88M / 172M  INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:01:18.161    88M / 172M  INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:01:20.293    88M / 172M  INFO    General                 (coverage_filling.hpp      :  82)   Processed 494800 edges\n",
      "  0:01:20.311    52M / 172M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:01:20.311    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 6\n",
      "  0:01:20.311    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 41\n",
      "  0:01:20.311    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 41. Coverage mad: 11.8608\n",
      "  0:01:20.311    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:01:20.345    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:01:20.436    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:01:20.725    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:01:21.419    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:01:21.902    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 40.4747. Fitted coverage std. dev: 10.9026\n",
      "  0:01:21.903    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.512008\n",
      "  0:01:21.903    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 16\n",
      "  0:01:21.903    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 16\n",
      "  0:01:21.903    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 4487959\n",
      "  0:01:21.903    52M / 172M  INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 40.4747\n",
      "  0:01:21.903    52M / 172M  INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 16\n",
      "  0:01:21.903    52M / 172M  INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:01:21.903    52M / 172M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Gap Closer (id: early_gapcloser)\n",
      "  0:01:21.905    61M / 172M  INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:01:21.986    66M / 172M  INFO    General                 (gap_closer.cpp            : 101)   Total edges in tip neighborhood: 134528 out of 494800, length: 1226132\n",
      "  0:01:21.996    68M / 172M  INFO    General                 (edge_index.hpp            : 167)   Using small index (max_id = 497276)\n",
      "  0:01:22.097    68M / 172M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:22.554    69M / 172M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 1092761 kmers, 790000 bytes occupied (5.78352 bits per kmer).\n",
      "  0:01:22.554    78M / 172M  INFO    General                 (edge_index_builders.hpp   : 265)   Collecting edge information from graph, this takes a while.\n",
      "  0:01:26.479    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:01:30.990    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:01:35.472    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:01:39.999    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:01:44.442    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:01:48.872    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:02:11.117    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:02:16.617    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:02:16.622    62M / 172M  INFO    General                 (gap_closer.cpp            : 452)   Initializing gap closer\n",
      "  0:02:16.622    62M / 172M  INFO   GapCloser                (gap_closer.cpp            : 359)   Closing short gaps\n",
      "  0:02:16.827    62M / 178M  INFO   GapCloser                (gap_closer.cpp            : 393)   Closing short gaps complete: filled 0 gaps after checking 20 candidates\n",
      "  0:02:16.834    62M / 178M  INFO    General                 (gap_closer.cpp            : 456)   Gap closer done\n",
      "  0:02:16.835    52M / 178M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:02:16.835    52M / 178M  INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:02:16.835    52M / 178M  INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:02:16.835    52M / 178M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:02:16.865    52M / 178M  INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:02:16.865    52M / 178M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n",
      "  0:02:17.621    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 119766 times\n",
      "  0:02:17.621    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n",
      "  0:02:17.628    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 92 times\n",
      "  0:02:17.628    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:02:17.638    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 7236 times\n",
      "  0:02:17.639    45M / 179M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:02:17.639    45M / 179M  INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:02:17.639    45M / 179M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:02:17.639    45M / 179M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:02:17.639    45M / 179M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.640    45M / 179M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 19 times\n",
      "  0:02:17.640    45M / 179M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.739    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 447 times\n",
      "  0:02:17.739    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.740    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:17.740    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:02:17.740    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.740    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 3 times\n",
      "  0:02:17.740    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.740    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:17.740    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.741    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 13 times\n",
      "  0:02:17.741    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:02:17.741    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.741    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 1 times\n",
      "  0:02:17.741    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.741    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 8 times\n",
      "  0:02:17.741    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.743    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 11 times\n",
      "  0:02:17.743    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:02:17.743    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.743    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 3 times\n",
      "  0:02:17.743    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.743    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 7 times\n",
      "  0:02:17.743    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.743    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:02:17.743    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:02:17.743    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.743    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:17.743    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.743    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:17.743    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 3 times\n",
      "  0:02:17.744    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 1 times\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:02:17.744    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:02:17.744    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:17.744    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:17.744    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.746    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:02:17.746    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:02:17.746    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.746    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:17.746    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.746    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:17.746    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.746    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:02:17.746    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:02:17.746    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.747    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:17.747    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.747    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:17.747    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.748    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:17.748    48M / 180M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Gap Closer (id: late_gapcloser)\n",
      "  0:02:17.750    57M / 180M  INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:02:17.751    57M / 180M  INFO    General                 (gap_closer.cpp            : 101)   Total edges in tip neighborhood: 15 out of 882, length: 167704\n",
      "  0:02:17.751    57M / 180M  INFO    General                 (edge_index.hpp            : 167)   Using small index (max_id = 497276)\n",
      "  0:02:17.756    57M / 180M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:02:17.785    57M / 180M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 166932 kmers, 121312 bytes occupied (5.81372 bits per kmer).\n",
      "  0:02:17.786    58M / 180M  INFO    General                 (edge_index_builders.hpp   : 265)   Collecting edge information from graph, this takes a while.\n",
      "  0:02:20.355    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:02:22.854    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:02:25.348    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:02:27.873    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:02:30.366    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:02:32.868    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:02:45.352    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:02:48.461    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:02:48.463    57M / 180M  INFO    General                 (gap_closer.cpp            : 452)   Initializing gap closer\n",
      "  0:02:48.464    57M / 180M  INFO   GapCloser                (gap_closer.cpp            : 359)   Closing short gaps\n",
      "  0:02:48.464    57M / 180M  INFO   GapCloser                (gap_closer.cpp            : 393)   Closing short gaps complete: filled 1 gaps after checking 7 candidates\n",
      "  0:02:48.465    57M / 180M  INFO    General                 (gap_closer.cpp            : 456)   Gap closer done\n",
      "  0:02:48.466    48M / 180M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
      "  0:02:48.466    48M / 180M  INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\n",
      "  0:02:48.466    48M / 180M  INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\n",
      "  0:02:48.466    48M / 180M  INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\n",
      "  0:02:48.466    48M / 180M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:02:48.466    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:48.467    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:48.467    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:48.468    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:48.468    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:48.469    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:48.469    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:48.470    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:48.470    48M / 180M  INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:02:48.471    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:02:48.472    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 1 times\n",
      "  0:02:48.472    48M / 180M  INFO    General                 (simplification.cpp        : 491)   Counting average coverage\n",
      "  0:02:48.472    48M / 180M  INFO    General                 (simplification.cpp        : 497)   Average coverage = 41.1665\n",
      "  0:02:48.472    48M / 180M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Mismatch Correction (id: mismatch_correction)\n",
      "  0:02:48.472    48M / 180M  INFO    General                 (graph_pack.cpp            :  67)   Index refill\n",
      "  0:02:48.472    48M / 180M  INFO    General                 (edge_index.hpp            : 156)   Using small index (max_id = 497276)\n",
      "  0:02:48.643    48M / 180M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:02:49.667    51M / 180M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 4555847 kmers, 3291264 bytes occupied (5.77941 bits per kmer).\n",
      "  0:02:49.672    87M / 185M  INFO    General                 (edge_index_builders.hpp   : 252)   Collecting edge information from graph, this takes a while.\n",
      "  0:02:50.684    87M / 185M  INFO    General                 (graph_pack.cpp            :  77)   Normalizing k-mer map. Total 56128 kmers to process\n",
      "  0:02:50.701    87M / 185M  INFO    General                 (graph_pack.cpp            :  79)   Normalizing done\n",
      "  0:02:50.701    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 400)   Collect potential mismatches\n",
      "  0:02:50.755    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 193)   Total 354 edges (out of 878) with 1752 potential mismatch positions (4.94915 positions per edge)\n",
      "  0:02:50.755    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 402)   Potential mismatches collected\n",
      "  0:02:51.131    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:02:51.534    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:02:51.928    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:02:52.326    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:02:52.722    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:02:53.115    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:02:55.067    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:02:58.962    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 4200000 reads\n",
      "  0:03:07.124    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 8400000 reads\n",
      "  0:03:10.054    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 9891902 reads processed\n",
      "  0:03:10.090    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 395)   All edges processed\n",
      "  0:03:10.090    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 450)   Corrected 3 nucleotides\n",
      "  0:03:10.090    87M / 185M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:03:10.090    87M / 185M  INFO    General                 (contig_output.hpp         :  16)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/before_rr.fasta\n",
      "  0:03:10.120    87M / 186M  INFO    General                 (contig_output_stage.cpp   : 151)   Writing GFA graph to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/assembly_graph_after_simplification.gfa\n",
      "  0:03:10.145    87M / 186M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Paired Information Counting (id: late_pair_info_count)\n",
      "  0:03:10.146    87M / 186M  INFO    General                 (graph_pack.cpp            :  77)   Normalizing k-mer map. Total 56246 kmers to process\n",
      "  0:03:10.159    87M / 186M  INFO    General                 (graph_pack.cpp            :  79)   Normalizing done\n",
      "  0:03:10.160    87M / 186M  INFO    General                 (pair_info_count.cpp       : 339)   Min edge length for estimation: 61753\n",
      "  0:03:10.160    87M / 186M  INFO    General                 (pair_info_count.cpp       : 350)   Estimating insert size for library #0\n",
      "  0:03:10.160    87M / 186M  INFO    General                 (pair_info_count.cpp       : 202)   Estimating insert size (takes a while)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:03:10.226   119M / 186M  INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  0:03:10.839   119M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:03:11.499   119M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:03:12.154   119M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:03:12.800   119M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:03:13.451   119M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:03:14.098   119M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:03:17.324   119M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:03:18.150   119M / 186M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:03:18.196   119M / 186M  INFO    General                 (pair_info_count.cpp       : 220)   Edge pairs: 3670\n",
      "  0:03:18.196   119M / 186M  INFO    General                 (pair_info_count.cpp       : 224)   1128696 paired reads (46.0942% of all) aligned to long edges\n",
      "  0:03:18.196    87M / 186M  INFO    General                 (pair_info_count.cpp       : 373)     Insert size = 215.4, deviation = 10.1891, left quantile = 202, right quantile = 229, read length = 100\n",
      "  0:03:18.196    87M / 186M  INFO    General                 (pair_info_count.cpp       : 390)   Filtering data for library #0\n",
      "  0:03:18.196    87M / 186M  INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  0:03:18.843    87M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:03:19.485    87M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:03:20.123    87M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:03:20.765    87M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:03:21.415    87M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:03:22.056    87M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:03:25.335    87M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:03:26.253    87M / 186M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:03:26.253    87M / 186M  INFO    General                 (pair_info_count.cpp       : 402)   Mapping library #0\n",
      "  0:03:26.253    87M / 186M  INFO    General                 (pair_info_count.cpp       : 404)   Mapping paired reads (takes a while)\n",
      "  0:03:26.253    87M / 186M  INFO    General                 (pair_info_count.cpp       : 302)   Left insert size quantile 202, right insert size quantile 229, filtering threshold 2, rounding threshold 0\n",
      "  0:03:26.255    96M / 186M  INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  0:03:27.040    96M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:03:27.764    96M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:03:28.460    96M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:03:29.161    96M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:03:29.848    97M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:03:30.532    97M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:03:33.983    97M / 186M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:03:34.835    97M / 186M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:03:34.838    88M / 186M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Distance Estimation (id: distance_estimation)\n",
      "  0:03:34.838    88M / 186M  INFO    General                 (distance_estimation.cpp   : 179)   Processing library #0\n",
      "  0:03:34.838    88M / 186M  INFO    General                 (distance_estimation.cpp   : 150)   Weight Filter Done\n",
      "  0:03:34.838    88M / 186M  INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:03:34.842    88M / 186M  INFO    General                 (distance_estimation.cpp   :  37)   Filtering info\n",
      "  0:03:34.842    88M / 186M  INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; library index size: 3698\n",
      "  0:03:34.843    88M / 186M  INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering; library index size: 3234\n",
      "  0:03:34.843    88M / 186M  INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:03:34.843    88M / 186M  INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:03:34.843    88M / 186M  INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:03:34.849    88M / 186M  INFO   PairInfoImprover         (pair_info_improver.hpp    : 105)   Paired info stats: missing = 203; contradictional = 4\n",
      "  0:03:34.854    88M / 186M  INFO   PairInfoImprover         (pair_info_improver.hpp    : 105)   Paired info stats: missing = 0; contradictional = 0\n",
      "  0:03:34.854    88M / 186M  INFO    General                 (distance_estimation.cpp   : 104)   Filling scaffolding index\n",
      "  0:03:34.854    88M / 186M  INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:03:34.958    88M / 186M  INFO    General                 (distance_estimation.cpp   :  37)   Filtering info\n",
      "  0:03:34.958    88M / 186M  INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; library index size: 891\n",
      "  0:03:34.958    88M / 186M  INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering; library index size: 891\n",
      "  0:03:34.958    88M / 186M  INFO    General                 (distance_estimation.cpp   : 186)   Clearing raw paired index\n",
      "  0:03:34.958    87M / 186M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Repeat Resolving (id: repeat_resolving)\n",
      "  0:03:34.958    87M / 186M  INFO    General                 (repeat_resolving.cpp      :  87)   Using Path-Extend repeat resolving\n",
      "  0:03:34.958    87M / 186M  INFO    General                 (launcher.cpp              : 600)   ExSPAnder repeat resolving tool started\n",
      "  0:03:34.959    87M / 186M  INFO    General                 (coverage_uniformity_analyz:  25)   genomic coverage is 40.29 calculated of length 4525364\n",
      "  0:03:34.960    97M / 186M  INFO    General                 (launcher.cpp              : 418)   Creating main extenders, unique edge length = 2000\n",
      "  0:03:34.960    97M / 186M  INFO    General                 (extenders_logic.cpp       : 334)   Estimated coverage of library #0 is 41.1665\n",
      "  0:03:34.960    97M / 186M  INFO    General                 (extenders_logic.cpp       : 345)   Creating extender; library index size: 3636\n",
      "  0:03:34.960    97M / 186M  INFO    General                 (extenders_logic.cpp       : 334)   Estimated coverage of library #0 is 41.1665\n",
      "  0:03:34.960    97M / 186M  INFO    General                 (extenders_logic.cpp       : 345)   Creating extender; library index size: 3636\n",
      "  0:03:34.961    97M / 186M  INFO    General                 (extenders_logic.cpp       : 543)   Using 1 paired-end library\n",
      "  0:03:34.961    97M / 186M  INFO    General                 (extenders_logic.cpp       : 544)   Using 1 paired-end scaffolding library\n",
      "  0:03:34.961    97M / 186M  INFO    General                 (extenders_logic.cpp       : 545)   Using 0 single read libraries\n",
      "  0:03:34.961    97M / 186M  INFO    General                 (launcher.cpp              : 447)   Total number of extenders is 3\n",
      "  0:03:34.961    97M / 186M  INFO    General                 (path_extenders.cpp        :  36)   Processed 0 paths from 394 (0%)\n",
      "  0:03:34.966    99M / 186M  INFO    General                 (path_extenders.cpp        :  36)   Processed 40 paths from 394 (10%)\n",
      "  0:03:34.968   100M / 186M  INFO    General                 (path_extenders.cpp        :  36)   Processed 80 paths from 394 (20%)\n",
      "  0:03:34.970   101M / 186M  INFO    General                 (path_extenders.cpp        :  36)   Processed 120 paths from 394 (30%)\n",
      "  0:03:34.970   101M / 186M  INFO    General                 (path_extenders.cpp        :  34)   Processed 128 paths from 394 (32%)\n",
      "  0:03:34.971   102M / 186M  INFO    General                 (path_extenders.cpp        :  36)   Processed 160 paths from 394 (40%)\n",
      "  0:03:34.972   102M / 186M  INFO    General                 (path_extenders.cpp        :  36)   Processed 200 paths from 394 (50%)\n",
      "  0:03:34.972   103M / 186M  INFO    General                 (path_extenders.cpp        :  36)   Processed 240 paths from 394 (60%)\n",
      "  0:03:34.973   103M / 186M  INFO    General                 (path_extenders.cpp        :  34)   Processed 256 paths from 394 (64%)\n",
      "  0:03:34.973   103M / 186M  INFO    General                 (path_extenders.cpp        :  36)   Processed 280 paths from 394 (71%)\n",
      "  0:03:34.973   103M / 186M  INFO    General                 (path_extenders.cpp        :  36)   Processed 320 paths from 394 (81%)\n",
      "  0:03:34.973   104M / 186M  INFO    General                 (path_extenders.cpp        :  36)   Processed 360 paths from 394 (91%)\n",
      "  0:03:34.973   104M / 186M  INFO    General                 (launcher.cpp              : 250)   Finalizing paths\n",
      "  0:03:34.973   104M / 186M  INFO    General                 (launcher.cpp              : 252)   Deduplicating paths\n",
      "  0:03:34.974   102M / 186M  INFO    General                 (launcher.cpp              : 256)   Paths deduplicated\n",
      "  0:03:34.974   102M / 186M  INFO   PEResolver               (pe_resolver.cpp           :  60)   Removing overlaps\n",
      "  0:03:34.974   102M / 186M  INFO   PEResolver               (pe_resolver.cpp           :  63)   Sorting paths\n",
      "  0:03:34.974   102M / 186M  INFO   PEResolver               (pe_resolver.cpp           :  70)   Marking overlaps\n",
      "  0:03:34.974   102M / 186M  INFO   OverlapRemover           (overlap_remover.hpp       : 116)   Marking start/end overlaps\n",
      "  0:03:34.975   102M / 186M  INFO   OverlapRemover           (overlap_remover.hpp       : 119)   Marking remaining overlaps\n",
      "  0:03:34.976   102M / 186M  INFO   PEResolver               (pe_resolver.cpp           :  73)   Splitting paths\n",
      "  0:03:34.977   110M / 187M  INFO   PEResolver               (pe_resolver.cpp           :  78)   Deduplicating paths\n",
      "  0:03:34.977   101M / 187M  INFO   PEResolver               (pe_resolver.cpp           :  80)   Overlaps removed\n",
      "  0:03:34.978   102M / 187M  INFO    General                 (launcher.cpp              : 273)   Paths finalized\n",
      "  0:03:34.978   102M / 187M  INFO    General                 (launcher.cpp              : 454)   Closing gaps in paths\n",
      "  0:03:34.978   106M / 187M  INFO    General                 (launcher.cpp              : 484)   Gap closing completed\n",
      "  0:03:34.978   106M / 187M  INFO    General                 (launcher.cpp              : 302)   Traversing tandem repeats\n",
      "  0:03:34.979   106M / 187M  INFO    General                 (launcher.cpp              : 312)   Traversed 5 loops\n",
      "  0:03:34.979   106M / 187M  INFO    General                 (launcher.cpp              : 250)   Finalizing paths\n",
      "  0:03:34.979   106M / 187M  INFO    General                 (launcher.cpp              : 252)   Deduplicating paths\n",
      "  0:03:34.979   106M / 187M  INFO    General                 (launcher.cpp              : 256)   Paths deduplicated\n",
      "  0:03:34.979   106M / 187M  INFO   PEResolver               (pe_resolver.cpp           :  60)   Removing overlaps\n",
      "  0:03:34.979   106M / 187M  INFO   PEResolver               (pe_resolver.cpp           :  63)   Sorting paths\n",
      "  0:03:34.979   106M / 187M  INFO   PEResolver               (pe_resolver.cpp           :  70)   Marking overlaps\n",
      "  0:03:34.979   106M / 187M  INFO   OverlapRemover           (overlap_remover.hpp       : 116)   Marking start/end overlaps\n",
      "  0:03:34.980   106M / 187M  INFO   OverlapRemover           (overlap_remover.hpp       : 119)   Marking remaining overlaps\n",
      "  0:03:34.980   106M / 187M  INFO   PEResolver               (pe_resolver.cpp           :  73)   Splitting paths\n",
      "  0:03:34.980   106M / 187M  INFO   PEResolver               (pe_resolver.cpp           :  78)   Deduplicating paths\n",
      "  0:03:34.980   106M / 187M  INFO   PEResolver               (pe_resolver.cpp           :  80)   Overlaps removed\n",
      "  0:03:34.980   106M / 187M  INFO    General                 (launcher.cpp              : 273)   Paths finalized\n",
      "  0:03:34.980   106M / 187M  INFO    General                 (launcher.cpp              : 664)   ExSPAnder repeat resolving tool finished\n",
      "  0:03:34.981    91M / 187M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:03:34.981    91M / 187M  INFO    General                 (contig_output.hpp         :  16)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/before_rr.fasta\n",
      "  0:03:35.002    91M / 187M  INFO    General                 (contig_output_stage.cpp   : 151)   Writing GFA graph to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/assembly_graph_with_scaffolds.gfa\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:03:35.026    91M / 187M  INFO    General                 (contig_output_stage.cpp   : 165)   Outputting FastG graph to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/assembly_graph.fastg\n",
      "  0:03:35.091    91M / 187M  INFO    General                 (contig_output_stage.cpp   : 196)   Breaking scaffolds\n",
      "  0:03:35.102   101M / 187M  INFO    General                 (contig_output_stage.cpp   :  98)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/final_contigs.fasta\n",
      "  0:03:35.122   101M / 187M  INFO    General                 (contig_output_stage.cpp   : 104)   Outputting FastG paths to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/final_contigs.paths\n",
      "  0:03:35.132    96M / 187M  INFO    General                 (contig_output_stage.cpp   :  98)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/scaffolds.fasta\n",
      "  0:03:35.150    96M / 187M  INFO    General                 (contig_output_stage.cpp   : 104)   Outputting FastG paths to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/scaffolds.paths\n",
      "  0:03:35.151    96M / 187M  INFO    General                 (contig_output_stage.cpp   : 111)   Populating GFA with scaffold paths\n",
      "  0:03:35.151    91M / 187M  INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  0:03:35.154     1M / 187M  INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 3 minutes 35 seconds\n",
      "\n",
      "===== K55 finished. \n",
      "\n",
      "\n",
      "===== Copy files started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/miniconda3/bin/python /Users/olga/Downloads/SPAdes-3.15.2/share/spades/spades_pipeline/scripts/copy_files.py /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/before_rr.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/before_rr.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/assembly_graph_after_simplification.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/assembly_graph_after_simplification.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/final_contigs.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/contigs.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/first_pe_contigs.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/first_pe_contigs.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/strain_graph.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/strain_graph.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/scaffolds.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/scaffolds.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/scaffolds.paths /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/scaffolds.paths /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/assembly_graph_with_scaffolds.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/assembly_graph_with_scaffolds.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/assembly_graph.fastg /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/assembly_graph.fastg /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/K55/final_contigs.paths /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/contigs.paths\n",
      "\n",
      "\n",
      "===== Copy files finished. \n",
      "\n",
      "\n",
      "===== Assembling finished. \n",
      "\n",
      "\n",
      "===== Breaking scaffolds started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/miniconda3/bin/python /Users/olga/Downloads/SPAdes-3.15.2/share/spades/spades_pipeline/scripts/breaking_scaffolds_script.py --result_scaffolds_filename /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/scaffolds.fasta --misc_dir /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/misc --threshold_for_breaking_scaffolds 3\n",
      "\n",
      "\n",
      "===== Breaking scaffolds finished. \n",
      "\n",
      "\n",
      "===== Terminate started. \n",
      "\n",
      "\n",
      "===== Terminate finished. \n",
      "\n",
      " * Corrected reads are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/corrected/\n",
      " * Assembled contigs are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/contigs.fasta\n",
      " * Assembled scaffolds are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/scaffolds.fasta\n",
      " * Paths in the assembly graph corresponding to the contigs are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/scaffolds.paths\n",
      " * Assembly graph is in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/assembly_graph_with_scaffolds.gfa\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n"
     ]
    }
   ],
   "source": [
    "! /Users/olga/Downloads/SPAdes-3.15.2/bin/spades.py -1 ./homework6/illumina.100x.1.fq -2 ./homework6/illumina.100x.2.fq -o illumina"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3. Гибридная сборка"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Собираем с помощью Spades"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "== Warning ==  output dir is not empty! Please, clean output directory before run.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "== Warning ==  No assembly mode was specified! If you intend to assemble high-coverage multi-cell/isolate data, use '--isolate' option.\n",
      "\n",
      "\n",
      "Command line: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades.py\t-1\t/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq\t-2\t/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq\t--pacbio\t/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/pacbio_10x.fq\t-o\t/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.15.2\n",
      "  Python version: 3.7.7\n",
      "  OS: Darwin-20.3.0-x86_64-i386-64bit\n",
      "\n",
      "Output dir: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Standard mode\n",
      "  For multi-cell/isolate data we recommend to use '--isolate' option; for single-cell MDA data use '--sc'; for metagenomic data use '--meta'; for RNA-Seq use '--rna'.\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq']\n",
      "      right reads: ['/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "    Library number: 2, library type: pacbio\n",
      "      left reads: not specified\n",
      "      right reads: not specified\n",
      "      interlaced reads: not specified\n",
      "      single reads: ['/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/pacbio_10x.fq']\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: automatic selection based on read length\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 250\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/spades.log\n",
      "\n",
      "/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq: max reads length: 100\n",
      "/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq: max reads length: 100\n",
      "\n",
      "Reads length: 100\n",
      "\n",
      "\n",
      "===== Before start started. \n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-hammer /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  76)   Loading config from /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/corrected/configs/config.info\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (hammer_tools.cpp          :  38)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 16 files using 1 threads. This might take a while.\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 4194304\n",
      "  0:00:00.001   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq\n",
      "  0:00:09.809   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 467314 reads\n",
      "  0:00:19.224   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 935009 reads\n",
      "  0:00:28.537   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 1402861 reads\n",
      "  0:00:37.773   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 1870646 reads\n",
      "  0:00:46.983   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2339377 reads\n",
      "  0:00:50.249   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2500000 reads\n",
      "  0:00:50.249   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq\n",
      "  0:00:59.344   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2974795 reads\n",
      "  0:01:26.595   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 4400968 reads\n",
      "  0:01:38.120   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 5000000 reads processed\n",
      "  0:01:38.120     1M / 162M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:01:41.836     1M / 170M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 38919786 kmers in total.\n",
      "  0:01:41.836     1M / 170M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:45.188    30M / 195M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 38919786 kmers, 28121944 bytes occupied (5.78049 bits per kmer).\n",
      "  0:01:45.188    30M / 195M  INFO   K-mer Counting           (kmer_data.cpp             : 354)   Arranging kmers in hash map order\n",
      "  0:01:47.616   631M / 789M  INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:05:53.035   631M / 789M  INFO    General                 (main.cpp                  : 155)   Extracting clusters:\n",
      "  0:05:53.035   631M / 789M  INFO    General                 (concurrent_dsu.cpp        :  18)   Connecting to root\n",
      "  0:05:53.791   631M / 789M  INFO    General                 (concurrent_dsu.cpp        :  34)   Calculating counts\n",
      "  0:06:05.910   903M / 1061M INFO    General                 (concurrent_dsu.cpp        :  63)   Writing down entries\n",
      "  0:06:18.373   631M / 789M  INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 13651605\n",
      "  0:06:18.393   330M / 492M  INFO   K-mer Counting           (kmer_data.cpp             : 371)   Collecting K-mer information, this takes a while.\n",
      "  0:06:18.775  1223M / 1383M INFO   K-mer Counting           (kmer_data.cpp             : 377)   Processing /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq\n",
      "  0:11:02.374  1223M / 1383M INFO   K-mer Counting           (kmer_data.cpp             : 377)   Processing /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq\n",
      "  0:15:41.830  1223M / 1383M INFO   K-mer Counting           (kmer_data.cpp             : 384)   Collection done, postprocessing.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:15:42.067  1223M / 1383M INFO   K-mer Counting           (kmer_data.cpp             : 398)   There are 38919786 kmers in total. Among them 27377044 (70.3422%) are singletons.\n",
      "  0:15:42.067  1223M / 1383M INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n",
      "  0:16:25.199  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering done. Total 26 non-read kmers were generated.\n",
      "  0:16:25.199  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)   Subclustering statistics:\n",
      "  0:16:25.199  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton hamming clusters: 5541662. Among them 963110 (17.3794%) are good\n",
      "  0:16:25.199  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total singleton subclusters: 13747. Among them 13599 (98.9234%) are good\n",
      "  0:16:25.199  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Total non-singleton subcluster centers: 8197112. Among them 6855795 (83.6367%) are good\n",
      "  0:16:25.199  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average size of non-trivial subcluster: 4.07194 kmers\n",
      "  0:16:25.199  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Average number of sub-clusters per non-singleton cluster: 1.01244\n",
      "  0:16:25.199  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Total solid k-mers: 7832504\n",
      "  0:16:25.199  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 658)     Substitution probabilities: [4,4]((0.928457,0.0421442,0.022541,0.00685828),(0.0141904,0.967828,0.0090854,0.00889584),(0.00886597,0.00903879,0.967867,0.0142285),(0.00683121,0.022511,0.0420995,0.928558))\n",
      "  0:16:25.207  1223M / 1384M INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:16:25.207  1223M / 1384M INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 1 threads.\n",
      "  0:18:52.728  1223M / 1384M INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 39977 new k-mers.\n",
      "  0:21:19.720  1223M / 1384M INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 790 new k-mers.\n",
      "  0:23:47.245  1223M / 1384M INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 38 new k-mers.\n",
      "  0:26:15.522  1223M / 1384M INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 3 produced 0 new k-mers.\n",
      "  0:26:15.522  1223M / 1384M INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:26:15.522  1223M / 1384M INFO    General                 (hammer_tools.cpp          : 222)   Starting read correction in 1 threads.\n",
      "  0:26:15.522  1223M / 1384M INFO    General                 (hammer_tools.cpp          : 235)   Correcting pair of reads: /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq and /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq\n",
      "  0:26:15.691  1289M / 1394M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 0 of 100000 reads.\n",
      "  0:26:22.376  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 0\n",
      "  0:26:22.600  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 0\n",
      "  0:26:22.749  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 1 of 100000 reads.\n",
      "  0:26:29.675  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 1\n",
      "  0:26:29.889  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 1\n",
      "  0:26:30.033  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 2 of 100000 reads.\n",
      "  0:26:36.321  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 2\n",
      "  0:26:36.544  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 2\n",
      "  0:26:36.689  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 3 of 100000 reads.\n",
      "  0:26:43.080  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 3\n",
      "  0:26:43.324  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 3\n",
      "  0:26:43.478  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 4 of 100000 reads.\n",
      "  0:26:50.372  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 4\n",
      "  0:26:50.647  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 4\n",
      "  0:26:50.806  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 5 of 100000 reads.\n",
      "  0:26:58.298  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 5\n",
      "  0:26:58.583  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 5\n",
      "  0:26:58.751  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 6 of 100000 reads.\n",
      "  0:27:06.241  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 6\n",
      "  0:27:06.488  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 6\n",
      "  0:27:06.639  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 7 of 100000 reads.\n",
      "  0:27:13.634  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 7\n",
      "  0:27:13.887  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 7\n",
      "  0:27:14.037  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 8 of 100000 reads.\n",
      "  0:27:20.917  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 8\n",
      "  0:27:21.170  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 8\n",
      "  0:27:21.321  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 9 of 100000 reads.\n",
      "  0:27:28.345  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 9\n",
      "  0:27:28.594  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 9\n",
      "  0:27:28.742  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 10 of 100000 reads.\n",
      "  0:27:35.576  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 10\n",
      "  0:27:35.832  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 10\n",
      "  0:27:35.984  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 11 of 100000 reads.\n",
      "  0:27:42.831  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 11\n",
      "  0:27:43.079  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 11\n",
      "  0:27:43.228  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 12 of 100000 reads.\n",
      "  0:27:50.114  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 12\n",
      "  0:27:50.359  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 12\n",
      "  0:27:50.511  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 13 of 100000 reads.\n",
      "  0:27:57.420  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 13\n",
      "  0:27:57.665  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 13\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:27:57.815  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 14 of 100000 reads.\n",
      "  0:28:04.726  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 14\n",
      "  0:28:04.969  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 14\n",
      "  0:28:05.121  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 15 of 100000 reads.\n",
      "  0:28:12.222  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 15\n",
      "  0:28:12.474  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 15\n",
      "  0:28:12.626  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 16 of 100000 reads.\n",
      "  0:28:19.593  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 16\n",
      "  0:28:19.842  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 16\n",
      "  0:28:19.992  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 17 of 100000 reads.\n",
      "  0:28:26.768  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 17\n",
      "  0:28:27.017  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 17\n",
      "  0:28:27.166  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 18 of 100000 reads.\n",
      "  0:28:34.245  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 18\n",
      "  0:28:34.505  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 18\n",
      "  0:28:34.676  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 19 of 100000 reads.\n",
      "  0:28:41.863  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 19\n",
      "  0:28:42.118  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 19\n",
      "  0:28:42.268  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 20 of 100000 reads.\n",
      "  0:28:49.199  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 20\n",
      "  0:28:49.462  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 20\n",
      "  0:28:49.612  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 21 of 100000 reads.\n",
      "  0:28:56.458  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 21\n",
      "  0:28:56.706  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 21\n",
      "  0:28:56.868  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 22 of 100000 reads.\n",
      "  0:29:04.368  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 22\n",
      "  0:29:04.637  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 22\n",
      "  0:29:04.798  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 23 of 100000 reads.\n",
      "  0:29:11.897  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 23\n",
      "  0:29:12.181  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 23\n",
      "  0:29:12.345  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 24 of 100000 reads.\n",
      "  0:29:20.225  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 24\n",
      "  0:29:20.535  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 24\n",
      "  0:29:20.543  1223M / 1396M INFO    General                 (hammer_tools.cpp          : 276)   Correction done. Changed 1639742 bases in 1194871 reads.\n",
      "  0:29:20.543  1223M / 1396M INFO    General                 (hammer_tools.cpp          : 277)   Failed to correct 44220 bases out of 458864877.\n",
      "  0:29:20.637     1M / 208M  INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/corrected/corrected.yaml\n",
      "  0:29:20.648     1M / 208M  INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== corrected reads compression started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/miniconda3/bin/python /Users/olga/Downloads/SPAdes-3.15.2/share/spades/spades_pipeline/scripts/compress_all.py --input_file /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/corrected/corrected.yaml --ext_python_modules_home /Users/olga/Downloads/SPAdes-3.15.2/share/spades --max_threads 16 --output_dir /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/corrected --not_used_yaml_file /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/corrected/dont_correct.yaml --gzip_output\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "== Files to compress: ['/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/corrected/illumina.100x.1.00.0_0.cor.fastq', '/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/corrected/illumina.100x.2.00.0_0.cor.fastq', '/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/corrected/illumina.100x._unpaired.00.0_0.cor.fastq']\n",
      "== Files compression is finished\n",
      "== Info about datasets not used in error correction stage is loaded\n",
      "== Dataset yaml file is updated\n",
      "\n",
      "===== corrected reads compression finished. \n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started. \n",
      "\n",
      "\n",
      "===== K21 started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-core /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K21/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  :  99)   Loaded config from /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K21/configs/config.info\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 109)   Assembling dataset (/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/dataset.info) with K=21\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  72)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  73)   Converting paired reads\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:00.066    22M / 22M   INFO    General                 (binary_converter.cpp      :  96)   16384 reads processed\n",
      "  0:00:00.131    25M / 25M   INFO    General                 (binary_converter.cpp      :  96)   32768 reads processed\n",
      "  0:00:00.346    22M / 25M   INFO    General                 (binary_converter.cpp      :  96)   65536 reads processed\n",
      "  0:00:00.689    25M / 31M   INFO    General                 (binary_converter.cpp      :  96)   131072 reads processed\n",
      "  0:00:01.460    21M / 31M   INFO    General                 (binary_converter.cpp      :  96)   262144 reads processed\n",
      "  0:00:02.852    24M / 31M   INFO    General                 (binary_converter.cpp      :  96)   524288 reads processed\n",
      "  0:00:05.697    28M / 31M   INFO    General                 (binary_converter.cpp      :  96)   1048576 reads processed\n",
      "  0:00:11.456    28M / 31M   INFO    General                 (binary_converter.cpp      :  96)   2097152 reads processed\n",
      "  0:00:13.521    19M / 31M   INFO    General                 (binary_converter.cpp      : 111)   2448672 reads written\n",
      "  0:00:13.521     1M / 31M   INFO    General                 (read_converter.cpp        :  81)   Converting single reads\n",
      "  0:00:13.555    10M / 31M   INFO    General                 (binary_converter.cpp      :  96)   16384 reads processed\n",
      "  0:00:13.587    12M / 31M   INFO    General                 (binary_converter.cpp      :  96)   32768 reads processed\n",
      "  0:00:13.659     9M / 31M   INFO    General                 (binary_converter.cpp      : 111)   48607 reads written\n",
      "  0:00:13.659     1M / 31M   INFO    General                 (read_converter.cpp        :  87)   Converting merged reads\n",
      "  0:00:13.660     9M / 31M   INFO    General                 (binary_converter.cpp      : 111)   0 reads written\n",
      "  0:00:13.660     1M / 31M   INFO    General                 (read_converter.cpp        :  72)   Converting reads to binary format for library #1 (takes a while)\n",
      "  0:00:13.660     1M / 31M   INFO    General                 (read_converter.cpp        :  73)   Converting paired reads\n",
      "  0:00:13.661    19M / 31M   INFO    General                 (binary_converter.cpp      : 111)   0 reads written\n",
      "  0:00:13.661     1M / 31M   INFO    General                 (read_converter.cpp        :  81)   Converting single reads\n",
      "  0:00:14.431     9M / 75M   INFO    General                 (binary_converter.cpp      : 111)   10367 reads written\n",
      "  0:00:14.432     1M / 75M   INFO    General                 (read_converter.cpp        :  87)   Converting merged reads\n",
      "  0:00:14.432     9M / 75M   INFO    General                 (binary_converter.cpp      : 111)   0 reads written\n",
      "  0:00:14.432     1M / 75M   INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:14.433     1M / 75M   INFO    General                 (construction.cpp          : 153)   Max read length 100\n",
      "  0:00:14.433     1M / 75M   INFO    General                 (construction.cpp          : 159)   Average read length 92.7147\n",
      "  0:00:14.433     1M / 75M   INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:14.433     1M / 75M   INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:14.433     1M / 75M   INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:14.433     1M / 75M   INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:14.433     1M / 75M   INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 6710886\n",
      "  0:00:25.122   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 1873811 reads\n",
      "  0:00:35.357   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 3734243 reads\n",
      "  0:00:45.544   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 5595232 reads\n",
      "  0:00:55.079   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 7456112 reads\n",
      "  0:01:04.521   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 9318931 reads\n",
      "  0:01:07.638   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 9891902 reads\n",
      "  0:01:07.638     1M / 173M  INFO    General                 (kmer_splitters.hpp        : 131)   Used 9891902 reads\n",
      "  0:01:07.667     1M / 173M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:01:08.134     1M / 173M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5715581 kmers in total.\n",
      "  0:01:08.134     1M / 173M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:01:08.136     1M / 173M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:01:08.136     1M / 173M  INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:01:08.136     1M / 173M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:01:08.136     1M / 173M  INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:01:08.136     1M / 173M  INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 6710886\n",
      "  0:01:09.935   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 194)   Processed 5715581 kmers\n",
      "  0:01:09.935   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 199)   Used 5715581 kmers.\n",
      "  0:01:09.935     1M / 217M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:01:10.114     1M / 217M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5708731 kmers in total.\n",
      "  0:01:10.114     1M / 217M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:10.548     5M / 217M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5708731 kmers, 4130592 bytes occupied (5.78846 bits per kmer).\n",
      "  0:01:10.548     5M / 217M  INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n",
      "  0:01:10.682    11M / 217M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:01:12.429    11M / 217M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:01:12.429    11M / 173M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Early tip clipping (id: construction:early_tip_clipper)\n",
      "  0:01:12.430    11M / 173M  INFO    General                 (construction.cpp          : 301)   Early tip clipper length bound set as (RL - K)\n",
      "  0:01:12.430    11M / 173M  INFO   Early tip clipping       (early_simplification.hpp  :  47)   Early tip clipping\n",
      "  0:01:14.100    21M / 217M  INFO   Early tip clipping       (early_simplification.hpp  :  82)   #tipped junctions: 123493\n",
      "  0:01:14.139    21M / 217M  INFO   Early tip clipping       (early_simplification.hpp  :  93)   Clipped tips: 124241\n",
      "  0:01:14.140    11M / 173M  INFO   Early tip clipping       (early_simplification.hpp  :  49)   978561 22-mers were removed by early tip clipper\n",
      "  0:01:14.140    11M / 173M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:01:14.143    11M / 173M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:01:16.324    13M / 217M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 25730 sequences extracted\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:01:17.094    13M / 173M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:01:17.488    13M / 217M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 4 loops collected\n",
      "  0:01:17.490    13M / 173M  INFO    General                 (debruijn_graph_constructor: 487)   Total 51468 edges to create\n",
      "  0:01:17.490    15M / 173M  INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:01:17.501    16M / 174M  INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:01:17.504    16M / 174M  INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:01:17.504    16M / 174M  INFO    General                 (debruijn_graph_constructor: 503)   Total 18884 vertices to create\n",
      "  0:01:17.504    18M / 174M  INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:01:17.506    16M / 174M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:01:17.506    16M / 174M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:17.939    20M / 174M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5715581 kmers, 4135544 bytes occupied (5.78845 bits per kmer).\n",
      "  0:01:17.940    42M / 174M  INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:02:14.430    42M / 174M  INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:02:15.846    42M / 174M  INFO    General                 (coverage_filling.hpp      :  82)   Processed 51447 edges\n",
      "  0:02:15.859     6M / 174M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:02:15.859     6M / 174M  INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 16\n",
      "  0:02:15.859     6M / 174M  INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 76\n",
      "  0:02:15.859     6M / 174M  INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 77. Coverage mad: 14.826\n",
      "  0:02:15.859     6M / 174M  INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:02:15.923     6M / 174M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:02:16.083     6M / 174M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:02:16.618     6M / 174M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:02:17.852     6M / 174M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:02:17.986     6M / 174M  INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 76.252. Fitted coverage std. dev: 14.5028\n",
      "  0:02:17.988     6M / 174M  INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.863705\n",
      "  0:02:17.988     6M / 174M  INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 37\n",
      "  0:02:17.988     6M / 174M  INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 37\n",
      "  0:02:17.988     6M / 174M  INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 4507902\n",
      "  0:02:17.988     6M / 174M  INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 76.252\n",
      "  0:02:17.988     6M / 174M  INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 37\n",
      "  0:02:17.988     6M / 174M  INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:02:17.988     6M / 174M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:02:17.988     6M / 174M  INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:02:17.988     6M / 174M  INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:02:17.988     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:02:17.990     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:02:17.990     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n",
      "  0:02:18.006     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 2371 times\n",
      "  0:02:18.006     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n",
      "  0:02:18.073     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 4029 times\n",
      "  0:02:18.073     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:02:18.074     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 308 times\n",
      "  0:02:18.074     6M / 174M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:02:18.074     6M / 174M  INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:02:18.074     6M / 174M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:02:18.074     6M / 174M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:02:18.074     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:18.077     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 128 times\n",
      "  0:02:18.077     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:18.145     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 1810 times\n",
      "  0:02:18.145     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:18.149     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 143 times\n",
      "  0:02:18.149     6M / 174M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:02:18.149     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:18.149     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 5 times\n",
      "  0:02:18.150     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:18.151     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 36 times\n",
      "  0:02:18.151     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:18.152     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 68 times\n",
      "  0:02:18.152     6M / 174M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:02:18.152     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:18.152     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 3 times\n",
      "  0:02:18.152     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:18.153     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 23 times\n",
      "  0:02:18.153     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:18.153     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 16 times\n",
      "  0:02:18.153     6M / 174M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:02:18.153     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:18.153     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:18.153     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 7 times\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 10 times\n",
      "  0:02:18.154     6M / 174M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 2 times\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 4 times\n",
      "  0:02:18.154     6M / 174M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 4 times\n",
      "  0:02:18.154     6M / 174M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 1 times\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 2 times\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 3 times\n",
      "  0:02:18.154     6M / 174M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 7 times\n",
      "  0:02:18.154     6M / 174M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:02:18.154     6M / 174M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 6 times\n",
      "  0:02:18.154     6M / 174M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:02:18.154     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:18.155     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:18.155     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:18.157     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 3 times\n",
      "  0:02:18.157     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:18.158     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:18.158     6M / 174M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:02:18.158     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:18.158     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:18.158     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:18.158     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:18.158     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:18.158     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:18.158     6M / 174M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
      "  0:02:18.158     6M / 174M  INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\n",
      "  0:02:18.158     6M / 174M  INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\n",
      "  0:02:18.158     6M / 174M  INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\n",
      "  0:02:18.158     6M / 174M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:02:18.158     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:18.158     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:18.158     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:18.160     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:18.160     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:18.161     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:18.161     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:02:18.164     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:18.164     6M / 174M  INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:02:18.164     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:02:18.165     6M / 174M  INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 7 times\n",
      "  0:02:18.165     6M / 174M  INFO    General                 (simplification.cpp        : 491)   Counting average coverage\n",
      "  0:02:18.165     6M / 174M  INFO    General                 (simplification.cpp        : 497)   Average coverage = 78.0189\n",
      "  0:02:18.165     6M / 174M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:02:18.165     6M / 174M  INFO    General                 (read_converter.cpp        : 102)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K21/simplified_contigs\n",
      "  0:02:18.169     8M / 174M  INFO    General                 (binary_converter.cpp      : 111)   2571 reads written\n",
      "  0:02:18.169     6M / 174M  INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  0:02:18.170     1M / 174M  INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 2 minutes 18 seconds\n",
      "\n",
      "===== K21 finished. \n",
      "\n",
      "\n",
      "===== K33 started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-core /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K33/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  :  99)   Loaded config from /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K33/configs/config.info\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 109)   Assembling dataset (/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/dataset.info) with K=33\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 118)   Contigs from previous K will be used: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K21/simplified_contigs\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (construction.cpp          : 153)   Max read length 100\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (construction.cpp          : 159)   Average read length 92.7147\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 3355443\n",
      "  0:00:05.688   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 1132634 reads\n",
      "  0:00:11.044   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 2249495 reads\n",
      "  0:00:16.423   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 3368100 reads\n",
      "  0:00:21.785   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 4487032 reads\n",
      "  0:00:27.127   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 5607299 reads\n",
      "  0:00:32.468   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 6726813 reads\n",
      "  0:00:37.785   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 7846654 reads\n",
      "  0:00:43.122   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 8966769 reads\n",
      "  0:00:48.252   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 9897044 reads\n",
      "  0:00:48.252     1M / 158M  INFO    General                 (kmer_splitters.hpp        : 131)   Used 9897044 reads\n",
      "  0:00:48.285     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:49.315     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5756125 kmers in total.\n",
      "  0:00:49.315     1M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:00:49.317     1M / 158M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:00:49.317     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:49.318     1M / 158M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:49.318     1M / 158M  INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:49.318     1M / 158M  INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 3355443\n",
      "  0:00:51.471   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 194)   Processed 5756125 kmers\n",
      "  0:00:51.471   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 199)   Used 5756125 kmers.\n",
      "  0:00:51.471     1M / 246M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:51.791     1M / 246M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5754745 kmers in total.\n",
      "  0:00:51.791     1M / 246M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:52.317     5M / 246M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5754745 kmers, 4163816 bytes occupied (5.78836 bits per kmer).\n",
      "  0:00:52.317     5M / 246M  INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n",
      "  0:00:52.590    11M / 246M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:00:54.441    11M / 246M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:54.442    11M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Early tip clipping (id: construction:early_tip_clipper)\n",
      "  0:00:54.442    11M / 158M  INFO    General                 (construction.cpp          : 301)   Early tip clipper length bound set as (RL - K)\n",
      "  0:00:54.442    11M / 158M  INFO   Early tip clipping       (early_simplification.hpp  :  47)   Early tip clipping\n",
      "  0:00:56.200    21M / 246M  INFO   Early tip clipping       (early_simplification.hpp  :  82)   #tipped junctions: 124481\n",
      "  0:00:56.245    21M / 246M  INFO   Early tip clipping       (early_simplification.hpp  :  93)   Clipped tips: 125222\n",
      "  0:00:56.246    11M / 158M  INFO   Early tip clipping       (early_simplification.hpp  :  49)   1029171 34-mers were removed by early tip clipper\n",
      "  0:00:56.246    11M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:00:56.255    11M / 158M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:00:58.623    12M / 248M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 15754 sequences extracted\n",
      "  0:00:59.642    12M / 160M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:01:00.127    12M / 248M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 2 loops collected\n",
      "  0:01:00.130    12M / 160M  INFO    General                 (debruijn_graph_constructor: 487)   Total 31512 edges to create\n",
      "  0:01:00.130    14M / 160M  INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:01:00.139    14M / 160M  INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:01:00.140    14M / 160M  INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:01:00.141    14M / 160M  INFO    General                 (debruijn_graph_constructor: 503)   Total 14376 vertices to create\n",
      "  0:01:00.141    16M / 160M  INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:01:00.142    15M / 160M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:01:00.142    15M / 160M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:00.730    19M / 160M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5756125 kmers, 4164864 bytes occupied (5.78843 bits per kmer).\n",
      "  0:01:00.733    41M / 163M  INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:01:50.793    41M / 163M  INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:01:52.278    41M / 163M  INFO    General                 (coverage_filling.hpp      :  82)   Processed 31510 edges\n",
      "  0:01:52.297     5M / 163M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:01:52.297     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 13\n",
      "  0:01:52.297     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 64\n",
      "  0:01:52.297     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 64. Coverage mad: 14.826\n",
      "  0:01:52.297     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:01:52.351     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:01:52.490     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:01:52.934     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:01:53.982     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:01:54.642     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 63.4837. Fitted coverage std. dev: 13.4198\n",
      "  0:01:54.644     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.693899\n",
      "  0:01:54.644     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 29\n",
      "  0:01:54.644     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 29\n",
      "  0:01:54.644     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 4507034\n",
      "  0:01:54.644     5M / 163M  INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 63.4837\n",
      "  0:01:54.644     5M / 163M  INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 29\n",
      "  0:01:54.644     5M / 163M  INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:01:54.644     5M / 163M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:01:54.644     5M / 163M  INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:01:54.644     5M / 163M  INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:01:54.644     5M / 163M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:01:54.645     5M / 163M  INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:01:54.645     5M / 163M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n",
      "  0:01:54.675     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 2699 times\n",
      "  0:01:54.675     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n",
      "  0:01:54.710     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 1350 times\n",
      "  0:01:54.710     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:01:54.712     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 1603 times\n",
      "  0:01:54.712     5M / 164M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:01:54.712     5M / 164M  INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:01:54.712     5M / 164M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:01:54.712     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:01:54.712     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:54.713     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 59 times\n",
      "  0:01:54.713     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:01:54.785     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 1018 times\r\n",
      "  0:01:54.785     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:01:54.787     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 42 times\r\n",
      "  0:01:54.787     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\r\n",
      "  0:01:54.787     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:01:54.787     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 5 times\r\n",
      "  0:01:54.787     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:01:54.788     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 17 times\r\n",
      "  0:01:54.788     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:01:54.790     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 52 times\r\n",
      "  0:01:54.790     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\r\n",
      "  0:01:54.790     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:01:54.790     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 1 times\r\n",
      "  0:01:54.790     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:01:54.792     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 27 times\r\n",
      "  0:01:54.792     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:01:54.792     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 10 times\r\n",
      "  0:01:54.792     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\r\n",
      "  0:01:54.792     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:01:54.792     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:01:54.792     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:01:54.793     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 9 times\r\n",
      "  0:01:54.793     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:01:54.793     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 8 times\r\n",
      "  0:01:54.793     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\r\n",
      "  0:01:54.793     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:01:54.793     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:01:54.793     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:01:54.793     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 5 times\r\n",
      "  0:01:54.793     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 6 times\r\n",
      "  0:01:54.794     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\r\n",
      "  0:01:54.794     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 3 times\r\n",
      "  0:01:54.794     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\r\n",
      "  0:01:54.794     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\r\n",
      "  0:01:54.794     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\r\n",
      "  0:01:54.794     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:01:54.794     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:01:54.795     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 2 times\r\n",
      "  0:01:54.795     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:01:54.795     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\r\n",
      "  0:01:54.795     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 12\r\n",
      "  0:01:54.795     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:01:54.795     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:01:54.795     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:01:54.795     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\r\n",
      "  0:01:54.795     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:01:54.796     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\r\n",
      "  0:01:54.796     5M / 164M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\r\n",
      "  0:01:54.796     5M / 165M  INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\r\n",
      "  0:01:54.796     5M / 165M  INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\r\n",
      "  0:01:54.796     5M / 165M  INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\r\n",
      "  0:01:54.796     5M / 165M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\r\n",
      "  0:01:54.796     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:01:54.796     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:01:54.796     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:01:54.797     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\r\n",
      "  0:01:54.797     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:01:54.797     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:01:54.797     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:01:54.797     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:54.797     5M / 165M  INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:01:54.798     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:01:54.798     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 5 times\n",
      "  0:01:54.798     5M / 165M  INFO    General                 (simplification.cpp        : 491)   Counting average coverage\n",
      "  0:01:54.798     5M / 165M  INFO    General                 (simplification.cpp        : 497)   Average coverage = 64.7522\n",
      "  0:01:54.798     5M / 165M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:01:54.798     5M / 165M  INFO    General                 (read_converter.cpp        : 102)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K33/simplified_contigs\n",
      "  0:01:54.802     7M / 165M  INFO    General                 (binary_converter.cpp      : 111)   1188 reads written\n",
      "  0:01:54.803     5M / 165M  INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  0:01:54.803     1M / 165M  INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 1 minutes 54 seconds\n",
      "\n",
      "===== K33 finished. \n",
      "\n",
      "\n",
      "===== K55 started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-core /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  :  99)   Loaded config from /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/configs/config.info\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 109)   Assembling dataset (/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/dataset.info) with K=55\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 242)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 118)   Contigs from previous K will be used: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K33/simplified_contigs\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 153)   Max read length 100\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 159)   Average read length 92.7147\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 3355443\n",
      "  0:00:05.857   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 1768213 reads\n",
      "  0:00:11.383   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 3518079 reads\n",
      "  0:00:16.900   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 5268519 reads\n",
      "  0:00:22.413   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 7021543 reads\n",
      "  0:00:28.484   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 8774696 reads\n",
      "  0:00:33.058   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 9894278 reads\n",
      "  0:00:33.058     1M / 158M  INFO    General                 (kmer_splitters.hpp        : 131)   Used 9894278 reads\n",
      "  0:00:33.086     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:33.800     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5692221 kmers in total.\n",
      "  0:00:33.800     1M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:00:33.802     1M / 158M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:00:33.802     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:33.802     1M / 158M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:33.802     1M / 158M  INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:33.802     1M / 158M  INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 3355443\n",
      "  0:00:35.924   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 194)   Processed 5692221 kmers\n",
      "  0:00:35.924   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 199)   Used 5692221 kmers.\n",
      "  0:00:35.924     1M / 245M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:36.238     1M / 245M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5698724 kmers in total.\n",
      "  0:00:36.238     1M / 245M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:36.750     5M / 245M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5698724 kmers, 4123376 bytes occupied (5.78849 bits per kmer).\n",
      "  0:00:36.750     5M / 245M  INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:37.018    11M / 245M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:00:38.870    11M / 245M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:38.871    11M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:00:38.881    11M / 158M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:00:41.755    23M / 248M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 247400 sequences extracted\n",
      "  0:00:42.836    23M / 161M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:00:43.254    23M / 248M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:43.259    23M / 161M  INFO    General                 (debruijn_graph_constructor: 487)   Total 494800 edges to create\n",
      "  0:00:43.259    42M / 161M  INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:00:43.396    50M / 172M  INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:00:43.431    50M / 172M  INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:00:43.433    52M / 172M  INFO    General                 (debruijn_graph_constructor: 503)   Total 253903 vertices to create\n",
      "  0:00:43.433    76M / 172M  INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:00:43.473    62M / 172M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:00:43.473    62M / 172M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:43.984    66M / 172M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5692221 kmers, 4118704 bytes occupied (5.78854 bits per kmer).\n",
      "  0:00:43.985    88M / 172M  INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:01:17.418    88M / 172M  INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:01:19.491    88M / 172M  INFO    General                 (coverage_filling.hpp      :  82)   Processed 494800 edges\n",
      "  0:01:19.510    52M / 172M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:01:19.510    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 6\n",
      "  0:01:19.510    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 41\n",
      "  0:01:19.510    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 41. Coverage mad: 11.8608\n",
      "  0:01:19.510    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:01:19.544    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:01:19.636    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:01:19.924    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:01:20.602    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:01:21.083    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 40.4747. Fitted coverage std. dev: 10.9026\n",
      "  0:01:21.084    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.512008\n",
      "  0:01:21.084    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 16\n",
      "  0:01:21.085    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 16\n",
      "  0:01:21.085    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 4487959\n",
      "  0:01:21.085    52M / 172M  INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 40.4747\n",
      "  0:01:21.085    52M / 172M  INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 16\n",
      "  0:01:21.085    52M / 172M  INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:01:21.085    52M / 172M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Gap Closer (id: early_gapcloser)\n",
      "  0:01:21.088    61M / 172M  INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:01:21.187    66M / 172M  INFO    General                 (gap_closer.cpp            : 101)   Total edges in tip neighborhood: 134528 out of 494800, length: 1226132\n",
      "  0:01:21.196    68M / 172M  INFO    General                 (edge_index.hpp            : 167)   Using small index (max_id = 497276)\n",
      "  0:01:21.299    68M / 172M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:21.838    69M / 172M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 1092761 kmers, 790000 bytes occupied (5.78352 bits per kmer).\n",
      "  0:01:21.839    78M / 172M  INFO    General                 (edge_index_builders.hpp   : 265)   Collecting edge information from graph, this takes a while.\n",
      "  0:01:26.531    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:01:30.887    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:01:35.235    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:01:39.614    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:01:44.440    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:01:48.800    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:02:10.448    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:02:15.888    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:02:15.893    62M / 172M  INFO    General                 (gap_closer.cpp            : 452)   Initializing gap closer\n",
      "  0:02:15.893    62M / 172M  INFO   GapCloser                (gap_closer.cpp            : 359)   Closing short gaps\n",
      "  0:02:16.107    62M / 178M  INFO   GapCloser                (gap_closer.cpp            : 393)   Closing short gaps complete: filled 0 gaps after checking 20 candidates\n",
      "  0:02:16.113    62M / 178M  INFO    General                 (gap_closer.cpp            : 456)   Gap closer done\n",
      "  0:02:16.114    52M / 178M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:02:16.114    52M / 178M  INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:02:16.115    52M / 178M  INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:02:16.115    52M / 178M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:02:16.143    52M / 178M  INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:02:16.143    52M / 178M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:02:16.904    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 119766 times\n",
      "  0:02:16.904    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n",
      "  0:02:16.911    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 92 times\n",
      "  0:02:16.911    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:02:16.920    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 7236 times\n",
      "  0:02:16.920    45M / 179M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:02:16.920    45M / 179M  INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:02:16.920    45M / 179M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:02:16.920    45M / 179M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:02:16.920    45M / 179M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:16.922    45M / 179M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 19 times\n",
      "  0:02:16.922    45M / 179M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.018    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 447 times\n",
      "  0:02:17.018    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.019    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:17.019    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:02:17.019    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.019    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 3 times\n",
      "  0:02:17.019    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.019    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:17.019    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.020    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 13 times\n",
      "  0:02:17.020    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:02:17.020    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.020    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 1 times\n",
      "  0:02:17.020    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.021    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 8 times\n",
      "  0:02:17.021    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.022    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 11 times\n",
      "  0:02:17.022    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:02:17.022    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.022    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 3 times\n",
      "  0:02:17.022    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.023    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 7 times\n",
      "  0:02:17.023    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.023    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:02:17.023    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:02:17.023    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.023    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:17.023    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.023    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:17.023    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.023    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 3 times\n",
      "  0:02:17.023    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:02:17.023    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.023    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:17.023    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.024    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 1 times\n",
      "  0:02:17.024    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.024    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:02:17.024    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:02:17.024    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.024    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:17.024    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.024    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:17.024    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.024    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:02:17.024    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:02:17.024    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.024    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:17.024    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.024    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:17.024    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.024    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:17.024    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:02:17.024    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.024    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:17.024    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.024    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:17.024    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.025    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:02:17.025    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:02:17.025    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.025    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:17.025    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.025    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:17.025    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.025    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:02:17.025    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:02:17.025    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:17.026    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:17.026    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:17.027    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:17.027    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:17.028    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:17.028    48M / 180M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Gap Closer (id: late_gapcloser)\n",
      "  0:02:17.030    57M / 180M  INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:02:17.031    57M / 180M  INFO    General                 (gap_closer.cpp            : 101)   Total edges in tip neighborhood: 15 out of 882, length: 167704\n",
      "  0:02:17.031    57M / 180M  INFO    General                 (edge_index.hpp            : 167)   Using small index (max_id = 497276)\n",
      "  0:02:17.036    57M / 180M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:02:17.068    57M / 180M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 166932 kmers, 121312 bytes occupied (5.81372 bits per kmer).\n",
      "  0:02:17.068    58M / 180M  INFO    General                 (edge_index_builders.hpp   : 265)   Collecting edge information from graph, this takes a while.\n",
      "  0:02:19.596    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:02:22.096    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:02:24.562    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:02:27.036    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:02:29.501    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:02:31.969    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:02:44.140    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:02:47.172    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:02:47.174    57M / 180M  INFO    General                 (gap_closer.cpp            : 452)   Initializing gap closer\n",
      "  0:02:47.175    57M / 180M  INFO   GapCloser                (gap_closer.cpp            : 359)   Closing short gaps\n",
      "  0:02:47.175    57M / 180M  INFO   GapCloser                (gap_closer.cpp            : 393)   Closing short gaps complete: filled 1 gaps after checking 7 candidates\n",
      "  0:02:47.176    57M / 180M  INFO    General                 (gap_closer.cpp            : 456)   Gap closer done\n",
      "  0:02:47.177    48M / 180M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
      "  0:02:47.177    48M / 180M  INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\n",
      "  0:02:47.177    48M / 180M  INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\n",
      "  0:02:47.177    48M / 180M  INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\n",
      "  0:02:47.177    48M / 180M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:02:47.177    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:47.178    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:47.178    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:47.178    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:47.178    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:47.179    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:47.179    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:47.180    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:47.180    48M / 180M  INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:02:47.181    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:02:47.181    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 1 times\n",
      "  0:02:47.181    48M / 180M  INFO    General                 (simplification.cpp        : 491)   Counting average coverage\n",
      "  0:02:47.182    48M / 180M  INFO    General                 (simplification.cpp        : 497)   Average coverage = 41.1665\n",
      "  0:02:47.182    48M / 180M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Mismatch Correction (id: mismatch_correction)\n",
      "  0:02:47.182    48M / 180M  INFO    General                 (graph_pack.cpp            :  67)   Index refill\n",
      "  0:02:47.182    48M / 180M  INFO    General                 (edge_index.hpp            : 156)   Using small index (max_id = 497276)\n",
      "  0:02:47.349    48M / 180M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:02:48.359    51M / 180M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 4555847 kmers, 3291264 bytes occupied (5.77941 bits per kmer).\n",
      "  0:02:48.364    87M / 185M  INFO    General                 (edge_index_builders.hpp   : 252)   Collecting edge information from graph, this takes a while.\n",
      "  0:02:49.222    87M / 185M  INFO    General                 (graph_pack.cpp            :  77)   Normalizing k-mer map. Total 56128 kmers to process\n",
      "  0:02:49.236    87M / 185M  INFO    General                 (graph_pack.cpp            :  79)   Normalizing done\n",
      "  0:02:49.236    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 400)   Collect potential mismatches\n",
      "  0:02:49.292    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 193)   Total 354 edges (out of 878) with 1752 potential mismatch positions (4.94915 positions per edge)\n",
      "  0:02:49.292    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 402)   Potential mismatches collected\n",
      "  0:02:49.696    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:02:50.124    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:02:50.549    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:02:50.975    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:02:51.401    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:02:51.830    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:02:53.979    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:02:58.239    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 4200000 reads\n",
      "  0:03:07.233    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 8400000 reads\n",
      "  0:03:10.400    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 9891902 reads processed\n",
      "  0:03:10.432    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 395)   All edges processed\n",
      "  0:03:10.432    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 450)   Corrected 3 nucleotides\n",
      "  0:03:10.432    87M / 185M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:03:10.432    87M / 186M  INFO    General                 (contig_output.hpp         :  16)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/before_rr.fasta\n",
      "  0:03:10.460    87M / 186M  INFO    General                 (contig_output_stage.cpp   : 151)   Writing GFA graph to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/assembly_graph_after_simplification.gfa\n",
      "  0:03:10.484    87M / 186M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Hybrid Aligning (id: hybrid_aligning)\n",
      "  0:03:10.484    87M / 186M  INFO   HybridAligning           (hybrid_aligning.cpp       : 341)   Hybrid library detected: #1\n",
      "  0:03:10.484    87M / 186M  INFO    General                 (hybrid_aligning.cpp       : 294)   Aligning long reads with bwa-mem based aligner\n",
      "  0:03:10.554   100M / 186M  INFO    General                 (bwa_index.cpp             : 173)   Using BWA IS algorithm\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:03:12.336   214M / 259M  INFO    General                 (hybrid_aligning.cpp       : 274)   Prepared batch 0 of 10367 reads.\n",
      "  0:05:35.690   215M / 260M  INFO    General                 (hybrid_aligning.cpp       : 239)   Read batch of size: 10367 processed; 10052 of them longer than 500; among long reads aligned: 9890; paths of more than one edge received: 595\n",
      "  0:05:35.690   215M / 261M  INFO    General                 (hybrid_aligning.cpp       : 279)   Processed 10367 reads\n",
      "  0:05:35.692    96M / 261M  INFO    General                 (hybrid_aligning.cpp       : 308)   For library of long reads\n",
      "  0:05:35.692    96M / 261M  INFO   StatsCounter             (pacbio_read_structures.hpp: 163)   Median fraction of present seeds in maximal alignmnent among reads aligned to the graph: 0\n",
      "  0:05:35.692    96M / 261M  INFO    General                 (hybrid_aligning.cpp       : 310)   Aligning of long reads finished\n",
      "  0:05:35.692    87M / 261M  INFO   HybridAligning           (hybrid_aligning.cpp       : 388)   Padding gaps\n",
      "  0:05:35.692    87M / 261M  INFO   HybridAligning           (hybrid_aligning.cpp       : 392)   Min gap weight set to 2\n",
      "  0:05:35.692    87M / 261M  INFO    General                 (hybrid_aligning.cpp       : 146)   Closing gaps with long reads\n",
      "  0:05:35.724    87M / 261M  INFO   MultiGapJoiner           (hybrid_gap_closer.hpp     : 531)   Closed 3 gaps\n",
      "  0:05:35.758    87M / 261M  INFO    General                 (hybrid_aligning.cpp       : 167)   Closing gaps with long reads finished\n",
      "  0:05:35.758    87M / 261M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Paired Information Counting (id: late_pair_info_count)\n",
      "  0:05:35.761    87M / 261M  INFO    General                 (graph_pack.cpp            :  77)   Normalizing k-mer map. Total 56246 kmers to process\n",
      "  0:05:35.776    87M / 262M  INFO    General                 (graph_pack.cpp            :  79)   Normalizing done\n",
      "  0:05:35.776    87M / 262M  INFO    General                 (pair_info_count.cpp       : 339)   Min edge length for estimation: 61753\n",
      "  0:05:35.776    87M / 262M  INFO    General                 (pair_info_count.cpp       : 350)   Estimating insert size for library #0\n",
      "  0:05:35.776    87M / 262M  INFO    General                 (pair_info_count.cpp       : 202)   Estimating insert size (takes a while)\n",
      "  0:05:35.858   119M / 293M  INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  0:05:36.458   119M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:05:37.101   119M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:05:37.739   119M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:05:38.385   119M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:05:39.026   119M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:05:39.661   119M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:05:42.840   119M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:05:43.662   119M / 293M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:05:43.709   119M / 293M  INFO    General                 (pair_info_count.cpp       : 220)   Edge pairs: 3658\n",
      "  0:05:43.709   119M / 293M  INFO    General                 (pair_info_count.cpp       : 224)   1128696 paired reads (46.0942% of all) aligned to long edges\n",
      "  0:05:43.709    87M / 293M  INFO    General                 (pair_info_count.cpp       : 373)     Insert size = 215.4, deviation = 10.1891, left quantile = 202, right quantile = 229, read length = 100\n",
      "  0:05:43.709    87M / 293M  INFO    General                 (pair_info_count.cpp       : 390)   Filtering data for library #0\n",
      "  0:05:43.710    87M / 293M  INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  0:05:44.346    87M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:05:44.977    87M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:05:45.604    87M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:05:46.233    87M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:05:46.857    87M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:05:47.486    87M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:05:50.585    87M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:05:51.368    87M / 293M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:05:51.369    87M / 293M  INFO    General                 (pair_info_count.cpp       : 402)   Mapping library #0\n",
      "  0:05:51.369    87M / 293M  INFO    General                 (pair_info_count.cpp       : 404)   Mapping paired reads (takes a while)\n",
      "  0:05:51.369    87M / 293M  INFO    General                 (pair_info_count.cpp       : 302)   Left insert size quantile 202, right insert size quantile 229, filtering threshold 2, rounding threshold 0\n",
      "  0:05:51.371    96M / 293M  INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  0:05:52.063    97M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:05:52.750    97M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:05:53.431    97M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:05:54.114    97M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:05:54.795    97M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:05:55.473    97M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:05:58.825    97M / 293M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:05:59.668    97M / 293M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:05:59.672    88M / 293M  INFO    General                 (pair_info_count.cpp       : 343)   Library #1 was mapped earlier on hybrid aligning stage, skipping\n",
      "  0:05:59.672    88M / 293M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Distance Estimation (id: distance_estimation)\n",
      "  0:05:59.672    88M / 293M  INFO    General                 (distance_estimation.cpp   : 179)   Processing library #0\n",
      "  0:05:59.672    88M / 293M  INFO    General                 (distance_estimation.cpp   : 150)   Weight Filter Done\n",
      "  0:05:59.672    88M / 293M  INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:05:59.676    88M / 293M  INFO    General                 (distance_estimation.cpp   :  37)   Filtering info\n",
      "  0:05:59.676    88M / 293M  INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; library index size: 3698\n",
      "  0:05:59.677    88M / 293M  INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering; library index size: 3228\n",
      "  0:05:59.677    88M / 293M  INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:05:59.677    88M / 293M  INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:05:59.677    88M / 293M  INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:05:59.685    88M / 293M  INFO   PairInfoImprover         (pair_info_improver.hpp    : 105)   Paired info stats: missing = 202; contradictional = 8\n",
      "  0:05:59.691    88M / 293M  INFO   PairInfoImprover         (pair_info_improver.hpp    : 105)   Paired info stats: missing = 0; contradictional = 0\n",
      "  0:05:59.691    88M / 293M  INFO    General                 (distance_estimation.cpp   : 104)   Filling scaffolding index\n",
      "  0:05:59.691    88M / 293M  INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:05:59.796    88M / 294M  INFO    General                 (distance_estimation.cpp   :  37)   Filtering info\n",
      "  0:05:59.796    88M / 294M  INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; library index size: 879\n",
      "  0:05:59.797    88M / 294M  INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering; library index size: 879\n",
      "  0:05:59.797    88M / 294M  INFO    General                 (distance_estimation.cpp   : 186)   Clearing raw paired index\n",
      "  0:05:59.797    88M / 294M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Repeat Resolving (id: repeat_resolving)\n",
      "  0:05:59.797    88M / 294M  INFO    General                 (repeat_resolving.cpp      :  87)   Using Path-Extend repeat resolving\n",
      "  0:05:59.797    88M / 294M  INFO    General                 (launcher.cpp              : 600)   ExSPAnder repeat resolving tool started\n",
      "  0:05:59.798    88M / 294M  INFO    General                 (coverage_uniformity_analyz:  25)   genomic coverage is 40.29 calculated of length 4525454\n",
      "  0:05:59.798    88M / 294M  INFO    General                 (launcher.cpp              : 186)   Autodetecting unique edge set parameters...\n",
      "  0:05:59.798    88M / 294M  INFO    General                 (launcher.cpp              : 188)   Minimal unique edge length set to the smallest MP library IS: 2000\n",
      "  0:05:59.799    88M / 294M  INFO    General                 (coverage_uniformity_analyz:  25)   genomic coverage is 40.29 calculated of length 4503448\n",
      "  0:05:59.799    88M / 294M  INFO    General                 (launcher.cpp              : 194)   median coverage for edges longer than 2000 is 40.29 uniformity 99%\n",
      "  0:05:59.800    88M / 294M  INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  69)   With length cutoff: 2000, median long edge coverage: 40.29, and maximal unique coverage: 0.5\n",
      "  0:05:59.800    88M / 294M  INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  70)   Unique edges quantity: 230, unique edges length 9000952, total edges length 9111874\n",
      "  0:05:59.803    97M / 296M  INFO    General                 (launcher.cpp              : 418)   Creating main extenders, unique edge length = 2000\n",
      "  0:05:59.803    97M / 297M  INFO    General                 (launcher.cpp              : 342)   filling path container\n",
      "  0:05:59.803   104M / 297M  INFO    General                 (extenders_logic.cpp       : 334)   Estimated coverage of library #0 is 41.1665\n",
      "  0:05:59.803   104M / 297M  INFO    General                 (extenders_logic.cpp       : 345)   Creating extender; library index size: 3624\n",
      "  0:05:59.803   104M / 297M  INFO    General                 (extenders_logic.cpp       : 334)   Estimated coverage of library #0 is 41.1665\n",
      "  0:05:59.803   104M / 297M  INFO    General                 (extenders_logic.cpp       : 345)   Creating extender; library index size: 3624\n",
      "  0:05:59.804   104M / 297M  INFO    General                 (extenders_logic.cpp       :  47)   resolvable_repeat_length_bound set to 21547\n",
      "  0:05:59.806   104M / 297M  INFO    General                 (extenders_logic.cpp       : 543)   Using 1 paired-end library\n",
      "  0:05:59.806   104M / 297M  INFO    General                 (extenders_logic.cpp       : 544)   Using 1 paired-end scaffolding library\n",
      "  0:05:59.806   104M / 297M  INFO    General                 (extenders_logic.cpp       : 545)   Using 1 single read library\n",
      "  0:05:59.806   104M / 297M  INFO    General                 (launcher.cpp              : 385)   Filling backbone edges for long reads scaffolding...\n",
      "  0:05:59.806   104M / 297M  INFO    General                 (launcher.cpp              : 404)    with coverage\n",
      "  0:05:59.807   104M / 297M  INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  69)   With length cutoff: 500, median long edge coverage: 40.29, and maximal unique coverage: 0.5\n",
      "  0:05:59.807   104M / 297M  INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  70)   Unique edges quantity: 258, unique edges length 9034102, total edges length 9111874\n",
      "  0:05:59.807   104M / 297M  INFO    General                 (launcher.cpp              : 407)   258 unique edges\n",
      "  0:05:59.807   104M / 297M  INFO    General                 (extenders_logic.cpp       : 420)   Creating scaffolding extender for lib 1\n",
      "  0:05:59.807   104M / 297M  INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  50)   ExtensionChooser2015 created\n",
      "  0:05:59.807   104M / 297M  INFO    General                 (extenders_logic.cpp       : 446)   Using 1 long reads scaffolding library\n",
      "  0:05:59.807   104M / 297M  INFO    General                 (launcher.cpp              : 447)   Total number of extenders is 5\n",
      "  0:05:59.807   104M / 297M  INFO    General                 (path_extenders.cpp        :  36)   Processed 0 paths from 391 (0%)\n",
      "  0:05:59.814   106M / 297M  INFO    General                 (path_extenders.cpp        :  36)   Processed 40 paths from 391 (10%)\n",
      "  0:05:59.817   107M / 297M  INFO    General                 (path_extenders.cpp        :  36)   Processed 80 paths from 391 (20%)\n",
      "  0:05:59.817   107M / 297M  INFO    General                 (path_extenders.cpp        :  36)   Processed 120 paths from 391 (30%)\n",
      "  0:05:59.818   107M / 297M  INFO    General                 (path_extenders.cpp        :  34)   Processed 128 paths from 391 (32%)\n",
      "  0:05:59.819   108M / 297M  INFO    General                 (path_extenders.cpp        :  36)   Processed 160 paths from 391 (40%)\n",
      "  0:05:59.819   108M / 297M  INFO    General                 (path_extenders.cpp        :  36)   Processed 200 paths from 391 (51%)\n",
      "  0:05:59.820   109M / 297M  INFO    General                 (path_extenders.cpp        :  36)   Processed 240 paths from 391 (61%)\n",
      "  0:05:59.821   109M / 297M  INFO    General                 (path_extenders.cpp        :  34)   Processed 256 paths from 391 (65%)\n",
      "  0:05:59.821   109M / 297M  INFO    General                 (path_extenders.cpp        :  36)   Processed 280 paths from 391 (71%)\n",
      "  0:05:59.821   109M / 297M  INFO    General                 (path_extenders.cpp        :  36)   Processed 320 paths from 391 (81%)\n",
      "  0:05:59.821   109M / 297M  INFO    General                 (path_extenders.cpp        :  36)   Processed 360 paths from 391 (92%)\n",
      "  0:05:59.821   109M / 297M  INFO    General                 (launcher.cpp              : 250)   Finalizing paths\n",
      "  0:05:59.821   109M / 297M  INFO    General                 (launcher.cpp              : 252)   Deduplicating paths\n",
      "  0:05:59.822   109M / 297M  INFO    General                 (launcher.cpp              : 256)   Paths deduplicated\n",
      "  0:05:59.822   109M / 297M  INFO   PEResolver               (pe_resolver.cpp           :  60)   Removing overlaps\n",
      "  0:05:59.822   109M / 297M  INFO   PEResolver               (pe_resolver.cpp           :  63)   Sorting paths\n",
      "  0:05:59.822   109M / 297M  INFO   PEResolver               (pe_resolver.cpp           :  70)   Marking overlaps\n",
      "  0:05:59.822   109M / 297M  INFO   OverlapRemover           (overlap_remover.hpp       : 116)   Marking start/end overlaps\n",
      "  0:05:59.822   109M / 297M  INFO   OverlapRemover           (overlap_remover.hpp       : 119)   Marking remaining overlaps\n",
      "  0:05:59.823   109M / 297M  INFO   PEResolver               (pe_resolver.cpp           :  73)   Splitting paths\n",
      "  0:05:59.824   114M / 297M  INFO   PEResolver               (pe_resolver.cpp           :  78)   Deduplicating paths\n",
      "  0:05:59.824   108M / 297M  INFO   PEResolver               (pe_resolver.cpp           :  80)   Overlaps removed\n",
      "  0:05:59.824   108M / 297M  INFO    General                 (launcher.cpp              : 273)   Paths finalized\n",
      "  0:05:59.824   108M / 297M  INFO    General                 (launcher.cpp              : 454)   Closing gaps in paths\n",
      "  0:05:59.825   111M / 297M  INFO    General                 (launcher.cpp              : 484)   Gap closing completed\n",
      "  0:05:59.825   112M / 297M  INFO    General                 (launcher.cpp              : 302)   Traversing tandem repeats\n",
      "  0:05:59.826   112M / 297M  INFO    General                 (launcher.cpp              : 312)   Traversed 5 loops\n",
      "  0:05:59.826   112M / 297M  INFO    General                 (launcher.cpp              : 250)   Finalizing paths\n",
      "  0:05:59.826   112M / 297M  INFO    General                 (launcher.cpp              : 252)   Deduplicating paths\n",
      "  0:05:59.826   111M / 297M  INFO    General                 (launcher.cpp              : 256)   Paths deduplicated\n",
      "  0:05:59.826   111M / 297M  INFO   PEResolver               (pe_resolver.cpp           :  60)   Removing overlaps\n",
      "  0:05:59.826   111M / 297M  INFO   PEResolver               (pe_resolver.cpp           :  63)   Sorting paths\n",
      "  0:05:59.826   111M / 297M  INFO   PEResolver               (pe_resolver.cpp           :  70)   Marking overlaps\n",
      "  0:05:59.826   111M / 297M  INFO   OverlapRemover           (overlap_remover.hpp       : 116)   Marking start/end overlaps\n",
      "  0:05:59.827   111M / 297M  INFO   OverlapRemover           (overlap_remover.hpp       : 119)   Marking remaining overlaps\n",
      "  0:05:59.827   111M / 297M  INFO   PEResolver               (pe_resolver.cpp           :  73)   Splitting paths\n",
      "  0:05:59.827   112M / 297M  INFO   PEResolver               (pe_resolver.cpp           :  78)   Deduplicating paths\n",
      "  0:05:59.827   111M / 297M  INFO   PEResolver               (pe_resolver.cpp           :  80)   Overlaps removed\n",
      "  0:05:59.827   112M / 297M  INFO    General                 (launcher.cpp              : 273)   Paths finalized\n",
      "  0:05:59.827   112M / 297M  INFO    General                 (launcher.cpp              : 664)   ExSPAnder repeat resolving tool finished\n",
      "  0:05:59.828    91M / 297M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:05:59.829    91M / 297M  INFO    General                 (contig_output.hpp         :  16)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/before_rr.fasta\n",
      "  0:05:59.850    91M / 297M  INFO    General                 (contig_output_stage.cpp   : 151)   Writing GFA graph to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/assembly_graph_with_scaffolds.gfa\n",
      "  0:05:59.874    91M / 297M  INFO    General                 (contig_output_stage.cpp   : 165)   Outputting FastG graph to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/assembly_graph.fastg\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:05:59.939    91M / 297M  INFO    General                 (contig_output_stage.cpp   : 196)   Breaking scaffolds\n",
      "  0:05:59.949   100M / 298M  INFO    General                 (contig_output_stage.cpp   :  98)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/final_contigs.fasta\n",
      "  0:05:59.968   100M / 298M  INFO    General                 (contig_output_stage.cpp   : 104)   Outputting FastG paths to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/final_contigs.paths\n",
      "  0:05:59.978    96M / 298M  INFO    General                 (contig_output_stage.cpp   :  98)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/scaffolds.fasta\n",
      "  0:06:00.000    96M / 298M  INFO    General                 (contig_output_stage.cpp   : 104)   Outputting FastG paths to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/scaffolds.paths\n",
      "  0:06:00.000    96M / 298M  INFO    General                 (contig_output_stage.cpp   : 111)   Populating GFA with scaffold paths\n",
      "  0:06:00.001    91M / 298M  INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  0:06:00.004     1M / 257M  INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 6 minutes 0 seconds\n",
      "\n",
      "===== K55 finished. \n",
      "\n",
      "\n",
      "===== Copy files started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/miniconda3/bin/python /Users/olga/Downloads/SPAdes-3.15.2/share/spades/spades_pipeline/scripts/copy_files.py /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/before_rr.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/before_rr.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/assembly_graph_after_simplification.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/assembly_graph_after_simplification.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/final_contigs.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/contigs.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/first_pe_contigs.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/first_pe_contigs.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/strain_graph.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/strain_graph.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/scaffolds.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/scaffolds.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/scaffolds.paths /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/scaffolds.paths /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/assembly_graph_with_scaffolds.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/assembly_graph_with_scaffolds.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/assembly_graph.fastg /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/assembly_graph.fastg /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/K55/final_contigs.paths /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/contigs.paths\n",
      "\n",
      "\n",
      "===== Copy files finished. \n",
      "\n",
      "\n",
      "===== Assembling finished. \n",
      "\n",
      "\n",
      "===== Breaking scaffolds started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/miniconda3/bin/python /Users/olga/Downloads/SPAdes-3.15.2/share/spades/spades_pipeline/scripts/breaking_scaffolds_script.py --result_scaffolds_filename /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/scaffolds.fasta --misc_dir /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/misc --threshold_for_breaking_scaffolds 3\n",
      "\n",
      "\n",
      "===== Breaking scaffolds finished. \n",
      "\n",
      "\n",
      "===== Terminate started. \n",
      "\n",
      "\n",
      "===== Terminate finished. \n",
      "\n",
      " * Corrected reads are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/corrected/\n",
      " * Assembled contigs are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/contigs.fasta\n",
      " * Assembled scaffolds are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/scaffolds.fasta\n",
      " * Paths in the assembly graph corresponding to the contigs are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/scaffolds.paths\n",
      " * Assembly graph is in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/assembly_graph_with_scaffolds.gfa\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio10/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n"
     ]
    }
   ],
   "source": [
    "! /Users/olga/Downloads/SPAdes-3.15.2/bin/spades.py -1 ./homework6/illumina.100x.1.fq -2 ./homework6/illumina.100x.2.fq --pacbio ./homework6/pacbio_10x.fq -o illumina_pacbio10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "== Warning ==  No assembly mode was specified! If you intend to assemble high-coverage multi-cell/isolate data, use '--isolate' option.\n",
      "\n",
      "\n",
      "Command line: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades.py\t-1\t/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq\t-2\t/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq\t--pacbio\t/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/pacbio_20x.fq\t-o\t/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.15.2\n",
      "  Python version: 3.7.7\n",
      "  OS: Darwin-20.3.0-x86_64-i386-64bit\n",
      "\n",
      "Output dir: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Standard mode\n",
      "  For multi-cell/isolate data we recommend to use '--isolate' option; for single-cell MDA data use '--sc'; for metagenomic data use '--meta'; for RNA-Seq use '--rna'.\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq']\n",
      "      right reads: ['/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "    Library number: 2, library type: pacbio\n",
      "      left reads: not specified\n",
      "      right reads: not specified\n",
      "      interlaced reads: not specified\n",
      "      single reads: ['/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/pacbio_20x.fq']\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: automatic selection based on read length\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 250\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/spades.log\n",
      "\n",
      "/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq: max reads length: 100\n",
      "/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq: max reads length: 100\n",
      "\n",
      "Reads length: 100\n",
      "\n",
      "\n",
      "===== Before start started. \n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-hammer /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  76)   Loading config from /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/corrected/configs/config.info\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (hammer_tools.cpp          :  38)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 16 files using 1 threads. This might take a while.\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 4194304\n",
      "  0:00:00.000   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq\n",
      "  0:00:09.370   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 467314 reads\n",
      "  0:00:18.225   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 935009 reads\n",
      "  0:00:27.064   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 1402861 reads\n",
      "  0:00:35.894   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 1870646 reads\n",
      "  0:00:44.733   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2339377 reads\n",
      "  0:00:47.920   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2500000 reads\n",
      "  0:00:47.920   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq\n",
      "  0:00:56.858   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2974795 reads\n",
      "  0:01:23.733   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 4400968 reads\n",
      "  0:01:35.148   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 5000000 reads processed\n",
      "  0:01:35.149     1M / 162M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:01:39.723     1M / 170M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 38919786 kmers in total.\n",
      "  0:01:39.724     1M / 170M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:42.975    30M / 195M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 38919786 kmers, 28121944 bytes occupied (5.78049 bits per kmer).\n",
      "  0:01:42.976    30M / 195M  INFO   K-mer Counting           (kmer_data.cpp             : 354)   Arranging kmers in hash map order\n",
      "  0:01:45.163   631M / 789M  INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:05:44.674   631M / 789M  INFO    General                 (main.cpp                  : 155)   Extracting clusters:\n",
      "  0:05:44.675   631M / 789M  INFO    General                 (concurrent_dsu.cpp        :  18)   Connecting to root\n",
      "  0:05:45.411   631M / 789M  INFO    General                 (concurrent_dsu.cpp        :  34)   Calculating counts\n",
      "  0:05:57.439   903M / 1061M INFO    General                 (concurrent_dsu.cpp        :  63)   Writing down entries\n",
      "  0:06:09.574   631M / 789M  INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 13651605\n",
      "  0:06:09.591   330M / 492M  INFO   K-mer Counting           (kmer_data.cpp             : 371)   Collecting K-mer information, this takes a while.\n",
      "  0:06:09.974  1223M / 1383M INFO   K-mer Counting           (kmer_data.cpp             : 377)   Processing /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq\n",
      "  0:10:54.863  1223M / 1383M INFO   K-mer Counting           (kmer_data.cpp             : 377)   Processing /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq\n",
      "  0:15:30.420  1223M / 1383M INFO   K-mer Counting           (kmer_data.cpp             : 384)   Collection done, postprocessing.\n",
      "  0:15:30.656  1223M / 1383M INFO   K-mer Counting           (kmer_data.cpp             : 398)   There are 38919786 kmers in total. Among them 27377044 (70.3422%) are singletons.\n",
      "  0:15:30.656  1223M / 1383M INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:16:13.702  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering done. Total 26 non-read kmers were generated.\n",
      "  0:16:13.702  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)   Subclustering statistics:\n",
      "  0:16:13.702  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton hamming clusters: 5541662. Among them 963110 (17.3794%) are good\n",
      "  0:16:13.702  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total singleton subclusters: 13747. Among them 13599 (98.9234%) are good\n",
      "  0:16:13.702  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Total non-singleton subcluster centers: 8197112. Among them 6855795 (83.6367%) are good\n",
      "  0:16:13.702  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average size of non-trivial subcluster: 4.07194 kmers\n",
      "  0:16:13.702  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Average number of sub-clusters per non-singleton cluster: 1.01244\n",
      "  0:16:13.702  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Total solid k-mers: 7832504\n",
      "  0:16:13.702  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 658)     Substitution probabilities: [4,4]((0.928457,0.0421442,0.022541,0.00685828),(0.0141904,0.967828,0.0090854,0.00889584),(0.00886597,0.00903879,0.967867,0.0142285),(0.00683121,0.022511,0.0420995,0.928558))\n",
      "  0:16:13.709  1223M / 1384M INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:16:13.709  1223M / 1384M INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 1 threads.\n",
      "  0:18:50.372  1223M / 1384M INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 39977 new k-mers.\n",
      "  0:21:26.918  1223M / 1384M INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 790 new k-mers.\n",
      "  0:23:53.537  1223M / 1384M INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 38 new k-mers.\n",
      "  0:26:18.632  1223M / 1384M INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 3 produced 0 new k-mers.\n",
      "  0:26:18.632  1223M / 1384M INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:26:18.632  1223M / 1384M INFO    General                 (hammer_tools.cpp          : 222)   Starting read correction in 1 threads.\n",
      "  0:26:18.632  1223M / 1384M INFO    General                 (hammer_tools.cpp          : 235)   Correcting pair of reads: /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq and /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq\n",
      "  0:26:18.803  1289M / 1394M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 0 of 100000 reads.\n",
      "  0:26:25.093  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 0\n",
      "  0:26:25.308  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 0\n",
      "  0:26:25.451  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 1 of 100000 reads.\n",
      "  0:26:31.705  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 1\n",
      "  0:26:31.926  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 1\n",
      "  0:26:32.069  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 2 of 100000 reads.\n",
      "  0:26:38.265  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 2\n",
      "  0:26:38.479  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 2\n",
      "  0:26:38.622  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 3 of 100000 reads.\n",
      "  0:26:44.863  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 3\n",
      "  0:26:45.082  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 3\n",
      "  0:26:45.225  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 4 of 100000 reads.\n",
      "  0:26:51.422  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 4\n",
      "  0:26:51.656  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 4\n",
      "  0:26:51.803  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 5 of 100000 reads.\n",
      "  0:26:57.977  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 5\n",
      "  0:26:58.207  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 5\n",
      "  0:26:58.350  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 6 of 100000 reads.\n",
      "  0:27:04.590  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 6\n",
      "  0:27:04.822  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 6\n",
      "  0:27:04.965  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 7 of 100000 reads.\n",
      "  0:27:11.150  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 7\n",
      "  0:27:11.379  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 7\n",
      "  0:27:11.522  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 8 of 100000 reads.\n",
      "  0:27:17.689  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 8\n",
      "  0:27:17.919  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 8\n",
      "  0:27:18.063  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 9 of 100000 reads.\n",
      "  0:27:24.645  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 9\n",
      "  0:27:24.890  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 9\n",
      "  0:27:25.040  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 10 of 100000 reads.\n",
      "  0:27:31.796  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 10\n",
      "  0:27:32.032  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 10\n",
      "  0:27:32.174  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 11 of 100000 reads.\n",
      "  0:27:38.380  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 11\n",
      "  0:27:38.614  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 11\n",
      "  0:27:38.755  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 12 of 100000 reads.\n",
      "  0:27:44.975  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 12\n",
      "  0:27:45.206  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 12\n",
      "  0:27:45.349  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 13 of 100000 reads.\n",
      "  0:27:51.536  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 13\n",
      "  0:27:51.779  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 13\n",
      "  0:27:51.923  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 14 of 100000 reads.\n",
      "  0:27:58.107  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:27:58.341  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 14\n",
      "  0:27:58.485  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 15 of 100000 reads.\n",
      "  0:28:04.698  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 15\n",
      "  0:28:04.933  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 15\n",
      "  0:28:05.075  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 16 of 100000 reads.\n",
      "  0:28:11.244  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 16\n",
      "  0:28:11.477  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 16\n",
      "  0:28:11.620  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 17 of 100000 reads.\n",
      "  0:28:17.766  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 17\n",
      "  0:28:17.998  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 17\n",
      "  0:28:18.139  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 18 of 100000 reads.\n",
      "  0:28:24.357  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 18\n",
      "  0:28:24.589  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 18\n",
      "  0:28:24.732  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 19 of 100000 reads.\n",
      "  0:28:30.896  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 19\n",
      "  0:28:31.122  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 19\n",
      "  0:28:31.265  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 20 of 100000 reads.\n",
      "  0:28:37.429  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 20\n",
      "  0:28:37.659  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 20\n",
      "  0:28:37.803  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 21 of 100000 reads.\n",
      "  0:28:44.012  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 21\n",
      "  0:28:44.239  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 21\n",
      "  0:28:44.381  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 22 of 100000 reads.\n",
      "  0:28:50.601  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 22\n",
      "  0:28:50.839  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 22\n",
      "  0:28:50.982  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 23 of 100000 reads.\n",
      "  0:28:57.149  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 23\n",
      "  0:28:57.384  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 23\n",
      "  0:28:57.526  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 24 of 100000 reads.\n",
      "  0:29:03.737  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 24\n",
      "  0:29:03.971  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 24\n",
      "  0:29:03.980  1223M / 1396M INFO    General                 (hammer_tools.cpp          : 276)   Correction done. Changed 1639742 bases in 1194871 reads.\n",
      "  0:29:03.980  1223M / 1396M INFO    General                 (hammer_tools.cpp          : 277)   Failed to correct 44220 bases out of 458864877.\n",
      "  0:29:04.052     1M / 208M  INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/corrected/corrected.yaml\n",
      "  0:29:04.053     1M / 208M  INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== corrected reads compression started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/miniconda3/bin/python /Users/olga/Downloads/SPAdes-3.15.2/share/spades/spades_pipeline/scripts/compress_all.py --input_file /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/corrected/corrected.yaml --ext_python_modules_home /Users/olga/Downloads/SPAdes-3.15.2/share/spades --max_threads 16 --output_dir /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/corrected --not_used_yaml_file /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/corrected/dont_correct.yaml --gzip_output\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "== Files to compress: ['/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/corrected/illumina.100x.1.00.0_0.cor.fastq', '/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/corrected/illumina.100x.2.00.0_0.cor.fastq', '/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/corrected/illumina.100x._unpaired.00.0_0.cor.fastq']\n",
      "== Files compression is finished\n",
      "== Info about datasets not used in error correction stage is loaded\n",
      "== Dataset yaml file is updated\n",
      "\n",
      "===== corrected reads compression finished. \n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started. \n",
      "\n",
      "\n",
      "===== K21 started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-core /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K21/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  :  99)   Loaded config from /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K21/configs/config.info\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 109)   Assembling dataset (/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/dataset.info) with K=21\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:00:00.001     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.002     1M / 4M    INFO    General                 (read_converter.cpp        :  72)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.002     1M / 4M    INFO    General                 (read_converter.cpp        :  73)   Converting paired reads\n",
      "  0:00:00.070    22M / 22M   INFO    General                 (binary_converter.cpp      :  96)   16384 reads processed\n",
      "  0:00:00.134    25M / 25M   INFO    General                 (binary_converter.cpp      :  96)   32768 reads processed\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:00.345    22M / 25M   INFO    General                 (binary_converter.cpp      :  96)   65536 reads processed\n",
      "  0:00:00.670    25M / 31M   INFO    General                 (binary_converter.cpp      :  96)   131072 reads processed\n",
      "  0:00:01.400    21M / 31M   INFO    General                 (binary_converter.cpp      :  96)   262144 reads processed\n",
      "  0:00:02.734    24M / 31M   INFO    General                 (binary_converter.cpp      :  96)   524288 reads processed\n",
      "  0:00:05.415    28M / 31M   INFO    General                 (binary_converter.cpp      :  96)   1048576 reads processed\n",
      "  0:00:10.851    28M / 31M   INFO    General                 (binary_converter.cpp      :  96)   2097152 reads processed\n",
      "  0:00:12.749    19M / 31M   INFO    General                 (binary_converter.cpp      : 111)   2448672 reads written\n",
      "  0:00:12.749     1M / 31M   INFO    General                 (read_converter.cpp        :  81)   Converting single reads\n",
      "  0:00:12.781    10M / 31M   INFO    General                 (binary_converter.cpp      :  96)   16384 reads processed\n",
      "  0:00:12.813    12M / 31M   INFO    General                 (binary_converter.cpp      :  96)   32768 reads processed\n",
      "  0:00:12.880     9M / 31M   INFO    General                 (binary_converter.cpp      : 111)   48607 reads written\n",
      "  0:00:12.880     1M / 31M   INFO    General                 (read_converter.cpp        :  87)   Converting merged reads\n",
      "  0:00:12.880     9M / 31M   INFO    General                 (binary_converter.cpp      : 111)   0 reads written\n",
      "  0:00:12.881     1M / 31M   INFO    General                 (read_converter.cpp        :  72)   Converting reads to binary format for library #1 (takes a while)\n",
      "  0:00:12.881     1M / 31M   INFO    General                 (read_converter.cpp        :  73)   Converting paired reads\n",
      "  0:00:12.882    19M / 31M   INFO    General                 (binary_converter.cpp      : 111)   0 reads written\n",
      "  0:00:12.882     1M / 31M   INFO    General                 (read_converter.cpp        :  81)   Converting single reads\n",
      "  0:00:13.400    98M / 108M  INFO    General                 (binary_converter.cpp      :  96)   16384 reads processed\n",
      "  0:00:14.304     9M / 132M  INFO    General                 (binary_converter.cpp      : 111)   20829 reads written\n",
      "  0:00:14.304     1M / 132M  INFO    General                 (read_converter.cpp        :  87)   Converting merged reads\n",
      "  0:00:14.304     9M / 132M  INFO    General                 (binary_converter.cpp      : 111)   0 reads written\n",
      "  0:00:14.304     1M / 132M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:14.306     1M / 132M  INFO    General                 (construction.cpp          : 153)   Max read length 100\n",
      "  0:00:14.306     1M / 132M  INFO    General                 (construction.cpp          : 159)   Average read length 92.7147\n",
      "  0:00:14.306     1M / 132M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:14.306     1M / 132M  INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:14.306     1M / 132M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:14.306     1M / 132M  INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:14.306     1M / 132M  INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 6710886\n",
      "  0:00:24.400   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 1873811 reads\n",
      "  0:00:34.238   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 3734243 reads\n",
      "  0:00:44.083   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 5595232 reads\n",
      "  0:00:53.804   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 7456112 reads\n",
      "  0:01:03.504   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 9318931 reads\n",
      "  0:01:06.704   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 9891902 reads\n",
      "  0:01:06.704     1M / 186M  INFO    General                 (kmer_splitters.hpp        : 131)   Used 9891902 reads\n",
      "  0:01:06.734     1M / 186M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:01:07.218     1M / 186M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5715581 kmers in total.\n",
      "  0:01:07.218     1M / 186M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:01:07.219     1M / 186M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:01:07.219     1M / 186M  INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:01:07.219     1M / 186M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:01:07.219     1M / 186M  INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:01:07.219     1M / 186M  INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 6710886\n",
      "  0:01:09.056   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 194)   Processed 5715581 kmers\n",
      "  0:01:09.056   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 199)   Used 5715581 kmers.\n",
      "  0:01:09.056     1M / 229M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:01:09.238     1M / 229M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5708731 kmers in total.\n",
      "  0:01:09.238     1M / 229M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:09.681     5M / 229M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5708731 kmers, 4130592 bytes occupied (5.78846 bits per kmer).\n",
      "  0:01:09.681     5M / 229M  INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n",
      "  0:01:09.822    11M / 229M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:01:11.713    11M / 229M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:01:11.714    11M / 186M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Early tip clipping (id: construction:early_tip_clipper)\n",
      "  0:01:11.714    11M / 186M  INFO    General                 (construction.cpp          : 301)   Early tip clipper length bound set as (RL - K)\n",
      "  0:01:11.714    11M / 186M  INFO   Early tip clipping       (early_simplification.hpp  :  47)   Early tip clipping\n",
      "  0:01:13.490    21M / 229M  INFO   Early tip clipping       (early_simplification.hpp  :  82)   #tipped junctions: 123493\n",
      "  0:01:13.532    21M / 229M  INFO   Early tip clipping       (early_simplification.hpp  :  93)   Clipped tips: 124241\n",
      "  0:01:13.533    11M / 186M  INFO   Early tip clipping       (early_simplification.hpp  :  49)   978561 22-mers were removed by early tip clipper\n",
      "  0:01:13.533    11M / 186M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:01:13.537    11M / 186M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:01:15.836    13M / 229M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 25730 sequences extracted\n",
      "  0:01:16.639    13M / 186M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:01:17.049    13M / 229M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 4 loops collected\n",
      "  0:01:17.051    13M / 186M  INFO    General                 (debruijn_graph_constructor: 487)   Total 51468 edges to create\n",
      "  0:01:17.051    15M / 186M  INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:01:17.062    16M / 186M  INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:01:17.064    16M / 186M  INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:01:17.065    16M / 186M  INFO    General                 (debruijn_graph_constructor: 503)   Total 18884 vertices to create\n",
      "  0:01:17.065    18M / 186M  INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:01:17.067    16M / 186M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:01:17.067    16M / 186M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:17.513    20M / 186M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5715581 kmers, 4135544 bytes occupied (5.78845 bits per kmer).\n",
      "  0:01:17.514    42M / 186M  INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:02:21.952    42M / 186M  INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:02:23.431    42M / 186M  INFO    General                 (coverage_filling.hpp      :  82)   Processed 51447 edges\n",
      "  0:02:23.446     6M / 186M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:02:23.447     6M / 186M  INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 16\n",
      "  0:02:23.447     6M / 186M  INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 76\n",
      "  0:02:23.447     6M / 186M  INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 77. Coverage mad: 14.826\n",
      "  0:02:23.447     6M / 186M  INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:02:23.512     6M / 186M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:02:23.678     6M / 186M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:02:24.228     6M / 186M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:02:25.500     6M / 186M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:02:25.638     6M / 186M  INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 76.252. Fitted coverage std. dev: 14.5028\n",
      "  0:02:25.640     6M / 186M  INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.863705\n",
      "  0:02:25.640     6M / 186M  INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 37\n",
      "  0:02:25.640     6M / 186M  INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 37\n",
      "  0:02:25.640     6M / 186M  INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 4507902\n",
      "  0:02:25.640     6M / 186M  INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 76.252\n",
      "  0:02:25.640     6M / 186M  INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 37\n",
      "  0:02:25.640     6M / 186M  INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:02:25.640     6M / 186M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:02:25.641     6M / 186M  INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:02:25.643     6M / 186M  INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:02:25.644     6M / 186M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:02:25.647     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:02:25.647     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n",
      "  0:02:25.665     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 2371 times\n",
      "  0:02:25.665     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n",
      "  0:02:25.736     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 4029 times\n",
      "  0:02:25.736     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:02:25.738     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 308 times\n",
      "  0:02:25.738     6M / 187M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:02:25.738     6M / 187M  INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:02:25.738     6M / 187M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:02:25.738     6M / 187M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:02:25.738     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:25.741     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 128 times\n",
      "  0:02:25.741     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:25.812     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 1810 times\n",
      "  0:02:25.812     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:25.816     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 143 times\n",
      "  0:02:25.816     6M / 187M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:02:25.816     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:25.817     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 5 times\n",
      "  0:02:25.817     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:25.818     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 36 times\n",
      "  0:02:25.818     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:25.819     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 68 times\n",
      "  0:02:25.819     6M / 187M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:02:25.819     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:25.820     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 3 times\n",
      "  0:02:25.820     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:25.821     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 23 times\n",
      "  0:02:25.821     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:02:25.821     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 16 times\r\n",
      "  0:02:25.821     6M / 187M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\r\n",
      "  0:02:25.821     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:02:25.821     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:02:25.821     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:02:25.821     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 7 times\r\n",
      "  0:02:25.821     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 10 times\r\n",
      "  0:02:25.822     6M / 187M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 2 times\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 4 times\r\n",
      "  0:02:25.822     6M / 187M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 4 times\r\n",
      "  0:02:25.822     6M / 187M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 1 times\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 2 times\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 3 times\r\n",
      "  0:02:25.822     6M / 187M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\r\n",
      "  0:02:25.822     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:02:25.823     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 7 times\r\n",
      "  0:02:25.823     6M / 187M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\r\n",
      "  0:02:25.823     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:02:25.823     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:02:25.823     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:02:25.823     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\r\n",
      "  0:02:25.823     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:02:25.823     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\r\n",
      "  0:02:25.823     6M / 187M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\r\n",
      "  0:02:25.823     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:02:25.823     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:02:25.823     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:02:25.823     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\r\n",
      "  0:02:25.823     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:02:25.823     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 6 times\r\n",
      "  0:02:25.823     6M / 187M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\r\n",
      "  0:02:25.823     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:02:25.823     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:02:25.823     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:02:25.826     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 3 times\r\n",
      "  0:02:25.826     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:02:25.826     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\r\n",
      "  0:02:25.826     6M / 187M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 12\r\n",
      "  0:02:25.826     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:02:25.826     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:02:25.826     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:02:25.826     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\r\n",
      "  0:02:25.826     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\r\n",
      "  0:02:25.826     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\r\n",
      "  0:02:25.827     6M / 187M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\r\n",
      "  0:02:25.827     6M / 187M  INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\r\n",
      "  0:02:25.827     6M / 187M  INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\r\n",
      "  0:02:25.827     6M / 187M  INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\r\n",
      "  0:02:25.827     6M / 187M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\r\n",
      "  0:02:25.827     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:02:25.828     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:02:25.828     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:02:25.830     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\r\n",
      "  0:02:25.830     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\r\n",
      "  0:02:25.830     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\r\n",
      "  0:02:25.830     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\r\n",
      "  0:02:25.833     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\r\n",
      "  0:02:25.833     6M / 187M  INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\r\n",
      "  0:02:25.833     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\r\n",
      "  0:02:25.833     6M / 187M  INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 7 times\r\n",
      "  0:02:25.833     6M / 187M  INFO    General                 (simplification.cpp        : 491)   Counting average coverage\r\n",
      "  0:02:25.834     6M / 187M  INFO    General                 (simplification.cpp        : 497)   Average coverage = 78.0189\r\n",
      "  0:02:25.834     6M / 187M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\r\n",
      "  0:02:25.834     6M / 187M  INFO    General                 (read_converter.cpp        : 102)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K21/simplified_contigs\r\n",
      "  0:02:25.838     8M / 187M  INFO    General                 (binary_converter.cpp      : 111)   2571 reads written\r\n",
      "  0:02:25.838     6M / 187M  INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\r\n",
      "  0:02:25.839     1M / 187M  INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 2 minutes 25 seconds\r\n",
      "\r\n",
      "===== K21 finished. \r\n",
      "\r\n",
      "\r\n",
      "===== K33 started. \r\n",
      "\r\n",
      "\r\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-core /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K33/configs/config.info\r\n",
      "\r\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  :  99)   Loaded config from /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K33/configs/config.info\r\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\r\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\r\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\r\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 109)   Assembling dataset (/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/dataset.info) with K=33\r\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\r\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 212)   SPAdes started\r\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\r\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\r\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\r\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\r\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\r\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\r\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\r\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\r\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\r\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 118)   Contigs from previous K will be used: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K21/simplified_contigs\r\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 153)   Max read length 100\r\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 159)   Average read length 92.7147\r\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\r\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\r\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\r\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\r\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 3355443\r\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:05.822   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 1132634 reads\n",
      "  0:00:11.294   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 2249495 reads\n",
      "  0:00:16.752   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 3368100 reads\n",
      "  0:00:22.237   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 4487032 reads\n",
      "  0:00:27.689   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 5607299 reads\n",
      "  0:00:33.143   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 6726813 reads\n",
      "  0:00:38.596   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 7846654 reads\n",
      "  0:00:44.164   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 8966769 reads\n",
      "  0:00:49.413   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 9897044 reads\n",
      "  0:00:49.413     1M / 158M  INFO    General                 (kmer_splitters.hpp        : 131)   Used 9897044 reads\n",
      "  0:00:49.442     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:50.547     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5756125 kmers in total.\n",
      "  0:00:50.547     1M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:00:50.549     1M / 158M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:00:50.549     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:50.549     1M / 158M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:50.549     1M / 158M  INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:50.549     1M / 158M  INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 3355443\n",
      "  0:00:52.756   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 194)   Processed 5756125 kmers\n",
      "  0:00:52.756   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 199)   Used 5756125 kmers.\n",
      "  0:00:52.756     1M / 246M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:53.075     1M / 246M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5754745 kmers in total.\n",
      "  0:00:53.075     1M / 246M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:53.616     5M / 246M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5754745 kmers, 4163816 bytes occupied (5.78836 bits per kmer).\n",
      "  0:00:53.616     5M / 246M  INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n",
      "  0:00:53.885    11M / 246M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:00:55.883    11M / 246M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:55.884    11M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Early tip clipping (id: construction:early_tip_clipper)\n",
      "  0:00:55.884    11M / 158M  INFO    General                 (construction.cpp          : 301)   Early tip clipper length bound set as (RL - K)\n",
      "  0:00:55.884    11M / 158M  INFO   Early tip clipping       (early_simplification.hpp  :  47)   Early tip clipping\n",
      "  0:00:57.738    21M / 246M  INFO   Early tip clipping       (early_simplification.hpp  :  82)   #tipped junctions: 124481\n",
      "  0:00:57.780    21M / 246M  INFO   Early tip clipping       (early_simplification.hpp  :  93)   Clipped tips: 125222\n",
      "  0:00:57.781    11M / 158M  INFO   Early tip clipping       (early_simplification.hpp  :  49)   1029171 34-mers were removed by early tip clipper\n",
      "  0:00:57.781    11M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:00:57.792    11M / 158M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:01:00.119    12M / 248M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 15754 sequences extracted\n",
      "  0:01:00.974    12M / 160M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:01:01.407    12M / 248M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 2 loops collected\n",
      "  0:01:01.409    12M / 160M  INFO    General                 (debruijn_graph_constructor: 487)   Total 31512 edges to create\n",
      "  0:01:01.409    14M / 160M  INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:01:01.416    14M / 160M  INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:01:01.418    14M / 160M  INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:01:01.418    14M / 160M  INFO    General                 (debruijn_graph_constructor: 503)   Total 14376 vertices to create\n",
      "  0:01:01.418    16M / 160M  INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:01:01.420    15M / 160M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:01:01.420    15M / 160M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:01.952    19M / 160M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5756125 kmers, 4164864 bytes occupied (5.78843 bits per kmer).\n",
      "  0:01:01.954    41M / 163M  INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:01:53.125    41M / 163M  INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:01:54.650    41M / 163M  INFO    General                 (coverage_filling.hpp      :  82)   Processed 31510 edges\n",
      "  0:01:54.669     5M / 163M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:01:54.669     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 13\n",
      "  0:01:54.669     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 64\n",
      "  0:01:54.669     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 64. Coverage mad: 14.826\n",
      "  0:01:54.669     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:01:54.723     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:01:54.865     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:01:55.323     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:01:56.407     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:01:57.085     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 63.4837. Fitted coverage std. dev: 13.4198\n",
      "  0:01:57.087     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.693899\n",
      "  0:01:57.087     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 29\n",
      "  0:01:57.087     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 29\n",
      "  0:01:57.087     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 4507034\n",
      "  0:01:57.087     5M / 163M  INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 63.4837\n",
      "  0:01:57.087     5M / 163M  INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 29\n",
      "  0:01:57.087     5M / 163M  INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:01:57.087     5M / 163M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:01:57.087     5M / 163M  INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:01:57.087     5M / 163M  INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:01:57.087     5M / 163M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:01:57.089     5M / 163M  INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:01:57.089     5M / 163M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n",
      "  0:01:57.124     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 2699 times\n",
      "  0:01:57.124     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n",
      "  0:01:57.161     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 1350 times\n",
      "  0:01:57.162     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:01:57.163     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 1603 times\n",
      "  0:01:57.163     5M / 164M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:01:57.163     5M / 164M  INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:01:57.163     5M / 164M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:01:57.163     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:01:57.163     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:57.165     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 59 times\n",
      "  0:01:57.165     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:57.230     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 1018 times\n",
      "  0:01:57.230     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:57.232     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 42 times\n",
      "  0:01:57.232     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:01:57.232     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:57.233     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 5 times\n",
      "  0:01:57.233     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:57.233     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 17 times\n",
      "  0:01:57.233     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:57.235     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 52 times\n",
      "  0:01:57.235     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:01:57.235     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:57.235     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 1 times\n",
      "  0:01:57.235     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:57.237     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 27 times\n",
      "  0:01:57.237     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:57.238     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 10 times\n",
      "  0:01:57.238     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:01:57.238     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:57.238     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:57.238     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:57.238     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 9 times\n",
      "  0:01:57.238     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:57.238     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 8 times\n",
      "  0:01:57.238     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:01:57.238     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:57.238     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:57.238     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:57.238     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 5 times\n",
      "  0:01:57.238     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 6 times\n",
      "  0:01:57.239     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:01:57.239     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 3 times\n",
      "  0:01:57.239     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:01:57.239     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:01:57.239     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:01:57.239     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:01:57.239     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:57.240     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:57.240     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:57.240     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 2 times\n",
      "  0:01:57.240     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:57.241     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:01:57.241     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:01:57.241     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:57.241     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:57.241     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:57.241     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:57.241     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:57.241     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:01:57.241     5M / 164M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
      "  0:01:57.241     5M / 165M  INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\n",
      "  0:01:57.241     5M / 165M  INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\n",
      "  0:01:57.241     5M / 165M  INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\n",
      "  0:01:57.241     5M / 165M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:01:57.241     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:57.241     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:57.241     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:57.242     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:57.242     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:57.242     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:57.242     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:01:57.243     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:57.243     5M / 165M  INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:01:57.244     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:01:57.244     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 5 times\n",
      "  0:01:57.244     5M / 165M  INFO    General                 (simplification.cpp        : 491)   Counting average coverage\n",
      "  0:01:57.244     5M / 165M  INFO    General                 (simplification.cpp        : 497)   Average coverage = 64.7522\n",
      "  0:01:57.244     5M / 165M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:01:57.244     5M / 165M  INFO    General                 (read_converter.cpp        : 102)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K33/simplified_contigs\n",
      "  0:01:57.248     7M / 165M  INFO    General                 (binary_converter.cpp      : 111)   1188 reads written\n",
      "  0:01:57.248     5M / 165M  INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  0:01:57.249     1M / 165M  INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 1 minutes 57 seconds\n",
      "\n",
      "===== K33 finished. \n",
      "\n",
      "\n",
      "===== K55 started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-core /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  :  99)   Loaded config from /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/configs/config.info\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 109)   Assembling dataset (/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/dataset.info) with K=55\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 242)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 118)   Contigs from previous K will be used: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K33/simplified_contigs\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 153)   Max read length 100\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 159)   Average read length 92.7147\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 3355443\n",
      "  0:00:06.031   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 1768213 reads\n",
      "  0:00:11.709   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 3518079 reads\n",
      "  0:00:17.383   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 5268519 reads\n",
      "  0:00:23.084   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 7021543 reads\n",
      "  0:00:28.818   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 8774696 reads\n",
      "  0:00:33.231   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 9894278 reads\n",
      "  0:00:33.231     1M / 158M  INFO    General                 (kmer_splitters.hpp        : 131)   Used 9894278 reads\n",
      "  0:00:33.261     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:33.995     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5692221 kmers in total.\n",
      "  0:00:33.995     1M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:00:33.997     1M / 158M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:00:33.997     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:33.998     1M / 158M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:33.998     1M / 158M  INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:33.998     1M / 158M  INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 3355443\n",
      "  0:00:36.186   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 194)   Processed 5692221 kmers\n",
      "  0:00:36.186   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 199)   Used 5692221 kmers.\n",
      "  0:00:36.186     1M / 245M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:36.501     1M / 245M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5698724 kmers in total.\n",
      "  0:00:36.501     1M / 245M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:37.037     5M / 245M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5698724 kmers, 4123376 bytes occupied (5.78849 bits per kmer).\n",
      "  0:00:37.037     5M / 245M  INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:37.311    11M / 245M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:00:39.318    11M / 245M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:39.320    11M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:00:39.329    11M / 158M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:00:42.346    23M / 248M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 247400 sequences extracted\n",
      "  0:00:43.516    23M / 161M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:00:43.949    23M / 248M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:43.955    23M / 161M  INFO    General                 (debruijn_graph_constructor: 487)   Total 494800 edges to create\n",
      "  0:00:43.955    42M / 161M  INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:00:44.107    50M / 172M  INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:00:44.140    50M / 172M  INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:00:44.143    52M / 172M  INFO    General                 (debruijn_graph_constructor: 503)   Total 253903 vertices to create\n",
      "  0:00:44.143    76M / 172M  INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:00:44.174    62M / 172M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:00:44.174    62M / 172M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:44.723    66M / 172M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5692221 kmers, 4118704 bytes occupied (5.78854 bits per kmer).\n",
      "  0:00:44.724    88M / 172M  INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:01:19.614    88M / 172M  INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:01:21.788    88M / 172M  INFO    General                 (coverage_filling.hpp      :  82)   Processed 494800 edges\n",
      "  0:01:21.807    52M / 172M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:01:21.807    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 6\n",
      "  0:01:21.807    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 41\n",
      "  0:01:21.807    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 41. Coverage mad: 11.8608\n",
      "  0:01:21.807    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:01:21.842    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:01:21.942    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:01:22.244    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:01:22.948    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:01:23.446    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 40.4747. Fitted coverage std. dev: 10.9026\n",
      "  0:01:23.447    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.512008\n",
      "  0:01:23.447    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 16\n",
      "  0:01:23.447    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 16\n",
      "  0:01:23.447    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 4487959\n",
      "  0:01:23.447    52M / 172M  INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 40.4747\n",
      "  0:01:23.447    52M / 172M  INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 16\n",
      "  0:01:23.447    52M / 172M  INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:01:23.447    52M / 172M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Gap Closer (id: early_gapcloser)\n",
      "  0:01:23.449    61M / 172M  INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:01:23.538    66M / 172M  INFO    General                 (gap_closer.cpp            : 101)   Total edges in tip neighborhood: 134528 out of 494800, length: 1226132\n",
      "  0:01:23.547    68M / 172M  INFO    General                 (edge_index.hpp            : 167)   Using small index (max_id = 497276)\n",
      "  0:01:23.649    68M / 172M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:24.119    69M / 172M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 1092761 kmers, 790000 bytes occupied (5.78352 bits per kmer).\n",
      "  0:01:24.120    78M / 172M  INFO    General                 (edge_index_builders.hpp   : 265)   Collecting edge information from graph, this takes a while.\n",
      "  0:01:29.047    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:01:33.527    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:01:37.999    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:01:42.203    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:01:46.461    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:01:50.932    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:02:13.117    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:02:18.669    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:02:18.674    62M / 172M  INFO    General                 (gap_closer.cpp            : 452)   Initializing gap closer\n",
      "  0:02:18.674    62M / 172M  INFO   GapCloser                (gap_closer.cpp            : 359)   Closing short gaps\n",
      "  0:02:18.873    62M / 178M  INFO   GapCloser                (gap_closer.cpp            : 393)   Closing short gaps complete: filled 0 gaps after checking 20 candidates\n",
      "  0:02:18.879    62M / 178M  INFO    General                 (gap_closer.cpp            : 456)   Gap closer done\n",
      "  0:02:18.880    52M / 178M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:02:18.880    52M / 178M  INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:02:18.880    52M / 178M  INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:02:18.880    52M / 178M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:02:18.908    52M / 178M  INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:02:18.908    52M / 178M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:02:19.669    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 119766 times\n",
      "  0:02:19.669    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n",
      "  0:02:19.676    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 92 times\n",
      "  0:02:19.676    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:02:19.685    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 7236 times\n",
      "  0:02:19.686    45M / 179M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:02:19.686    45M / 179M  INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:02:19.686    45M / 179M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:02:19.686    45M / 179M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:02:19.686    45M / 179M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.687    45M / 179M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 19 times\n",
      "  0:02:19.687    45M / 179M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.788    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 447 times\n",
      "  0:02:19.788    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.789    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:19.789    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:02:19.789    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.789    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 3 times\n",
      "  0:02:19.789    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.789    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:19.789    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.790    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 13 times\n",
      "  0:02:19.790    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:02:19.790    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.790    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 1 times\n",
      "  0:02:19.790    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.791    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 8 times\n",
      "  0:02:19.791    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.792    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 11 times\n",
      "  0:02:19.792    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:02:19.792    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.792    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 3 times\n",
      "  0:02:19.792    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.793    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 7 times\n",
      "  0:02:19.793    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.793    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:02:19.793    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:02:19.793    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.793    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:19.793    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.793    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:19.793    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.793    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 3 times\n",
      "  0:02:19.793    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:02:19.793    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.793    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:19.793    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.794    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 1 times\n",
      "  0:02:19.794    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.794    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:02:19.794    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:02:19.794    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.794    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:19.794    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.794    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:19.794    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.794    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:02:19.794    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:02:19.794    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.794    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:19.794    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.794    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:19.794    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.794    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:19.794    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:02:19.794    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.794    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:19.794    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.794    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:19.794    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.795    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:02:19.795    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:02:19.795    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.795    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:19.795    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.795    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:19.795    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.795    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:02:19.795    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:02:19.795    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.796    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:19.796    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.797    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:19.797    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.798    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:19.798    48M / 180M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Gap Closer (id: late_gapcloser)\n",
      "  0:02:19.800    57M / 180M  INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:02:19.801    57M / 180M  INFO    General                 (gap_closer.cpp            : 101)   Total edges in tip neighborhood: 15 out of 882, length: 167704\n",
      "  0:02:19.801    57M / 180M  INFO    General                 (edge_index.hpp            : 167)   Using small index (max_id = 497276)\n",
      "  0:02:19.806    57M / 180M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:02:19.836    57M / 180M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 166932 kmers, 121312 bytes occupied (5.81372 bits per kmer).\n",
      "  0:02:19.836    58M / 180M  INFO    General                 (edge_index_builders.hpp   : 265)   Collecting edge information from graph, this takes a while.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:02:22.822    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:02:25.813    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:02:28.764    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:02:31.713    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:02:34.686    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:02:37.617    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:02:52.254    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:02:55.913    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:02:55.915    57M / 180M  INFO    General                 (gap_closer.cpp            : 452)   Initializing gap closer\n",
      "  0:02:55.916    57M / 180M  INFO   GapCloser                (gap_closer.cpp            : 359)   Closing short gaps\n",
      "  0:02:55.917    57M / 180M  INFO   GapCloser                (gap_closer.cpp            : 393)   Closing short gaps complete: filled 1 gaps after checking 7 candidates\n",
      "  0:02:55.917    57M / 180M  INFO    General                 (gap_closer.cpp            : 456)   Gap closer done\n",
      "  0:02:55.918    48M / 180M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
      "  0:02:55.918    48M / 180M  INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\n",
      "  0:02:55.918    48M / 180M  INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\n",
      "  0:02:55.918    48M / 180M  INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\n",
      "  0:02:55.918    48M / 180M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:02:55.918    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:55.919    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:55.919    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:55.920    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:55.920    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:55.921    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:55.921    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:55.921    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:55.921    48M / 180M  INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:02:55.922    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:02:55.923    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 1 times\n",
      "  0:02:55.923    48M / 180M  INFO    General                 (simplification.cpp        : 491)   Counting average coverage\n",
      "  0:02:55.923    48M / 180M  INFO    General                 (simplification.cpp        : 497)   Average coverage = 41.1665\n",
      "  0:02:55.923    48M / 180M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Mismatch Correction (id: mismatch_correction)\n",
      "  0:02:55.923    48M / 180M  INFO    General                 (graph_pack.cpp            :  67)   Index refill\n",
      "  0:02:55.923    48M / 180M  INFO    General                 (edge_index.hpp            : 156)   Using small index (max_id = 497276)\n",
      "  0:02:56.095    48M / 180M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:02:57.311    51M / 180M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 4555847 kmers, 3291264 bytes occupied (5.77941 bits per kmer).\n",
      "  0:02:57.316    87M / 185M  INFO    General                 (edge_index_builders.hpp   : 252)   Collecting edge information from graph, this takes a while.\n",
      "  0:02:58.159    87M / 185M  INFO    General                 (graph_pack.cpp            :  77)   Normalizing k-mer map. Total 56128 kmers to process\n",
      "  0:02:58.175    87M / 185M  INFO    General                 (graph_pack.cpp            :  79)   Normalizing done\n",
      "  0:02:58.175    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 400)   Collect potential mismatches\n",
      "  0:02:58.234    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 193)   Total 354 edges (out of 878) with 1752 potential mismatch positions (4.94915 positions per edge)\n",
      "  0:02:58.234    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 402)   Potential mismatches collected\n",
      "  0:02:58.653    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:02:59.103    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:02:59.544    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:02:59.991    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:03:00.444    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:03:00.885    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:03:03.107    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:03:07.523    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 4200000 reads\n",
      "  0:03:16.680    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 8400000 reads\n",
      "  0:03:19.835    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 9891902 reads processed\n",
      "  0:03:19.869    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 395)   All edges processed\n",
      "  0:03:19.870    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 450)   Corrected 3 nucleotides\n",
      "  0:03:19.870    87M / 185M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:03:19.870    87M / 185M  INFO    General                 (contig_output.hpp         :  16)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/before_rr.fasta\n",
      "  0:03:19.898    87M / 186M  INFO    General                 (contig_output_stage.cpp   : 151)   Writing GFA graph to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/assembly_graph_after_simplification.gfa\n",
      "  0:03:19.920    87M / 186M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Hybrid Aligning (id: hybrid_aligning)\n",
      "  0:03:19.920    87M / 186M  INFO   HybridAligning           (hybrid_aligning.cpp       : 341)   Hybrid library detected: #1\n",
      "  0:03:19.921    87M / 186M  INFO    General                 (hybrid_aligning.cpp       : 294)   Aligning long reads with bwa-mem based aligner\n",
      "  0:03:19.990   100M / 186M  INFO    General                 (bwa_index.cpp             : 173)   Using BWA IS algorithm\n",
      "  0:03:22.064   329M / 373M  INFO    General                 (hybrid_aligning.cpp       : 274)   Prepared batch 0 of 20829 reads.\n",
      "  0:08:30.617   330M / 376M  INFO    General                 (hybrid_aligning.cpp       : 239)   Read batch of size: 20829 processed; 20135 of them longer than 500; among long reads aligned: 19812; paths of more than one edge received: 1248\n",
      "  0:08:30.617   330M / 376M  INFO    General                 (hybrid_aligning.cpp       : 279)   Processed 20829 reads\n",
      "  0:08:30.619    96M / 376M  INFO    General                 (hybrid_aligning.cpp       : 308)   For library of long reads\n",
      "  0:08:30.619    96M / 376M  INFO   StatsCounter             (pacbio_read_structures.hpp: 163)   Median fraction of present seeds in maximal alignmnent among reads aligned to the graph: 0\n",
      "  0:08:30.619    96M / 376M  INFO    General                 (hybrid_aligning.cpp       : 310)   Aligning of long reads finished\n",
      "  0:08:30.620    87M / 376M  INFO   HybridAligning           (hybrid_aligning.cpp       : 388)   Padding gaps\n",
      "  0:08:30.620    87M / 376M  INFO   HybridAligning           (hybrid_aligning.cpp       : 392)   Min gap weight set to 2\n",
      "  0:08:30.620    87M / 376M  INFO    General                 (hybrid_aligning.cpp       : 146)   Closing gaps with long reads\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:08:30.653    87M / 376M  INFO   MultiGapJoiner           (hybrid_gap_closer.hpp     : 531)   Closed 3 gaps\n",
      "  0:08:30.683    87M / 376M  INFO    General                 (hybrid_aligning.cpp       : 167)   Closing gaps with long reads finished\n",
      "  0:08:30.683    87M / 376M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Paired Information Counting (id: late_pair_info_count)\n",
      "  0:08:30.686    87M / 376M  INFO    General                 (graph_pack.cpp            :  77)   Normalizing k-mer map. Total 56246 kmers to process\n",
      "  0:08:30.702    87M / 376M  INFO    General                 (graph_pack.cpp            :  79)   Normalizing done\n",
      "  0:08:30.703    87M / 376M  INFO    General                 (pair_info_count.cpp       : 339)   Min edge length for estimation: 61753\n",
      "  0:08:30.703    87M / 376M  INFO    General                 (pair_info_count.cpp       : 350)   Estimating insert size for library #0\n",
      "  0:08:30.703    87M / 376M  INFO    General                 (pair_info_count.cpp       : 202)   Estimating insert size (takes a while)\n",
      "  0:08:30.773   119M / 377M  INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  0:08:31.402   119M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:08:32.039   119M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:08:32.677   119M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:08:33.315   119M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:08:33.951   119M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:08:34.584   119M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:08:37.741   119M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:08:38.557   119M / 377M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:08:38.604   119M / 377M  INFO    General                 (pair_info_count.cpp       : 220)   Edge pairs: 3658\n",
      "  0:08:38.604   119M / 377M  INFO    General                 (pair_info_count.cpp       : 224)   1128696 paired reads (46.0942% of all) aligned to long edges\n",
      "  0:08:38.605    87M / 377M  INFO    General                 (pair_info_count.cpp       : 373)     Insert size = 215.4, deviation = 10.1891, left quantile = 202, right quantile = 229, read length = 100\n",
      "  0:08:38.605    87M / 377M  INFO    General                 (pair_info_count.cpp       : 390)   Filtering data for library #0\n",
      "  0:08:38.605    87M / 377M  INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  0:08:39.235    87M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:08:39.865    87M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:08:40.493    87M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:08:41.123    87M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:08:41.744    87M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:08:42.366    87M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:08:45.474    87M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:08:46.251    87M / 377M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:08:46.251    87M / 377M  INFO    General                 (pair_info_count.cpp       : 402)   Mapping library #0\n",
      "  0:08:46.251    87M / 377M  INFO    General                 (pair_info_count.cpp       : 404)   Mapping paired reads (takes a while)\n",
      "  0:08:46.251    87M / 377M  INFO    General                 (pair_info_count.cpp       : 302)   Left insert size quantile 202, right insert size quantile 229, filtering threshold 2, rounding threshold 0\n",
      "  0:08:46.253    96M / 377M  INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  0:08:46.932    97M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:08:47.615    97M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:08:48.290    97M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:08:48.970    97M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:08:49.651    97M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:08:50.329    97M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:08:53.736    97M / 377M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:08:54.583    97M / 377M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:08:54.586    88M / 377M  INFO    General                 (pair_info_count.cpp       : 343)   Library #1 was mapped earlier on hybrid aligning stage, skipping\n",
      "  0:08:54.586    88M / 377M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Distance Estimation (id: distance_estimation)\n",
      "  0:08:54.586    88M / 377M  INFO    General                 (distance_estimation.cpp   : 179)   Processing library #0\n",
      "  0:08:54.586    88M / 377M  INFO    General                 (distance_estimation.cpp   : 150)   Weight Filter Done\n",
      "  0:08:54.587    88M / 377M  INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:08:54.593    88M / 377M  INFO    General                 (distance_estimation.cpp   :  37)   Filtering info\n",
      "  0:08:54.593    88M / 377M  INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; library index size: 3698\n",
      "  0:08:54.594    88M / 377M  INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering; library index size: 3228\n",
      "  0:08:54.594    88M / 377M  INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:08:54.594    88M / 377M  INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:08:54.594    88M / 377M  INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:08:54.600    88M / 378M  INFO   PairInfoImprover         (pair_info_improver.hpp    : 105)   Paired info stats: missing = 202; contradictional = 8\n",
      "  0:08:54.606    88M / 378M  INFO   PairInfoImprover         (pair_info_improver.hpp    : 105)   Paired info stats: missing = 0; contradictional = 0\n",
      "  0:08:54.606    88M / 378M  INFO    General                 (distance_estimation.cpp   : 104)   Filling scaffolding index\n",
      "  0:08:54.606    88M / 378M  INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:08:54.715    88M / 378M  INFO    General                 (distance_estimation.cpp   :  37)   Filtering info\n",
      "  0:08:54.715    88M / 378M  INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; library index size: 879\n",
      "  0:08:54.715    88M / 378M  INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering; library index size: 879\n",
      "  0:08:54.715    88M / 378M  INFO    General                 (distance_estimation.cpp   : 186)   Clearing raw paired index\n",
      "  0:08:54.716    88M / 378M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Repeat Resolving (id: repeat_resolving)\n",
      "  0:08:54.716    88M / 378M  INFO    General                 (repeat_resolving.cpp      :  87)   Using Path-Extend repeat resolving\n",
      "  0:08:54.716    88M / 378M  INFO    General                 (launcher.cpp              : 600)   ExSPAnder repeat resolving tool started\n",
      "  0:08:54.717    88M / 378M  INFO    General                 (coverage_uniformity_analyz:  25)   genomic coverage is 40.29 calculated of length 4525456\n",
      "  0:08:54.718    88M / 378M  INFO    General                 (launcher.cpp              : 186)   Autodetecting unique edge set parameters...\n",
      "  0:08:54.718    88M / 378M  INFO    General                 (launcher.cpp              : 188)   Minimal unique edge length set to the smallest MP library IS: 2000\n",
      "  0:08:54.718    88M / 378M  INFO    General                 (coverage_uniformity_analyz:  25)   genomic coverage is 40.29 calculated of length 4503450\n",
      "  0:08:54.719    88M / 378M  INFO    General                 (launcher.cpp              : 194)   median coverage for edges longer than 2000 is 40.29 uniformity 99%\n",
      "  0:08:54.720    88M / 378M  INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  69)   With length cutoff: 2000, median long edge coverage: 40.29, and maximal unique coverage: 0.5\n",
      "  0:08:54.720    88M / 378M  INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  70)   Unique edges quantity: 230, unique edges length 9000956, total edges length 9111878\n",
      "  0:08:54.722    97M / 379M  INFO    General                 (launcher.cpp              : 418)   Creating main extenders, unique edge length = 2000\n",
      "  0:08:54.722    97M / 379M  INFO    General                 (launcher.cpp              : 342)   filling path container\n",
      "  0:08:54.724   106M / 379M  INFO    General                 (extenders_logic.cpp       : 334)   Estimated coverage of library #0 is 41.1665\n",
      "  0:08:54.724   106M / 379M  INFO    General                 (extenders_logic.cpp       : 345)   Creating extender; library index size: 3624\n",
      "  0:08:54.725   106M / 379M  INFO    General                 (extenders_logic.cpp       : 334)   Estimated coverage of library #0 is 41.1665\n",
      "  0:08:54.725   106M / 379M  INFO    General                 (extenders_logic.cpp       : 345)   Creating extender; library index size: 3624\n",
      "  0:08:54.726   106M / 379M  INFO    General                 (extenders_logic.cpp       :  47)   resolvable_repeat_length_bound set to 22609\n",
      "  0:08:54.728   106M / 379M  INFO    General                 (extenders_logic.cpp       : 543)   Using 1 paired-end library\n",
      "  0:08:54.728   106M / 379M  INFO    General                 (extenders_logic.cpp       : 544)   Using 1 paired-end scaffolding library\n",
      "  0:08:54.728   106M / 379M  INFO    General                 (extenders_logic.cpp       : 545)   Using 1 single read library\n",
      "  0:08:54.728   106M / 379M  INFO    General                 (launcher.cpp              : 385)   Filling backbone edges for long reads scaffolding...\n",
      "  0:08:54.728   106M / 379M  INFO    General                 (launcher.cpp              : 404)    with coverage\n",
      "  0:08:54.729   106M / 379M  INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  69)   With length cutoff: 500, median long edge coverage: 40.29, and maximal unique coverage: 0.5\n",
      "  0:08:54.729   106M / 379M  INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  70)   Unique edges quantity: 258, unique edges length 9034106, total edges length 9111878\n",
      "  0:08:54.729   106M / 379M  INFO    General                 (launcher.cpp              : 407)   258 unique edges\n",
      "  0:08:54.729   106M / 379M  INFO    General                 (extenders_logic.cpp       : 420)   Creating scaffolding extender for lib 1\n",
      "  0:08:54.729   106M / 379M  INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  50)   ExtensionChooser2015 created\n",
      "  0:08:54.729   106M / 379M  INFO    General                 (extenders_logic.cpp       : 446)   Using 1 long reads scaffolding library\n",
      "  0:08:54.729   106M / 379M  INFO    General                 (launcher.cpp              : 447)   Total number of extenders is 5\n",
      "  0:08:54.730   106M / 379M  INFO    General                 (path_extenders.cpp        :  36)   Processed 0 paths from 391 (0%)\n",
      "  0:08:54.737   108M / 379M  INFO    General                 (path_extenders.cpp        :  36)   Processed 40 paths from 391 (10%)\n",
      "  0:08:54.739   109M / 379M  INFO    General                 (path_extenders.cpp        :  36)   Processed 80 paths from 391 (20%)\n",
      "  0:08:54.740   109M / 379M  INFO    General                 (path_extenders.cpp        :  36)   Processed 120 paths from 391 (30%)\n",
      "  0:08:54.740   109M / 379M  INFO    General                 (path_extenders.cpp        :  34)   Processed 128 paths from 391 (32%)\n",
      "  0:08:54.741   110M / 379M  INFO    General                 (path_extenders.cpp        :  36)   Processed 160 paths from 391 (40%)\n",
      "  0:08:54.742   110M / 379M  INFO    General                 (path_extenders.cpp        :  36)   Processed 200 paths from 391 (51%)\n",
      "  0:08:54.743   111M / 379M  INFO    General                 (path_extenders.cpp        :  36)   Processed 240 paths from 391 (61%)\n",
      "  0:08:54.743   111M / 379M  INFO    General                 (path_extenders.cpp        :  34)   Processed 256 paths from 391 (65%)\n",
      "  0:08:54.743   111M / 379M  INFO    General                 (path_extenders.cpp        :  36)   Processed 280 paths from 391 (71%)\n",
      "  0:08:54.744   111M / 379M  INFO    General                 (path_extenders.cpp        :  36)   Processed 320 paths from 391 (81%)\n",
      "  0:08:54.744   111M / 379M  INFO    General                 (path_extenders.cpp        :  36)   Processed 360 paths from 391 (92%)\n",
      "  0:08:54.744   111M / 379M  INFO    General                 (launcher.cpp              : 250)   Finalizing paths\n",
      "  0:08:54.744   111M / 379M  INFO    General                 (launcher.cpp              : 252)   Deduplicating paths\n",
      "  0:08:54.744   111M / 379M  INFO    General                 (launcher.cpp              : 256)   Paths deduplicated\n",
      "  0:08:54.744   111M / 379M  INFO   PEResolver               (pe_resolver.cpp           :  60)   Removing overlaps\n",
      "  0:08:54.744   111M / 379M  INFO   PEResolver               (pe_resolver.cpp           :  63)   Sorting paths\n",
      "  0:08:54.744   111M / 379M  INFO   PEResolver               (pe_resolver.cpp           :  70)   Marking overlaps\n",
      "  0:08:54.744   111M / 379M  INFO   OverlapRemover           (overlap_remover.hpp       : 116)   Marking start/end overlaps\n",
      "  0:08:54.745   111M / 379M  INFO   OverlapRemover           (overlap_remover.hpp       : 119)   Marking remaining overlaps\n",
      "  0:08:54.746   111M / 379M  INFO   PEResolver               (pe_resolver.cpp           :  73)   Splitting paths\n",
      "  0:08:54.747   116M / 379M  INFO   PEResolver               (pe_resolver.cpp           :  78)   Deduplicating paths\n",
      "  0:08:54.747   110M / 379M  INFO   PEResolver               (pe_resolver.cpp           :  80)   Overlaps removed\n",
      "  0:08:54.747   111M / 379M  INFO    General                 (launcher.cpp              : 273)   Paths finalized\n",
      "  0:08:54.747   111M / 379M  INFO    General                 (launcher.cpp              : 454)   Closing gaps in paths\n",
      "  0:08:54.748   113M / 379M  INFO    General                 (launcher.cpp              : 484)   Gap closing completed\n",
      "  0:08:54.748   113M / 379M  INFO    General                 (launcher.cpp              : 302)   Traversing tandem repeats\n",
      "  0:08:54.749   114M / 379M  INFO    General                 (launcher.cpp              : 312)   Traversed 5 loops\n",
      "  0:08:54.749   114M / 379M  INFO    General                 (launcher.cpp              : 250)   Finalizing paths\n",
      "  0:08:54.749   114M / 379M  INFO    General                 (launcher.cpp              : 252)   Deduplicating paths\n",
      "  0:08:54.749   113M / 379M  INFO    General                 (launcher.cpp              : 256)   Paths deduplicated\n",
      "  0:08:54.749   113M / 379M  INFO   PEResolver               (pe_resolver.cpp           :  60)   Removing overlaps\n",
      "  0:08:54.749   113M / 379M  INFO   PEResolver               (pe_resolver.cpp           :  63)   Sorting paths\n",
      "  0:08:54.749   113M / 379M  INFO   PEResolver               (pe_resolver.cpp           :  70)   Marking overlaps\n",
      "  0:08:54.749   113M / 379M  INFO   OverlapRemover           (overlap_remover.hpp       : 116)   Marking start/end overlaps\n",
      "  0:08:54.749   113M / 379M  INFO   OverlapRemover           (overlap_remover.hpp       : 119)   Marking remaining overlaps\n",
      "  0:08:54.749   113M / 379M  INFO   PEResolver               (pe_resolver.cpp           :  73)   Splitting paths\n",
      "  0:08:54.750   114M / 379M  INFO   PEResolver               (pe_resolver.cpp           :  78)   Deduplicating paths\n",
      "  0:08:54.750   113M / 379M  INFO   PEResolver               (pe_resolver.cpp           :  80)   Overlaps removed\n",
      "  0:08:54.750   114M / 379M  INFO    General                 (launcher.cpp              : 273)   Paths finalized\n",
      "  0:08:54.750   114M / 379M  INFO    General                 (launcher.cpp              : 664)   ExSPAnder repeat resolving tool finished\n",
      "  0:08:54.751    91M / 379M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:08:54.751    91M / 379M  INFO    General                 (contig_output.hpp         :  16)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/before_rr.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:08:54.772    91M / 379M  INFO    General                 (contig_output_stage.cpp   : 151)   Writing GFA graph to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/assembly_graph_with_scaffolds.gfa\n",
      "  0:08:54.795    91M / 379M  INFO    General                 (contig_output_stage.cpp   : 165)   Outputting FastG graph to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/assembly_graph.fastg\n",
      "  0:08:54.861    91M / 379M  INFO    General                 (contig_output_stage.cpp   : 196)   Breaking scaffolds\n",
      "  0:08:54.872   100M / 379M  INFO    General                 (contig_output_stage.cpp   :  98)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/final_contigs.fasta\n",
      "  0:08:54.894   100M / 379M  INFO    General                 (contig_output_stage.cpp   : 104)   Outputting FastG paths to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/final_contigs.paths\n",
      "  0:08:54.905    95M / 379M  INFO    General                 (contig_output_stage.cpp   :  98)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/scaffolds.fasta\n",
      "  0:08:54.924    95M / 379M  INFO    General                 (contig_output_stage.cpp   : 104)   Outputting FastG paths to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/scaffolds.paths\n",
      "  0:08:54.924    95M / 379M  INFO    General                 (contig_output_stage.cpp   : 111)   Populating GFA with scaffold paths\n",
      "  0:08:54.925    91M / 379M  INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  0:08:54.928     1M / 379M  INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 8 minutes 54 seconds\n",
      "\n",
      "===== K55 finished. \n",
      "\n",
      "\n",
      "===== Copy files started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/miniconda3/bin/python /Users/olga/Downloads/SPAdes-3.15.2/share/spades/spades_pipeline/scripts/copy_files.py /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/before_rr.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/before_rr.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/assembly_graph_after_simplification.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/assembly_graph_after_simplification.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/final_contigs.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/contigs.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/first_pe_contigs.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/first_pe_contigs.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/strain_graph.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/strain_graph.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/scaffolds.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/scaffolds.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/scaffolds.paths /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/scaffolds.paths /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/assembly_graph_with_scaffolds.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/assembly_graph_with_scaffolds.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/assembly_graph.fastg /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/assembly_graph.fastg /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/K55/final_contigs.paths /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/contigs.paths\n",
      "\n",
      "\n",
      "===== Copy files finished. \n",
      "\n",
      "\n",
      "===== Assembling finished. \n",
      "\n",
      "\n",
      "===== Breaking scaffolds started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/miniconda3/bin/python /Users/olga/Downloads/SPAdes-3.15.2/share/spades/spades_pipeline/scripts/breaking_scaffolds_script.py --result_scaffolds_filename /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/scaffolds.fasta --misc_dir /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/misc --threshold_for_breaking_scaffolds 3\n",
      "\n",
      "\n",
      "===== Breaking scaffolds finished. \n",
      "\n",
      "\n",
      "===== Terminate started. \n",
      "\n",
      "\n",
      "===== Terminate finished. \n",
      "\n",
      " * Corrected reads are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/corrected/\n",
      " * Assembled contigs are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/contigs.fasta\n",
      " * Assembled scaffolds are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/scaffolds.fasta\n",
      " * Paths in the assembly graph corresponding to the contigs are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/scaffolds.paths\n",
      " * Assembly graph is in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/assembly_graph_with_scaffolds.gfa\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio20/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n"
     ]
    }
   ],
   "source": [
    "! /Users/olga/Downloads/SPAdes-3.15.2/bin/spades.py -1 ./homework6/illumina.100x.1.fq -2 ./homework6/illumina.100x.2.fq --pacbio ./homework6/pacbio_20x.fq -o illumina_pacbio20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "== Warning ==  No assembly mode was specified! If you intend to assemble high-coverage multi-cell/isolate data, use '--isolate' option.\n",
      "\n",
      "\n",
      "Command line: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades.py\t-1\t/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq\t-2\t/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq\t--pacbio\t/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/pacbio_40x.fq\t-o\t/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.15.2\n",
      "  Python version: 3.7.7\n",
      "  OS: Darwin-20.3.0-x86_64-i386-64bit\n",
      "\n",
      "Output dir: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Standard mode\n",
      "  For multi-cell/isolate data we recommend to use '--isolate' option; for single-cell MDA data use '--sc'; for metagenomic data use '--meta'; for RNA-Seq use '--rna'.\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq']\n",
      "      right reads: ['/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "    Library number: 2, library type: pacbio\n",
      "      left reads: not specified\n",
      "      right reads: not specified\n",
      "      interlaced reads: not specified\n",
      "      single reads: ['/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/pacbio_40x.fq']\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: automatic selection based on read length\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 250\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/spades.log\n",
      "\n",
      "/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq: max reads length: 100\n",
      "/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq: max reads length: 100\n",
      "\n",
      "Reads length: 100\n",
      "\n",
      "\n",
      "===== Before start started. \n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-hammer /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  76)   Loading config from /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/corrected/configs/config.info\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (hammer_tools.cpp          :  38)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 16 files using 1 threads. This might take a while.\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 4194304\n",
      "  0:00:00.001   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq\n",
      "  0:00:09.816   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 467314 reads\n",
      "  0:00:19.134   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 935009 reads\n",
      "  0:00:28.499   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 1402861 reads\n",
      "  0:00:37.828   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 1870646 reads\n",
      "  0:00:47.149   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2339377 reads\n",
      "  0:00:50.476   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2500000 reads\n",
      "  0:00:50.476   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq\n",
      "  0:00:59.941   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2974795 reads\n",
      "  0:01:28.275   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 4400968 reads\n",
      "  0:01:40.228   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 5000000 reads processed\n",
      "  0:01:40.228     1M / 162M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:01:44.017     1M / 170M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 38919786 kmers in total.\n",
      "  0:01:44.017     1M / 170M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:47.668    30M / 195M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 38919786 kmers, 28121944 bytes occupied (5.78049 bits per kmer).\n",
      "  0:01:47.668    30M / 195M  INFO   K-mer Counting           (kmer_data.cpp             : 354)   Arranging kmers in hash map order\n",
      "  0:01:50.030   631M / 789M  INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:05:59.593   631M / 789M  INFO    General                 (main.cpp                  : 155)   Extracting clusters:\n",
      "  0:05:59.594   631M / 789M  INFO    General                 (concurrent_dsu.cpp        :  18)   Connecting to root\n",
      "  0:06:00.374   631M / 789M  INFO    General                 (concurrent_dsu.cpp        :  34)   Calculating counts\n",
      "  0:06:12.437   903M / 1061M INFO    General                 (concurrent_dsu.cpp        :  63)   Writing down entries\n",
      "  0:06:24.098   631M / 789M  INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 13651605\n",
      "  0:06:24.115   330M / 492M  INFO   K-mer Counting           (kmer_data.cpp             : 371)   Collecting K-mer information, this takes a while.\n",
      "  0:06:24.479  1223M / 1383M INFO   K-mer Counting           (kmer_data.cpp             : 377)   Processing /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq\n",
      "  0:11:07.243  1223M / 1383M INFO   K-mer Counting           (kmer_data.cpp             : 377)   Processing /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq\n",
      "  0:15:43.713  1223M / 1383M INFO   K-mer Counting           (kmer_data.cpp             : 384)   Collection done, postprocessing.\n",
      "  0:15:43.947  1223M / 1383M INFO   K-mer Counting           (kmer_data.cpp             : 398)   There are 38919786 kmers in total. Among them 27377044 (70.3422%) are singletons.\n",
      "  0:15:43.947  1223M / 1383M INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:16:26.595  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering done. Total 26 non-read kmers were generated.\n",
      "  0:16:26.595  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)   Subclustering statistics:\n",
      "  0:16:26.595  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton hamming clusters: 5541662. Among them 963110 (17.3794%) are good\n",
      "  0:16:26.595  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total singleton subclusters: 13747. Among them 13599 (98.9234%) are good\n",
      "  0:16:26.595  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Total non-singleton subcluster centers: 8197112. Among them 6855795 (83.6367%) are good\n",
      "  0:16:26.595  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average size of non-trivial subcluster: 4.07194 kmers\n",
      "  0:16:26.595  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Average number of sub-clusters per non-singleton cluster: 1.01244\n",
      "  0:16:26.595  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Total solid k-mers: 7832504\n",
      "  0:16:26.595  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 658)     Substitution probabilities: [4,4]((0.928457,0.0421442,0.022541,0.00685828),(0.0141904,0.967828,0.0090854,0.00889584),(0.00886597,0.00903879,0.967867,0.0142285),(0.00683121,0.022511,0.0420995,0.928558))\n",
      "  0:16:26.603  1223M / 1384M INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:16:26.603  1223M / 1384M INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 1 threads.\n",
      "  0:19:59.215  1223M / 1384M INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 39977 new k-mers.\n",
      "  0:22:27.841  1223M / 1384M INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 790 new k-mers.\n",
      "  0:24:55.929  1223M / 1384M INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 38 new k-mers.\n",
      "  0:27:24.785  1223M / 1384M INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 3 produced 0 new k-mers.\n",
      "  0:27:24.785  1223M / 1384M INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:27:24.785  1223M / 1384M INFO    General                 (hammer_tools.cpp          : 222)   Starting read correction in 1 threads.\n",
      "  0:27:24.785  1223M / 1384M INFO    General                 (hammer_tools.cpp          : 235)   Correcting pair of reads: /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq and /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq\n",
      "  0:27:24.955  1289M / 1394M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 0 of 100000 reads.\n",
      "  0:27:31.282  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 0\n",
      "  0:27:31.496  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 0\n",
      "  0:27:31.639  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 1 of 100000 reads.\n",
      "  0:27:37.965  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 1\n",
      "  0:27:38.187  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 1\n",
      "  0:27:38.331  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 2 of 100000 reads.\n",
      "  0:27:44.540  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 2\n",
      "  0:27:44.766  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 2\n",
      "  0:27:44.907  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 3 of 100000 reads.\n",
      "  0:27:51.190  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 3\n",
      "  0:27:51.410  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 3\n",
      "  0:27:51.555  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 4 of 100000 reads.\n",
      "  0:27:57.771  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 4\n",
      "  0:27:58.055  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 4\n",
      "  0:27:58.198  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 5 of 100000 reads.\n",
      "  0:28:04.403  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 5\n",
      "  0:28:04.632  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 5\n",
      "  0:28:04.777  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 6 of 100000 reads.\n",
      "  0:28:11.015  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 6\n",
      "  0:28:11.246  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 6\n",
      "  0:28:11.389  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 7 of 100000 reads.\n",
      "  0:28:17.598  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 7\n",
      "  0:28:17.829  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 7\n",
      "  0:28:17.974  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 8 of 100000 reads.\n",
      "  0:28:24.151  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 8\n",
      "  0:28:24.380  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 8\n",
      "  0:28:24.522  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 9 of 100000 reads.\n",
      "  0:28:30.748  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 9\n",
      "  0:28:30.992  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 9\n",
      "  0:28:31.135  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 10 of 100000 reads.\n",
      "  0:28:37.317  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 10\n",
      "  0:28:37.549  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 10\n",
      "  0:28:37.691  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 11 of 100000 reads.\n",
      "  0:28:43.866  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 11\n",
      "  0:28:44.099  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 11\n",
      "  0:28:44.241  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 12 of 100000 reads.\n",
      "  0:28:50.480  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 12\n",
      "  0:28:50.718  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 12\n",
      "  0:28:50.865  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 13 of 100000 reads.\n",
      "  0:28:57.073  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 13\n",
      "  0:28:57.312  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 13\n",
      "  0:28:57.455  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 14 of 100000 reads.\n",
      "  0:29:03.675  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:29:03.922  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 14\n",
      "  0:29:04.065  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 15 of 100000 reads.\n",
      "  0:29:10.268  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 15\n",
      "  0:29:10.507  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 15\n",
      "  0:29:10.649  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 16 of 100000 reads.\n",
      "  0:29:16.836  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 16\n",
      "  0:29:17.088  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 16\n",
      "  0:29:17.231  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 17 of 100000 reads.\n",
      "  0:29:23.405  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 17\n",
      "  0:29:23.646  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 17\n",
      "  0:29:23.788  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 18 of 100000 reads.\n",
      "  0:29:30.015  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 18\n",
      "  0:29:30.247  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 18\n",
      "  0:29:30.390  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 19 of 100000 reads.\n",
      "  0:29:36.607  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 19\n",
      "  0:29:36.850  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 19\n",
      "  0:29:37.002  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 20 of 100000 reads.\n",
      "  0:29:43.330  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 20\n",
      "  0:29:43.558  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 20\n",
      "  0:29:43.699  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 21 of 100000 reads.\n",
      "  0:29:50.018  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 21\n",
      "  0:29:50.261  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 21\n",
      "  0:29:50.412  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 22 of 100000 reads.\n",
      "  0:29:56.901  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 22\n",
      "  0:29:57.152  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 22\n",
      "  0:29:57.302  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 23 of 100000 reads.\n",
      "  0:30:03.767  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 23\n",
      "  0:30:04.011  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 23\n",
      "  0:30:04.158  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 24 of 100000 reads.\n",
      "  0:30:10.659  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 24\n",
      "  0:30:10.895  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 24\n",
      "  0:30:10.904  1223M / 1396M INFO    General                 (hammer_tools.cpp          : 276)   Correction done. Changed 1639742 bases in 1194871 reads.\n",
      "  0:30:10.904  1223M / 1396M INFO    General                 (hammer_tools.cpp          : 277)   Failed to correct 44220 bases out of 458864877.\n",
      "  0:30:10.978     1M / 208M  INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/corrected/corrected.yaml\n",
      "  0:30:10.979     1M / 208M  INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== corrected reads compression started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/miniconda3/bin/python /Users/olga/Downloads/SPAdes-3.15.2/share/spades/spades_pipeline/scripts/compress_all.py --input_file /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/corrected/corrected.yaml --ext_python_modules_home /Users/olga/Downloads/SPAdes-3.15.2/share/spades --max_threads 16 --output_dir /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/corrected --not_used_yaml_file /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/corrected/dont_correct.yaml --gzip_output\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "== Files to compress: ['/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/corrected/illumina.100x.1.00.0_0.cor.fastq', '/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/corrected/illumina.100x.2.00.0_0.cor.fastq', '/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/corrected/illumina.100x._unpaired.00.0_0.cor.fastq']\n",
      "== Files compression is finished\n",
      "== Info about datasets not used in error correction stage is loaded\n",
      "== Dataset yaml file is updated\n",
      "\n",
      "===== corrected reads compression finished. \n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started. \n",
      "\n",
      "\n",
      "===== K21 started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-core /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K21/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  :  99)   Loaded config from /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K21/configs/config.info\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 109)   Assembling dataset (/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/dataset.info) with K=21\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  72)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  73)   Converting paired reads\n",
      "  0:00:00.065    22M / 22M   INFO    General                 (binary_converter.cpp      :  96)   16384 reads processed\n",
      "  0:00:00.134    25M / 25M   INFO    General                 (binary_converter.cpp      :  96)   32768 reads processed\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:00.344    22M / 25M   INFO    General                 (binary_converter.cpp      :  96)   65536 reads processed\n",
      "  0:00:00.679    25M / 31M   INFO    General                 (binary_converter.cpp      :  96)   131072 reads processed\n",
      "  0:00:01.413    21M / 31M   INFO    General                 (binary_converter.cpp      :  96)   262144 reads processed\n",
      "  0:00:02.808    24M / 31M   INFO    General                 (binary_converter.cpp      :  96)   524288 reads processed\n",
      "  0:00:05.550    28M / 31M   INFO    General                 (binary_converter.cpp      :  96)   1048576 reads processed\n",
      "  0:00:11.126    28M / 31M   INFO    General                 (binary_converter.cpp      :  96)   2097152 reads processed\n",
      "  0:00:13.075    19M / 31M   INFO    General                 (binary_converter.cpp      : 111)   2448672 reads written\n",
      "  0:00:13.075     1M / 31M   INFO    General                 (read_converter.cpp        :  81)   Converting single reads\n",
      "  0:00:13.105    10M / 31M   INFO    General                 (binary_converter.cpp      :  96)   16384 reads processed\n",
      "  0:00:13.138    12M / 31M   INFO    General                 (binary_converter.cpp      :  96)   32768 reads processed\n",
      "  0:00:13.205     9M / 31M   INFO    General                 (binary_converter.cpp      : 111)   48607 reads written\n",
      "  0:00:13.206     1M / 31M   INFO    General                 (read_converter.cpp        :  87)   Converting merged reads\n",
      "  0:00:13.206     9M / 31M   INFO    General                 (binary_converter.cpp      : 111)   0 reads written\n",
      "  0:00:13.207     1M / 31M   INFO    General                 (read_converter.cpp        :  72)   Converting reads to binary format for library #1 (takes a while)\n",
      "  0:00:13.207     1M / 31M   INFO    General                 (read_converter.cpp        :  73)   Converting paired reads\n",
      "  0:00:13.207    19M / 31M   INFO    General                 (binary_converter.cpp      : 111)   0 reads written\n",
      "  0:00:13.207     1M / 31M   INFO    General                 (read_converter.cpp        :  81)   Converting single reads\n",
      "  0:00:13.732    98M / 108M  INFO    General                 (binary_converter.cpp      :  96)   16384 reads processed\n",
      "  0:00:14.259   189M / 199M  INFO    General                 (binary_converter.cpp      :  96)   32768 reads processed\n",
      "  0:00:16.039     9M / 247M  INFO    General                 (binary_converter.cpp      : 111)   41447 reads written\n",
      "  0:00:16.040     1M / 247M  INFO    General                 (read_converter.cpp        :  87)   Converting merged reads\n",
      "  0:00:16.040     9M / 247M  INFO    General                 (binary_converter.cpp      : 111)   0 reads written\n",
      "  0:00:16.041     1M / 245M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:16.042     1M / 245M  INFO    General                 (construction.cpp          : 153)   Max read length 100\n",
      "  0:00:16.042     1M / 245M  INFO    General                 (construction.cpp          : 159)   Average read length 92.7147\n",
      "  0:00:16.042     1M / 245M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:16.042     1M / 245M  INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:16.042     1M / 245M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:16.042     1M / 245M  INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:16.042     1M / 245M  INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 6710886\n",
      "  0:00:26.187   601M / 663M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 1873811 reads\n",
      "  0:00:36.099   601M / 663M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 3734243 reads\n",
      "  0:00:46.000   601M / 663M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 5595232 reads\n",
      "  0:00:55.891   601M / 663M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 7456112 reads\n",
      "  0:01:05.780   601M / 663M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 9318931 reads\n",
      "  0:01:09.032   601M / 663M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 9891902 reads\n",
      "  0:01:09.032     1M / 255M  INFO    General                 (kmer_splitters.hpp        : 131)   Used 9891902 reads\n",
      "  0:01:09.065     1M / 255M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:01:09.561     1M / 255M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5715581 kmers in total.\n",
      "  0:01:09.561     1M / 255M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:01:09.563     1M / 255M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:01:09.563     1M / 255M  INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:01:09.563     1M / 255M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:01:09.563     1M / 255M  INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:01:09.563     1M / 255M  INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 6710886\n",
      "  0:01:11.435   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 194)   Processed 5715581 kmers\n",
      "  0:01:11.435   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 199)   Used 5715581 kmers.\n",
      "  0:01:11.435     1M / 299M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:01:11.617     1M / 299M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5708731 kmers in total.\n",
      "  0:01:11.617     1M / 299M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:12.086     5M / 299M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5708731 kmers, 4130592 bytes occupied (5.78846 bits per kmer).\n",
      "  0:01:12.086     5M / 299M  INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n",
      "  0:01:12.231    11M / 299M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:01:14.053    11M / 299M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:01:14.054    11M / 255M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Early tip clipping (id: construction:early_tip_clipper)\n",
      "  0:01:14.054    11M / 255M  INFO    General                 (construction.cpp          : 301)   Early tip clipper length bound set as (RL - K)\n",
      "  0:01:14.054    11M / 255M  INFO   Early tip clipping       (early_simplification.hpp  :  47)   Early tip clipping\n",
      "  0:01:15.801    21M / 299M  INFO   Early tip clipping       (early_simplification.hpp  :  82)   #tipped junctions: 123493\n",
      "  0:01:15.846    21M / 299M  INFO   Early tip clipping       (early_simplification.hpp  :  93)   Clipped tips: 124241\n",
      "  0:01:15.847    11M / 255M  INFO   Early tip clipping       (early_simplification.hpp  :  49)   978561 22-mers were removed by early tip clipper\n",
      "  0:01:15.847    11M / 255M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:01:15.851    11M / 255M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:01:18.132    13M / 299M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 25730 sequences extracted\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:01:18.953    13M / 256M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:01:19.368    13M / 299M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 4 loops collected\n",
      "  0:01:19.370    13M / 256M  INFO    General                 (debruijn_graph_constructor: 487)   Total 51468 edges to create\n",
      "  0:01:19.370    15M / 256M  INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:01:19.380    16M / 256M  INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:01:19.383    16M / 256M  INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:01:19.383    16M / 256M  INFO    General                 (debruijn_graph_constructor: 503)   Total 18884 vertices to create\n",
      "  0:01:19.383    18M / 256M  INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:01:19.385    16M / 256M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:01:19.385    16M / 256M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:19.848    20M / 256M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5715581 kmers, 4135544 bytes occupied (5.78845 bits per kmer).\n",
      "  0:01:19.849    42M / 256M  INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:02:17.556    42M / 256M  INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:02:19.095    42M / 256M  INFO    General                 (coverage_filling.hpp      :  82)   Processed 51447 edges\n",
      "  0:02:19.108     6M / 256M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:02:19.108     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 16\n",
      "  0:02:19.108     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 76\n",
      "  0:02:19.108     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 77. Coverage mad: 14.826\n",
      "  0:02:19.108     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:02:19.173     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:02:19.338     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:02:19.895     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:02:21.191     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:02:21.330     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 76.252. Fitted coverage std. dev: 14.5028\n",
      "  0:02:21.332     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.863705\n",
      "  0:02:21.332     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 37\n",
      "  0:02:21.332     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 37\n",
      "  0:02:21.332     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 4507902\n",
      "  0:02:21.332     6M / 256M  INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 76.252\n",
      "  0:02:21.332     6M / 256M  INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 37\n",
      "  0:02:21.332     6M / 256M  INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:02:21.332     6M / 256M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:02:21.332     6M / 256M  INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:02:21.332     6M / 256M  INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:02:21.332     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:02:21.334     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:02:21.334     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n",
      "  0:02:21.353     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 2371 times\n",
      "  0:02:21.353     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n",
      "  0:02:21.423     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 4029 times\n",
      "  0:02:21.424     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:02:21.425     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 308 times\n",
      "  0:02:21.425     6M / 256M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:02:21.425     6M / 256M  INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:02:21.425     6M / 256M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:02:21.425     6M / 256M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:02:21.425     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:21.427     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 128 times\n",
      "  0:02:21.427     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:21.498     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 1810 times\n",
      "  0:02:21.498     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:21.502     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 143 times\n",
      "  0:02:21.502     6M / 256M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:02:21.502     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:21.503     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 5 times\n",
      "  0:02:21.503     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:21.505     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 36 times\n",
      "  0:02:21.505     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:21.506     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 68 times\n",
      "  0:02:21.506     6M / 256M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:02:21.506     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:21.506     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 3 times\n",
      "  0:02:21.506     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:21.507     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 23 times\n",
      "  0:02:21.507     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 16 times\n",
      "  0:02:21.508     6M / 256M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 7 times\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 10 times\n",
      "  0:02:21.508     6M / 256M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 2 times\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 4 times\n",
      "  0:02:21.508     6M / 256M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 4 times\n",
      "  0:02:21.508     6M / 256M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 1 times\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 2 times\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 3 times\n",
      "  0:02:21.508     6M / 256M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 7 times\n",
      "  0:02:21.508     6M / 256M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:21.508     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:21.509     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:21.509     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:21.509     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:02:21.509     6M / 256M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:02:21.509     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:21.509     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:21.509     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:21.509     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:21.509     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:21.509     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 6 times\n",
      "  0:02:21.509     6M / 256M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:02:21.509     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:21.509     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:21.509     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:21.511     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 3 times\n",
      "  0:02:21.511     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:21.512     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:21.512     6M / 256M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:02:21.512     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:21.512     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:21.512     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:21.512     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:21.512     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:21.512     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:21.512     6M / 256M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
      "  0:02:21.512     6M / 256M  INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\n",
      "  0:02:21.512     6M / 257M  INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\n",
      "  0:02:21.512     6M / 257M  INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\n",
      "  0:02:21.512     6M / 257M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:02:21.512     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:21.512     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:21.512     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:21.515     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:21.515     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:21.515     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:21.515     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:21.518     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:21.518     6M / 257M  INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:02:21.518     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:02:21.519     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 7 times\n",
      "  0:02:21.519     6M / 257M  INFO    General                 (simplification.cpp        : 491)   Counting average coverage\n",
      "  0:02:21.519     6M / 257M  INFO    General                 (simplification.cpp        : 497)   Average coverage = 78.0189\n",
      "  0:02:21.519     6M / 257M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:02:21.519     6M / 257M  INFO    General                 (read_converter.cpp        : 102)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K21/simplified_contigs\n",
      "  0:02:21.524     8M / 257M  INFO    General                 (binary_converter.cpp      : 111)   2571 reads written\n",
      "  0:02:21.524     6M / 257M  INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  0:02:21.525     1M / 257M  INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 2 minutes 21 seconds\n",
      "\n",
      "===== K21 finished. \n",
      "\n",
      "\n",
      "===== K33 started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-core /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K33/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  :  99)   Loaded config from /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K33/configs/config.info\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 109)   Assembling dataset (/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/dataset.info) with K=33\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 118)   Contigs from previous K will be used: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K21/simplified_contigs\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 153)   Max read length 100\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 159)   Average read length 92.7147\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 3355443\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:05.974   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 1132634 reads\n",
      "  0:00:11.562   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 2249495 reads\n",
      "  0:00:17.152   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 3368100 reads\n",
      "  0:00:22.741   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 4487032 reads\n",
      "  0:00:28.337   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 5607299 reads\n",
      "  0:00:33.932   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 6726813 reads\n",
      "  0:00:39.542   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 7846654 reads\n",
      "  0:00:45.144   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 8966769 reads\n",
      "  0:00:50.516   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 9897044 reads\n",
      "  0:00:50.516     1M / 158M  INFO    General                 (kmer_splitters.hpp        : 131)   Used 9897044 reads\n",
      "  0:00:50.547     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:52.128     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5756125 kmers in total.\n",
      "  0:00:52.128     1M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:00:52.130     1M / 158M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:00:52.130     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:52.130     1M / 158M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:52.130     1M / 158M  INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:52.130     1M / 158M  INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 3355443\n",
      "  0:00:54.380   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 194)   Processed 5756125 kmers\n",
      "  0:00:54.380   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 199)   Used 5756125 kmers.\n",
      "  0:00:54.380     1M / 246M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:54.711     1M / 246M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5754745 kmers in total.\n",
      "  0:00:54.711     1M / 246M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:55.284     5M / 246M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5754745 kmers, 4163816 bytes occupied (5.78836 bits per kmer).\n",
      "  0:00:55.284     5M / 246M  INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n",
      "  0:00:55.572    11M / 246M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:00:57.518    11M / 246M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:57.520    11M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Early tip clipping (id: construction:early_tip_clipper)\n",
      "  0:00:57.520    11M / 158M  INFO    General                 (construction.cpp          : 301)   Early tip clipper length bound set as (RL - K)\n",
      "  0:00:57.520    11M / 158M  INFO   Early tip clipping       (early_simplification.hpp  :  47)   Early tip clipping\n",
      "  0:00:59.572    21M / 246M  INFO   Early tip clipping       (early_simplification.hpp  :  82)   #tipped junctions: 124481\n",
      "  0:00:59.625    21M / 246M  INFO   Early tip clipping       (early_simplification.hpp  :  93)   Clipped tips: 125222\n",
      "  0:00:59.627    11M / 158M  INFO   Early tip clipping       (early_simplification.hpp  :  49)   1029171 34-mers were removed by early tip clipper\n",
      "  0:00:59.627    11M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:00:59.637    11M / 158M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:01:02.182    12M / 248M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 15754 sequences extracted\n",
      "  0:01:03.065    12M / 160M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:01:03.490    12M / 248M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 2 loops collected\n",
      "  0:01:03.492    12M / 160M  INFO    General                 (debruijn_graph_constructor: 487)   Total 31512 edges to create\n",
      "  0:01:03.493    14M / 160M  INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:01:03.500    14M / 160M  INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:01:03.502    14M / 160M  INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:01:03.502    14M / 160M  INFO    General                 (debruijn_graph_constructor: 503)   Total 14376 vertices to create\n",
      "  0:01:03.502    16M / 160M  INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:01:03.503    15M / 160M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:01:03.503    15M / 160M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:04.094    19M / 160M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5756125 kmers, 4164864 bytes occupied (5.78843 bits per kmer).\n",
      "  0:01:04.096    41M / 163M  INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:01:58.829    41M / 163M  INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:02:00.457    41M / 163M  INFO    General                 (coverage_filling.hpp      :  82)   Processed 31510 edges\n",
      "  0:02:00.477     5M / 163M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:02:00.478     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 13\n",
      "  0:02:00.478     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 64\n",
      "  0:02:00.478     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 64. Coverage mad: 14.826\n",
      "  0:02:00.478     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:02:00.530     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:02:00.678     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:02:01.156     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:02:02.271     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:02:02.999     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 63.4837. Fitted coverage std. dev: 13.4198\n",
      "  0:02:03.001     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.693899\n",
      "  0:02:03.001     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 29\n",
      "  0:02:03.001     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 29\n",
      "  0:02:03.001     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 4507034\n",
      "  0:02:03.001     5M / 163M  INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 63.4837\n",
      "  0:02:03.001     5M / 163M  INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 29\n",
      "  0:02:03.001     5M / 163M  INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:02:03.001     5M / 163M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:02:03.001     5M / 163M  INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:02:03.001     5M / 163M  INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:02:03.001     5M / 163M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:02:03.002     5M / 163M  INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:02:03.002     5M / 163M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n",
      "  0:02:03.036     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 2699 times\n",
      "  0:02:03.036     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:02:03.076     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 1350 times\n",
      "  0:02:03.076     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:02:03.078     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 1603 times\n",
      "  0:02:03.079     5M / 164M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:02:03.079     5M / 164M  INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:02:03.079     5M / 164M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:02:03.079     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:02:03.079     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:03.080     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 59 times\n",
      "  0:02:03.080     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:03.146     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 1018 times\n",
      "  0:02:03.146     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:03.148     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 42 times\n",
      "  0:02:03.148     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:02:03.148     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:03.149     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 5 times\n",
      "  0:02:03.149     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:03.149     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 17 times\n",
      "  0:02:03.149     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:03.151     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 52 times\n",
      "  0:02:03.151     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:02:03.151     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:03.151     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 1 times\n",
      "  0:02:03.151     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:03.153     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 27 times\n",
      "  0:02:03.153     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:03.154     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 10 times\n",
      "  0:02:03.154     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:02:03.154     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:03.154     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:03.154     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:03.154     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 9 times\n",
      "  0:02:03.154     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:03.154     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 8 times\n",
      "  0:02:03.154     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:02:03.155     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:03.155     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:03.155     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:03.155     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 5 times\n",
      "  0:02:03.155     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:03.155     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 6 times\n",
      "  0:02:03.155     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:02:03.155     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:03.155     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:03.155     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:03.155     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:03.155     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:03.155     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:03.155     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:02:03.155     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:03.155     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:03.155     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:03.155     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:03.155     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:03.155     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 3 times\n",
      "  0:02:03.156     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:02:03.156     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:02:03.156     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:02:03.156     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:03.156     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:03.157     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 2 times\n",
      "  0:02:03.157     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:03.157     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:03.157     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:02:03.157     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:03.157     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:03.157     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:03.157     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:03.157     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:03.157     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:03.157     5M / 164M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
      "  0:02:03.157     5M / 165M  INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\n",
      "  0:02:03.157     5M / 165M  INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\n",
      "  0:02:03.157     5M / 165M  INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\n",
      "  0:02:03.157     5M / 165M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:02:03.157     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:03.158     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:03.158     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:03.159     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:03.159     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:03.159     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:03.159     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:03.160     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:03.160     5M / 165M  INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:02:03.160     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:02:03.160     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 5 times\n",
      "  0:02:03.160     5M / 165M  INFO    General                 (simplification.cpp        : 491)   Counting average coverage\n",
      "  0:02:03.160     5M / 165M  INFO    General                 (simplification.cpp        : 497)   Average coverage = 64.7522\n",
      "  0:02:03.160     5M / 165M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:02:03.160     5M / 165M  INFO    General                 (read_converter.cpp        : 102)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K33/simplified_contigs\n",
      "  0:02:03.164     7M / 165M  INFO    General                 (binary_converter.cpp      : 111)   1188 reads written\n",
      "  0:02:03.164     5M / 165M  INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  0:02:03.165     1M / 165M  INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 2 minutes 3 seconds\n",
      "\n",
      "===== K33 finished. \n",
      "\n",
      "\n",
      "===== K55 started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-core /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  :  99)   Loaded config from /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/configs/config.info\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 109)   Assembling dataset (/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/dataset.info) with K=55\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 242)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 118)   Contigs from previous K will be used: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K33/simplified_contigs\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 153)   Max read length 100\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (construction.cpp          : 159)   Average read length 92.7147\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 3355443\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:06.264   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 1768213 reads\n",
      "  0:00:12.024   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 3518079 reads\n",
      "  0:00:17.589   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 5268519 reads\n",
      "  0:00:23.524   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 7021543 reads\n",
      "  0:00:29.390   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 8774696 reads\n",
      "  0:00:33.923   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 9894278 reads\n",
      "  0:00:33.923     1M / 158M  INFO    General                 (kmer_splitters.hpp        : 131)   Used 9894278 reads\n",
      "  0:00:33.955     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:35.580     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5692221 kmers in total.\n",
      "  0:00:35.580     1M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:00:35.582     1M / 158M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:00:35.582     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:35.582     1M / 158M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:35.582     1M / 158M  INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:35.582     1M / 158M  INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 3355443\n",
      "  0:00:37.874   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 194)   Processed 5692221 kmers\n",
      "  0:00:37.874   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 199)   Used 5692221 kmers.\n",
      "  0:00:37.874     1M / 245M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:38.218     1M / 245M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5698724 kmers in total.\n",
      "  0:00:38.218     1M / 245M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:38.807     5M / 245M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5698724 kmers, 4123376 bytes occupied (5.78849 bits per kmer).\n",
      "  0:00:38.807     5M / 245M  INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n",
      "  0:00:39.090    11M / 245M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:00:40.994    11M / 245M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:40.995    11M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:00:41.005    11M / 158M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:00:43.905    23M / 248M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 247400 sequences extracted\n",
      "  0:00:44.996    23M / 161M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:00:45.417    23M / 248M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:45.422    23M / 161M  INFO    General                 (debruijn_graph_constructor: 487)   Total 494800 edges to create\n",
      "  0:00:45.422    42M / 161M  INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:00:45.555    50M / 172M  INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:00:45.588    50M / 172M  INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:00:45.590    52M / 172M  INFO    General                 (debruijn_graph_constructor: 503)   Total 253903 vertices to create\n",
      "  0:00:45.590    76M / 172M  INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:00:45.627    62M / 172M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:00:45.627    62M / 172M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:46.138    66M / 172M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5692221 kmers, 4118704 bytes occupied (5.78854 bits per kmer).\n",
      "  0:00:46.140    88M / 172M  INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:01:22.222    88M / 172M  INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:01:24.544    88M / 172M  INFO    General                 (coverage_filling.hpp      :  82)   Processed 494800 edges\n",
      "  0:01:24.565    52M / 172M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:01:24.565    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 6\n",
      "  0:01:24.565    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 41\n",
      "  0:01:24.565    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 41. Coverage mad: 11.8608\n",
      "  0:01:24.565    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:01:24.600    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:01:24.693    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:01:24.990    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:01:25.710    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:01:26.216    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 40.4747. Fitted coverage std. dev: 10.9026\n",
      "  0:01:26.217    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.512008\n",
      "  0:01:26.217    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 16\n",
      "  0:01:26.217    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 16\n",
      "  0:01:26.217    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 4487959\n",
      "  0:01:26.217    52M / 172M  INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 40.4747\n",
      "  0:01:26.217    52M / 172M  INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 16\n",
      "  0:01:26.217    52M / 172M  INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:01:26.217    52M / 172M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Gap Closer (id: early_gapcloser)\n",
      "  0:01:26.219    61M / 172M  INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:01:26.312    66M / 172M  INFO    General                 (gap_closer.cpp            : 101)   Total edges in tip neighborhood: 134528 out of 494800, length: 1226132\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:01:26.321    68M / 172M  INFO    General                 (edge_index.hpp            : 167)   Using small index (max_id = 497276)\n",
      "  0:01:26.427    68M / 172M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:27.055    69M / 172M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 1092761 kmers, 790000 bytes occupied (5.78352 bits per kmer).\n",
      "  0:01:27.057    78M / 172M  INFO    General                 (edge_index_builders.hpp   : 265)   Collecting edge information from graph, this takes a while.\n",
      "  0:01:31.942    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:01:36.485    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:01:40.870    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:01:45.324    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:01:49.756    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:01:54.227    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:02:16.922    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:02:22.406    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:02:22.412    62M / 172M  INFO    General                 (gap_closer.cpp            : 452)   Initializing gap closer\n",
      "  0:02:22.412    62M / 172M  INFO   GapCloser                (gap_closer.cpp            : 359)   Closing short gaps\n",
      "  0:02:22.599    62M / 178M  INFO   GapCloser                (gap_closer.cpp            : 393)   Closing short gaps complete: filled 0 gaps after checking 20 candidates\n",
      "  0:02:22.605    62M / 178M  INFO    General                 (gap_closer.cpp            : 456)   Gap closer done\n",
      "  0:02:22.605    52M / 178M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:02:22.605    52M / 178M  INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:02:22.606    52M / 178M  INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:02:22.606    52M / 178M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:02:22.634    52M / 178M  INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:02:22.634    52M / 178M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n",
      "  0:02:23.393    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 119766 times\n",
      "  0:02:23.394    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n",
      "  0:02:23.401    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 92 times\n",
      "  0:02:23.401    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:02:23.410    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 7236 times\n",
      "  0:02:23.411    45M / 179M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:02:23.411    45M / 179M  INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:02:23.411    45M / 179M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:02:23.411    45M / 179M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:02:23.411    45M / 179M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:23.412    45M / 179M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 19 times\n",
      "  0:02:23.412    45M / 179M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:23.505    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 447 times\n",
      "  0:02:23.505    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:23.506    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:23.506    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:02:23.506    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:23.506    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 3 times\n",
      "  0:02:23.506    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:23.506    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:23.507    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:23.507    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 13 times\n",
      "  0:02:23.507    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:02:23.507    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:23.507    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 1 times\n",
      "  0:02:23.507    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:23.508    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 8 times\n",
      "  0:02:23.508    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:23.509    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 11 times\n",
      "  0:02:23.509    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:02:23.509    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:23.509    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 3 times\n",
      "  0:02:23.509    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:23.510    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 7 times\n",
      "  0:02:23.510    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:23.510    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:02:23.510    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:02:23.510    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:23.510    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:23.510    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:23.510    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:23.510    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:23.510    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 3 times\n",
      "  0:02:23.510    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:02:23.510    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 1 times\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:02:23.511    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:02:23.511    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:23.511    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:23.511    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:23.512    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:02:23.512    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:02:23.512    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:23.512    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:23.512    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:23.512    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:23.512    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:23.512    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:02:23.512    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:02:23.512    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:23.513    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:23.513    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:23.514    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:23.514    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:23.515    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:23.515    48M / 180M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Gap Closer (id: late_gapcloser)\n",
      "  0:02:23.517    57M / 180M  INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:02:23.517    57M / 180M  INFO    General                 (gap_closer.cpp            : 101)   Total edges in tip neighborhood: 15 out of 882, length: 167704\n",
      "  0:02:23.517    57M / 180M  INFO    General                 (edge_index.hpp            : 167)   Using small index (max_id = 497276)\n",
      "  0:02:23.523    57M / 180M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:02:23.552    57M / 180M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 166932 kmers, 121312 bytes occupied (5.81372 bits per kmer).\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:02:23.553    58M / 180M  INFO    General                 (edge_index_builders.hpp   : 265)   Collecting edge information from graph, this takes a while.\n",
      "  0:02:26.890    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:02:30.398    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:02:33.704    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:02:36.860    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:02:39.701    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:02:42.456    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:02:57.734    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:03:01.541    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:03:01.543    57M / 180M  INFO    General                 (gap_closer.cpp            : 452)   Initializing gap closer\n",
      "  0:03:01.544    57M / 180M  INFO   GapCloser                (gap_closer.cpp            : 359)   Closing short gaps\n",
      "  0:03:01.544    57M / 180M  INFO   GapCloser                (gap_closer.cpp            : 393)   Closing short gaps complete: filled 1 gaps after checking 7 candidates\n",
      "  0:03:01.545    57M / 180M  INFO    General                 (gap_closer.cpp            : 456)   Gap closer done\n",
      "  0:03:01.546    48M / 180M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
      "  0:03:01.546    48M / 180M  INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\n",
      "  0:03:01.546    48M / 180M  INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\n",
      "  0:03:01.546    48M / 180M  INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\n",
      "  0:03:01.546    48M / 180M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:03:01.546    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:03:01.547    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:03:01.547    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:03:01.548    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:03:01.548    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:03:01.549    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:03:01.549    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:03:01.550    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:03:01.550    48M / 180M  INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:03:01.550    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:03:01.551    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 1 times\n",
      "  0:03:01.551    48M / 180M  INFO    General                 (simplification.cpp        : 491)   Counting average coverage\n",
      "  0:03:01.552    48M / 180M  INFO    General                 (simplification.cpp        : 497)   Average coverage = 41.1665\n",
      "  0:03:01.552    48M / 180M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Mismatch Correction (id: mismatch_correction)\n",
      "  0:03:01.552    48M / 180M  INFO    General                 (graph_pack.cpp            :  67)   Index refill\n",
      "  0:03:01.552    48M / 180M  INFO    General                 (edge_index.hpp            : 156)   Using small index (max_id = 497276)\n",
      "  0:03:01.728    48M / 180M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:03:03.071    51M / 180M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 4555847 kmers, 3291264 bytes occupied (5.77941 bits per kmer).\n",
      "  0:03:03.077    87M / 185M  INFO    General                 (edge_index_builders.hpp   : 252)   Collecting edge information from graph, this takes a while.\n",
      "  0:03:04.057    87M / 185M  INFO    General                 (graph_pack.cpp            :  77)   Normalizing k-mer map. Total 56128 kmers to process\n",
      "  0:03:04.075    87M / 185M  INFO    General                 (graph_pack.cpp            :  79)   Normalizing done\n",
      "  0:03:04.075    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 400)   Collect potential mismatches\n",
      "  0:03:04.134    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 193)   Total 354 edges (out of 878) with 1752 potential mismatch positions (4.94915 positions per edge)\n",
      "  0:03:04.134    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 402)   Potential mismatches collected\n",
      "  0:03:04.581    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:03:05.052    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:03:05.517    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:03:06.022    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:03:06.518    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:03:06.974    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:03:09.315    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:03:13.884    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 4200000 reads\n",
      "  0:03:23.626    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 8400000 reads\n",
      "  0:03:26.991    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 9891902 reads processed\n",
      "  0:03:27.024    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 395)   All edges processed\n",
      "  0:03:27.024    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 450)   Corrected 3 nucleotides\n",
      "  0:03:27.024    87M / 185M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:03:27.024    87M / 185M  INFO    General                 (contig_output.hpp         :  16)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/before_rr.fasta\n",
      "  0:03:27.054    87M / 186M  INFO    General                 (contig_output_stage.cpp   : 151)   Writing GFA graph to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/assembly_graph_after_simplification.gfa\n",
      "  0:03:27.076    87M / 186M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Hybrid Aligning (id: hybrid_aligning)\n",
      "  0:03:27.076    87M / 186M  INFO   HybridAligning           (hybrid_aligning.cpp       : 341)   Hybrid library detected: #1\n",
      "  0:03:27.076    87M / 186M  INFO    General                 (hybrid_aligning.cpp       : 294)   Aligning long reads with bwa-mem based aligner\n",
      "  0:03:27.142   100M / 186M  INFO    General                 (bwa_index.cpp             : 173)   Using BWA IS algorithm\n",
      "  0:03:30.306   560M / 604M  INFO    General                 (hybrid_aligning.cpp       : 274)   Prepared batch 0 of 41447 reads.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:13:07.177   561M / 606M  INFO    General                 (hybrid_aligning.cpp       : 239)   Read batch of size: 41447 processed; 40140 of them longer than 500; among long reads aligned: 39532; paths of more than one edge received: 2639\n",
      "  0:13:07.178   561M / 606M  INFO    General                 (hybrid_aligning.cpp       : 279)   Processed 41447 reads\n",
      "  0:13:07.184    96M / 606M  INFO    General                 (hybrid_aligning.cpp       : 308)   For library of long reads\n",
      "  0:13:07.184    96M / 606M  INFO   StatsCounter             (pacbio_read_structures.hpp: 163)   Median fraction of present seeds in maximal alignmnent among reads aligned to the graph: 0\n",
      "  0:13:07.184    96M / 606M  INFO    General                 (hybrid_aligning.cpp       : 310)   Aligning of long reads finished\n",
      "  0:13:07.184    87M / 606M  INFO   HybridAligning           (hybrid_aligning.cpp       : 388)   Padding gaps\n",
      "  0:13:07.184    87M / 606M  INFO   HybridAligning           (hybrid_aligning.cpp       : 392)   Min gap weight set to 2\n",
      "  0:13:07.184    87M / 606M  INFO    General                 (hybrid_aligning.cpp       : 146)   Closing gaps with long reads\n",
      "  0:13:07.218    87M / 607M  INFO   MultiGapJoiner           (hybrid_gap_closer.hpp     : 531)   Closed 3 gaps\n",
      "  0:13:07.251    87M / 607M  INFO    General                 (hybrid_aligning.cpp       : 167)   Closing gaps with long reads finished\n",
      "  0:13:07.251    87M / 363M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Paired Information Counting (id: late_pair_info_count)\n",
      "  0:13:07.273    87M / 363M  INFO    General                 (graph_pack.cpp            :  77)   Normalizing k-mer map. Total 56246 kmers to process\n",
      "  0:13:07.295    87M / 363M  INFO    General                 (graph_pack.cpp            :  79)   Normalizing done\n",
      "  0:13:07.296    87M / 363M  INFO    General                 (pair_info_count.cpp       : 339)   Min edge length for estimation: 61753\n",
      "  0:13:07.296    87M / 363M  INFO    General                 (pair_info_count.cpp       : 350)   Estimating insert size for library #0\n",
      "  0:13:07.296    87M / 363M  INFO    General                 (pair_info_count.cpp       : 202)   Estimating insert size (takes a while)\n",
      "  0:13:07.365   119M / 365M  INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  0:13:07.969   119M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:13:08.618   119M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:13:09.260   119M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:13:09.901   119M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:13:10.541   119M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:13:11.180   119M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:13:14.355   119M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:13:15.171   119M / 365M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:13:15.218   119M / 365M  INFO    General                 (pair_info_count.cpp       : 220)   Edge pairs: 3658\n",
      "  0:13:15.218   119M / 365M  INFO    General                 (pair_info_count.cpp       : 224)   1128696 paired reads (46.0942% of all) aligned to long edges\n",
      "  0:13:15.218    87M / 365M  INFO    General                 (pair_info_count.cpp       : 373)     Insert size = 215.4, deviation = 10.1891, left quantile = 202, right quantile = 229, read length = 100\n",
      "  0:13:15.218    87M / 365M  INFO    General                 (pair_info_count.cpp       : 390)   Filtering data for library #0\n",
      "  0:13:15.218    87M / 365M  INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  0:13:15.854    87M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:13:16.480    87M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:13:17.105    87M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:13:17.735    87M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:13:18.363    87M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:13:18.991    87M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:13:22.099    87M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:13:22.886    87M / 365M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:13:22.886    87M / 365M  INFO    General                 (pair_info_count.cpp       : 402)   Mapping library #0\n",
      "  0:13:22.886    87M / 365M  INFO    General                 (pair_info_count.cpp       : 404)   Mapping paired reads (takes a while)\n",
      "  0:13:22.886    87M / 365M  INFO    General                 (pair_info_count.cpp       : 302)   Left insert size quantile 202, right insert size quantile 229, filtering threshold 2, rounding threshold 0\n",
      "  0:13:22.889    96M / 365M  INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  0:13:23.573    97M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:13:24.254    97M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:13:24.933    97M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:13:25.617    97M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:13:26.298    97M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:13:26.981    97M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:13:30.366    97M / 365M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:13:31.220    97M / 365M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:13:31.223    88M / 365M  INFO    General                 (pair_info_count.cpp       : 343)   Library #1 was mapped earlier on hybrid aligning stage, skipping\n",
      "  0:13:31.223    88M / 365M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Distance Estimation (id: distance_estimation)\n",
      "  0:13:31.223    88M / 365M  INFO    General                 (distance_estimation.cpp   : 179)   Processing library #0\n",
      "  0:13:31.223    88M / 365M  INFO    General                 (distance_estimation.cpp   : 150)   Weight Filter Done\n",
      "  0:13:31.223    88M / 365M  INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  0:13:31.227    88M / 365M  INFO    General                 (distance_estimation.cpp   :  37)   Filtering info\n",
      "  0:13:31.227    88M / 365M  INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; library index size: 3698\n",
      "  0:13:31.228    88M / 365M  INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering; library index size: 3228\n",
      "  0:13:31.228    88M / 365M  INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  0:13:31.228    88M / 365M  INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  0:13:31.228    88M / 365M  INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n",
      "  0:13:31.235    88M / 365M  INFO   PairInfoImprover         (pair_info_improver.hpp    : 105)   Paired info stats: missing = 202; contradictional = 8\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:13:31.241    88M / 365M  INFO   PairInfoImprover         (pair_info_improver.hpp    : 105)   Paired info stats: missing = 0; contradictional = 0\n",
      "  0:13:31.241    88M / 365M  INFO    General                 (distance_estimation.cpp   : 104)   Filling scaffolding index\n",
      "  0:13:31.241    88M / 365M  INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  0:13:31.357    88M / 365M  INFO    General                 (distance_estimation.cpp   :  37)   Filtering info\n",
      "  0:13:31.357    88M / 365M  INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; library index size: 879\n",
      "  0:13:31.357    88M / 365M  INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering; library index size: 879\n",
      "  0:13:31.357    88M / 365M  INFO    General                 (distance_estimation.cpp   : 186)   Clearing raw paired index\n",
      "  0:13:31.358    88M / 365M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Repeat Resolving (id: repeat_resolving)\n",
      "  0:13:31.358    88M / 365M  INFO    General                 (repeat_resolving.cpp      :  87)   Using Path-Extend repeat resolving\n",
      "  0:13:31.358    88M / 365M  INFO    General                 (launcher.cpp              : 600)   ExSPAnder repeat resolving tool started\n",
      "  0:13:31.359    88M / 365M  INFO    General                 (coverage_uniformity_analyz:  25)   genomic coverage is 40.29 calculated of length 4525458\n",
      "  0:13:31.359    88M / 365M  INFO    General                 (launcher.cpp              : 186)   Autodetecting unique edge set parameters...\n",
      "  0:13:31.359    88M / 365M  INFO    General                 (launcher.cpp              : 188)   Minimal unique edge length set to the smallest MP library IS: 2000\n",
      "  0:13:31.359    88M / 365M  INFO    General                 (coverage_uniformity_analyz:  25)   genomic coverage is 40.29 calculated of length 4503452\n",
      "  0:13:31.360    88M / 365M  INFO    General                 (launcher.cpp              : 194)   median coverage for edges longer than 2000 is 40.29 uniformity 99%\n",
      "  0:13:31.361    88M / 365M  INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  69)   With length cutoff: 2000, median long edge coverage: 40.29, and maximal unique coverage: 0.5\n",
      "  0:13:31.361    88M / 365M  INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  70)   Unique edges quantity: 230, unique edges length 9000960, total edges length 9111882\n",
      "  0:13:31.362    97M / 366M  INFO    General                 (launcher.cpp              : 418)   Creating main extenders, unique edge length = 2000\n",
      "  0:13:31.362    97M / 366M  INFO    General                 (launcher.cpp              : 342)   filling path container\n",
      "  0:13:31.364   108M / 366M  INFO    General                 (extenders_logic.cpp       : 334)   Estimated coverage of library #0 is 41.1665\n",
      "  0:13:31.364   108M / 366M  INFO    General                 (extenders_logic.cpp       : 345)   Creating extender; library index size: 3624\n",
      "  0:13:31.364   108M / 366M  INFO    General                 (extenders_logic.cpp       : 334)   Estimated coverage of library #0 is 41.1665\n",
      "  0:13:31.364   108M / 366M  INFO    General                 (extenders_logic.cpp       : 345)   Creating extender; library index size: 3624\n",
      "  0:13:31.364   108M / 366M  INFO    General                 (extenders_logic.cpp       :  47)   resolvable_repeat_length_bound set to 22460\n",
      "  0:13:31.366   108M / 366M  INFO    General                 (extenders_logic.cpp       : 543)   Using 1 paired-end library\n",
      "  0:13:31.366   108M / 366M  INFO    General                 (extenders_logic.cpp       : 544)   Using 1 paired-end scaffolding library\n",
      "  0:13:31.366   108M / 366M  INFO    General                 (extenders_logic.cpp       : 545)   Using 1 single read library\n",
      "  0:13:31.367   108M / 366M  INFO    General                 (launcher.cpp              : 385)   Filling backbone edges for long reads scaffolding...\n",
      "  0:13:31.367   108M / 366M  INFO    General                 (launcher.cpp              : 404)    with coverage\n",
      "  0:13:31.367   108M / 366M  INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  69)   With length cutoff: 500, median long edge coverage: 40.29, and maximal unique coverage: 0.5\n",
      "  0:13:31.367   108M / 366M  INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  70)   Unique edges quantity: 258, unique edges length 9034110, total edges length 9111882\n",
      "  0:13:31.367   108M / 366M  INFO    General                 (launcher.cpp              : 407)   258 unique edges\n",
      "  0:13:31.367   108M / 366M  INFO    General                 (extenders_logic.cpp       : 420)   Creating scaffolding extender for lib 1\n",
      "  0:13:31.367   108M / 366M  INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  50)   ExtensionChooser2015 created\n",
      "  0:13:31.368   108M / 366M  INFO    General                 (extenders_logic.cpp       : 446)   Using 1 long reads scaffolding library\n",
      "  0:13:31.368   108M / 366M  INFO    General                 (launcher.cpp              : 447)   Total number of extenders is 5\n",
      "  0:13:31.368   108M / 366M  INFO    General                 (path_extenders.cpp        :  36)   Processed 0 paths from 391 (0%)\n",
      "  0:13:31.376   111M / 366M  INFO    General                 (path_extenders.cpp        :  36)   Processed 40 paths from 391 (10%)\n",
      "  0:13:31.378   111M / 366M  INFO    General                 (path_extenders.cpp        :  36)   Processed 80 paths from 391 (20%)\n",
      "  0:13:31.379   112M / 366M  INFO    General                 (path_extenders.cpp        :  36)   Processed 120 paths from 391 (30%)\n",
      "  0:13:31.380   112M / 366M  INFO    General                 (path_extenders.cpp        :  34)   Processed 128 paths from 391 (32%)\n",
      "  0:13:31.381   112M / 366M  INFO    General                 (path_extenders.cpp        :  36)   Processed 160 paths from 391 (40%)\n",
      "  0:13:31.381   113M / 366M  INFO    General                 (path_extenders.cpp        :  36)   Processed 200 paths from 391 (51%)\n",
      "  0:13:31.382   113M / 366M  INFO    General                 (path_extenders.cpp        :  36)   Processed 240 paths from 391 (61%)\n",
      "  0:13:31.383   114M / 366M  INFO    General                 (path_extenders.cpp        :  34)   Processed 256 paths from 391 (65%)\n",
      "  0:13:31.383   114M / 366M  INFO    General                 (path_extenders.cpp        :  36)   Processed 280 paths from 391 (71%)\n",
      "  0:13:31.383   114M / 366M  INFO    General                 (path_extenders.cpp        :  36)   Processed 320 paths from 391 (81%)\n",
      "  0:13:31.383   114M / 366M  INFO    General                 (path_extenders.cpp        :  36)   Processed 360 paths from 391 (92%)\n",
      "  0:13:31.383   114M / 366M  INFO    General                 (launcher.cpp              : 250)   Finalizing paths\n",
      "  0:13:31.383   114M / 366M  INFO    General                 (launcher.cpp              : 252)   Deduplicating paths\n",
      "  0:13:31.384   113M / 366M  INFO    General                 (launcher.cpp              : 256)   Paths deduplicated\n",
      "  0:13:31.384   113M / 366M  INFO   PEResolver               (pe_resolver.cpp           :  60)   Removing overlaps\n",
      "  0:13:31.384   113M / 366M  INFO   PEResolver               (pe_resolver.cpp           :  63)   Sorting paths\n",
      "  0:13:31.384   113M / 366M  INFO   PEResolver               (pe_resolver.cpp           :  70)   Marking overlaps\n",
      "  0:13:31.384   113M / 366M  INFO   OverlapRemover           (overlap_remover.hpp       : 116)   Marking start/end overlaps\n",
      "  0:13:31.385   113M / 366M  INFO   OverlapRemover           (overlap_remover.hpp       : 119)   Marking remaining overlaps\n",
      "  0:13:31.386   113M / 366M  INFO   PEResolver               (pe_resolver.cpp           :  73)   Splitting paths\n",
      "  0:13:31.386   118M / 366M  INFO   PEResolver               (pe_resolver.cpp           :  78)   Deduplicating paths\n",
      "  0:13:31.387   112M / 366M  INFO   PEResolver               (pe_resolver.cpp           :  80)   Overlaps removed\n",
      "  0:13:31.387   113M / 366M  INFO    General                 (launcher.cpp              : 273)   Paths finalized\n",
      "  0:13:31.387   113M / 366M  INFO    General                 (launcher.cpp              : 454)   Closing gaps in paths\n",
      "  0:13:31.387   116M / 366M  INFO    General                 (launcher.cpp              : 484)   Gap closing completed\n",
      "  0:13:31.388   116M / 366M  INFO    General                 (launcher.cpp              : 302)   Traversing tandem repeats\n",
      "  0:13:31.389   116M / 366M  INFO    General                 (launcher.cpp              : 312)   Traversed 5 loops\n",
      "  0:13:31.389   116M / 366M  INFO    General                 (launcher.cpp              : 250)   Finalizing paths\n",
      "  0:13:31.389   116M / 366M  INFO    General                 (launcher.cpp              : 252)   Deduplicating paths\n",
      "  0:13:31.389   116M / 366M  INFO    General                 (launcher.cpp              : 256)   Paths deduplicated\n",
      "  0:13:31.389   116M / 366M  INFO   PEResolver               (pe_resolver.cpp           :  60)   Removing overlaps\n",
      "  0:13:31.389   116M / 366M  INFO   PEResolver               (pe_resolver.cpp           :  63)   Sorting paths\n",
      "  0:13:31.389   116M / 366M  INFO   PEResolver               (pe_resolver.cpp           :  70)   Marking overlaps\n",
      "  0:13:31.389   116M / 366M  INFO   OverlapRemover           (overlap_remover.hpp       : 116)   Marking start/end overlaps\n",
      "  0:13:31.389   116M / 366M  INFO   OverlapRemover           (overlap_remover.hpp       : 119)   Marking remaining overlaps\n",
      "  0:13:31.389   116M / 366M  INFO   PEResolver               (pe_resolver.cpp           :  73)   Splitting paths\n",
      "  0:13:31.389   116M / 366M  INFO   PEResolver               (pe_resolver.cpp           :  78)   Deduplicating paths\n",
      "  0:13:31.389   116M / 366M  INFO   PEResolver               (pe_resolver.cpp           :  80)   Overlaps removed\n",
      "  0:13:31.390   116M / 366M  INFO    General                 (launcher.cpp              : 273)   Paths finalized\n",
      "  0:13:31.390   116M / 366M  INFO    General                 (launcher.cpp              : 664)   ExSPAnder repeat resolving tool finished\n",
      "  0:13:31.391    91M / 366M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:13:31.391    91M / 366M  INFO    General                 (contig_output.hpp         :  16)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/before_rr.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:13:31.413    91M / 366M  INFO    General                 (contig_output_stage.cpp   : 151)   Writing GFA graph to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/assembly_graph_with_scaffolds.gfa\n",
      "  0:13:31.435    91M / 366M  INFO    General                 (contig_output_stage.cpp   : 165)   Outputting FastG graph to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/assembly_graph.fastg\n",
      "  0:13:31.498    91M / 366M  INFO    General                 (contig_output_stage.cpp   : 196)   Breaking scaffolds\n",
      "  0:13:31.509    99M / 366M  INFO    General                 (contig_output_stage.cpp   :  98)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/final_contigs.fasta\n",
      "  0:13:31.529    99M / 366M  INFO    General                 (contig_output_stage.cpp   : 104)   Outputting FastG paths to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/final_contigs.paths\n",
      "  0:13:31.539    95M / 366M  INFO    General                 (contig_output_stage.cpp   :  98)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/scaffolds.fasta\n",
      "  0:13:31.557    95M / 366M  INFO    General                 (contig_output_stage.cpp   : 104)   Outputting FastG paths to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/scaffolds.paths\n",
      "  0:13:31.557    95M / 366M  INFO    General                 (contig_output_stage.cpp   : 111)   Populating GFA with scaffold paths\n",
      "  0:13:31.558    91M / 366M  INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  0:13:31.561     1M / 366M  INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 13 minutes 31 seconds\n",
      "\n",
      "===== K55 finished. \n",
      "\n",
      "\n",
      "===== Copy files started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/miniconda3/bin/python /Users/olga/Downloads/SPAdes-3.15.2/share/spades/spades_pipeline/scripts/copy_files.py /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/before_rr.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/before_rr.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/assembly_graph_after_simplification.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/assembly_graph_after_simplification.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/final_contigs.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/contigs.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/first_pe_contigs.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/first_pe_contigs.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/strain_graph.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/strain_graph.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/scaffolds.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/scaffolds.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/scaffolds.paths /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/scaffolds.paths /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/assembly_graph_with_scaffolds.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/assembly_graph_with_scaffolds.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/assembly_graph.fastg /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/assembly_graph.fastg /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/K55/final_contigs.paths /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/contigs.paths\n",
      "\n",
      "\n",
      "===== Copy files finished. \n",
      "\n",
      "\n",
      "===== Assembling finished. \n",
      "\n",
      "\n",
      "===== Breaking scaffolds started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/miniconda3/bin/python /Users/olga/Downloads/SPAdes-3.15.2/share/spades/spades_pipeline/scripts/breaking_scaffolds_script.py --result_scaffolds_filename /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/scaffolds.fasta --misc_dir /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/misc --threshold_for_breaking_scaffolds 3\n",
      "\n",
      "\n",
      "===== Breaking scaffolds finished. \n",
      "\n",
      "\n",
      "===== Terminate started. \n",
      "\n",
      "\n",
      "===== Terminate finished. \n",
      "\n",
      " * Corrected reads are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/corrected/\n",
      " * Assembled contigs are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/contigs.fasta\n",
      " * Assembled scaffolds are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/scaffolds.fasta\n",
      " * Paths in the assembly graph corresponding to the contigs are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/scaffolds.paths\n",
      " * Assembly graph is in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/assembly_graph_with_scaffolds.gfa\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio40/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n"
     ]
    }
   ],
   "source": [
    "! /Users/olga/Downloads/SPAdes-3.15.2/bin/spades.py -1 ./homework6/illumina.100x.1.fq -2 ./homework6/illumina.100x.2.fq --pacbio ./homework6/pacbio_40x.fq -o illumina_pacbio40"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "== Warning ==  No assembly mode was specified! If you intend to assemble high-coverage multi-cell/isolate data, use '--isolate' option.\n",
      "\n",
      "\n",
      "Command line: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades.py\t-1\t/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq\t-2\t/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq\t--pacbio\t/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/pacbio_80x.fq\t-o\t/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80\t\n",
      "\n",
      "System information:\n",
      "  SPAdes version: 3.15.2\n",
      "  Python version: 3.7.7\n",
      "  OS: Darwin-20.3.0-x86_64-i386-64bit\n",
      "\n",
      "Output dir: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80\n",
      "Mode: read error correction and assembling\n",
      "Debug mode is turned OFF\n",
      "\n",
      "Dataset parameters:\n",
      "  Standard mode\n",
      "  For multi-cell/isolate data we recommend to use '--isolate' option; for single-cell MDA data use '--sc'; for metagenomic data use '--meta'; for RNA-Seq use '--rna'.\n",
      "  Reads:\n",
      "    Library number: 1, library type: paired-end\n",
      "      orientation: fr\n",
      "      left reads: ['/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq']\n",
      "      right reads: ['/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq']\n",
      "      interlaced reads: not specified\n",
      "      single reads: not specified\n",
      "      merged reads: not specified\n",
      "    Library number: 2, library type: pacbio\n",
      "      left reads: not specified\n",
      "      right reads: not specified\n",
      "      interlaced reads: not specified\n",
      "      single reads: ['/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/pacbio_80x.fq']\n",
      "      merged reads: not specified\n",
      "Read error correction parameters:\n",
      "  Iterations: 1\n",
      "  PHRED offset will be auto-detected\n",
      "  Corrected reads will be compressed\n",
      "Assembly parameters:\n",
      "  k: automatic selection based on read length\n",
      "  Repeat resolution is enabled\n",
      "  Mismatch careful mode is turned OFF\n",
      "  MismatchCorrector will be SKIPPED\n",
      "  Coverage cutoff is turned OFF\n",
      "Other parameters:\n",
      "  Dir for temp files: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/tmp\n",
      "  Threads: 16\n",
      "  Memory limit (in Gb): 250\n",
      "\n",
      "\n",
      "======= SPAdes pipeline started. Log can be found here: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/spades.log\n",
      "\n",
      "/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq: max reads length: 100\n",
      "/Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq: max reads length: 100\n",
      "\n",
      "Reads length: 100\n",
      "\n",
      "\n",
      "===== Before start started. \n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "===== Read error correction started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-hammer /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/corrected/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  75)   Starting BayesHammer, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  76)   Loading config from /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/corrected/configs/config.info\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  78)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  86)   Trying to determine PHRED offset\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  :  92)   Determined value is 33\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (hammer_tools.cpp          :  38)   Hamming graph threshold tau=1, k=21, subkmer positions = [ 0 10 ]\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (main.cpp                  : 113)   Size of aux. kmer data 24 bytes\n",
      "     === ITERATION 0 begins ===\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 16 files using 1 threads. This might take a while.\n",
      "  0:00:00.000     1M / 2M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:00.001     1M / 2M    INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 4194304\n",
      "  0:00:00.001   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq\n",
      "  0:00:09.369   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 467314 reads\n",
      "  0:00:18.200   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 935009 reads\n",
      "  0:00:27.051   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 1402861 reads\n",
      "  0:00:35.920   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 1870646 reads\n",
      "  0:00:44.740   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2339377 reads\n",
      "  0:00:47.900   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2500000 reads\n",
      "  0:00:47.900   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             :  97)   Processing /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq\n",
      "  0:00:56.846   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 2974795 reads\n",
      "  0:01:23.920   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 107)   Processed 4400968 reads\n",
      "  0:04:19.567   577M / 577M  INFO   K-mer Splitting          (kmer_data.cpp             : 112)   Total 5000000 reads processed\n",
      "  0:04:19.567     1M / 162M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:04:23.863     1M / 170M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 38919786 kmers in total.\n",
      "  0:04:23.863     1M / 170M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:04:27.085    30M / 195M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 38919786 kmers, 28121944 bytes occupied (5.78049 bits per kmer).\n",
      "  0:04:27.085    30M / 195M  INFO   K-mer Counting           (kmer_data.cpp             : 354)   Arranging kmers in hash map order\n",
      "  0:04:42.152   631M / 789M  INFO    General                 (main.cpp                  : 148)   Clustering Hamming graph.\n",
      "  0:38:13.839   631M / 789M  INFO    General                 (main.cpp                  : 155)   Extracting clusters:\n",
      "  0:38:13.839   631M / 789M  INFO    General                 (concurrent_dsu.cpp        :  18)   Connecting to root\n",
      "  0:38:14.579   631M / 789M  INFO    General                 (concurrent_dsu.cpp        :  34)   Calculating counts\n",
      "  0:38:26.586   903M / 1061M INFO    General                 (concurrent_dsu.cpp        :  63)   Writing down entries\n",
      "  0:38:38.233   631M / 789M  INFO    General                 (main.cpp                  : 167)   Clustering done. Total clusters: 13651605\n",
      "  0:38:38.250   330M / 492M  INFO   K-mer Counting           (kmer_data.cpp             : 371)   Collecting K-mer information, this takes a while.\n",
      "  0:38:38.613  1223M / 1383M INFO   K-mer Counting           (kmer_data.cpp             : 377)   Processing /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq\n",
      "  0:43:20.420  1223M / 1383M INFO   K-mer Counting           (kmer_data.cpp             : 377)   Processing /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq\n",
      "  0:47:55.406  1223M / 1383M INFO   K-mer Counting           (kmer_data.cpp             : 384)   Collection done, postprocessing.\n",
      "  0:47:55.639  1223M / 1383M INFO   K-mer Counting           (kmer_data.cpp             : 398)   There are 38919786 kmers in total. Among them 27377044 (70.3422%) are singletons.\n",
      "  0:47:55.639  1223M / 1383M INFO    General                 (main.cpp                  : 173)   Subclustering Hamming graph\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:48:38.385  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 650)   Subclustering done. Total 26 non-read kmers were generated.\n",
      "  0:48:38.385  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 651)   Subclustering statistics:\n",
      "  0:48:38.385  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 652)     Total singleton hamming clusters: 5541662. Among them 963110 (17.3794%) are good\n",
      "  0:48:38.385  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 653)     Total singleton subclusters: 13747. Among them 13599 (98.9234%) are good\n",
      "  0:48:38.385  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 654)     Total non-singleton subcluster centers: 8197112. Among them 6855795 (83.6367%) are good\n",
      "  0:48:38.385  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 655)     Average size of non-trivial subcluster: 4.07194 kmers\n",
      "  0:48:38.385  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 656)     Average number of sub-clusters per non-singleton cluster: 1.01244\n",
      "  0:48:38.385  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 657)     Total solid k-mers: 7832504\n",
      "  0:48:38.385  1223M / 1488M INFO   Hamming Subclustering    (kmer_cluster.cpp          : 658)     Substitution probabilities: [4,4]((0.928457,0.0421442,0.022541,0.00685828),(0.0141904,0.967828,0.0090854,0.00889584),(0.00886597,0.00903879,0.967867,0.0142285),(0.00683121,0.022511,0.0420995,0.928558))\n",
      "  0:48:38.394  1223M / 1384M INFO    General                 (main.cpp                  : 178)   Finished clustering.\n",
      "  0:48:38.394  1223M / 1384M INFO    General                 (main.cpp                  : 197)   Starting solid k-mers expansion in 1 threads.\n",
      "  0:51:05.762  1223M / 1384M INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 0 produced 39977 new k-mers.\n",
      "  0:53:34.382  1223M / 1384M INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 1 produced 790 new k-mers.\n",
      "  0:56:11.482  1223M / 1384M INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 2 produced 38 new k-mers.\n",
      "  0:58:46.643  1223M / 1384M INFO    General                 (main.cpp                  : 218)   Solid k-mers iteration 3 produced 0 new k-mers.\n",
      "  0:58:46.643  1223M / 1384M INFO    General                 (main.cpp                  : 222)   Solid k-mers finalized\n",
      "  0:58:46.643  1223M / 1384M INFO    General                 (hammer_tools.cpp          : 222)   Starting read correction in 1 threads.\n",
      "  0:58:46.643  1223M / 1384M INFO    General                 (hammer_tools.cpp          : 235)   Correcting pair of reads: /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.1.fq and /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/illumina.100x.2.fq\n",
      "  0:58:46.811  1289M / 1394M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 0 of 100000 reads.\n",
      "  0:58:53.548  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 0\n",
      "  0:58:53.766  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 0\n",
      "  0:58:53.907  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 1 of 100000 reads.\n",
      "  0:59:00.463  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 1\n",
      "  0:59:00.692  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 1\n",
      "  0:59:00.841  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 2 of 100000 reads.\n",
      "  0:59:07.595  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 2\n",
      "  0:59:07.834  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 2\n",
      "  0:59:07.979  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 3 of 100000 reads.\n",
      "  0:59:14.316  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 3\n",
      "  0:59:14.552  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 3\n",
      "  0:59:14.696  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 4 of 100000 reads.\n",
      "  0:59:21.287  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 4\n",
      "  0:59:21.526  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 4\n",
      "  0:59:21.671  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 5 of 100000 reads.\n",
      "  0:59:27.947  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 5\n",
      "  0:59:28.189  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 5\n",
      "  0:59:28.333  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 6 of 100000 reads.\n",
      "  0:59:34.667  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 6\n",
      "  0:59:34.899  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 6\n",
      "  0:59:35.046  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 7 of 100000 reads.\n",
      "  0:59:41.347  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 7\n",
      "  0:59:41.583  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 7\n",
      "  0:59:41.728  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 8 of 100000 reads.\n",
      "  0:59:48.510  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 8\n",
      "  0:59:48.751  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 8\n",
      "  0:59:48.896  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 9 of 100000 reads.\n",
      "  0:59:55.449  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 9\n",
      "  0:59:55.674  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 9\n",
      "  0:59:55.823  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 10 of 100000 reads.\n",
      "  1:00:02.097  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 10\n",
      "  1:00:02.335  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 10\n",
      "  1:00:02.480  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 11 of 100000 reads.\n",
      "  1:00:08.734  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 11\n",
      "  1:00:08.981  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 11\n",
      "  1:00:09.124  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 12 of 100000 reads.\n",
      "  1:00:15.435  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 12\n",
      "  1:00:15.669  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 12\n",
      "  1:00:15.813  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 13 of 100000 reads.\n",
      "  1:00:22.086  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 13\n",
      "  1:00:22.324  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 13\n",
      "  1:00:22.469  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 14 of 100000 reads.\n",
      "  1:00:28.758  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 14\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  1:00:29.011  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 14\n",
      "  1:00:29.157  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 15 of 100000 reads.\n",
      "  1:00:35.464  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 15\n",
      "  1:00:35.705  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 15\n",
      "  1:00:35.849  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 16 of 100000 reads.\n",
      "  1:00:42.113  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 16\n",
      "  1:00:42.355  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 16\n",
      "  1:00:42.500  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 17 of 100000 reads.\n",
      "  1:00:48.750  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 17\n",
      "  1:00:48.997  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 17\n",
      "  1:00:49.140  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 18 of 100000 reads.\n",
      "  1:00:55.590  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 18\n",
      "  1:00:55.846  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 18\n",
      "  1:00:55.997  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 19 of 100000 reads.\n",
      "  1:01:02.672  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 19\n",
      "  1:01:02.916  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 19\n",
      "  1:01:03.061  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 20 of 100000 reads.\n",
      "  1:01:09.312  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 20\n",
      "  1:01:09.553  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 20\n",
      "  1:01:09.696  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 21 of 100000 reads.\n",
      "  1:01:16.021  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 21\n",
      "  1:01:16.260  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 21\n",
      "  1:01:16.405  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 22 of 100000 reads.\n",
      "  1:01:22.703  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 22\n",
      "  1:01:22.945  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 22\n",
      "  1:01:23.090  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 23 of 100000 reads.\n",
      "  1:01:29.355  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 23\n",
      "  1:01:29.600  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 23\n",
      "  1:01:29.749  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 170)   Prepared batch 24 of 100000 reads.\n",
      "  1:01:36.448  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 177)   Processed batch 24\n",
      "  1:01:36.695  1293M / 1396M INFO    General                 (hammer_tools.cpp          : 187)   Written batch 24\n",
      "  1:01:36.706  1223M / 1396M INFO    General                 (hammer_tools.cpp          : 276)   Correction done. Changed 1639742 bases in 1194871 reads.\n",
      "  1:01:36.706  1223M / 1396M INFO    General                 (hammer_tools.cpp          : 277)   Failed to correct 44220 bases out of 458864877.\n",
      "  1:01:36.785     1M / 208M  INFO    General                 (main.cpp                  : 255)   Saving corrected dataset description to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/corrected/corrected.yaml\n",
      "  1:01:36.785     1M / 208M  INFO    General                 (main.cpp                  : 262)   All done. Exiting.\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== corrected reads compression started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/miniconda3/bin/python /Users/olga/Downloads/SPAdes-3.15.2/share/spades/spades_pipeline/scripts/compress_all.py --input_file /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/corrected/corrected.yaml --ext_python_modules_home /Users/olga/Downloads/SPAdes-3.15.2/share/spades --max_threads 16 --output_dir /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/corrected --not_used_yaml_file /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/corrected/dont_correct.yaml --gzip_output\n",
      "\n",
      "== Compressing corrected reads (with gzip)\n",
      "== Files to compress: ['/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/corrected/illumina.100x.1.00.0_0.cor.fastq', '/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/corrected/illumina.100x.2.00.0_0.cor.fastq', '/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/corrected/illumina.100x._unpaired.00.0_0.cor.fastq']\n",
      "== Files compression is finished\n",
      "== Info about datasets not used in error correction stage is loaded\n",
      "== Dataset yaml file is updated\n",
      "\n",
      "===== corrected reads compression finished. \n",
      "\n",
      "\n",
      "===== Read error correction finished. \n",
      "\n",
      "\n",
      "===== Assembling started. \n",
      "\n",
      "\n",
      "===== K21 started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-core /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K21/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  :  99)   Loaded config from /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K21/configs/config.info\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 109)   Assembling dataset (/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/dataset.info) with K=21\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:00:00.001     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.002     1M / 4M    INFO    General                 (read_converter.cpp        :  72)   Converting reads to binary format for library #0 (takes a while)\n",
      "  0:00:00.002     1M / 4M    INFO    General                 (read_converter.cpp        :  73)   Converting paired reads\n",
      "  0:00:00.071    22M / 22M   INFO    General                 (binary_converter.cpp      :  96)   16384 reads processed\n",
      "  0:00:00.137    25M / 25M   INFO    General                 (binary_converter.cpp      :  96)   32768 reads processed\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:00.345    22M / 25M   INFO    General                 (binary_converter.cpp      :  96)   65536 reads processed\n",
      "  0:00:00.670    25M / 31M   INFO    General                 (binary_converter.cpp      :  96)   131072 reads processed\n",
      "  0:00:01.401    21M / 31M   INFO    General                 (binary_converter.cpp      :  96)   262144 reads processed\n",
      "  0:00:02.734    24M / 31M   INFO    General                 (binary_converter.cpp      :  96)   524288 reads processed\n",
      "  0:00:05.414    28M / 31M   INFO    General                 (binary_converter.cpp      :  96)   1048576 reads processed\n",
      "  0:00:10.839    28M / 31M   INFO    General                 (binary_converter.cpp      :  96)   2097152 reads processed\n",
      "  0:00:12.734    19M / 31M   INFO    General                 (binary_converter.cpp      : 111)   2448672 reads written\n",
      "  0:00:12.735     1M / 31M   INFO    General                 (read_converter.cpp        :  81)   Converting single reads\n",
      "  0:00:12.770    10M / 31M   INFO    General                 (binary_converter.cpp      :  96)   16384 reads processed\n",
      "  0:00:12.800    12M / 31M   INFO    General                 (binary_converter.cpp      :  96)   32768 reads processed\n",
      "  0:00:12.870     9M / 31M   INFO    General                 (binary_converter.cpp      : 111)   48607 reads written\n",
      "  0:00:12.870     1M / 31M   INFO    General                 (read_converter.cpp        :  87)   Converting merged reads\n",
      "  0:00:12.870     9M / 31M   INFO    General                 (binary_converter.cpp      : 111)   0 reads written\n",
      "  0:00:12.871     1M / 31M   INFO    General                 (read_converter.cpp        :  72)   Converting reads to binary format for library #1 (takes a while)\n",
      "  0:00:12.871     1M / 31M   INFO    General                 (read_converter.cpp        :  73)   Converting paired reads\n",
      "  0:00:12.871    19M / 31M   INFO    General                 (binary_converter.cpp      : 111)   0 reads written\n",
      "  0:00:12.871     1M / 31M   INFO    General                 (read_converter.cpp        :  81)   Converting single reads\n",
      "  0:00:13.380    98M / 108M  INFO    General                 (binary_converter.cpp      :  96)   16384 reads processed\n",
      "  0:00:13.876   187M / 197M  INFO    General                 (binary_converter.cpp      :  96)   32768 reads processed\n",
      "  0:00:16.625    96M / 327M  INFO    General                 (binary_converter.cpp      :  96)   65536 reads processed\n",
      "  0:00:18.347     9M / 421M  INFO    General                 (binary_converter.cpp      : 111)   82783 reads written\n",
      "  0:00:18.347     1M / 421M  INFO    General                 (read_converter.cpp        :  87)   Converting merged reads\n",
      "  0:00:18.348     9M / 421M  INFO    General                 (binary_converter.cpp      : 111)   0 reads written\n",
      "  0:00:18.353     1M / 246M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:18.366     1M / 246M  INFO    General                 (construction.cpp          : 153)   Max read length 100\n",
      "  0:00:18.366     1M / 246M  INFO    General                 (construction.cpp          : 159)   Average read length 92.7147\n",
      "  0:00:18.366     1M / 246M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:18.367     1M / 246M  INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:18.367     1M / 246M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:18.367     1M / 246M  INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:18.367     1M / 246M  INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 6710886\n",
      "  0:00:28.284   601M / 663M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 1873811 reads\n",
      "  0:00:38.144   601M / 664M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 3734243 reads\n",
      "  0:00:47.641   601M / 664M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 5595232 reads\n",
      "  0:00:57.155   601M / 664M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 7456112 reads\n",
      "  0:01:06.668   601M / 664M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 9318931 reads\n",
      "  0:01:09.810   601M / 664M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 9891902 reads\n",
      "  0:01:09.810     1M / 255M  INFO    General                 (kmer_splitters.hpp        : 131)   Used 9891902 reads\n",
      "  0:01:09.844     1M / 255M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:01:10.804     1M / 255M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5715581 kmers in total.\n",
      "  0:01:10.804     1M / 255M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:01:10.805     1M / 256M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:01:10.805     1M / 256M  INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:01:10.805     1M / 256M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:01:10.805     1M / 256M  INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:01:10.805     1M / 256M  INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 6710886\n",
      "  0:01:12.606   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 194)   Processed 5715581 kmers\n",
      "  0:01:12.606   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 199)   Used 5715581 kmers.\n",
      "  0:01:12.606     1M / 299M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:01:12.793     1M / 299M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5708731 kmers in total.\n",
      "  0:01:12.793     1M / 299M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:13.223     5M / 299M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5708731 kmers, 4130592 bytes occupied (5.78846 bits per kmer).\n",
      "  0:01:13.223     5M / 299M  INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n",
      "  0:01:13.364    11M / 299M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:01:15.413    11M / 299M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:01:15.414    11M / 256M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Early tip clipping (id: construction:early_tip_clipper)\n",
      "  0:01:15.414    11M / 256M  INFO    General                 (construction.cpp          : 301)   Early tip clipper length bound set as (RL - K)\n",
      "  0:01:15.414    11M / 256M  INFO   Early tip clipping       (early_simplification.hpp  :  47)   Early tip clipping\n",
      "  0:01:17.123    21M / 299M  INFO   Early tip clipping       (early_simplification.hpp  :  82)   #tipped junctions: 123493\n",
      "  0:01:17.162    21M / 299M  INFO   Early tip clipping       (early_simplification.hpp  :  93)   Clipped tips: 124241\n",
      "  0:01:17.163    11M / 256M  INFO   Early tip clipping       (early_simplification.hpp  :  49)   978561 22-mers were removed by early tip clipper\n",
      "  0:01:17.163    11M / 256M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:01:17.167    11M / 256M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:01:19.372    13M / 299M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 25730 sequences extracted\n",
      "  0:01:20.149    13M / 256M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:01:20.545    13M / 299M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 4 loops collected\n",
      "  0:01:20.547    13M / 256M  INFO    General                 (debruijn_graph_constructor: 487)   Total 51468 edges to create\n",
      "  0:01:20.547    15M / 256M  INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:01:20.558    16M / 256M  INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:01:20.561    16M / 256M  INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:01:20.561    16M / 256M  INFO    General                 (debruijn_graph_constructor: 503)   Total 18884 vertices to create\n",
      "  0:01:20.561    18M / 256M  INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:01:20.565    16M / 256M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:01:20.565    16M / 256M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:21.003    20M / 256M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5715581 kmers, 4135544 bytes occupied (5.78845 bits per kmer).\n",
      "  0:01:21.004    42M / 256M  INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:02:21.106    42M / 256M  INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:02:22.544    42M / 256M  INFO    General                 (coverage_filling.hpp      :  82)   Processed 51447 edges\n",
      "  0:02:22.558     6M / 256M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:02:22.560     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 16\n",
      "  0:02:22.560     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 76\n",
      "  0:02:22.560     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 77. Coverage mad: 14.826\n",
      "  0:02:22.560     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:02:22.625     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:02:22.789     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:02:23.321     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:02:24.554     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:02:24.688     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 76.252. Fitted coverage std. dev: 14.5028\n",
      "  0:02:24.690     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.863705\n",
      "  0:02:24.690     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 37\n",
      "  0:02:24.690     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 37\n",
      "  0:02:24.690     6M / 256M  INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 4507902\n",
      "  0:02:24.690     6M / 256M  INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 76.252\n",
      "  0:02:24.690     6M / 256M  INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 37\n",
      "  0:02:24.690     6M / 256M  INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:02:24.690     6M / 256M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:02:24.692     6M / 256M  INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:02:24.694     6M / 256M  INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:02:24.694     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:02:24.697     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:02:24.697     6M / 256M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n",
      "  0:02:24.717     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 2371 times\n",
      "  0:02:24.717     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n",
      "  0:02:24.789     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 4029 times\n",
      "  0:02:24.789     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:02:24.790     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 308 times\n",
      "  0:02:24.791     6M / 257M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:02:24.791     6M / 257M  INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:02:24.791     6M / 257M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:02:24.791     6M / 257M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:02:24.791     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:24.794     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 128 times\n",
      "  0:02:24.794     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:24.866     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 1810 times\n",
      "  0:02:24.866     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:24.871     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 143 times\n",
      "  0:02:24.871     6M / 257M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:02:24.871     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:24.872     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 5 times\n",
      "  0:02:24.872     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:24.873     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 36 times\n",
      "  0:02:24.873     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:24.875     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 68 times\n",
      "  0:02:24.875     6M / 257M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:02:24.875     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:24.875     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 3 times\n",
      "  0:02:24.875     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:24.876     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 23 times\n",
      "  0:02:24.876     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:24.876     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 16 times\n",
      "  0:02:24.876     6M / 257M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:02:24.876     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:24.876     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:24.876     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:24.876     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 7 times\n",
      "  0:02:24.876     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 10 times\n",
      "  0:02:24.877     6M / 257M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 2 times\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 4 times\n",
      "  0:02:24.877     6M / 257M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 4 times\n",
      "  0:02:24.877     6M / 257M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 1 times\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 2 times\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 3 times\n",
      "  0:02:24.877     6M / 257M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 7 times\n",
      "  0:02:24.877     6M / 257M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:02:24.877     6M / 257M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 6 times\n",
      "  0:02:24.877     6M / 257M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:02:24.877     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:24.878     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:24.878     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:24.880     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 3 times\n",
      "  0:02:24.880     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:24.881     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:24.881     6M / 257M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:02:24.881     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:24.881     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:24.881     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:24.881     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:24.881     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:24.881     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:24.881     6M / 257M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
      "  0:02:24.881     6M / 257M  INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\n",
      "  0:02:24.882     6M / 257M  INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\n",
      "  0:02:24.882     6M / 257M  INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\n",
      "  0:02:24.882     6M / 257M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:02:24.882     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:24.883     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:24.883     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:24.887     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:24.887     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:24.888     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:24.888     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:24.890     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:24.890     6M / 257M  INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:02:24.890     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:02:24.891     6M / 257M  INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 7 times\n",
      "  0:02:24.891     6M / 257M  INFO    General                 (simplification.cpp        : 491)   Counting average coverage\n",
      "  0:02:24.891     6M / 257M  INFO    General                 (simplification.cpp        : 497)   Average coverage = 78.0189\n",
      "  0:02:24.891     6M / 257M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:02:24.892     6M / 257M  INFO    General                 (read_converter.cpp        : 102)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K21/simplified_contigs\n",
      "  0:02:24.896     8M / 257M  INFO    General                 (binary_converter.cpp      : 111)   2571 reads written\n",
      "  0:02:24.899     6M / 257M  INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  0:02:24.900     1M / 257M  INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 2 minutes 24 seconds\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "===== K21 finished. \n",
      "\n",
      "\n",
      "===== K33 started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-core /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K33/configs/config.info\n",
      "\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  :  99)   Loaded config from /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K33/configs/config.info\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 109)   Assembling dataset (/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/dataset.info) with K=33\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.001     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (construction.cpp          : 118)   Contigs from previous K will be used: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K21/simplified_contigs\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (construction.cpp          : 153)   Max read length 100\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (construction.cpp          : 159)   Average read length 92.7147\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:00.002     1M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:00.002     1M / 4M    INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:00.002     1M / 4M    INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 3355443\n",
      "  0:00:05.677   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 1132634 reads\n",
      "  0:00:11.013   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 2249495 reads\n",
      "  0:00:16.377   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 3368100 reads\n",
      "  0:00:21.718   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 4487032 reads\n",
      "  0:00:27.085   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 5607299 reads\n",
      "  0:00:32.413   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 6726813 reads\n",
      "  0:00:37.726   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 7846654 reads\n",
      "  0:00:43.060   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 8966769 reads\n",
      "  0:00:48.233   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 9897044 reads\n",
      "  0:00:48.233     1M / 158M  INFO    General                 (kmer_splitters.hpp        : 131)   Used 9897044 reads\n",
      "  0:00:48.265     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:50.604     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5756125 kmers in total.\n",
      "  0:00:50.605     1M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:00:50.608     1M / 158M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:00:50.608     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:50.608     1M / 158M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:50.608     1M / 158M  INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:50.608     1M / 158M  INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 3355443\n",
      "  0:00:52.748   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 194)   Processed 5756125 kmers\n",
      "  0:00:52.748   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 199)   Used 5756125 kmers.\n",
      "  0:00:52.748     1M / 246M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:53.075     1M / 246M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5754745 kmers in total.\n",
      "  0:00:53.075     1M / 246M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:53.594     5M / 246M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5754745 kmers, 4163816 bytes occupied (5.78836 bits per kmer).\n",
      "  0:00:53.594     5M / 246M  INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n",
      "  0:00:53.860    11M / 246M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:00:55.775    11M / 246M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:55.777    11M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Early tip clipping (id: construction:early_tip_clipper)\n",
      "  0:00:55.777    11M / 158M  INFO    General                 (construction.cpp          : 301)   Early tip clipper length bound set as (RL - K)\n",
      "  0:00:55.777    11M / 158M  INFO   Early tip clipping       (early_simplification.hpp  :  47)   Early tip clipping\n",
      "  0:00:57.523    21M / 246M  INFO   Early tip clipping       (early_simplification.hpp  :  82)   #tipped junctions: 124481\n",
      "  0:00:57.570    21M / 246M  INFO   Early tip clipping       (early_simplification.hpp  :  93)   Clipped tips: 125222\n",
      "  0:00:57.572    11M / 158M  INFO   Early tip clipping       (early_simplification.hpp  :  49)   1029171 34-mers were removed by early tip clipper\n",
      "  0:00:57.572    11M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:57.583    11M / 158M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:00:59.838    12M / 248M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 15754 sequences extracted\n",
      "  0:01:00.682    12M / 160M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:01:01.101    12M / 248M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 2 loops collected\n",
      "  0:01:01.103    12M / 160M  INFO    General                 (debruijn_graph_constructor: 487)   Total 31512 edges to create\n",
      "  0:01:01.104    14M / 160M  INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:01:01.111    14M / 160M  INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:01:01.112    14M / 160M  INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:01:01.112    14M / 160M  INFO    General                 (debruijn_graph_constructor: 503)   Total 14376 vertices to create\n",
      "  0:01:01.112    16M / 160M  INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:01:01.114    15M / 160M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n",
      "  0:01:01.114    15M / 160M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:01.640    19M / 160M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5756125 kmers, 4164864 bytes occupied (5.78843 bits per kmer).\n",
      "  0:01:01.641    41M / 163M  INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:01:52.698    41M / 163M  INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:01:54.205    41M / 163M  INFO    General                 (coverage_filling.hpp      :  82)   Processed 31510 edges\n",
      "  0:01:54.224     5M / 163M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:01:54.224     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 13\n",
      "  0:01:54.224     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 64\n",
      "  0:01:54.224     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 64. Coverage mad: 14.826\n",
      "  0:01:54.224     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:01:54.279     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:01:54.417     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:01:54.863     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:01:55.913     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:01:56.575     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 63.4837. Fitted coverage std. dev: 13.4198\n",
      "  0:01:56.577     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.693899\n",
      "  0:01:56.577     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 29\n",
      "  0:01:56.577     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 29\n",
      "  0:01:56.577     5M / 163M  INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 4507034\n",
      "  0:01:56.577     5M / 163M  INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 63.4837\n",
      "  0:01:56.577     5M / 163M  INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 29\n",
      "  0:01:56.577     5M / 163M  INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:01:56.577     5M / 163M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:01:56.577     5M / 163M  INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:01:56.577     5M / 163M  INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:01:56.577     5M / 163M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:01:56.578     5M / 163M  INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:01:56.578     5M / 163M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n",
      "  0:01:56.607     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 2699 times\n",
      "  0:01:56.607     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n",
      "  0:01:56.646     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 1350 times\n",
      "  0:01:56.647     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:01:56.649     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 1603 times\n",
      "  0:01:56.649     5M / 164M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:01:56.649     5M / 164M  INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:01:56.649     5M / 164M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:01:56.649     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:01:56.649     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:56.650     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 59 times\n",
      "  0:01:56.650     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:56.718     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 1018 times\n",
      "  0:01:56.718     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:56.720     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 42 times\n",
      "  0:01:56.720     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:01:56.720     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:56.721     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 5 times\n",
      "  0:01:56.721     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:56.721     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 17 times\n",
      "  0:01:56.721     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:56.723     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 52 times\n",
      "  0:01:56.723     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:01:56.723     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:56.723     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 1 times\n",
      "  0:01:56.723     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:56.725     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 27 times\n",
      "  0:01:56.725     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:56.725     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 10 times\n",
      "  0:01:56.725     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:01:56.725     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:56.725     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:56.725     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:56.726     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 9 times\n",
      "  0:01:56.726     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:56.726     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 8 times\n",
      "  0:01:56.726     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:01:56.726     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:56.726     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:56.726     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:56.726     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 5 times\n",
      "  0:01:56.726     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 6 times\n",
      "  0:01:56.727     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:01:56.727     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 3 times\n",
      "  0:01:56.727     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:01:56.727     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:01:56.727     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:01:56.727     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:56.727     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:56.728     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 2 times\n",
      "  0:01:56.728     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:56.729     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:01:56.729     5M / 164M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 12\n",
      "  0:01:56.729     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:56.729     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:56.729     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:56.729     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:56.729     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:01:56.729     5M / 164M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:01:56.729     5M / 164M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
      "  0:01:56.729     5M / 165M  INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\n",
      "  0:01:56.729     5M / 165M  INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\n",
      "  0:01:56.729     5M / 165M  INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\n",
      "  0:01:56.729     5M / 165M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:01:56.729     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:56.729     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:56.729     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:56.730     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:56.730     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:01:56.730     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:01:56.730     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:01:56.730     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:01:56.730     5M / 165M  INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:01:56.731     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:01:56.731     5M / 165M  INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 5 times\n",
      "  0:01:56.731     5M / 165M  INFO    General                 (simplification.cpp        : 491)   Counting average coverage\n",
      "  0:01:56.731     5M / 165M  INFO    General                 (simplification.cpp        : 497)   Average coverage = 64.7522\n",
      "  0:01:56.731     5M / 165M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:01:56.732     5M / 165M  INFO    General                 (read_converter.cpp        : 102)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K33/simplified_contigs\n",
      "  0:01:56.736     7M / 165M  INFO    General                 (binary_converter.cpp      : 111)   1188 reads written\n",
      "  0:01:56.736     5M / 165M  INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  0:01:56.737     1M / 165M  INFO    General                 (main.cpp                  : 136)   Assembling time: 0 hours 1 minutes 56 seconds\n",
      "\n",
      "===== K33 finished. \n",
      "\n",
      "\n",
      "===== K55 started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/Downloads/SPAdes-3.15.2/bin/spades-core /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/configs/config.info\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  :  99)   Loaded config from /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/configs/config.info\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (memory_limit.cpp          :  48)   Memory limit set to 250 Gb\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 107)   Starting SPAdes, built from N/A, git revision N/A\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 108)   Maximum k-mer length: 128\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 109)   Assembling dataset (/Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/dataset.info) with K=55\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (main.cpp                  : 110)   Maximum # of threads to use (adjusted due to OMP capabilities): 1\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 212)   SPAdes started\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 225)   Starting from stage: read_conversion\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 231)   Two-step repeat resolution disabled\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 672)   Graph created, vertex min_id: 3, edge min_id: 3\n",
      "  0:00:00.000     1M / 4M    INFO   GraphCore                (graph_core.hpp            : 673)   Vertex size: 48, edge size: 40\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (pipeline.cpp              : 242)   Will need read mapping, kmer mapper will be attached\n",
      "  0:00:00.000     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == Binary Read Conversion (id: read_conversion)\n",
      "  0:00:00.000     1M / 4M    INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (read_converter.cpp        :  53)   Binary reads detected\n",
      "  0:00:00.001     1M / 4M    INFO   StageManager             (stage.cpp                 : 185)   STAGE == de Bruijn graph construction (id: construction)\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (construction.cpp          : 118)   Contigs from previous K will be used: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K33/simplified_contigs\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (construction.cpp          : 153)   Max read length 100\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (construction.cpp          : 159)   Average read length 92.7147\n",
      "  0:00:00.001     1M / 4M    INFO    General                 (stage.cpp                 : 117)   PROCEDURE == k+1-mer counting (id: construction:kpomer_counting)\n",
      "  0:00:00.002     1M / 4M    INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:00.002     1M / 4M    INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:00.002     1M / 4M    INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:00.002     1M / 4M    INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 3355443\n",
      "  0:00:05.912   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 1768213 reads\n",
      "  0:00:11.449   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 3518079 reads\n",
      "  0:00:17.078   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 5268519 reads\n",
      "  0:00:22.641   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 7021543 reads\n",
      "  0:00:28.188   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 8774696 reads\n",
      "  0:00:32.487   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 125)   Processed 9894278 reads\n",
      "  0:00:32.487     1M / 158M  INFO    General                 (kmer_splitters.hpp        : 131)   Used 9894278 reads\n",
      "  0:00:32.525     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:34.092     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5692221 kmers in total.\n",
      "  0:00:34.092     1M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Extension index construction (id: construction:extension_index_construction)\n",
      "  0:00:34.094     1M / 158M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 438)   Building kmer index\n",
      "  0:00:34.094     1M / 158M  INFO    General                 (kmer_index_builder.hpp    : 243)   Splitting kmer instances into 10 files using 1 threads. This might take a while.\n",
      "  0:00:34.094     1M / 158M  INFO    General                 (file_limit.hpp            :  32)   Open file limit set to 2560\n",
      "  0:00:34.094     1M / 158M  INFO    General                 (kmer_splitter.hpp         :  93)   Memory available for splitting buffers: 83.3333 Gb\n",
      "  0:00:34.094     1M / 158M  INFO    General                 (kmer_splitter.hpp         : 101)   Using cell size of 3355443\n",
      "  0:00:36.228   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 194)   Processed 5692221 kmers\n",
      "  0:00:36.228   601M / 601M  INFO    General                 (kmer_splitters.hpp        : 199)   Used 5692221 kmers.\n",
      "  0:00:36.228     1M / 245M  INFO    General                 (kmer_index_builder.hpp    : 249)   Starting k-mer counting.\n",
      "  0:00:36.543     1M / 245M  INFO    General                 (kmer_index_builder.hpp    : 260)   K-mer counting done. There are 5698724 kmers in total.\n",
      "  0:00:36.543     1M / 245M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:37.054     5M / 245M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5698724 kmers, 4123376 bytes occupied (5.78849 bits per kmer).\n",
      "  0:00:37.054     5M / 245M  INFO    General                 (kmer_index_builder.hpp    : 169)   Merging final buckets.\n",
      "  0:00:37.320    11M / 245M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 100)   Building k-mer extensions from k+1-mers\n",
      "  0:00:39.196    11M / 245M  INFO   DeBruijnExtensionIndexBu (kmer_extension_index_build: 105)   Building k-mer extensions from k+1-mers finished.\n",
      "  0:00:39.198    11M / 158M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Condensing graph (id: construction:graph_condensing)\n",
      "  0:00:39.206    11M / 158M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 354)   Extracting unbranching paths\n",
      "  0:00:42.114    23M / 248M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 373)   Extracting unbranching paths finished. 247400 sequences extracted\n",
      "  0:00:43.200    23M / 161M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 309)   Collecting perfect loops\n",
      "  0:00:43.613    23M / 248M  INFO   UnbranchingPathExtractor (debruijn_graph_constructor: 342)   Collecting perfect loops finished. 0 loops collected\n",
      "  0:00:43.619    23M / 161M  INFO    General                 (debruijn_graph_constructor: 487)   Total 494800 edges to create\n",
      "  0:00:43.619    42M / 161M  INFO    General                 (debruijn_graph_constructor: 489)   Collecting link records\n",
      "  0:00:43.765    50M / 172M  INFO    General                 (debruijn_graph_constructor: 491)   Ordering link records\n",
      "  0:00:43.797    50M / 172M  INFO    General                 (debruijn_graph_constructor: 493)   Sorting done\n",
      "  0:00:43.799    52M / 172M  INFO    General                 (debruijn_graph_constructor: 503)   Total 253903 vertices to create\n",
      "  0:00:43.799    76M / 172M  INFO    General                 (debruijn_graph_constructor: 506)   Connecting the graph\n",
      "  0:00:43.844    62M / 172M  INFO    General                 (stage.cpp                 : 117)   PROCEDURE == Filling coverage indices (PHM) (id: construction:coverage_filling_phm)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:00:43.844    62M / 172M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:00:44.363    66M / 172M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 5692221 kmers, 4118704 bytes occupied (5.78854 bits per kmer).\n",
      "  0:00:44.364    88M / 172M  INFO    General                 (coverage_hash_map_builder.:  47)   Collecting k-mer coverage information from reads, this takes a while.\n",
      "  0:01:18.757    88M / 172M  INFO    General                 (construction.cpp          : 430)   Filling coverage and flanking coverage from PHM\n",
      "  0:01:20.907    88M / 172M  INFO    General                 (coverage_filling.hpp      :  82)   Processed 494800 edges\n",
      "  0:01:20.927    52M / 172M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == EC Threshold Finding (id: ec_threshold_finder)\n",
      "  0:01:20.927    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 181)   Kmer coverage valley at: 6\n",
      "  0:01:20.927    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 201)   K-mer histogram maximum: 41\n",
      "  0:01:20.927    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 237)   Estimated median coverage: 41. Coverage mad: 11.8608\n",
      "  0:01:20.927    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 259)   Fitting coverage model\n",
      "  0:01:20.963    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 2\n",
      "  0:01:21.052    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 4\n",
      "  0:01:21.342    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 8\n",
      "  0:01:22.025    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 295)   ... iteration 16\n",
      "  0:01:22.513    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 309)   Fitted mean coverage: 40.4747. Fitted coverage std. dev: 10.9026\n",
      "  0:01:22.514    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 334)   Probability of erroneous kmer at valley: 0.512008\n",
      "  0:01:22.514    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 358)   Preliminary threshold calculated as: 16\n",
      "  0:01:22.514    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 362)   Threshold adjusted to: 16\n",
      "  0:01:22.514    52M / 172M  INFO    General                 (kmer_coverage_model.cpp   : 375)   Estimated genome size (ignoring repeats): 4487959\n",
      "  0:01:22.514    52M / 172M  INFO    General                 (genomic_info_filler.cpp   :  55)   Mean coverage was calculated as 40.4747\n",
      "  0:01:22.514    52M / 172M  INFO    General                 (genomic_info_filler.cpp   :  70)   EC coverage threshold value was calculated as 16\n",
      "  0:01:22.514    52M / 172M  INFO    General                 (genomic_info_filler.cpp   :  71)   Trusted kmer low bound: 0\n",
      "  0:01:22.514    52M / 172M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Gap Closer (id: early_gapcloser)\n",
      "  0:01:22.518    61M / 172M  INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:01:22.611    66M / 172M  INFO    General                 (gap_closer.cpp            : 101)   Total edges in tip neighborhood: 134528 out of 494800, length: 1226132\n",
      "  0:01:22.620    68M / 172M  INFO    General                 (edge_index.hpp            : 167)   Using small index (max_id = 497276)\n",
      "  0:01:22.725    68M / 172M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:01:23.185    69M / 172M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 1092761 kmers, 790000 bytes occupied (5.78352 bits per kmer).\n",
      "  0:01:23.186    78M / 172M  INFO    General                 (edge_index_builders.hpp   : 265)   Collecting edge information from graph, this takes a while.\n",
      "  0:01:27.983    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:01:32.453    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:01:36.939    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:01:41.408    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:01:45.864    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:01:50.324    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:02:12.591    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:02:18.156    76M / 172M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:02:18.162    62M / 172M  INFO    General                 (gap_closer.cpp            : 452)   Initializing gap closer\n",
      "  0:02:18.162    62M / 172M  INFO   GapCloser                (gap_closer.cpp            : 359)   Closing short gaps\n",
      "  0:02:18.377    62M / 178M  INFO   GapCloser                (gap_closer.cpp            : 393)   Closing short gaps complete: filled 0 gaps after checking 20 candidates\n",
      "  0:02:18.383    62M / 178M  INFO    General                 (gap_closer.cpp            : 456)   Gap closer done\n",
      "  0:02:18.384    52M / 178M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Raw Simplification (id: raw_simplification)\n",
      "  0:02:18.384    52M / 178M  INFO    General                 (simplification.cpp        : 127)   PROCEDURE == Initial cleaning\n",
      "  0:02:18.385    52M / 178M  INFO    General                 (graph_simplification.hpp  : 662)   Flanking coverage based disconnection disabled\n",
      "  0:02:18.385    52M / 178M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Self conjugate edge remover\n",
      "  0:02:18.413    52M / 178M  INFO   Simplification           (parallel_processing.hpp   : 170)   Self conjugate edge remover triggered 0 times\n",
      "  0:02:18.413    52M / 178M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial tip clipper\n",
      "  0:02:19.166    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial tip clipper triggered 119766 times\n",
      "  0:02:19.166    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial ec remover\n",
      "  0:02:19.173    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial ec remover triggered 92 times\n",
      "  0:02:19.173    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Initial isolated edge remover\n",
      "  0:02:19.182    46M / 179M  INFO   Simplification           (parallel_processing.hpp   : 170)   Initial isolated edge remover triggered 7236 times\n",
      "  0:02:19.182    45M / 179M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification (id: simplification)\n",
      "  0:02:19.182    45M / 179M  INFO    General                 (simplification.cpp        : 368)   Graph simplification started\n",
      "  0:02:19.182    45M / 179M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:02:19.182    45M / 179M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 1\n",
      "  0:02:19.182    45M / 179M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.183    45M / 179M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 19 times\n",
      "  0:02:19.183    45M / 179M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.278    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 447 times\n",
      "  0:02:19.278    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.279    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:19.279    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 2\n",
      "  0:02:19.279    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.279    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 3 times\n",
      "  0:02:19.279    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.279    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:19.279    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.280    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 13 times\n",
      "  0:02:19.280    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 3\n",
      "  0:02:19.280    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.280    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 1 times\n",
      "  0:02:19.280    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.281    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 8 times\n",
      "  0:02:19.281    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.282    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 11 times\n",
      "  0:02:19.282    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 4\n",
      "  0:02:19.282    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.282    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 3 times\n",
      "  0:02:19.282    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.283    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 7 times\n",
      "  0:02:19.283    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.283    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:02:19.283    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 5\n",
      "  0:02:19.283    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.283    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:19.283    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.283    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:19.283    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.283    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 3 times\n",
      "  0:02:19.283    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 6\n",
      "  0:02:19.283    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.283    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 1 times\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:02:19.284    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 7\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:02:19.284    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 8\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:19.284    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 9\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:19.284    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.285    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 1 times\n",
      "  0:02:19.285    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 10\n",
      "  0:02:19.285    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.285    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:19.285    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.285    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:19.285    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.285    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 2 times\n",
      "  0:02:19.285    48M / 180M  INFO    General                 (simplification.cpp        : 373)   PROCEDURE == Simplification cycle, iteration 11\n",
      "  0:02:19.285    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:19.286    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:19.286    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:19.287    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:19.287    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Low coverage edge remover\n",
      "  0:02:19.288    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Low coverage edge remover triggered 0 times\n",
      "  0:02:19.288    48M / 180M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Gap Closer (id: late_gapcloser)\n",
      "  0:02:19.290    57M / 180M  INFO    General                 (edge_index.hpp            : 115)   Size of edge index entries: 12/8\n",
      "  0:02:19.291    57M / 180M  INFO    General                 (gap_closer.cpp            : 101)   Total edges in tip neighborhood: 15 out of 882, length: 167704\n",
      "  0:02:19.291    57M / 180M  INFO    General                 (edge_index.hpp            : 167)   Using small index (max_id = 497276)\n",
      "  0:02:19.296    57M / 180M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:02:19.327    57M / 180M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 166932 kmers, 121312 bytes occupied (5.81372 bits per kmer).\n",
      "  0:02:19.327    58M / 180M  INFO    General                 (edge_index_builders.hpp   : 265)   Collecting edge information from graph, this takes a while.\n",
      "  0:02:21.895    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:02:24.431    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:02:26.917    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:02:29.476    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:02:32.112    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:02:34.674    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:02:47.181    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:02:50.289    58M / 180M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  0:02:50.291    57M / 180M  INFO    General                 (gap_closer.cpp            : 452)   Initializing gap closer\n",
      "  0:02:50.292    57M / 180M  INFO   GapCloser                (gap_closer.cpp            : 359)   Closing short gaps\n",
      "  0:02:50.293    57M / 180M  INFO   GapCloser                (gap_closer.cpp            : 393)   Closing short gaps complete: filled 1 gaps after checking 7 candidates\n",
      "  0:02:50.294    57M / 180M  INFO    General                 (gap_closer.cpp            : 456)   Gap closer done\n",
      "  0:02:50.295    48M / 180M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Simplification Cleanup (id: simplification_cleanup)\n",
      "  0:02:50.295    48M / 180M  INFO    General                 (simplification.cpp        : 176)   PROCEDURE == Post simplification\n",
      "  0:02:50.295    48M / 180M  INFO    General                 (graph_simplification.hpp  : 448)   Disconnection of relatively low covered edges disabled\n",
      "  0:02:50.295    48M / 180M  INFO    General                 (graph_simplification.hpp  : 485)   Complex tip clipping disabled\n",
      "  0:02:50.295    48M / 180M  INFO    General                 (graph_simplification.hpp  : 634)   Creating parallel br instance\n",
      "  0:02:50.295    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:50.296    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:50.296    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:50.297    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:50.297    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Tip clipper\n",
      "  0:02:50.297    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Tip clipper triggered 0 times\n",
      "  0:02:50.297    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Bulge remover\n",
      "  0:02:50.298    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Bulge remover triggered 0 times\n",
      "  0:02:50.298    48M / 180M  INFO    General                 (simplification.cpp        : 327)   Disrupting self-conjugate edges\n",
      "  0:02:50.299    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 167)   Running Removing isolated edges\n",
      "  0:02:50.300    48M / 180M  INFO   Simplification           (parallel_processing.hpp   : 170)   Removing isolated edges triggered 1 times\n",
      "  0:02:50.300    48M / 180M  INFO    General                 (simplification.cpp        : 491)   Counting average coverage\n",
      "  0:02:50.300    48M / 180M  INFO    General                 (simplification.cpp        : 497)   Average coverage = 41.1665\n",
      "  0:02:50.300    48M / 180M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Mismatch Correction (id: mismatch_correction)\n",
      "  0:02:50.300    48M / 180M  INFO    General                 (graph_pack.cpp            :  67)   Index refill\n",
      "  0:02:50.301    48M / 180M  INFO    General                 (edge_index.hpp            : 156)   Using small index (max_id = 497276)\n",
      "  0:02:50.468    48M / 180M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 395)   Building perfect hash indices\n",
      "  0:02:51.499    51M / 180M  INFO   K-mer Index Building     (kmer_index_builder.hpp    : 431)   Index built. Total 4555847 kmers, 3291264 bytes occupied (5.77941 bits per kmer).\n",
      "  0:02:51.504    87M / 185M  INFO    General                 (edge_index_builders.hpp   : 252)   Collecting edge information from graph, this takes a while.\n",
      "  0:02:52.383    87M / 185M  INFO    General                 (graph_pack.cpp            :  77)   Normalizing k-mer map. Total 56128 kmers to process\n",
      "  0:02:52.398    87M / 185M  INFO    General                 (graph_pack.cpp            :  79)   Normalizing done\n",
      "  0:02:52.398    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 400)   Collect potential mismatches\n",
      "  0:02:52.455    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 193)   Total 354 edges (out of 878) with 1752 potential mismatch positions (4.94915 positions per edge)\n",
      "  0:02:52.455    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 402)   Potential mismatches collected\n",
      "  0:02:52.862    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  0:02:53.294    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  0:02:53.722    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  0:02:54.175    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  0:02:54.619    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  0:02:55.045    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  0:02:57.169    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  0:03:01.401    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 4200000 reads\n",
      "  0:03:10.311    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 8400000 reads\n",
      "  0:03:13.480    87M / 185M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 9891902 reads processed\n",
      "  0:03:13.513    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 395)   All edges processed\n",
      "  0:03:13.514    87M / 185M  INFO    General                 (mismatch_correction.cpp   : 450)   Corrected 3 nucleotides\n",
      "  0:03:13.514    87M / 185M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  0:03:13.514    87M / 185M  INFO    General                 (contig_output.hpp         :  16)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/before_rr.fasta\n",
      "  0:03:13.545    87M / 186M  INFO    General                 (contig_output_stage.cpp   : 151)   Writing GFA graph to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/assembly_graph_after_simplification.gfa\n",
      "  0:03:13.571    87M / 186M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Hybrid Aligning (id: hybrid_aligning)\n",
      "  0:03:13.571    87M / 186M  INFO   HybridAligning           (hybrid_aligning.cpp       : 341)   Hybrid library detected: #1\n",
      "  0:03:13.571    87M / 186M  INFO    General                 (hybrid_aligning.cpp       : 294)   Aligning long reads with bwa-mem based aligner\n",
      "  0:03:13.647   100M / 186M  INFO    General                 (bwa_index.cpp             : 173)   Using BWA IS algorithm\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  0:03:16.792   652M / 696M  INFO    General                 (hybrid_aligning.cpp       : 274)   Prepared batch 0 of 50000 reads.\n",
      "  0:31:33.471   653M / 699M  INFO    General                 (hybrid_aligning.cpp       : 239)   Read batch of size: 50000 processed; 48358 of them longer than 500; among long reads aligned: 47606; paths of more than one edge received: 2791\n",
      "  0:31:33.473   653M / 699M  INFO    General                 (hybrid_aligning.cpp       : 279)   Processed 50000 reads\n",
      "  0:31:34.535   467M / 751M  INFO    General                 (hybrid_aligning.cpp       : 274)   Prepared batch 1 of 32783 reads.\n",
      "  1:45:27.162   467M / 751M  INFO    General                 (hybrid_aligning.cpp       : 239)   Read batch of size: 32783 processed; 31706 of them longer than 500; among long reads aligned: 31186; paths of more than one edge received: 1850\n",
      "  1:45:27.165   467M / 751M  INFO    General                 (hybrid_aligning.cpp       : 279)   Processed 82783 reads\n",
      "  1:45:27.193    97M / 751M  INFO    General                 (hybrid_aligning.cpp       : 308)   For library of long reads\n",
      "  1:45:27.194    97M / 751M  INFO   StatsCounter             (pacbio_read_structures.hpp: 163)   Median fraction of present seeds in maximal alignmnent among reads aligned to the graph: 0\n",
      "  1:45:27.194    97M / 751M  INFO    General                 (hybrid_aligning.cpp       : 310)   Aligning of long reads finished\n",
      "  1:45:27.199    87M / 751M  INFO   HybridAligning           (hybrid_aligning.cpp       : 388)   Padding gaps\n",
      "  1:45:27.200    87M / 751M  INFO   HybridAligning           (hybrid_aligning.cpp       : 392)   Min gap weight set to 2\n",
      "  1:45:27.204    87M / 751M  INFO    General                 (hybrid_aligning.cpp       : 146)   Closing gaps with long reads\n",
      "  1:45:27.703    88M / 751M  INFO   MultiGapJoiner           (hybrid_gap_closer.hpp     : 531)   Closed 3 gaps\n",
      "  1:45:27.926    88M / 751M  INFO    General                 (hybrid_aligning.cpp       : 167)   Closing gaps with long reads finished\n",
      "  1:45:27.927    87M / 751M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Paired Information Counting (id: late_pair_info_count)\n",
      "  1:45:27.952    87M / 751M  INFO    General                 (graph_pack.cpp            :  77)   Normalizing k-mer map. Total 56246 kmers to process\n",
      "  1:45:28.078    87M / 751M  INFO    General                 (graph_pack.cpp            :  79)   Normalizing done\n",
      "  1:45:28.084    87M / 751M  INFO    General                 (pair_info_count.cpp       : 339)   Min edge length for estimation: 61753\n",
      "  1:45:28.084    87M / 751M  INFO    General                 (pair_info_count.cpp       : 350)   Estimating insert size for library #0\n",
      "  1:45:28.084    87M / 751M  INFO    General                 (pair_info_count.cpp       : 202)   Estimating insert size (takes a while)\n",
      "  1:45:28.736   120M / 753M  INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  1:45:33.721   120M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  1:45:38.827   120M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  1:45:43.893   120M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  1:45:48.858   120M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  1:45:53.858   120M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  1:45:58.809   120M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  1:47:40.818   120M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  1:47:41.672   120M / 753M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  1:47:41.717   120M / 753M  INFO    General                 (pair_info_count.cpp       : 220)   Edge pairs: 3658\n",
      "  1:47:41.717   120M / 753M  INFO    General                 (pair_info_count.cpp       : 224)   1128696 paired reads (46.0942% of all) aligned to long edges\n",
      "  1:47:41.717    87M / 753M  INFO    General                 (pair_info_count.cpp       : 373)     Insert size = 215.4, deviation = 10.1891, left quantile = 202, right quantile = 229, read length = 100\n",
      "  1:47:41.717    88M / 753M  INFO    General                 (pair_info_count.cpp       : 390)   Filtering data for library #0\n",
      "  1:47:41.717    88M / 753M  INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  1:47:42.364    88M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  1:47:42.999    88M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  1:47:43.650    88M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  1:47:44.309    88M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  1:47:44.962    88M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  1:47:45.603    88M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  1:47:48.756    88M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  1:47:53.069    88M / 753M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  1:47:53.069    88M / 753M  INFO    General                 (pair_info_count.cpp       : 402)   Mapping library #0\n",
      "  1:47:53.070    88M / 753M  INFO    General                 (pair_info_count.cpp       : 404)   Mapping paired reads (takes a while)\n",
      "  1:47:53.070    88M / 753M  INFO    General                 (pair_info_count.cpp       : 302)   Left insert size quantile 202, right insert size quantile 229, filtering threshold 2, rounding threshold 0\n",
      "  1:47:53.084    97M / 753M  INFO    General                 (pair_info_count.cpp       :  48)   Selecting usual mapper\n",
      "  1:47:58.516    97M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 200000 reads\n",
      "  1:48:06.088    97M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 400000 reads\n",
      "  1:48:11.862    97M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 600000 reads\n",
      "  1:48:19.135    97M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 800000 reads\n",
      "  1:48:24.769    97M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1000000 reads\n",
      "  1:48:30.580    97M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 1200000 reads\n",
      "  1:48:43.429    97M / 753M  INFO    General                 (sequence_mapper_notifier.h:  77)   Processed 2200000 reads\n",
      "  1:48:45.008    97M / 753M  INFO    General                 (sequence_mapper_notifier.h:  95)   Total 2448672 reads processed\n",
      "  1:48:45.013    88M / 753M  INFO    General                 (pair_info_count.cpp       : 343)   Library #1 was mapped earlier on hybrid aligning stage, skipping\n",
      "  1:48:45.013    88M / 753M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Distance Estimation (id: distance_estimation)\n",
      "  1:48:45.013    88M / 753M  INFO    General                 (distance_estimation.cpp   : 179)   Processing library #0\n",
      "  1:48:45.013    88M / 753M  INFO    General                 (distance_estimation.cpp   : 150)   Weight Filter Done\n",
      "  1:48:45.015    88M / 753M  INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SIMPLE distance estimator\n",
      "  1:48:45.025    88M / 753M  INFO    General                 (distance_estimation.cpp   :  37)   Filtering info\n",
      "  1:48:45.025    88M / 753M  INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; library index size: 3698\n",
      "  1:48:45.026    88M / 753M  INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering; library index size: 3228\n",
      "  1:48:45.027    88M / 753M  INFO    General                 (distance_estimation.cpp   : 156)   Refining clustered pair information\n",
      "  1:48:45.028    88M / 753M  INFO    General                 (distance_estimation.cpp   : 158)   The refining of clustered pair information has been finished\n",
      "  1:48:45.028    88M / 753M  INFO    General                 (distance_estimation.cpp   : 160)   Improving paired information\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  1:48:45.040    88M / 753M  INFO   PairInfoImprover         (pair_info_improver.hpp    : 105)   Paired info stats: missing = 202; contradictional = 8\n",
      "  1:48:45.053    88M / 753M  INFO   PairInfoImprover         (pair_info_improver.hpp    : 105)   Paired info stats: missing = 0; contradictional = 0\n",
      "  1:48:45.053    88M / 753M  INFO    General                 (distance_estimation.cpp   : 104)   Filling scaffolding index\n",
      "  1:48:45.054    88M / 753M  INFO   DistanceEstimator        (distance_estimation.hpp   : 116)   Using SMOOTHING distance estimator\n",
      "  1:48:45.274    88M / 753M  INFO    General                 (distance_estimation.cpp   :  37)   Filtering info\n",
      "  1:48:45.274    88M / 753M  INFO    General                 (pair_info_filters.hpp     : 242)   Start filtering; library index size: 879\n",
      "  1:48:45.274    88M / 753M  INFO    General                 (pair_info_filters.hpp     : 263)   Done filtering; library index size: 879\n",
      "  1:48:45.275    88M / 753M  INFO    General                 (distance_estimation.cpp   : 186)   Clearing raw paired index\n",
      "  1:48:45.275    88M / 753M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Repeat Resolving (id: repeat_resolving)\n",
      "  1:48:45.275    88M / 753M  INFO    General                 (repeat_resolving.cpp      :  87)   Using Path-Extend repeat resolving\n",
      "  1:48:45.276    88M / 753M  INFO    General                 (launcher.cpp              : 600)   ExSPAnder repeat resolving tool started\n",
      "  1:48:45.280    88M / 753M  INFO    General                 (coverage_uniformity_analyz:  25)   genomic coverage is 40.29 calculated of length 4525458\n",
      "  1:48:45.281    88M / 753M  INFO    General                 (launcher.cpp              : 186)   Autodetecting unique edge set parameters...\n",
      "  1:48:45.281    88M / 753M  INFO    General                 (launcher.cpp              : 188)   Minimal unique edge length set to the smallest MP library IS: 2000\n",
      "  1:48:45.282    88M / 753M  INFO    General                 (coverage_uniformity_analyz:  25)   genomic coverage is 40.29 calculated of length 4503452\n",
      "  1:48:45.283    88M / 753M  INFO    General                 (launcher.cpp              : 194)   median coverage for edges longer than 2000 is 40.29 uniformity 99%\n",
      "  1:48:45.284    88M / 753M  INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  69)   With length cutoff: 2000, median long edge coverage: 40.29, and maximal unique coverage: 0.5\n",
      "  1:48:45.284    88M / 753M  INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  70)   Unique edges quantity: 230, unique edges length 9000960, total edges length 9111882\n",
      "  1:48:45.288    97M / 753M  INFO    General                 (launcher.cpp              : 418)   Creating main extenders, unique edge length = 2000\n",
      "  1:48:45.288    97M / 753M  INFO    General                 (launcher.cpp              : 342)   filling path container\n",
      "  1:48:45.291   107M / 754M  INFO    General                 (extenders_logic.cpp       : 334)   Estimated coverage of library #0 is 41.1665\n",
      "  1:48:45.293   107M / 754M  INFO    General                 (extenders_logic.cpp       : 345)   Creating extender; library index size: 3624\n",
      "  1:48:45.293   107M / 754M  INFO    General                 (extenders_logic.cpp       : 334)   Estimated coverage of library #0 is 41.1665\n",
      "  1:48:45.293   107M / 754M  INFO    General                 (extenders_logic.cpp       : 345)   Creating extender; library index size: 3624\n",
      "  1:48:45.294   107M / 754M  INFO    General                 (extenders_logic.cpp       :  47)   resolvable_repeat_length_bound set to 22609\n",
      "  1:48:45.304   107M / 754M  INFO    General                 (extenders_logic.cpp       : 543)   Using 1 paired-end library\n",
      "  1:48:45.304   107M / 754M  INFO    General                 (extenders_logic.cpp       : 544)   Using 1 paired-end scaffolding library\n",
      "  1:48:45.304   107M / 754M  INFO    General                 (extenders_logic.cpp       : 545)   Using 1 single read library\n",
      "  1:48:45.305   107M / 754M  INFO    General                 (launcher.cpp              : 385)   Filling backbone edges for long reads scaffolding...\n",
      "  1:48:45.305   107M / 754M  INFO    General                 (launcher.cpp              : 404)    with coverage\n",
      "  1:48:45.306   107M / 754M  INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  69)   With length cutoff: 500, median long edge coverage: 40.29, and maximal unique coverage: 0.5\n",
      "  1:48:45.306   107M / 754M  INFO   ScaffoldingUniqueEdgeAna (scaff_supplementary.cpp   :  70)   Unique edges quantity: 258, unique edges length 9034110, total edges length 9111882\n",
      "  1:48:45.306   107M / 754M  INFO    General                 (launcher.cpp              : 407)   258 unique edges\n",
      "  1:48:45.306   107M / 754M  INFO    General                 (extenders_logic.cpp       : 420)   Creating scaffolding extender for lib 1\n",
      "  1:48:45.307   107M / 754M  INFO   ExtensionChooser2015     (extension_chooser2015.hpp :  50)   ExtensionChooser2015 created\n",
      "  1:48:45.308   107M / 754M  INFO    General                 (extenders_logic.cpp       : 446)   Using 1 long reads scaffolding library\n",
      "  1:48:45.308   107M / 754M  INFO    General                 (launcher.cpp              : 447)   Total number of extenders is 5\n",
      "  1:48:45.308   107M / 754M  INFO    General                 (path_extenders.cpp        :  36)   Processed 0 paths from 391 (0%)\n",
      "  1:48:45.323   109M / 754M  INFO    General                 (path_extenders.cpp        :  36)   Processed 40 paths from 391 (10%)\n",
      "  1:48:45.329   110M / 754M  INFO    General                 (path_extenders.cpp        :  36)   Processed 80 paths from 391 (20%)\n",
      "  1:48:45.331   110M / 754M  INFO    General                 (path_extenders.cpp        :  36)   Processed 120 paths from 391 (30%)\n",
      "  1:48:45.332   111M / 754M  INFO    General                 (path_extenders.cpp        :  34)   Processed 128 paths from 391 (32%)\n",
      "  1:48:45.333   111M / 754M  INFO    General                 (path_extenders.cpp        :  36)   Processed 160 paths from 391 (40%)\n",
      "  1:48:45.334   111M / 754M  INFO    General                 (path_extenders.cpp        :  36)   Processed 200 paths from 391 (51%)\n",
      "  1:48:45.336   112M / 754M  INFO    General                 (path_extenders.cpp        :  36)   Processed 240 paths from 391 (61%)\n",
      "  1:48:45.337   112M / 754M  INFO    General                 (path_extenders.cpp        :  34)   Processed 256 paths from 391 (65%)\n",
      "  1:48:45.337   112M / 754M  INFO    General                 (path_extenders.cpp        :  36)   Processed 280 paths from 391 (71%)\n",
      "  1:48:45.337   113M / 754M  INFO    General                 (path_extenders.cpp        :  36)   Processed 320 paths from 391 (81%)\n",
      "  1:48:45.338   113M / 754M  INFO    General                 (path_extenders.cpp        :  36)   Processed 360 paths from 391 (92%)\n",
      "  1:48:45.338   113M / 754M  INFO    General                 (launcher.cpp              : 250)   Finalizing paths\n",
      "  1:48:45.338   113M / 754M  INFO    General                 (launcher.cpp              : 252)   Deduplicating paths\n",
      "  1:48:45.339   112M / 754M  INFO    General                 (launcher.cpp              : 256)   Paths deduplicated\n",
      "  1:48:45.339   112M / 754M  INFO   PEResolver               (pe_resolver.cpp           :  60)   Removing overlaps\n",
      "  1:48:45.339   112M / 754M  INFO   PEResolver               (pe_resolver.cpp           :  63)   Sorting paths\n",
      "  1:48:45.339   112M / 754M  INFO   PEResolver               (pe_resolver.cpp           :  70)   Marking overlaps\n",
      "  1:48:45.339   112M / 754M  INFO   OverlapRemover           (overlap_remover.hpp       : 116)   Marking start/end overlaps\n",
      "  1:48:45.341   112M / 754M  INFO   OverlapRemover           (overlap_remover.hpp       : 119)   Marking remaining overlaps\n",
      "  1:48:45.342   112M / 754M  INFO   PEResolver               (pe_resolver.cpp           :  73)   Splitting paths\n",
      "  1:48:45.343   117M / 754M  INFO   PEResolver               (pe_resolver.cpp           :  78)   Deduplicating paths\n",
      "  1:48:45.344   111M / 754M  INFO   PEResolver               (pe_resolver.cpp           :  80)   Overlaps removed\n",
      "  1:48:45.345   112M / 754M  INFO    General                 (launcher.cpp              : 273)   Paths finalized\n",
      "  1:48:45.345   112M / 754M  INFO    General                 (launcher.cpp              : 454)   Closing gaps in paths\n",
      "  1:48:45.345   115M / 754M  INFO    General                 (launcher.cpp              : 484)   Gap closing completed\n",
      "  1:48:45.346   115M / 754M  INFO    General                 (launcher.cpp              : 302)   Traversing tandem repeats\n",
      "  1:48:45.349   115M / 754M  INFO    General                 (launcher.cpp              : 312)   Traversed 5 loops\n",
      "  1:48:45.349   115M / 754M  INFO    General                 (launcher.cpp              : 250)   Finalizing paths\n",
      "  1:48:45.349   115M / 754M  INFO    General                 (launcher.cpp              : 252)   Deduplicating paths\n",
      "  1:48:45.349   115M / 754M  INFO    General                 (launcher.cpp              : 256)   Paths deduplicated\n",
      "  1:48:45.349   115M / 754M  INFO   PEResolver               (pe_resolver.cpp           :  60)   Removing overlaps\n",
      "  1:48:45.349   115M / 754M  INFO   PEResolver               (pe_resolver.cpp           :  63)   Sorting paths\n",
      "  1:48:45.349   115M / 754M  INFO   PEResolver               (pe_resolver.cpp           :  70)   Marking overlaps\n",
      "  1:48:45.349   115M / 754M  INFO   OverlapRemover           (overlap_remover.hpp       : 116)   Marking start/end overlaps\n",
      "  1:48:45.349   115M / 754M  INFO   OverlapRemover           (overlap_remover.hpp       : 119)   Marking remaining overlaps\n",
      "  1:48:45.350   115M / 754M  INFO   PEResolver               (pe_resolver.cpp           :  73)   Splitting paths\n",
      "  1:48:45.350   115M / 754M  INFO   PEResolver               (pe_resolver.cpp           :  78)   Deduplicating paths\n",
      "  1:48:45.350   115M / 754M  INFO   PEResolver               (pe_resolver.cpp           :  80)   Overlaps removed\n",
      "  1:48:45.350   115M / 754M  INFO    General                 (launcher.cpp              : 273)   Paths finalized\n",
      "  1:48:45.350   115M / 754M  INFO    General                 (launcher.cpp              : 664)   ExSPAnder repeat resolving tool finished\n",
      "  1:48:45.352    91M / 754M  INFO   StageManager             (stage.cpp                 : 185)   STAGE == Contig Output (id: contig_output)\n",
      "  1:48:45.352    91M / 754M  INFO    General                 (contig_output.hpp         :  16)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/before_rr.fasta\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  1:48:45.394    91M / 754M  INFO    General                 (contig_output_stage.cpp   : 151)   Writing GFA graph to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/assembly_graph_with_scaffolds.gfa\n",
      "  1:48:45.436    91M / 754M  INFO    General                 (contig_output_stage.cpp   : 165)   Outputting FastG graph to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/assembly_graph.fastg\n",
      "  1:48:45.556    91M / 754M  INFO    General                 (contig_output_stage.cpp   : 196)   Breaking scaffolds\n",
      "  1:48:45.576   100M / 754M  INFO    General                 (contig_output_stage.cpp   :  98)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/final_contigs.fasta\n",
      "  1:48:45.614   100M / 754M  INFO    General                 (contig_output_stage.cpp   : 104)   Outputting FastG paths to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/final_contigs.paths\n",
      "  1:48:45.633    95M / 754M  INFO    General                 (contig_output_stage.cpp   :  98)   Outputting contigs to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/scaffolds.fasta\n",
      "  1:48:45.670    95M / 754M  INFO    General                 (contig_output_stage.cpp   : 104)   Outputting FastG paths to /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/scaffolds.paths\n",
      "  1:48:45.673    95M / 754M  INFO    General                 (contig_output_stage.cpp   : 111)   Populating GFA with scaffold paths\n",
      "  1:48:45.676    91M / 754M  INFO    General                 (pipeline.cpp              : 287)   SPAdes finished\n",
      "  1:48:45.681     1M / 505M  INFO    General                 (main.cpp                  : 136)   Assembling time: 1 hours 48 minutes 45 seconds\n",
      "\n",
      "===== K55 finished. \n",
      "\n",
      "\n",
      "===== Copy files started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/miniconda3/bin/python /Users/olga/Downloads/SPAdes-3.15.2/share/spades/spades_pipeline/scripts/copy_files.py /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/before_rr.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/before_rr.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/assembly_graph_after_simplification.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/assembly_graph_after_simplification.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/final_contigs.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/contigs.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/first_pe_contigs.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/first_pe_contigs.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/strain_graph.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/strain_graph.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/scaffolds.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/scaffolds.fasta /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/scaffolds.paths /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/scaffolds.paths /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/assembly_graph_with_scaffolds.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/assembly_graph_with_scaffolds.gfa /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/assembly_graph.fastg /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/assembly_graph.fastg /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/K55/final_contigs.paths /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/contigs.paths\n",
      "\n",
      "\n",
      "===== Copy files finished. \n",
      "\n",
      "\n",
      "===== Assembling finished. \n",
      "\n",
      "\n",
      "===== Breaking scaffolds started. \n",
      "\n",
      "\n",
      "== Running: /Users/olga/miniconda3/bin/python /Users/olga/Downloads/SPAdes-3.15.2/share/spades/spades_pipeline/scripts/breaking_scaffolds_script.py --result_scaffolds_filename /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/scaffolds.fasta --misc_dir /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/misc --threshold_for_breaking_scaffolds 3\n",
      "\n",
      "\n",
      "===== Breaking scaffolds finished. \n",
      "\n",
      "\n",
      "===== Terminate started. \n",
      "\n",
      "\n",
      "===== Terminate finished. \n",
      "\n",
      " * Corrected reads are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/corrected/\n",
      " * Assembled contigs are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/contigs.fasta\n",
      " * Assembled scaffolds are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/scaffolds.fasta\n",
      " * Paths in the assembly graph corresponding to the contigs are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/contigs.paths\n",
      " * Paths in the assembly graph corresponding to the scaffolds are in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/scaffolds.paths\n",
      " * Assembly graph is in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/assembly_graph.fastg\n",
      " * Assembly graph in GFA format is in /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/assembly_graph_with_scaffolds.gfa\n",
      "\n",
      "======= SPAdes pipeline finished.\n",
      "\n",
      "SPAdes log can be found here: /Users/olga/Documents/BioinformaticsInstitute/NGS/illumina_pacbio80/spades.log\n",
      "\n",
      "Thank you for using SPAdes!\n"
     ]
    }
   ],
   "source": [
    "! /Users/olga/Downloads/SPAdes-3.15.2/bin/spades.py -1 ./homework6/illumina.100x.1.fq -2 ./homework6/illumina.100x.2.fq --pacbio ./homework6/pacbio_80x.fq -o illumina_pacbio80"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 4. Сравним сборки с помощью quast"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/olga/Downloads/quast-5.0.2/quast.py pacbio10/pacbio10.contigs.fasta pacbio20/pacbio20.contigs.fasta pacbio40/pacbio40.contigs.fasta pacbio80/pacbio80.contigs.fasta illumina/contigs.fasta illumina_pacbio10/contigs.fasta illumina_pacbio20/contigs.fasta illumina_pacbio40/contigs.fasta illumina_pacbio80/contigs.fasta -R homework6/reference.fasta -o quast_result\n",
      "\n",
      "Version: 5.0.2, 0bb1dd1b\n",
      "\n",
      "System information:\n",
      "  OS: Darwin-20.3.0-x86_64-i386-64bit (macosx)\n",
      "  Python version: 3.7.7\n",
      "  CPUs number: 8\n",
      "\n",
      "Started: 2021-04-23 18:28:40\n",
      "\n",
      "Logging to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/quast.log\n",
      "NOTICE: Maximum number of threads is set to 2 (use --threads option to set it manually)\n",
      "\n",
      "CWD: /Users/olga/Documents/BioinformaticsInstitute/NGS\n",
      "Main parameters: \n",
      "  MODE: default, threads: 2, minimum contig length: 500, minimum alignment length: 65, \\\n",
      "  ambiguity: one, threshold for extensive misassembly size: 1000\n",
      "\n",
      "Reference:\n",
      "  /Users/olga/Documents/BioinformaticsInstitute/NGS/homework6/reference.fasta ==> reference\n",
      "\n",
      "Contigs:\n",
      "  Pre-processing...\n",
      "  1  pacbio10/pacbio10.contigs.fasta ==> pacbio10.contigs\n",
      "  2  pacbio20/pacbio20.contigs.fasta ==> pacbio20.contigs\n",
      "  3  pacbio40/pacbio40.contigs.fasta ==> pacbio40.contigs\n",
      "  4  pacbio80/pacbio80.contigs.fasta ==> pacbio80.contigs\n",
      "  5  illumina/contigs.fasta ==> illumina_contigs\n",
      "  6  illumina_pacbio10/contigs.fasta ==> illumina_pacbio10_contigs\n",
      "  7  illumina_pacbio20/contigs.fasta ==> illumina_pacbio20_contigs\n",
      "  8  illumina_pacbio40/contigs.fasta ==> illumina_pacbio40_contigs\n",
      "  9  illumina_pacbio80/contigs.fasta ==> illumina_pacbio80_contigs\n",
      "\n",
      "2021-04-23 19:42:41\n",
      "Running Basic statistics processor...\n",
      "  Reference genome:\n",
      "    reference.fasta, length = 4639675, num fragments = 1, GC % = 50.79\n",
      "  Contig files: \n",
      "    1  pacbio10.contigs\n",
      "    2  pacbio20.contigs\n",
      "    3  pacbio40.contigs\n",
      "    4  pacbio80.contigs\n",
      "    5  illumina_contigs\n",
      "    6  illumina_pacbio10_contigs\n",
      "    7  illumina_pacbio20_contigs\n",
      "    8  illumina_pacbio40_contigs\n",
      "    9  illumina_pacbio80_contigs\n",
      "  Calculating N50 and L50...\n",
      "    1  pacbio10.contigs, N50 = 49862, L50 = 25, Total length = 3540004, GC % = 50.40, # N's per 100 kbp =  0.00\n",
      "    2  pacbio20.contigs, N50 = 273171, L50 = 5, Total length = 4598820, GC % = 50.76, # N's per 100 kbp =  0.00\n",
      "    3  pacbio40.contigs, N50 = 4641141, L50 = 1, Total length = 4641141, GC % = 50.78, # N's per 100 kbp =  0.00\n",
      "    4  pacbio80.contigs, N50 = 4649321, L50 = 1, Total length = 4649321, GC % = 50.80, # N's per 100 kbp =  0.00\n",
      "    5  illumina_contigs, N50 = 129054, L50 = 14, Total length = 4555330, GC % = 50.74, # N's per 100 kbp =  0.00\n",
      "    6  illumina_pacbio10_contigs, N50 = 154542, L50 = 11, Total length = 4568726, GC % = 50.76, # N's per 100 kbp =  0.00\n",
      "    7  illumina_pacbio20_contigs, N50 = 154542, L50 = 11, Total length = 4569348, GC % = 50.76, # N's per 100 kbp =  0.00\n",
      "    8  illumina_pacbio40_contigs, N50 = 154542, L50 = 11, Total length = 4570120, GC % = 50.76, # N's per 100 kbp =  0.00\n",
      "    9  illumina_pacbio80_contigs, N50 = 154542, L50 = 11, Total length = 4571228, GC % = 50.76, # N's per 100 kbp =  0.00\n",
      "  Drawing Nx plot...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/basic_stats/Nx_plot.pdf\n",
      "  Drawing NGx plot...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/basic_stats/NGx_plot.pdf\n",
      "  Drawing cumulative plot...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/basic_stats/cumulative_plot.pdf\n",
      "  Drawing GC content plot...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/basic_stats/GC_content_plot.pdf\n",
      "  Drawing pacbio10.contigs GC content plot...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/basic_stats/pacbio10.contigs_GC_content_plot.pdf\n",
      "  Drawing pacbio20.contigs GC content plot...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/basic_stats/pacbio20.contigs_GC_content_plot.pdf\n",
      "  Drawing pacbio40.contigs GC content plot...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/basic_stats/pacbio40.contigs_GC_content_plot.pdf\n",
      "  Drawing pacbio80.contigs GC content plot...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/basic_stats/pacbio80.contigs_GC_content_plot.pdf\n",
      "  Drawing illumina_contigs GC content plot...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/basic_stats/illumina_contigs_GC_content_plot.pdf\n",
      "  Drawing illumina_pacbio10_contigs GC content plot...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/basic_stats/illumina_pacbio10_contigs_GC_content_plot.pdf\n",
      "  Drawing illumina_pacbio20_contigs GC content plot...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/basic_stats/illumina_pacbio20_contigs_GC_content_plot.pdf\n",
      "  Drawing illumina_pacbio40_contigs GC content plot...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/basic_stats/illumina_pacbio40_contigs_GC_content_plot.pdf\n",
      "  Drawing illumina_pacbio80_contigs GC content plot...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/basic_stats/illumina_pacbio80_contigs_GC_content_plot.pdf\n",
      "  Drawing Coverage histogram (bin size: 1x)...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/basic_stats/coverage_histogram.pdf\n",
      "  Drawing illumina_contigs coverage histogram (bin size: 1x)...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/basic_stats/illumina_contigs_coverage_histogram.pdf\n",
      "  Drawing illumina_pacbio10_contigs coverage histogram (bin size: 1x)...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/basic_stats/illumina_pacbio10_contigs_coverage_histogram.pdf\n",
      "  Drawing illumina_pacbio20_contigs coverage histogram (bin size: 1x)...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/basic_stats/illumina_pacbio20_contigs_coverage_histogram.pdf\n",
      "  Drawing illumina_pacbio40_contigs coverage histogram (bin size: 1x)...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/basic_stats/illumina_pacbio40_contigs_coverage_histogram.pdf\n",
      "  Drawing illumina_pacbio80_contigs coverage histogram (bin size: 1x)...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/basic_stats/illumina_pacbio80_contigs_coverage_histogram.pdf\n",
      "Done.\n",
      "\n",
      "2021-04-23 19:42:47\n",
      "Running Contig analyzer...\n",
      "  1  pacbio10.contigs\n",
      "  1  Logging to files /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/contigs_reports/contigs_report_pacbio10-contigs.stdout and contigs_report_pacbio10-contigs.stderr...\n",
      "  1  Aligning contigs to the reference\n",
      "  2  pacbio20.contigs\n",
      "  2  Logging to files /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/contigs_reports/contigs_report_pacbio20-contigs.stdout and contigs_report_pacbio20-contigs.stderr...\n",
      "  2  Aligning contigs to the reference\n",
      "  1  Analysis is finished.\n",
      "  3  pacbio40.contigs\n",
      "  3  Logging to files /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/contigs_reports/contigs_report_pacbio40-contigs.stdout and contigs_report_pacbio40-contigs.stderr...\n",
      "  3  Aligning contigs to the reference\n",
      "  2  Analysis is finished.\n",
      "  4  pacbio80.contigs\n",
      "  4  Logging to files /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/contigs_reports/contigs_report_pacbio80-contigs.stdout and contigs_report_pacbio80-contigs.stderr...\n",
      "  4  Aligning contigs to the reference\n",
      "  3  Analysis is finished.\n",
      "  5  illumina_contigs\n",
      "  5  Logging to files /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/contigs_reports/contigs_report_illumina_contigs.stdout and contigs_report_illumina_contigs.stderr...\n",
      "  5  Aligning contigs to the reference\n",
      "  4  Analysis is finished.\n",
      "  6  illumina_pacbio10_contigs\n",
      "  6  Logging to files /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/contigs_reports/contigs_report_illumina_pacbio10_contigs.stdout and contigs_report_illumina_pacbio10_contigs.stderr...\n",
      "  6  Aligning contigs to the reference\n",
      "  5  Analysis is finished.\n",
      "  7  illumina_pacbio20_contigs\n",
      "  7  Logging to files /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/contigs_reports/contigs_report_illumina_pacbio20_contigs.stdout and contigs_report_illumina_pacbio20_contigs.stderr...\n",
      "  7  Aligning contigs to the reference\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  6  Analysis is finished.\n",
      "  8  illumina_pacbio40_contigs\n",
      "  8  Logging to files /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/contigs_reports/contigs_report_illumina_pacbio40_contigs.stdout and contigs_report_illumina_pacbio40_contigs.stderr...\n",
      "  8  Aligning contigs to the reference\n",
      "  7  Analysis is finished.\n",
      "  9  illumina_pacbio80_contigs\n",
      "  9  Logging to files /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/contigs_reports/contigs_report_illumina_pacbio80_contigs.stdout and contigs_report_illumina_pacbio80_contigs.stderr...\n",
      "  9  Aligning contigs to the reference\n",
      "  8  Analysis is finished.\n",
      "  9  Analysis is finished.\n",
      "  Creating total report...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/contigs_reports/misassemblies_report.txt, misassemblies_report.tsv, and misassemblies_report.tex\n",
      "  Transposed version of total report...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/contigs_reports/transposed_report_misassemblies.txt, transposed_report_misassemblies.tsv, and transposed_report_misassemblies.tex\n",
      "  Creating total report...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/contigs_reports/unaligned_report.txt, unaligned_report.tsv, and unaligned_report.tex\n",
      "  Drawing misassemblies by types plot...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/contigs_reports/misassemblies_plot.pdf\n",
      "  Drawing misassemblies FRCurve plot...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/contigs_reports/misassemblies_frcurve_plot.pdf\n",
      "Done.\n",
      "\n",
      "2021-04-23 19:43:14\n",
      "Running NA-NGA calculation...\n",
      "  1  pacbio10.contigs, Largest alignment = 149598, NA50 = 47768, NGA50 = 39743, LA50 = 26, LGA50 = 38\n",
      "  2  pacbio20.contigs, Largest alignment = 517261, NA50 = 243589, NGA50 = 243589, LA50 = 6, LGA50 = 6\n",
      "  3  pacbio40.contigs, Largest alignment = 3025639, NA50 = 3025639, NGA50 = 3025639, LA50 = 1, LGA50 = 1\n",
      "  4  pacbio80.contigs, Largest alignment = 3023270, NA50 = 3023270, NGA50 = 3023270, LA50 = 1, LGA50 = 1\n",
      "  5  illumina_contigs, Largest alignment = 221546, NA50 = 129054, NGA50 = 129054, LA50 = 14, LGA50 = 14\n",
      "  6  illumina_pacbio10_contigs, Largest alignment = 236202, NA50 = 152878, NGA50 = 133254, LA50 = 12, LGA50 = 13\n",
      "  7  illumina_pacbio20_contigs, Largest alignment = 236202, NA50 = 152878, NGA50 = 133254, LA50 = 12, LGA50 = 13\n",
      "  8  illumina_pacbio40_contigs, Largest alignment = 236202, NA50 = 152878, NGA50 = 152878, LA50 = 12, LGA50 = 12\n",
      "  9  illumina_pacbio80_contigs, Largest alignment = 236106, NA50 = 152878, NGA50 = 133254, LA50 = 12, LGA50 = 13\n",
      "  Drawing cumulative plot...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/aligned_stats/cumulative_plot.pdf\n",
      "  Drawing NAx plot...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/aligned_stats/NAx_plot.pdf\n",
      "  Drawing NGAx plot...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/aligned_stats/NGAx_plot.pdf\n",
      "Done.\n",
      "\n",
      "2021-04-23 19:43:16\n",
      "Running Genome analyzer...\n",
      "  NOTICE: No file with genomic features were provided. Use the --features option if you want to specify it.\n",
      "\n",
      "  NOTICE: No file with operons were provided. Use the -O option if you want to specify it.\n",
      "  1  pacbio10.contigs\n",
      "  2  pacbio20.contigs\n",
      "  1  Analysis is finished.\n",
      "  3  pacbio40.contigs\n",
      "  2  Analysis is finished.\n",
      "  4  pacbio80.contigs\n",
      "  3  Analysis is finished.\n",
      "  5  illumina_contigs\n",
      "  4  Analysis is finished.\n",
      "  6  illumina_pacbio10_contigs\n",
      "  5  Analysis is finished.\n",
      "  7  illumina_pacbio20_contigs\n",
      "  6  Analysis is finished.\n",
      "  8  illumina_pacbio40_contigs\n",
      "  7  Analysis is finished.\n",
      "  9  illumina_pacbio80_contigs\n",
      "  8  Analysis is finished.\n",
      "  9  Analysis is finished.\n",
      "  Drawing Genome fraction, % histogram...\n",
      "    saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/genome_stats/genome_fraction_histogram.pdf\n",
      "Done.\n",
      "\n",
      "NOTICE: Genes are not predicted by default. Use --gene-finding or --glimmer option to enable it.\n",
      "\n",
      "2021-04-23 19:43:22\n",
      "Creating large visual summaries...\n",
      "This may take a while: press Ctrl-C to skip this step..\n",
      "  1 of 2: Creating Icarus viewers...\n",
      "  2 of 2: Creating PDF with all tables and plots...\n",
      "Done\n",
      "\n",
      "2021-04-23 19:43:30\n",
      "RESULTS:\n",
      "  Text versions of total report are saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/report.txt, report.tsv, and report.tex\n",
      "  Text versions of transposed total report are saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/transposed_report.txt, transposed_report.tsv, and transposed_report.tex\n",
      "  HTML version (interactive tables and plots) is saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/report.html\n",
      "  PDF version (tables and plots) is saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/report.pdf\n",
      "  Icarus (contig browser) is saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/icarus.html\n",
      "  Log is saved to /Users/olga/Documents/BioinformaticsInstitute/NGS/quast_result/quast.log\n",
      "\n",
      "Finished: 2021-04-23 19:43:30\n",
      "Elapsed time: 1:14:49.538681\n",
      "NOTICEs: 4; WARNINGs: 0; non-fatal ERRORs: 0\n",
      "\n",
      "Thank you for using QUAST!\n"
     ]
    }
   ],
   "source": [
    "! /Users/olga/Downloads/quast-5.0.2/quast.py \\\n",
    "pacbio10/pacbio10.contigs.fasta \\\n",
    "pacbio20/pacbio20.contigs.fasta \\\n",
    "pacbio40/pacbio40.contigs.fasta \\\n",
    "pacbio80/pacbio80.contigs.fasta \\\n",
    "illumina/contigs.fasta \\\n",
    "illumina_pacbio10/contigs.fasta \\\n",
    "illumina_pacbio20/contigs.fasta \\\n",
    "illumina_pacbio40/contigs.fasta \\\n",
    "illumina_pacbio80/contigs.fasta \\\n",
    "-R homework6/reference.fasta \\\n",
    "-o quast_result"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 5. Проанализируем результаты (на основе отчёта, сгенерированного quast)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Assembly</th>\n",
       "      <th>pacbio10.contigs</th>\n",
       "      <th>pacbio20.contigs</th>\n",
       "      <th>pacbio40.contigs</th>\n",
       "      <th>pacbio80.contigs</th>\n",
       "      <th>illumina_contigs</th>\n",
       "      <th>illumina_pacbio10_contigs</th>\n",
       "      <th>illumina_pacbio20_contigs</th>\n",
       "      <th>illumina_pacbio40_contigs</th>\n",
       "      <th>illumina_pacbio80_contigs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td># contigs (&gt;= 0 bp)</td>\n",
       "      <td>94</td>\n",
       "      <td>22</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>167</td>\n",
       "      <td>145</td>\n",
       "      <td>140</td>\n",
       "      <td>135</td>\n",
       "      <td>136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td># contigs (&gt;= 1000 bp)</td>\n",
       "      <td>94</td>\n",
       "      <td>22</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>87</td>\n",
       "      <td>63</td>\n",
       "      <td>62</td>\n",
       "      <td>59</td>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td># contigs (&gt;= 5000 bp)</td>\n",
       "      <td>92</td>\n",
       "      <td>22</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>62</td>\n",
       "      <td>46</td>\n",
       "      <td>47</td>\n",
       "      <td>44</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td># contigs (&gt;= 10000 bp)</td>\n",
       "      <td>89</td>\n",
       "      <td>21</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>57</td>\n",
       "      <td>45</td>\n",
       "      <td>46</td>\n",
       "      <td>44</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td># contigs (&gt;= 25000 bp)</td>\n",
       "      <td>55</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>47</td>\n",
       "      <td>39</td>\n",
       "      <td>39</td>\n",
       "      <td>38</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td># contigs (&gt;= 50000 bp)</td>\n",
       "      <td>24</td>\n",
       "      <td>18</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>31</td>\n",
       "      <td>29</td>\n",
       "      <td>30</td>\n",
       "      <td>30</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Total length (&gt;= 0 bp)</td>\n",
       "      <td>3540004</td>\n",
       "      <td>4598820</td>\n",
       "      <td>4641141</td>\n",
       "      <td>4649321</td>\n",
       "      <td>4567161</td>\n",
       "      <td>4580922</td>\n",
       "      <td>4581204</td>\n",
       "      <td>4581146</td>\n",
       "      <td>4582009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Total length (&gt;= 1000 bp)</td>\n",
       "      <td>3540004</td>\n",
       "      <td>4598820</td>\n",
       "      <td>4641141</td>\n",
       "      <td>4649321</td>\n",
       "      <td>4548539</td>\n",
       "      <td>4564350</td>\n",
       "      <td>4566832</td>\n",
       "      <td>4567604</td>\n",
       "      <td>4568712</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Total length (&gt;= 5000 bp)</td>\n",
       "      <td>3530802</td>\n",
       "      <td>4598820</td>\n",
       "      <td>4641141</td>\n",
       "      <td>4649321</td>\n",
       "      <td>4493976</td>\n",
       "      <td>4526103</td>\n",
       "      <td>4538061</td>\n",
       "      <td>4538834</td>\n",
       "      <td>4537248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Total length (&gt;= 10000 bp)</td>\n",
       "      <td>3506582</td>\n",
       "      <td>4591727</td>\n",
       "      <td>4641141</td>\n",
       "      <td>4649321</td>\n",
       "      <td>4458325</td>\n",
       "      <td>4518046</td>\n",
       "      <td>4529247</td>\n",
       "      <td>4538834</td>\n",
       "      <td>4537248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Total length (&gt;= 25000 bp)</td>\n",
       "      <td>2917321</td>\n",
       "      <td>4575443</td>\n",
       "      <td>4641141</td>\n",
       "      <td>4649321</td>\n",
       "      <td>4312386</td>\n",
       "      <td>4439862</td>\n",
       "      <td>4434562</td>\n",
       "      <td>4460648</td>\n",
       "      <td>4442027</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Total length (&gt;= 50000 bp)</td>\n",
       "      <td>1757051</td>\n",
       "      <td>4506249</td>\n",
       "      <td>4641141</td>\n",
       "      <td>4649321</td>\n",
       "      <td>3762852</td>\n",
       "      <td>4111348</td>\n",
       "      <td>4140604</td>\n",
       "      <td>4182994</td>\n",
       "      <td>4168651</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td># contigs</td>\n",
       "      <td>94</td>\n",
       "      <td>22</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>97</td>\n",
       "      <td>70</td>\n",
       "      <td>66</td>\n",
       "      <td>63</td>\n",
       "      <td>64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Largest contig</td>\n",
       "      <td>176275</td>\n",
       "      <td>751732</td>\n",
       "      <td>4641141</td>\n",
       "      <td>4649321</td>\n",
       "      <td>221601</td>\n",
       "      <td>349819</td>\n",
       "      <td>349819</td>\n",
       "      <td>349819</td>\n",
       "      <td>349819</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Total length</td>\n",
       "      <td>3540004</td>\n",
       "      <td>4598820</td>\n",
       "      <td>4641141</td>\n",
       "      <td>4649321</td>\n",
       "      <td>4555330</td>\n",
       "      <td>4568726</td>\n",
       "      <td>4569348</td>\n",
       "      <td>4570120</td>\n",
       "      <td>4571228</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Reference length</td>\n",
       "      <td>4639675</td>\n",
       "      <td>4639675</td>\n",
       "      <td>4639675</td>\n",
       "      <td>4639675</td>\n",
       "      <td>4639675</td>\n",
       "      <td>4639675</td>\n",
       "      <td>4639675</td>\n",
       "      <td>4639675</td>\n",
       "      <td>4639675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>GC (%)</td>\n",
       "      <td>50.40</td>\n",
       "      <td>50.76</td>\n",
       "      <td>50.78</td>\n",
       "      <td>50.80</td>\n",
       "      <td>50.74</td>\n",
       "      <td>50.76</td>\n",
       "      <td>50.76</td>\n",
       "      <td>50.76</td>\n",
       "      <td>50.76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Reference GC (%)</td>\n",
       "      <td>50.79</td>\n",
       "      <td>50.79</td>\n",
       "      <td>50.79</td>\n",
       "      <td>50.79</td>\n",
       "      <td>50.79</td>\n",
       "      <td>50.79</td>\n",
       "      <td>50.79</td>\n",
       "      <td>50.79</td>\n",
       "      <td>50.79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>N50</td>\n",
       "      <td>49862</td>\n",
       "      <td>273171</td>\n",
       "      <td>4641141</td>\n",
       "      <td>4649321</td>\n",
       "      <td>129054</td>\n",
       "      <td>154542</td>\n",
       "      <td>154542</td>\n",
       "      <td>154542</td>\n",
       "      <td>154542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>NG50</td>\n",
       "      <td>41894</td>\n",
       "      <td>273171</td>\n",
       "      <td>4641141</td>\n",
       "      <td>4649321</td>\n",
       "      <td>129054</td>\n",
       "      <td>154542</td>\n",
       "      <td>152878</td>\n",
       "      <td>152878</td>\n",
       "      <td>154542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>N75</td>\n",
       "      <td>30852</td>\n",
       "      <td>199323</td>\n",
       "      <td>4641141</td>\n",
       "      <td>4649321</td>\n",
       "      <td>60768</td>\n",
       "      <td>99738</td>\n",
       "      <td>99738</td>\n",
       "      <td>99738</td>\n",
       "      <td>99738</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>NG75</td>\n",
       "      <td>10787</td>\n",
       "      <td>199323</td>\n",
       "      <td>4641141</td>\n",
       "      <td>4649321</td>\n",
       "      <td>59669</td>\n",
       "      <td>95450</td>\n",
       "      <td>99738</td>\n",
       "      <td>99738</td>\n",
       "      <td>99738</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>L50</td>\n",
       "      <td>25</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>14</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>LG50</td>\n",
       "      <td>37</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>14</td>\n",
       "      <td>11</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>L75</td>\n",
       "      <td>46</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>26</td>\n",
       "      <td>20</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>LG75</td>\n",
       "      <td>87</td>\n",
       "      <td>10</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>27</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>21</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td># misassemblies</td>\n",
       "      <td>10</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td># misassembled contigs</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>Misassembled contigs length</td>\n",
       "      <td>357608</td>\n",
       "      <td>1189535</td>\n",
       "      <td>4641141</td>\n",
       "      <td>4649321</td>\n",
       "      <td>0</td>\n",
       "      <td>808786</td>\n",
       "      <td>349819</td>\n",
       "      <td>349819</td>\n",
       "      <td>855583</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td># local misassemblies</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>5</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td># scaffold gap ext. mis.</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td># scaffold gap loc. mis.</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td># unaligned mis. contigs</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td># unaligned contigs</td>\n",
       "      <td>0 + 0 part</td>\n",
       "      <td>0 + 0 part</td>\n",
       "      <td>0 + 0 part</td>\n",
       "      <td>0 + 0 part</td>\n",
       "      <td>1 + 0 part</td>\n",
       "      <td>1 + 0 part</td>\n",
       "      <td>1 + 0 part</td>\n",
       "      <td>1 + 0 part</td>\n",
       "      <td>1 + 0 part</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>Unaligned length</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>542</td>\n",
       "      <td>542</td>\n",
       "      <td>542</td>\n",
       "      <td>542</td>\n",
       "      <td>542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>Genome fraction (%)</td>\n",
       "      <td>76.336</td>\n",
       "      <td>98.850</td>\n",
       "      <td>99.995</td>\n",
       "      <td>99.998</td>\n",
       "      <td>98.142</td>\n",
       "      <td>98.439</td>\n",
       "      <td>98.457</td>\n",
       "      <td>98.476</td>\n",
       "      <td>98.499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>Duplication ratio</td>\n",
       "      <td>1.000</td>\n",
       "      <td>1.003</td>\n",
       "      <td>1.000</td>\n",
       "      <td>1.002</td>\n",
       "      <td>1.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>1.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td># N's per 100 kbp</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td># mismatches per 100 kbp</td>\n",
       "      <td>44.98</td>\n",
       "      <td>4.82</td>\n",
       "      <td>0.73</td>\n",
       "      <td>0.26</td>\n",
       "      <td>1.03</td>\n",
       "      <td>1.82</td>\n",
       "      <td>1.64</td>\n",
       "      <td>1.64</td>\n",
       "      <td>1.88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td># indels per 100 kbp</td>\n",
       "      <td>661.17</td>\n",
       "      <td>147.00</td>\n",
       "      <td>27.72</td>\n",
       "      <td>10.04</td>\n",
       "      <td>0.33</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.22</td>\n",
       "      <td>0.35</td>\n",
       "      <td>0.31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>Largest alignment</td>\n",
       "      <td>149598</td>\n",
       "      <td>517261</td>\n",
       "      <td>3025639</td>\n",
       "      <td>3023270</td>\n",
       "      <td>221546</td>\n",
       "      <td>236202</td>\n",
       "      <td>236202</td>\n",
       "      <td>236202</td>\n",
       "      <td>236106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>Total aligned length</td>\n",
       "      <td>3539867</td>\n",
       "      <td>4598808</td>\n",
       "      <td>4641139</td>\n",
       "      <td>4646042</td>\n",
       "      <td>4554465</td>\n",
       "      <td>4567861</td>\n",
       "      <td>4568479</td>\n",
       "      <td>4568802</td>\n",
       "      <td>4569910</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>NA50</td>\n",
       "      <td>47768</td>\n",
       "      <td>243589</td>\n",
       "      <td>3025639</td>\n",
       "      <td>3023270</td>\n",
       "      <td>129054</td>\n",
       "      <td>152878</td>\n",
       "      <td>152878</td>\n",
       "      <td>152878</td>\n",
       "      <td>152878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>NGA50</td>\n",
       "      <td>39743</td>\n",
       "      <td>243589</td>\n",
       "      <td>3025639</td>\n",
       "      <td>3023270</td>\n",
       "      <td>129054</td>\n",
       "      <td>133254</td>\n",
       "      <td>133254</td>\n",
       "      <td>152878</td>\n",
       "      <td>133254</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>NA75</td>\n",
       "      <td>28161</td>\n",
       "      <td>174377</td>\n",
       "      <td>939045</td>\n",
       "      <td>931200</td>\n",
       "      <td>60768</td>\n",
       "      <td>95400</td>\n",
       "      <td>99738</td>\n",
       "      <td>99738</td>\n",
       "      <td>95400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>NGA75</td>\n",
       "      <td>8918</td>\n",
       "      <td>174377</td>\n",
       "      <td>939045</td>\n",
       "      <td>931200</td>\n",
       "      <td>59669</td>\n",
       "      <td>88325</td>\n",
       "      <td>99738</td>\n",
       "      <td>99738</td>\n",
       "      <td>88325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>LA50</td>\n",
       "      <td>26</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>14</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>LGA50</td>\n",
       "      <td>38</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>14</td>\n",
       "      <td>13</td>\n",
       "      <td>13</td>\n",
       "      <td>12</td>\n",
       "      <td>13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>LA75</td>\n",
       "      <td>48</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>26</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>LGA75</td>\n",
       "      <td>93</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>27</td>\n",
       "      <td>23</td>\n",
       "      <td>22</td>\n",
       "      <td>22</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       Assembly pacbio10.contigs pacbio20.contigs  \\\n",
       "0           # contigs (>= 0 bp)               94               22   \n",
       "1        # contigs (>= 1000 bp)               94               22   \n",
       "2        # contigs (>= 5000 bp)               92               22   \n",
       "3       # contigs (>= 10000 bp)               89               21   \n",
       "4       # contigs (>= 25000 bp)               55               20   \n",
       "5       # contigs (>= 50000 bp)               24               18   \n",
       "6        Total length (>= 0 bp)          3540004          4598820   \n",
       "7     Total length (>= 1000 bp)          3540004          4598820   \n",
       "8     Total length (>= 5000 bp)          3530802          4598820   \n",
       "9    Total length (>= 10000 bp)          3506582          4591727   \n",
       "10   Total length (>= 25000 bp)          2917321          4575443   \n",
       "11   Total length (>= 50000 bp)          1757051          4506249   \n",
       "12                    # contigs               94               22   \n",
       "13               Largest contig           176275           751732   \n",
       "14                 Total length          3540004          4598820   \n",
       "15             Reference length          4639675          4639675   \n",
       "16                       GC (%)            50.40            50.76   \n",
       "17             Reference GC (%)            50.79            50.79   \n",
       "18                          N50            49862           273171   \n",
       "19                         NG50            41894           273171   \n",
       "20                          N75            30852           199323   \n",
       "21                         NG75            10787           199323   \n",
       "22                          L50               25                5   \n",
       "23                         LG50               37                5   \n",
       "24                          L75               46               10   \n",
       "25                         LG75               87               10   \n",
       "26              # misassemblies               10                8   \n",
       "27       # misassembled contigs                6                3   \n",
       "28  Misassembled contigs length           357608          1189535   \n",
       "29        # local misassemblies                1                3   \n",
       "30     # scaffold gap ext. mis.                0                0   \n",
       "31     # scaffold gap loc. mis.                0                0   \n",
       "32     # unaligned mis. contigs                0                0   \n",
       "33          # unaligned contigs       0 + 0 part       0 + 0 part   \n",
       "34             Unaligned length                0                0   \n",
       "35          Genome fraction (%)           76.336           98.850   \n",
       "36            Duplication ratio            1.000            1.003   \n",
       "37            # N's per 100 kbp             0.00             0.00   \n",
       "38     # mismatches per 100 kbp            44.98             4.82   \n",
       "39         # indels per 100 kbp           661.17           147.00   \n",
       "40            Largest alignment           149598           517261   \n",
       "41         Total aligned length          3539867          4598808   \n",
       "42                         NA50            47768           243589   \n",
       "43                        NGA50            39743           243589   \n",
       "44                         NA75            28161           174377   \n",
       "45                        NGA75             8918           174377   \n",
       "46                         LA50               26                6   \n",
       "47                        LGA50               38                6   \n",
       "48                         LA75               48               11   \n",
       "49                        LGA75               93               11   \n",
       "\n",
       "   pacbio40.contigs pacbio80.contigs illumina_contigs  \\\n",
       "0                 1                1              167   \n",
       "1                 1                1               87   \n",
       "2                 1                1               62   \n",
       "3                 1                1               57   \n",
       "4                 1                1               47   \n",
       "5                 1                1               31   \n",
       "6           4641141          4649321          4567161   \n",
       "7           4641141          4649321          4548539   \n",
       "8           4641141          4649321          4493976   \n",
       "9           4641141          4649321          4458325   \n",
       "10          4641141          4649321          4312386   \n",
       "11          4641141          4649321          3762852   \n",
       "12                1                1               97   \n",
       "13          4641141          4649321           221601   \n",
       "14          4641141          4649321          4555330   \n",
       "15          4639675          4639675          4639675   \n",
       "16            50.78            50.80            50.74   \n",
       "17            50.79            50.79            50.79   \n",
       "18          4641141          4649321           129054   \n",
       "19          4641141          4649321           129054   \n",
       "20          4641141          4649321            60768   \n",
       "21          4641141          4649321            59669   \n",
       "22                1                1               14   \n",
       "23                1                1               14   \n",
       "24                1                1               26   \n",
       "25                1                1               27   \n",
       "26                8                8                0   \n",
       "27                1                1                0   \n",
       "28          4641141          4649321                0   \n",
       "29                2                2                4   \n",
       "30                0                0                0   \n",
       "31                0                0                0   \n",
       "32                0                0                0   \n",
       "33       0 + 0 part       0 + 0 part       1 + 0 part   \n",
       "34                0                0              542   \n",
       "35           99.995           99.998           98.142   \n",
       "36            1.000            1.002            1.000   \n",
       "37             0.00             0.00             0.00   \n",
       "38             0.73             0.26             1.03   \n",
       "39            27.72            10.04             0.33   \n",
       "40          3025639          3023270           221546   \n",
       "41          4641139          4646042          4554465   \n",
       "42          3025639          3023270           129054   \n",
       "43          3025639          3023270           129054   \n",
       "44           939045           931200            60768   \n",
       "45           939045           931200            59669   \n",
       "46                1                1               14   \n",
       "47                1                1               14   \n",
       "48                2                2               26   \n",
       "49                2                2               27   \n",
       "\n",
       "   illumina_pacbio10_contigs illumina_pacbio20_contigs  \\\n",
       "0                        145                       140   \n",
       "1                         63                        62   \n",
       "2                         46                        47   \n",
       "3                         45                        46   \n",
       "4                         39                        39   \n",
       "5                         29                        30   \n",
       "6                    4580922                   4581204   \n",
       "7                    4564350                   4566832   \n",
       "8                    4526103                   4538061   \n",
       "9                    4518046                   4529247   \n",
       "10                   4439862                   4434562   \n",
       "11                   4111348                   4140604   \n",
       "12                        70                        66   \n",
       "13                    349819                    349819   \n",
       "14                   4568726                   4569348   \n",
       "15                   4639675                   4639675   \n",
       "16                     50.76                     50.76   \n",
       "17                     50.79                     50.79   \n",
       "18                    154542                    154542   \n",
       "19                    154542                    152878   \n",
       "20                     99738                     99738   \n",
       "21                     95450                     99738   \n",
       "22                        11                        11   \n",
       "23                        11                        12   \n",
       "24                        20                        21   \n",
       "25                        21                        21   \n",
       "26                         3                         1   \n",
       "27                         3                         1   \n",
       "28                    808786                    349819   \n",
       "29                         4                         4   \n",
       "30                         0                         0   \n",
       "31                         0                         0   \n",
       "32                         0                         0   \n",
       "33                1 + 0 part                1 + 0 part   \n",
       "34                       542                       542   \n",
       "35                    98.439                    98.457   \n",
       "36                     1.000                     1.000   \n",
       "37                      0.00                      0.00   \n",
       "38                      1.82                      1.64   \n",
       "39                      0.35                      0.22   \n",
       "40                    236202                    236202   \n",
       "41                   4567861                   4568479   \n",
       "42                    152878                    152878   \n",
       "43                    133254                    133254   \n",
       "44                     95400                     99738   \n",
       "45                     88325                     99738   \n",
       "46                        12                        12   \n",
       "47                        13                        13   \n",
       "48                        22                        22   \n",
       "49                        23                        22   \n",
       "\n",
       "   illumina_pacbio40_contigs illumina_pacbio80_contigs  \n",
       "0                        135                       136  \n",
       "1                         59                        60  \n",
       "2                         44                        44  \n",
       "3                         44                        44  \n",
       "4                         38                        37  \n",
       "5                         30                        29  \n",
       "6                    4581146                   4582009  \n",
       "7                    4567604                   4568712  \n",
       "8                    4538834                   4537248  \n",
       "9                    4538834                   4537248  \n",
       "10                   4460648                   4442027  \n",
       "11                   4182994                   4168651  \n",
       "12                        63                        64  \n",
       "13                    349819                    349819  \n",
       "14                   4570120                   4571228  \n",
       "15                   4639675                   4639675  \n",
       "16                     50.76                     50.76  \n",
       "17                     50.79                     50.79  \n",
       "18                    154542                    154542  \n",
       "19                    152878                    154542  \n",
       "20                     99738                     99738  \n",
       "21                     99738                     99738  \n",
       "22                        11                        11  \n",
       "23                        12                        11  \n",
       "24                        21                        20  \n",
       "25                        21                        20  \n",
       "26                         1                         3  \n",
       "27                         1                         3  \n",
       "28                    349819                    855583  \n",
       "29                         5                         6  \n",
       "30                         0                         0  \n",
       "31                         0                         0  \n",
       "32                         0                         0  \n",
       "33                1 + 0 part                1 + 0 part  \n",
       "34                       542                       542  \n",
       "35                    98.476                    98.499  \n",
       "36                     1.000                     1.000  \n",
       "37                      0.00                      0.00  \n",
       "38                      1.64                      1.88  \n",
       "39                      0.35                      0.31  \n",
       "40                    236202                    236106  \n",
       "41                   4568802                   4569910  \n",
       "42                    152878                    152878  \n",
       "43                    152878                    133254  \n",
       "44                     99738                     95400  \n",
       "45                     99738                     88325  \n",
       "46                        12                        12  \n",
       "47                        12                        13  \n",
       "48                        22                        22  \n",
       "49                        22                        23  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "quast_data = pd.read_csv('quast_result/report.tsv', sep='\\t')\n",
    "quast_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1) **# contigs:** лучшее - pacbio40, pacbio80, худшие значения - pacbio10, illumina <br>\n",
    "2) **Largest contig:** лучшее - pacbio40, pacbio80, худшие значения - pacbio10, illumina <br>\n",
    "3) **Total length:** лучшее - pacbio40, pacbio80, но остальные значения, кроме pacbio10, также хорошие <br>\n",
    "4) **N50:** лучшее - pacbio40, pacbio80, худшее значение - pacbio10, неплохое значение по сравнению с другими показывает pacbio20 <br>\n",
    "5) **NG50:** лучшее - pacbio40, pacbio80, худшее значение - pacbio10, неплохое значение по сравнению с другими показывает pacbio20 <br>\n",
    "6) **N75:** лучшее - pacbio40, pacbio80, худшее значение - pacbio10, неплохое значение по сравнению с другими показывает pacbio20 <br>\n",
    "7) **NG75:** лучшее - pacbio40, pacbio80, худшее значение - pacbio10, неплохое значение по сравнению с другими показывает pacbio20 <br>\n",
    "8) **L50:** лучшее - pacbio40, pacbio80, худшее значение - pacbio10, неплохое значение по сравнению с другими показывает pacbio20 <br>\n",
    "9) **LG50:** лучшее - pacbio40, pacbio80, худшее значение - pacbio10, неплохое значение по сравнению с другими показывает pacbio20 <br>\n",
    "10) **L75:** лучшее - pacbio40, pacbio80, худшее значение - pacbio10, неплохое значение по сравнению с другими показывает pacbio20 <br>\n",
    "11) **LG75:** лучшее - pacbio40, pacbio80, худшее значение - pacbio10, неплохое значение по сравнению с другими показывает pacbio20 <br>\n",
    "12) **# misassemblies:** здесь лучшие значения показывают illumina, illumina+pacbio20, illumina+pacbio40, худшие - все pacbio <br>\n",
    "13) **# misassembled contigs:** лучшее - pacbio40, pacbio80, illumina, illumina+pacbio20, illumina+pacbio40, остальные значения приемлемы <br>\n",
    "14) **# local misassemblies:** лучшее - pacbio10, pacbio40, pacbio80, остальные значения приемлемы <br>\n",
    "15) **# unaligned contigs & unaligned length:** лучшее - все pacbio, остальные значения также хорошие <br>\n",
    "16) **Genome fraction:** лучшее - pacbio40, pacbio80, остальные значения, кроме pacbio10, также очень хорошие <br>\n",
    "17) **Duplication ratio:** лучшие значения показали все, кроме pacbio20 и pacbio80 <br>\n",
    "18) **# mismatches per 100 kbp:** лучшее - pacbio40, pacbio80, приемлемое значение показывает illumina, худшее значение - pacbio10 <br>\n",
    "19) **# indels per 100 kbp:** лучшее - illumina и гибридные сборки, худшее значение - pacbio10 <br>\n",
    "20) **Largest alignment:** лучшее - pacbio40, pacbio80, худшее значение - pacbio10 <br>\n",
    "21) **Total aligned length:** лучшее - pacbio40, pacbio80, но остальные значения, кроме pacbio10, также хорошие <br>\n",
    "22) **NA50:** лучшее - pacbio40, pacbio80, худшее значение - pacbio10 <br>\n",
    "23) **NGA50:** лучшее - pacbio40, pacbio80, худшее значение - pacbio10 <br>\n",
    "24) **NA75:** лучшее - pacbio40, pacbio80, худшее значение - pacbio10 <br>\n",
    "25) **NGA75:** лучшее - pacbio40, pacbio80, худшее значение - pacbio10 <br>\n",
    "26) **LA50:** лучшее - pacbio40, pacbio80, худшее значение - pacbio10 <br>\n",
    "27) **LGA50:** лучшее - pacbio40, pacbio80, худшее значение - pacbio10 <br>\n",
    "28) **LA75:** лучшее - pacbio40, pacbio80, худшее значение - pacbio10 <br>\n",
    "29) **LGA75:** лучшее - pacbio40, pacbio80, худшее значение - pacbio10 <br>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Таким образом, лучшие сборки по метрикам - pacbio40, pacbio80, худшая - pacbio10 (видимо, из-за низкого покрытия)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
